






<p><br />
</p>


<p><strong><span><span class="smallcaps">Q-Chem</span></span> User’s Manual</strong><br />
<br />
</p>
<p><strong>Version 5.1 was edited by:</strong><br />
</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;">Dr. Andrew Gilbert</td>
</tr>
<tr class="even">
<td style="text-align: left;">Prof. John Herbert</td>
</tr>
</tbody>
</table>
<p><strong>Version 5.0 was edited by:</strong><br />
</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;">Dr. Andrew Gilbert</td>
</tr>
</tbody>
</table>
<p><strong>Version 4 editors:</strong><br />
</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;">Prof. John Herbert</td>
</tr>
<tr class="even">
<td style="text-align: left;">Prof. Anna Krylov</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dr. Narbe Mardirossian</td>
</tr>
<tr class="even">
<td style="text-align: left;">Prof. Martin Head-Gordon</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dr. Emil Proynov</td>
</tr>
<tr class="even">
<td style="text-align: left;">Dr. Andrew Gilbert</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Dr. Jing Kong</td>
</tr>
</tbody>
</table>
<p><strong>The contributions of individual developers to each version are highlighted in “New Features”,<br />
Section <a href="#sect:features" data-reference-type="ref" data-reference="sect:features">1.3</a></strong><br />
</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"><p><strong>Published by:</strong></p></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">Q-Chem, Inc.</td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Telephone:</td>
<td style="text-align: left;">(412) 687-0695</td>
</tr>
<tr class="odd">
<td style="text-align: left;">6601 Owens Dr.</td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Facsimile:</td>
<td style="text-align: left;">(412) 687-0698</td>
</tr>
<tr class="even">
<td style="text-align: left;">Suite 105</td>
<td style="text-align: left;"></td>
<td style="text-align: left;">email:</td>
<td style="text-align: left;"><a href="mailto:support@q-chem.com">support@q-chem.com</a></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Pleasanton, CA 94588</td>
<td style="text-align: left;"></td>
<td style="text-align: left;">website:</td>
<td style="text-align: left;"><a href="http://www.q-chem.com">www.q-chem.com</a></td>
</tr>
</tbody>
</table>
<p><br />
is a trademark of Q-Chem, Inc. All rights reserved.<br />
The information in this document applies to version 5.1 of <span><span class="smallcaps">Q-Chem</span></span>.<br />
This document version generated on .</p>

<p>Copyright 2000–2018 Q-Chem, Inc. This document is protected under the U.S. Copyright Act of 1976 and state trade secret laws. Unauthorized disclosure, reproduction, distribution, or use is prohibited and may violate federal and state laws.</p>




<h1 id="chap:intro">Introduction</h1>
<h2 id="about-this-manual">About This Manual</h2>
<h3 id="overview">Overview</h3>
<p>This manual is intended as a general-purpose user’s guide for <span><span class="smallcaps">Q-Chem</span></span>, a modern electronic structure program. The manual contains background information that describes <span><span class="smallcaps">Q-Chem</span></span> methods and user-selected parameters. It is assumed that the user has some familiarity with the Unix/Linux environment, an ASCII file editor, and a basic understanding of quantum chemistry.</p>
<p>After installing <span><span class="smallcaps">Q-Chem</span></span> and making necessary adjustments to your user account, it is recommended that particular attention be given to Chapters <a href="#chap:inputs" data-reference-type="ref" data-reference="chap:inputs">3</a> and <a href="#chap:scf" data-reference-type="ref" data-reference="chap:scf">4</a>. The latter, which describes <span><span class="smallcaps">Q-Chem</span></span>’s self-consistent field capabilities, has been formatted so that advanced users can quickly find the information they require while supplying new users with a moderate level of important background information. This format has been maintained throughout the manual, and every attempt has been made to guide the user forward and backward to other relevant information so that a logical progression through this manual is not necessary.</p>
<p>Documentation for <span><span class="smallcaps">IQmol</span></span>, a graphical user interface designed for use with <span><span class="smallcaps">Q-Chem</span></span>, can be found on the <a href="www.iqmol.org" class="uri">www.iqmol.org</a>. <span><span class="smallcaps">IQmol</span></span> functions as a molecular structure builder, as an interface for local or remote submission of <span><span class="smallcaps">Q-Chem</span></span> jobs, and as a post-calculation visualization program for densities and molecular orbitals.</p>
<h3 id="chapter-summaries">Chapter Summaries</h3>
<dl>
<dt>Ch. 1:</dt>
<dd><p>General overview of <span><span class="smallcaps">Q-Chem</span></span>’s features, contributors, and contact information.</p>
</dd>
<dt>Ch. 2:</dt>
<dd><p>Procedures to install, test, and run <span><span class="smallcaps">Q-Chem</span></span> on your machine.</p>
</dd>
<dt>Ch. 3:</dt>
<dd><p>Overview of the <span><span class="smallcaps">Q-Chem</span></span> command-line input.</p>
</dd>
<dt>Ch. 4:</dt>
<dd><p>Running ground-state self-consistent field calculations.</p>
</dd>
<dt>Ch. 5:</dt>
<dd><p>Details specific to running density functional theory (DFT) calculations.</p>
</dd>
<dt>Ch. 6:</dt>
<dd><p>Running post-Hartree-Fock correlated wave function calculations for ground states.</p>
</dd>
<dt>Ch. 7:</dt>
<dd><p>Running calculations for excited states and open-shell species.</p>
</dd>
<dt>Ch. 8:</dt>
<dd><p>Using <span><span class="smallcaps">Q-Chem</span></span>’s built-in basis sets, or specifying a user-defined basis set.</p>
</dd>
<dt>Ch. 9:</dt>
<dd><p>Using <span><span class="smallcaps">Q-Chem</span></span>’s effective core potential capabilities.</p>
</dd>
<dt>Ch. 10:</dt>
<dd><p>Options available for exploring potential energy surfaces, such as determining critical points (transition states and local minima on a single surface, or minimum-energy crossing points between surfaces) as well as <em>ab initio</em> molecular dynamics.</p>
</dd>
<dt>Ch. 11:</dt>
<dd><p>Molecular properties and <em>a posteriori</em> wave function analysis.</p>
</dd>
<dt>Ch. 12:</dt>
<dd><p>Methods for molecules in complex environments, including implicit solvation models, QM/MM models, the Effective Fragment Potential, and density embedding.</p>
</dd>
<dt>Ch. 13:</dt>
<dd><p>Fragment-based approaches for efficient calculations on large systems, calculation of non-covalent interactions, and energy decomposition analysis.</p>
</dd>
<dt>App. A:</dt>
<dd><p>Overview of the <span><span class="smallcaps">Optimize</span></span> package used for determining molecular geometry critical points.</p>
</dd>
<dt>App. B:</dt>
<dd><p>Overview of the <span><span class="smallcaps">AOInts</span></span> library, which contains some of the fastest two-electron integral code currently available.</p>
</dd>
<dt>App. C:</dt>
<dd><p>Quick-reference section containing an alphabetized list of job control variables.</p>
</dd>
</dl>
<h2 id="q-chem-inc."><span><span class="smallcaps">Q-Chem</span></span>, Inc.</h2>
<h3 id="contact-information-and-customer-support">Contact Information and Customer Support</h3>
<p>For general information regarding <span><span class="smallcaps">Q-Chem</span></span> program, visit <a href="http://www.q-chem.com">www.q-chem.com</a>. Full customer support is promptly provided via telephone or email (<a href="mailto:support@q-chem.com">support@q-chem.com</a>) for those customers who have purchased <span><span class="smallcaps">Q-Chem</span></span>’s “QMP” maintenance contract. In addition to free customer support, this contract provides discounts on future updates and releases of <span><span class="smallcaps">Q-Chem</span></span>. For details of the maintenance contract please see <a href="http://www.q-chem.com">www.q-chem.com</a>.</p>
<h3 id="about-the-company">About the Company</h3>
<p><span><span class="smallcaps">Q-Chem</span></span>, Inc. was founded in 1993 and was based in Pittsburgh, PA until 2013, when it relocated to Pleasanton, CA. <span><span class="smallcaps">Q-Chem</span></span>’s scientific contributors include leading quantum chemists around the world. The company is governed by the Board of Directors which currently consists of Peter Gill (Canberra), Anna Krylov (USC), John Herbert (Ohio State), and Hilary Pople. Fritz Schaefer (Georgia) is a Board Member Emeritus. Martin Head-Gordon is a Scientific Advisor to the Board. The close coupling between leading university research groups and <span><span class="smallcaps">Q-Chem</span></span> Inc. ensures that the methods and algorithms available in <span><span class="smallcaps">Q-Chem</span></span> are state-of-the-art.</p>
<p>In order to create this technology, the founders of <span><span class="smallcaps">Q-Chem</span></span>, Inc. built entirely new methodologies from the ground up, using the latest algorithms and modern programming techniques. Since 1993, well over 300 person-years have been devoted to the development of the <span><span class="smallcaps">Q-Chem</span></span> program. The author list of the program shows the full list of contributors to the current version, and the journal citations for <span><span class="smallcaps">Q-Chem</span></span> versions 2, 3, and 4<span class="citation" data-cites="QCHEM2 QCHEM3 QCHEM4"></span> illustrate the breadth of the <span><span class="smallcaps">Q-Chem</span></span> developer community. The current group of developers consist of more than 100 people in 9 countries. A brief history of <span><span class="smallcaps">Q-Chem</span></span> is given in the recent article <span><em>Q-Chem: An Engine for Innovation</em></span>.<span class="citation" data-cites="Krylov:2013a"></span></p>
<h3 id="company-mission">Company Mission</h3>
<p>The mission of <span><span class="smallcaps">Q-Chem</span></span>, Inc. is to develop, distribute, and support innovative and <span><em>sustainable</em></span> quantum chemistry software for industrial, government and academic researchers in the chemical, petrochemical, biochemical, pharmaceutical and material sciences.</p>
<h2 id="sect:features"><span><span class="smallcaps">Q-Chem</span></span> Features</h2>
<p>Quantum chemistry methods have proven invaluable for studying chemical and physical properties of molecules. The <span><span class="smallcaps">Q-Chem</span></span> system brings together a variety of advanced computational methods and tools in an integrated <em>ab initio</em> software package, greatly improving the speed and accuracy of calculations being performed. In addition, <span><span class="smallcaps">Q-Chem</span></span> will accommodate larger molecular structures than previously possible, with no loss in accuracy, thereby bringing the power of quantum chemistry to critical research projects for which this tool was previously unavailable. Below is a reverse-chronological listing of new features added to <span><span class="smallcaps">Q-Chem</span></span>.</p>
<h3 id="sect:features-5.1">New Features in <span><span class="smallcaps">Q-Chem</span></span> 5.1</h3>
<ul>
<li><p>Improved OpenMP parallelization for:</p>
<ul>
<li><p>SCF vibrational frequency calculations (Z. Gan)</p></li>
<li><p>RIMP2 gradient (F. Rob, Joonho Lee, X. Feng, &amp; E. Epifanovsky)</p></li>
</ul></li>
<li><p>Complete active space self-consistent field (CASSCF) and adaptive sampling CI (D. Levine, M. Head-Gordon)</p></li>
<li><p>Tkatchenko-Scheffler van der Waals method (Section <a href="#sect:TS" data-reference-type="ref" data-reference="sect:TS">[sect:TS]</a>) and many-body dispersion method (Section <a href="#sect:MBD" data-reference-type="ref" data-reference="sect:MBD">[sect:MBD]</a>) (D. Barton, Ka Un Lao, &amp; R. DiStasio)</p></li>
<li><p>Enhancements to the coupled-cluster package:</p>
<ul>
<li><p>Core/valence separation for EOM-CCSD core-level excited states (M. Vidal, A.I. Krylov, X. Feng, &amp; S. Coriani), Section <a href="#sub:cvs_eom" data-reference-type="ref" data-reference="sub:cvs_eom">[sub:cvs_eom]</a>.</p></li>
<li><p>NTO analysis of two-photon transitions (K. Nanda &amp; A.I. Krylov), Section <a href="#sec:2PA" data-reference-type="ref" data-reference="sec:2PA">[sec:2PA]</a>.</p></li>
<li><p>NTO analysis of the complex-valued EOM wave functions (A.I. Krylov, W. Skomorowski), Section <a href="#sec:EOMGRAD" data-reference-type="ref" data-reference="sec:EOMGRAD">[sec:EOMGRAD]</a>.</p></li>
<li><p>Analytic gradients for Cholesky-decomposed and resolution-of-identity CCSD and EOM-CCSD (X. Feng, A.I. Krylov).</p></li>
<li><p>Improved performance, reduced disk usage by coupled-cluster methods (E. Epifanovsky, I. Kaliman, &amp; X. Feng).</p></li>
</ul></li>
<li><p>New features in NTO analysis: Energies of NTOs (A.I. Krylov), Section <a href="#subsec:esanacc" data-reference-type="ref" data-reference="subsec:esanacc">[subsec:esanacc]</a>.</p></li>
<li><p>Finite-difference evaluation of non-linear properties (M. de Wergifosse &amp; A.I. Krylov), Section <a href="#sect:Romberg" data-reference-type="ref" data-reference="sect:Romberg">[sect:Romberg]</a>.</p></li>
<li><p>Poisson boundary conditions for SCF calculations (M. Coons &amp; J. Herbert), Section <a href="#sect:peq_bc" data-reference-type="ref" data-reference="sect:peq_bc">[sect:peq_bc]</a>.</p>
<ul>
<li><p>Enables quantum chemistry calculations in an arbitrary (anisotropic and inhomogeneous) dielectric environment.</p></li>
<li><p>Nonequilibrium solvent corrections for vertical ionization energies.</p></li>
</ul></li>
<li><p>Energy decomposition analysis (EDA):</p>
<ul>
<li><p>EDA based on symmetry-adapted perturbation theory and constrained DFT (SAPT/cDFT-EDA), Section <a href="#sect:SAPTEDA" data-reference-type="ref" data-reference="sect:SAPTEDA">[sect:SAPTEDA]</a> (Ka Un Lao, K. Fenk, &amp; J. Herbert)</p></li>
<li><p>ALMO-EDA for CIS and TDDFT/TDA excited states (Qinghui Ge, Yuezhi Mao, &amp; M. Head-Gordon)</p></li>
<li><p>Perturbative ALMO-CTA and COVP analysis in EDA2 (Yuezhi Mao &amp; M. Head-Gordon)</p></li>
</ul></li>
<li><p>Analytic derivative couplings for computing excitation/vibration energy couplings within the Frenkel-Davydov exciton model (A. Morrison &amp; J. Herbert), Section <a href="#subsec:AIFDEM-derivs" data-reference-type="ref" data-reference="subsec:AIFDEM-derivs">13.15.3</a>.</p></li>
<li><p>Hyperfine spin-spin couplings and nuclear electric quadrupole couplings, Section <a href="#sec:magnetman" data-reference-type="ref" data-reference="sec:magnetman">[sec:magnetman]</a> (E. Berquist &amp; D. Lambrecht)</p></li>
<li><p>Variational two-electron reduced-density-matrix (v2RDM) and v2RDM-driven complete active space self-consistent field (v2RDM-CASSCF) method (G. Gidofalvi, L. Koulias, J.W. Mullinax, &amp; A.E. DePrince III)</p></li>
<li><p>Frozen and restrained potential energy scans, Section <a href="#sec:pes-scan" data-reference-type="ref" data-reference="sec:pes-scan">[sec:pes-scan]</a> (Yihan Shao)</p></li>
<li><p>Extended ESP charge fitting procedure to the computation of RESP charges (Yihan Shao)</p></li>
</ul>
<h3 id="sect:features-5.0">New Features in <span><span class="smallcaps">Q-Chem</span></span> 5.0</h3>
<ul>
<li><p>Enhancements to the coupled-cluster package:</p>
<ul>
<li><p>Analytic gradients for Cholesky-decomposed CCSD and EOM-CCSD; efficiency improvement for canonical CCSD and EOM-CCSD gradients (X. Feng, E. Epifanovsky).</p></li>
<li><p>CAP-EOM-CCSD analytic gradients (Z. Benda and T.-C. Jagau) and Dyson orbitals for metastable states (T.-C. Jagau, A.I. Krylov), Section <a href="#sub:cc_cs_cap" data-reference-type="ref" data-reference="sub:cc_cs_cap">[sub:cc_cs_cap]</a>).</p></li>
<li><p>CAP-EOM-MP2 method (A. Kunitsa, K. Bravaya).</p></li>
<li><p>Evaluation of polarizabilities using CCSD and EOM-CCSD (EE and SF) wave functions using full derivative formulation (K. Nanda and A. Krylov, Section <a href="#sec:EOMpol" data-reference-type="ref" data-reference="sec:EOMpol">[sec:EOMpol]</a>).</p></li>
<li><p>Evaluation of <span class="math inline">⟨<em>S</em><sup>2</sup>⟩</span> for EOM-CCSD wave functions (X. Feng).</p></li>
<li><p>Evaluation of NACs for EOM-CCSD wave functions (S. Faraji, A. Krylov, E. Epifanovski, X. Feng, Section <a href="#sec:EOMNAC" data-reference-type="ref" data-reference="sec:EOMNAC">[sec:EOMNAC]</a>).</p></li>
<li><p>Efficiency improvement and new multicore-parallel code for (T) correction (I. Kaliman).</p></li>
<li><p>New coupled-cluster based methods for core states (A. Krylov).</p></li>
</ul></li>
<li><p>New capabilities for implicit solvation modeling:</p>
<ul>
<li><p>PCM capabilities for computing vertical excitation, ionization, and electron attachment energies at EOM-CC and MP2 levels (Section <a href="#sec:EOM-PCM" data-reference-type="ref" data-reference="sec:EOM-PCM">[sec:EOM-PCM]</a>).</p></li>
<li><p>State-specific equilibrium and non-equilibrium solvation for all orders and variants of ADC (J. M. Mewes and A. Dreuw; Section <a href="#subsec:ADC-PCM" data-reference-type="ref" data-reference="subsec:ADC-PCM">[subsec:ADC-PCM]</a>).</p></li>
<li><p>Poisson equation boundary conditions allowing use of an arbitrary, anisotropic dielectric function <span class="math inline"><em>ε</em>(<strong>r</strong>)</span>, with full treatment of volume polarization (M. P. Coons and J. M. Herbert; Section <a href="#sect:peq_bc" data-reference-type="ref" data-reference="sect:peq_bc">[sect:peq_bc]</a>).</p></li>
<li><p>Composite Model for Implicit Representation of Solvent (CMIRS), an accurate model for free energies of solvation (Section <a href="#subsec:nonels-PCM" data-reference-type="ref" data-reference="subsec:nonels-PCM">[subsec:nonels-PCM]</a>)</p></li>
</ul></li>
<li><p>New density functionals (N. Mardirossian and M. Head-Gordon; Section <a href="#sect:XCfuncs" data-reference-type="ref" data-reference="sect:XCfuncs">[sect:XCfuncs]</a>):</p>
<ul>
<li><p>GGA functionals: BEEF-vdW, HLE16, KT1, KT2, KT3, rVV10</p></li>
<li><p>Meta-GGA functionals: B97M-rV, BLOC, mBEEF, oTPSS, TM</p></li>
<li><p>Hybrids: CAM-QTP(00), CAM-QTP(01), HSE-HJS, LC-<span class="math inline"><em>ω</em></span>PBE08, MN15, rCAM-B3LYP, WC04, WP04</p></li>
<li><p>Double hybrids: B2GP-PLYP, DSD-PBEB95-D3, DSD-PBEP86-D3, DSD-PBEPBE-D3, LS1DH-PBE, PBE-QIDH, PTPSS-D3, PWPB95-D3</p></li>
<li><p>Grimme’s PBEh-3c “low-cost” composite method</p></li>
<li><p>rVV10 non-local correlation functional</p></li>
</ul></li>
<li><p>Additional DFT developments:</p>
<ul>
<li><p>New forms of DFT-D3 (J. Witte; Section <a href="#subsec:DFT-D" data-reference-type="ref" data-reference="subsec:DFT-D">[subsec:DFT-D]</a>).</p></li>
<li><p>New standard integration grids, SG-2 and SG-3 (S. Dasgupta and J. M. Herbert; Section <a href="#sect:standard-quad" data-reference-type="ref" data-reference="sect:standard-quad">[sect:standard-quad]</a>).</p></li>
<li><p>More efficient propagator algorithms for real-time TDDFT (Y. Zhu and J. M. Herbert; Section <a href="#sec:RT-TDDFT" data-reference-type="ref" data-reference="sec:RT-TDDFT">[sec:RT-TDDFT]</a>).</p></li>
</ul></li>
<li><p>New integral package for for computing effective core potential (ECP) integrals (S. C. McKenzie, E. Epifanovsky; Chapter <a href="#chap:ecp" data-reference-type="ref" data-reference="chap:ecp">9</a>).</p>
<ul>
<li><p>More efficient analytic algorithms for energies and first derivatives.</p></li>
<li><p>Support for arbitrary projector angular momentum.</p></li>
<li><p>Support up to <span class="math inline"><em>h</em></span> angular momentum in the basis set.</p></li>
</ul></li>
<li><p>Analytic derivative couplings for the <span><em>ab initio</em></span> Frenkel-Davydov exciton model (A. F. Morrison and J. M. Herbert; Section <a href="#subsec:AIFDEM-derivs" data-reference-type="ref" data-reference="subsec:AIFDEM-derivs">13.15.3</a>).</p></li>
<li><p>New ALMO-based energy decomposition analysis (EDA) methods:</p>
<ul>
<li><p>The second-generation ALMO-EDA methods for DFT (P. R. Horn, Y. Mao and M. Head-Gordon; Section <a href="#sect:eda2" data-reference-type="ref" data-reference="sect:eda2">[sect:eda2]</a>)</p></li>
<li><p>The extension of ALMO-EDA to RIMP2 theory (J. Thirman and M. Head-Gordon; Section <a href="#sect:mp2_eda" data-reference-type="ref" data-reference="sect:mp2_eda">[sect:mp2_eda]</a>)</p></li>
<li><p>The “adiabatic&quot; EDA method for decomposing changes in molecular properties (Y. Mao, P. R. Horn and M. Head-Gordon; Section <a href="#sect:adiabatic_eda" data-reference-type="ref" data-reference="sect:adiabatic_eda">[sect:adiabatic_eda]</a>)</p></li>
</ul></li>
<li><p>Wave function correlation capabilities:</p>
<ul>
<li><p>Coupled cluster valence bond (CCVB) method for describing open-shell molecules with strong spin correlations (D. W. Small and M. Head-Gordon; Section <a href="#subsec:CCVB" data-reference-type="ref" data-reference="subsec:CCVB">[subsec:CCVB]</a>).</p></li>
<li><p>Implementation of coupled-cluster valence bond with singles and doubles (CCVB-SD) for closed-shell species (J. Lee, D. W. Small and M. Head-Gordon; Section <a href="#sect:ccvbsd" data-reference-type="ref" data-reference="sect:ccvbsd">[sect:ccvbsd]</a>).</p></li>
</ul></li>
</ul>
<p><strong>Note: </strong></p>
<p>Several important changes in<span><span class="smallcaps">Q-Chem</span></span>’s default settings have occurred since version 4.4.</p>
<ul>
<li><p>Core electrons are now frozen by default in most post-Hartree-Fock calculations; see Section <a href="#sec:core" data-reference-type="ref" data-reference="sec:core">[sec:core]</a>.</p></li>
<li><p>The keywords for calculation of SOCs and NACs were renamed for consistency between different methods.</p></li>
<li><p>Some newer density functionals now use either the SG-2 or SG-3 quadrature grid by default, whereas all functionals used SG-1 by default in v. 4.4. Table <a href="#table:SG" data-reference-type="ref" data-reference="table:SG">[table:SG]</a> lists the default grid for various classes of functionals.</p></li>
</ul>
<h3 id="sect:features-4.4">New Features in <span><span class="smallcaps">Q-Chem</span></span> 4.4</h3>
<ul>
<li><p>occ-RI-K algorithm for the evaluation of exact exchange in energy and force calculations (S. Manzer, F. Rob and M. Head-Gordon; Section <a href="#subsec:occrik" data-reference-type="ref" data-reference="subsec:occrik">[subsec:occrik]</a>)</p></li>
<li><p>Combinatorially-optimized exchange-correlation functionals (N. Mardirossian and M. Head-Gordon; Section <a href="#sect:XCfuncs" data-reference-type="ref" data-reference="sect:XCfuncs">[sect:XCfuncs]</a>):</p>
<ul>
<li><p><span class="math inline"><em>ω</em></span>B97M-V (range-separated hybrid, meta-GGA functional with VV10 non-local correlation)</p></li>
<li><p>B97M-V (meta-GGA functional with VV10 non-local correlation)</p></li>
<li><p><span class="math inline"><em>ω</em></span>B97X-V (range-separated hybrid functional with VV10 non-local correlation)</p></li>
</ul></li>
<li><p>Implementation of new exchange-correlation functionals from the literature (N. Mardirossian and M. Head-Gordon; Section <a href="#sect:XCfuncs" data-reference-type="ref" data-reference="sect:XCfuncs">[sect:XCfuncs]</a>). These include:</p>
<ul>
<li><p>MGGA_MS0, MGGA_MS1, MGGA_MS2, MGGA_MS2h, MGGA_MVS, MGGA_MVSh, PKZB, revTPSS, revTPSSh, SCAN, SCAN0, PBEsol, revPBE, revPBE0</p></li>
<li><p>N12, N12-SX, GAM, MN12-L, MN12-SX, MN15-L, dlDF</p></li>
<li><p>VV10, LC-VV10</p></li>
<li><p>B97-K, B97-D3(0), B97-3, <span class="math inline"><em>τ</em></span>-HCTH, <span class="math inline"><em>τ</em></span>-HCTHh</p></li>
<li><p>SRC1-R1, SRC1-R2, SRC2-R1, SRC2-R2</p></li>
<li><p>B1LYP, B1PW91, MPW1K, LRC-BOP, BHH, BB1K, PW6B95, PWB6K, B2PLYP</p></li>
</ul></li>
<li><p>Hessian-free minimum point verification (S. M. Sharada and M. Head-Gordon; Section <a href="#sect:approx_hess" data-reference-type="ref" data-reference="sect:approx_hess">[sect:approx_hess]</a>)</p></li>
<li><p>Exciton-based excited-state models:</p>
<ul>
<li><p><em>Ab initio</em> Frenkel-Davydov model for coupled excitations in multi-chromophore systems (A. F. Morrison and J. M. Herbert; Section <a href="#sect:AIFDEM" data-reference-type="ref" data-reference="sect:AIFDEM">13.15</a>).</p></li>
<li><p>TDDFT for molecular interactions [TDDFT(MI)], a set of local excitation approximations for efficient TDDFT calculations in multi-chromophore systems and for single chromophores in the presence of explicit solvent molecules (J. Liu and J. M. Herbert; Section <a href="#sec:TDDFT(MI)" data-reference-type="ref" data-reference="sec:TDDFT(MI)">13.16</a>).</p></li>
</ul></li>
<li><p>Improvements to many-body and XSAPT methods (K. U. Lao and J. M. Herbert)</p>
<ul>
<li><p>MPI-parallelized many-body expansion with analytic gradient (Section <a href="#sec:mbe" data-reference-type="ref" data-reference="sec:mbe">13.14</a>).</p></li>
<li><p>Efficient atomic orbital implementation of XSAPT for both closed- and open-shell systems (Section <a href="#subsec:AO-XSAPT(KS)+D" data-reference-type="ref" data-reference="subsec:AO-XSAPT(KS)+D">13.12.2</a>).</p></li>
</ul></li>
<li><p>Thermostats for <em>ab initio</em> molecular dynamics (R. P. Steele and J. M. Herbert).</p></li>
<li><p>Analytic energy gradient for the Ewald summation in QM/MM calculations (Z. C. Holden and J. M. Herbert)</p></li>
<li><p>Zeolite QM/MM methods (J. Gomes and M. Head-Gordon).</p></li>
<li><p>EOM-MP2 methods for excitation, ionization and electron attachment energies (A. Kunitsa and K. Bravaya; Section <a href="#sec:EOMMP2" data-reference-type="ref" data-reference="sec:EOMMP2">[sec:EOMMP2]</a>).</p></li>
<li><p>Evaluation of polarizabilities using CCSD and EOM-CCSD wave functions (Section <a href="#sec:EOMpol" data-reference-type="ref" data-reference="sec:EOMpol">[sec:EOMpol]</a>, K. Nanda and A. I. Krylov)</p></li>
<li><p>Distributed-memory parallel implementation of CC and EOM-CC methods and performance improvements in disk-based algorithms (E. Epifanovsky, I. Kaliman, and A. I. Krylov)</p></li>
<li><p>Improvements to the maximum overlap method (MOM) for SCF calculations (A. T. B. Gilbert; Section <a href="#sect:mom" data-reference-type="ref" data-reference="sect:mom">[sect:mom]</a>).</p></li>
<li><p>Non-equilibrium PCM method to describe solvent effects in ADC excited-state calculations (J.-M. Mewes and A. Dreuw; Section <a href="#subsec:ADC-PCM" data-reference-type="ref" data-reference="subsec:ADC-PCM">[subsec:ADC-PCM]</a>).</p></li>
<li><p>Spin-flip ADC method (D. Lefrancois and A. Dreuw; Section <a href="#sect:SF-ADC" data-reference-type="ref" data-reference="sect:SF-ADC">[sect:SF-ADC]</a>).</p></li>
</ul>
<h3 id="sect:features-4.3">New Features in <span><span class="smallcaps">Q-Chem</span></span> 4.3</h3>
<ul>
<li><p>Analytic derivative couplings (<em>i.e.</em>, non-adiabatic couplings) between electronic states computed at the CIS, spin-flip CIS, TDDFT, and spin-flip TDDFT levels (S. Fatehi, Q. Ou, J. E. Subotnik, X. Zhang, and J. M. Herbert; Section <a href="#sec:MECPs" data-reference-type="ref" data-reference="sec:MECPs">[sec:MECPs]</a>).</p></li>
<li><p>A third-generation (“+D3”) dispersion potential for XSAPT (K. U. Lao and J. M. Herbert; Section <a href="#sect:XSAPT" data-reference-type="ref" data-reference="sect:XSAPT">13.12</a>).</p></li>
<li><p>Non-equilibrium PCM for computing vertical excitation energies (at the TDDFT level) and ionization energies in solution (Z.-Q. You and J. M. Herbert; Section <a href="#subsubsec:noneq-PCM" data-reference-type="ref" data-reference="subsubsec:noneq-PCM">[subsubsec:noneq-PCM]</a>).</p></li>
<li><p>Spin-orbit couplings between electronic states for CC and EOM-CC wave functions (E. Epifanovsky, J. Gauss, and A. I. Krylov; Section <a href="#sec:EOMSOC" data-reference-type="ref" data-reference="sec:EOMSOC">[sec:EOMSOC]</a>).</p></li>
<li><p>PARI-K method for evaluation of exact exchange, which affords dramatic speed-ups for triple-<span class="math inline"><em>ζ</em></span> and larger basis sets in hybrid DFT calculations (S. Manzer and M. Head-Gordon).</p></li>
<li><p>Transition moments and cross sections for two-photon absorption using EOM-CC wave functions (K. Nanda and A. I. Krylov; Section <a href="#sec:2PA" data-reference-type="ref" data-reference="sec:2PA">[sec:2PA]</a>).</p></li>
<li><p>New excited-state analysis for ADC and CC/EOM-CC methods (M. Wormit; Section <a href="#subsec:esanacc" data-reference-type="ref" data-reference="subsec:esanacc">[subsec:esanacc]</a>).</p></li>
<li><p>New Dyson orbital code for EOM-IP-CCSD and EOM-EA-CCSD (A. Gunina and A. I. Krylov; Section <a href="#sec:Dyson" data-reference-type="ref" data-reference="sec:Dyson">[sec:Dyson]</a>).</p></li>
<li><p>Transition moments, state dipole moments, and Dyson orbitals for CAP-EOM-CCSD (T.-C. Jagau and A. I. Krylov; Sections <a href="#sub:cc_cs_cap" data-reference-type="ref" data-reference="sub:cc_cs_cap">[sub:cc_cs_cap]</a> and <a href="#sec:Dyson" data-reference-type="ref" data-reference="sec:Dyson">[sec:Dyson]</a>).</p></li>
<li><p>TAO-DFT: Thermally-assisted-occupation density functional theory (J.-D. Chai; Section <a href="#sect:tao-dft" data-reference-type="ref" data-reference="sect:tao-dft">[sect:tao-dft]</a>).</p></li>
<li><p>MP2[V], a dual basis method that approximates the MP2 energy (J. Deng and A. Gilbert).</p></li>
<li><p>Iterative Hirshfeld population analysis for charged systems, and CM5 semi-empirical charge scheme (K. U. Lao and J. M. Herbert; Section <a href="#subsec:pop_anal" data-reference-type="ref" data-reference="subsec:pop_anal">[subsec:pop_anal]</a>).</p></li>
<li><p>New DFT functionals: (Section <a href="#sect:XCfuncs" data-reference-type="ref" data-reference="sect:XCfuncs">[sect:XCfuncs]</a>):</p>
<ul>
<li><p>Long-range corrected functionals with empirical dispersion-: <span class="math inline"><em>ω</em></span>M05-D, <span class="math inline"><em>ω</em></span>B97X-D3 and <span class="math inline"><em>ω</em></span>M06-D3 (Y.-S. Lin, K. Hui, and J.-D. Chai.</p></li>
<li><p>PBE0_DH and PBE0_2 double-hybrid functionals (K. Hui and J.-D. Chai; Section <a href="#subsec:dhdft" data-reference-type="ref" data-reference="subsec:dhdft">[subsec:dhdft]</a>).</p></li>
<li><p>AK13 (K. Hui and J.-D. Chai).</p></li>
<li><p>LFAs asymptotic correction scheme (P.-T. Fang and J.-D. Chai).</p></li>
</ul></li>
<li><p>LDA/GGA fundamental gap using a frozen-orbital approximation (K. Hui and J.-D. Chai; Section <a href="#sect:foaDD" data-reference-type="ref" data-reference="sect:foaDD">[sect:foaDD]</a>).</p></li>
</ul>
<h3 id="sect:features-4.2">New Features in <span><span class="smallcaps">Q-Chem</span></span> 4.2</h3>
<ul>
<li><p>Input file changes:</p>
<ul>
<li><p>New keyword simplifies input in most cases by replacing the pair of keywords and (see Chapter <a href="#chap:scf" data-reference-type="ref" data-reference="chap:scf">4</a>).</p></li>
<li><p>Keywords for requesting excited-state calculations have been modified and simplified (see Chapter <a href="#chap:excited" data-reference-type="ref" data-reference="chap:excited">7</a> for details).</p></li>
<li><p>Keywords for solvation models have been modified and simplified (see Section <a href="#sect:solvent" data-reference-type="ref" data-reference="sect:solvent">[sect:solvent]</a> for details).</p></li>
</ul></li>
<li><p>New features for NMR calculations including spin-spin couplings (J. Kussmann, A. Luenser, and C. Ochsenfeld; Section <a href="#sec:magnetic:NMR" data-reference-type="ref" data-reference="sec:magnetic:NMR">[sec:magnetic:NMR]</a>).</p></li>
<li><p>New built-in basis sets (see Chapter <a href="#chap:basis" data-reference-type="ref" data-reference="chap:basis">8</a>).</p></li>
<li><p>New features and performance improvements in EOM-CC:</p>
<ul>
<li><p>EOM-CC methods extended to treat meta-stable electronic states (resonances) via complex scaling and complex absorbing potentials (D. Zuev, T.-C. Jagau, Y. Shao, and A. I. Krylov; Section <a href="#sub:cc_cs_cap" data-reference-type="ref" data-reference="sub:cc_cs_cap">[sub:cc_cs_cap]</a>).</p></li>
<li><p>New features added to EOM-CC iterative solvers, such as methods for interior eigenvalues and user-specified guesses (D. Zuev; Section <a href="#sub:cc_diagonalizer" data-reference-type="ref" data-reference="sub:cc_diagonalizer">[sub:cc_diagonalizer]</a>).</p></li>
<li><p>Multi-threaded parallel code for (EOM-)CC gradients and improved CCSD(T) performance.</p></li>
</ul></li>
<li><p>New features and performance improvements in ADC methods (M. Wormit, A. Dreuw):</p>
<ul>
<li><p>RI-ADC can tackle much larger systems at reduced cost (Section <a href="#sec:ADC_RI" data-reference-type="ref" data-reference="sec:ADC_RI">[sec:ADC_RI]</a>).</p></li>
<li><p>SOS-ADC methods (Section <a href="#sect:SOS-ADC" data-reference-type="ref" data-reference="sect:SOS-ADC">[sect:SOS-ADC]</a>).</p></li>
<li><p>State-to-state properties for ADC (Section <a href="#sect:ADC_prop" data-reference-type="ref" data-reference="sect:ADC_prop">[sect:ADC_prop]</a>).</p></li>
</ul></li>
<li><p>SM12 implicit solvation model (A. V. Marenich, D. G. Truhlar, and Y. Shao; Section <a href="#subsubsec:SM8" data-reference-type="ref" data-reference="subsubsec:SM8">[subsubsec:SM8]</a>).</p></li>
<li><p>Interface to NBO v. 6 (Section <a href="#sect:NBO" data-reference-type="ref" data-reference="sect:NBO">[sect:NBO]</a>).</p></li>
<li><p>Optimization of MECPs between electronic states at the SOS-CIS(D) and TDDFT levels (X. Zhang and J. M. Herbert; Section <a href="#sect:MECP" data-reference-type="ref" data-reference="sect:MECP">[sect:MECP]</a>).</p></li>
<li><p>ROKS method for <span class="math inline"><em>Δ</em></span>SCF calculations of excited states (T. Kowalczyk and T. Van Voorhis; Section <a href="#sect:ROKS" data-reference-type="ref" data-reference="sect:ROKS">[sect:ROKS]</a>).</p></li>
<li><p>Fragment-based initial guess for SCF methods (Section <a href="#sect:FRAGMOguess" data-reference-type="ref" data-reference="sect:FRAGMOguess">[sect:FRAGMOguess]</a>).</p></li>
<li><p>Pseudo-fractional occupation number method for improved SCF convergence in small-gap systems (D. S. Lambrecht; Section <a href="#sect:small-gap" data-reference-type="ref" data-reference="sect:small-gap">[sect:small-gap]</a>).</p></li>
<li><p>Density embedding scheme (B. J. Albrecht, E. Berquist, and D. S. Lambrecht; Section <a href="#sect:den_emb" data-reference-type="ref" data-reference="sect:den_emb">[sect:den_emb]</a>).</p></li>
<li><p>New features and enhancements in fragment-based many-body expansion methods (K. U. Lao and J. M. Herbert):</p>
<ul>
<li><p>XSAPT(KS)+D: A dispersion corrected version of symmetry-adapted perturbation theory for fast and accurate calculation of interaction energies in non-covalent clusters (Section <a href="#sect:XSAPT" data-reference-type="ref" data-reference="sect:XSAPT">13.12</a>).</p></li>
<li><p>Many-body expansion and fragment molecular orbital (FMO) methods for clusters (Section <a href="#sec:mbe" data-reference-type="ref" data-reference="sec:mbe">13.14</a>).</p></li>
</ul></li>
<li><p>Periodic boundary conditions with proper Ewald summation, for energies only (Z. C. Holden and J. M. Herbert; Section <a href="#sec:QMMM" data-reference-type="ref" data-reference="sec:QMMM">[sec:QMMM]</a>).</p></li>
</ul>
<h3 id="sect:features-4.1">New Features in <span><span class="smallcaps">Q-Chem</span></span> 4.1</h3>
<ul>
<li><p>Fundamental algorithms:</p>
<ul>
<li><p>Improved parallel performance at all levels including new OpenMP capabilities for Hartree-Fock, DFT, MP2, and coupled cluster theory (Z. Gan, E. Epifanovsky, M. Goldey, and Y. Shao; Section <a href="#sec:parallel" data-reference-type="ref" data-reference="sec:parallel">2.8</a>).</p></li>
<li><p>Significantly enhanced ECP capabilities, including gradients and frequencies in all basis sets for which the energy can be evaluated (Y. Shao and M. Head-Gordon; Chap. <a href="#chap:ecp" data-reference-type="ref" data-reference="chap:ecp">9</a>).</p></li>
</ul></li>
<li><p>SCF and DFT capabilities:</p>
<ul>
<li><p>TDDFT energy with the M06, M08, and M11 series of functionals.</p></li>
<li><p>XYGJ-OS analytical energy gradient.</p></li>
<li><p>TDDFT/C-PCM excitation energies, gradient, and Hessian (J. Liu and W. Liang; Section <a href="#subsec:tddft-cPCM" data-reference-type="ref" data-reference="subsec:tddft-cPCM">[subsec:tddft-cPCM]</a>).</p></li>
<li><p>Additional features in the maximum overlap method (MOM) approach for converging difficult SCF calculations (N. A. Besley; Section <a href="#subsec:MOM" data-reference-type="ref" data-reference="subsec:MOM">[subsec:MOM]</a>).</p></li>
</ul></li>
<li><p>Wave function correlation capabilities:</p>
<ul>
<li><p>RI and Cholesky decomposition implementation of all CC and EOM-CC methods enabling applications to larger systems with reduced disk and memory requirements and improved performance (E. Epifanovsky, X. Feng, D. Zuev, Y. Shao, and A. I. Krylov; Sections <a href="#sec:ri-CC" data-reference-type="ref" data-reference="sec:ri-CC">[sec:ri-CC]</a> and <a href="#sec:CD-CC" data-reference-type="ref" data-reference="sec:CD-CC">[sec:CD-CC]</a>).</p></li>
<li><p>Attenuated MP2 theory in the aug-cc-pVDZ and aug-cc-pVTZ basis sets, which truncates two-electron integrals to cancel basis set superposition error, yielding results for intermolecular interactions that are much more accurate than standard MP2 in the same basis set (M. Goldey and M. Head-Gordon; Section <a href="#sect:attmp2" data-reference-type="ref" data-reference="sect:attmp2">[sect:attmp2]</a>).</p></li>
<li><p>Extended RAS-<span class="math inline"><em>n</em></span>SF methodology for ground and excited states involving strong non-dynamical correlation (P. M. Zimmerman, D. Casanova, and M. Head-Gordon; Section <a href="#sect:exrasci" data-reference-type="ref" data-reference="sect:exrasci">[sect:exrasci]</a>).</p></li>
<li><p>Coupled cluster valence bond (CCVB) method for describing molecules with strong spin correlations (D. W. Small and M. Head-Gordon; Section <a href="#subsec:CCVB" data-reference-type="ref" data-reference="subsec:CCVB">[subsec:CCVB]</a>).</p></li>
</ul></li>
<li><p>Searching and scanning potential energy surfaces:</p>
<ul>
<li><p>Potential energy surface scans (Y. Shao; Section <a href="#sec:pes-scan" data-reference-type="ref" data-reference="sec:pes-scan">[sec:pes-scan]</a>).</p></li>
<li><p>Improvements in automatic transition structure searching via the “freezing string” method, including the ability to perform such calculations without a Hessian calculation (S. M. Sharada and M. Head-Gordon; Section <a href="#sect:approx_hess" data-reference-type="ref" data-reference="sect:approx_hess">[sect:approx_hess]</a>).</p></li>
<li><p>Enhancements to partial Hessian vibrational analysis (N. A. Besley; Section <a href="#partial-hess" data-reference-type="ref" data-reference="partial-hess">[partial-hess]</a>).</p></li>
</ul></li>
<li><p>Calculating and characterizing inter- and intramolecular interactions</p>
<ul>
<li><p>Extension of EFP to macromolecules: fEFP approach (A. Laurent, D. Ghosh, A. I. Krylov, and L. V. Slipchenko; Section <a href="#sec:fefp" data-reference-type="ref" data-reference="sec:fefp">[sec:fefp]</a>).</p></li>
<li><p>Symmetry-adapted perturbation theory level at the “SAPT0” level, for intermolecular interaction energy decomposition analysis into physically-meaningful components such as electrostatics, induction, dispersion, and exchange. An RI version is also available (L. D. Jacobson, J. M. Herbert; Section <a href="#sec:sapt" data-reference-type="ref" data-reference="sec:sapt">13.11</a>).</p></li>
<li><p>The “explicit polarization” (XPol) monomer-based SCF calculations to compute many-body polarization effects in linear-scaling time via charge embedding (Section <a href="#sect:xpol" data-reference-type="ref" data-reference="sect:xpol">[sect:xpol]</a>), which can be combined either with empirical potentials (<em>e.g.</em>, Lennard-Jones) for the non-polarization parts of the intermolecular interactions, or better yet, with SAPT for an <em>ab initio</em> approach called XSAPT that extends SAPT to systems containing more that two monomers (L. D. Jacobson and J. M. Herbert; Section <a href="#sect:XSAPT" data-reference-type="ref" data-reference="sect:XSAPT">13.12</a>).</p></li>
<li><p>Extension of the absolutely-localized molecular orbital (ALMO)-based energy decomposition analysis to unrestricted cases (P. R. Horn and M. Head-Gordon; Section <a href="#sect:EDA-CTA" data-reference-type="ref" data-reference="sect:EDA-CTA">13.5</a>).</p></li>
<li><p>Calculation of the populations of “effectively unpaired electrons” in low-spin state using DFT, a new method of evaluating localized atomic magnetic moments within Kohn-Sham without symmetry breaking, and Mayer-type bond order analysis with inclusion of static correlation effects (E. I. Proynov; Section <a href="#sect:odd_electron" data-reference-type="ref" data-reference="sect:odd_electron">[sect:odd_electron]</a>).</p></li>
</ul></li>
<li><p>Quantum transport calculations including electron transmission functions and electron tunneling currents under applied bias voltage (B. D. Dunietz and N. Sergueev; Section <a href="#sect:transport" data-reference-type="ref" data-reference="sect:transport">[sect:transport]</a>).</p></li>
<li><p>Searchable online version of the <span><span class="smallcaps">Q-Chem</span></span> PDF manual (J. M. Herbert and E. Epifanovsky).</p></li>
</ul>
<h3 id="sect:features-4.0.1">New Features in <span><span class="smallcaps">Q-Chem</span></span> 4.0.1</h3>
<ul>
<li><p>Remote submission capability in <span><span class="smallcaps">IQmol</span></span> (A. T. B. Gilbert).</p></li>
<li><p>Scaled nuclear charge and charge-cage stabilization capabilities (T. Kús and A. I. Krylov; Section <a href="#sec:CAGE" data-reference-type="ref" data-reference="sec:CAGE">[sec:CAGE]</a>).</p></li>
<li><p>Calculations of excited state properties including transition dipole moments between different excited states in CIS and TDDFT as well as couplings for electron and energy transfer (Z.-Q. You and C.-P. Hsu; Section <a href="#sec:ElCoupl" data-reference-type="ref" data-reference="sec:ElCoupl">[sec:ElCoupl]</a>).</p></li>
</ul>
<h3 id="sect:features-4.0">New Features in <span><span class="smallcaps">Q-Chem</span></span> 4.0</h3>
<ul>
<li><p>New exchange-correlation functionals (Section <a href="#sect:XCfuncs" data-reference-type="ref" data-reference="sect:XCfuncs">[sect:XCfuncs]</a>):</p>
<ul>
<li><p>Density-functional dispersion using Becke and Johnson’s XDM model in an efficient, analytic form (Z. Gan, E. I. Proynov, and J. Kong; Section <a href="#sect:XDM" data-reference-type="ref" data-reference="sect:XDM">[sect:XDM]</a>).</p></li>
<li><p>Van der Waals density functionals vdW-DF-04 and vdW-DF-10 of Langreth and coworkers (O. Vydrov; Section <a href="#sect:VV10" data-reference-type="ref" data-reference="sect:VV10">[sect:VV10]</a>).</p></li>
<li><p>VV09 and VV10, new analytic dispersion functionals (O. Vydrov, T. Van Voorhis; Section <a href="#sect:VV10" data-reference-type="ref" data-reference="sect:VV10">[sect:VV10]</a>)</p></li>
<li><p>DFT-D3 empirical dispersion methods for non-covalent interactions (S.-P. Mao and J.-D. Chai; Section <a href="#subsec:DFT-D" data-reference-type="ref" data-reference="subsec:DFT-D">[subsec:DFT-D]</a>).</p></li>
<li><p><span class="math inline"><em>ω</em></span>B97X-2, a double-hybrid functional based on the long-range corrected B97 functional, with improved accounting for medium- and long-range interactions (J.-D. Chai and M. Head-Gordon; Section <a href="#subsec:dhdft" data-reference-type="ref" data-reference="subsec:dhdft">[subsec:dhdft]</a>).</p></li>
<li><p>XYGJ-OS, a double-hybrid functional for predictions of non-bonded interactions and thermochemistry at nearly chemical accuracy (X. Xu, W. A. Goddard, and Y. Jung; Section <a href="#subsec:dhdft" data-reference-type="ref" data-reference="subsec:dhdft">[subsec:dhdft]</a>).</p></li>
<li><p>Short-range corrected functional for calculation of near-edge X-ray absorption spectra (N. A. Besley; Section <a href="#subsec:TDDFT-Xray" data-reference-type="ref" data-reference="subsec:TDDFT-Xray">[subsec:TDDFT-Xray]</a>).</p></li>
<li><p>LB94 asymptotically-corrected exchange-correlation functional for TDDFT (Y.-C. Su and J.-D. Chai; Section <a href="#sect:LB94" data-reference-type="ref" data-reference="sect:LB94">[sect:LB94]</a>).</p></li>
<li><p>Non-dynamical correlation in DFT with an efficient RI implementation of the Becke05 model in a fully analytic formulation (E. I. Proynov, Y. Shao, F. Liu, and J. Kong; Section <a href="#sect:XCfuncs" data-reference-type="ref" data-reference="sect:XCfuncs">[sect:XCfuncs]</a>).</p></li>
<li><p>TPSS and its hybrid version TPSSh, and rPW86 (F. Liu and O. Vydrov).</p></li>
<li><p>Double-hybrid functional B2PLYP-D (J.-D. Chai).</p></li>
<li><p>Hyper-GGA functional MCY2 from Mori-Sánchez, Cohen, and Yang (F. Liu).</p></li>
<li><p>SOGGA, SOGGA11 and SOGGA11-X family of GGA functionals (R. Peverati, Y. Zhao, and D. G. Truhlar).</p></li>
<li><p>M08-HX and M08-SO suites of high HF exchange meta-GGA functionals (Y. Zhao and D. G. Truhlar).</p></li>
<li><p>M11-L and M11 suites of meta-GGA functionals (R. Peverati, Y. Zhao, D. G. Truhlar).</p></li>
</ul></li>
<li><p>Improved DFT algorithms:</p>
<ul>
<li><p>Multi-resolution exchange-correlation (mrXC) for fast calculation of grid-based XC quadrature (S. T. Brown, C.-M. Chang, and J. Kong; Section <a href="#sect:mrXC" data-reference-type="ref" data-reference="sect:mrXC">[sect:mrXC]</a>).</p></li>
<li><p>Efficient computation of the XC part of the dual basis DFT (Z. Gan and J. Kong; Section <a href="#sect:projection" data-reference-type="ref" data-reference="sect:projection">[sect:projection]</a>).</p></li>
<li><p>Fast DFT calculation with “triple jumps” between different sizes of basis set and grid, and different levels of functional (J. Deng, A. T. B. Gilbert, and P. M. W. Gill; Section <a href="#sect:HFPC" data-reference-type="ref" data-reference="sect:HFPC">[sect:HFPC]</a>).</p></li>
<li><p>Faster DFT and HF calculation with an atomic resolution-of-identity algorithm (A. Sodt and M. Head-Gordon; Section <a href="#sect:ri-DFT" data-reference-type="ref" data-reference="sect:ri-DFT">[sect:ri-DFT]</a>).</p></li>
</ul></li>
<li><p>Post-Hartree–Fock methods:</p>
<ul>
<li><p>Significantly enhanced coupled-cluster code rewritten for better performance on multi-core architectures, including energy and gradient calculations with CCSD and energy calculations with EOM-EE/SF/IP/EA-CCSD, and CCSD(T) energy calculations (E. Epifanovsky, M. Wormit, T. Kús, A. Landau, D. Zuev, K. Khistyaev, I. Kaliman, A. I. Krylov, and A. Dreuw; Chaps. <a href="#chap:correlation" data-reference-type="ref" data-reference="chap:correlation">6</a> and <a href="#chap:excited" data-reference-type="ref" data-reference="chap:excited">7</a>).</p></li>
<li><p>Fast and accurate coupled-cluster calculations with frozen natural orbitals (A. Landau, D. Zuev, and A. I. Krylov; Section <a href="#sec:FNOCC" data-reference-type="ref" data-reference="sec:FNOCC">[sec:FNOCC]</a>).</p></li>
<li><p>Correlated excited states with the perturbation-theory based, size-consistent ADC scheme (M. Wormit and A. Dreuw; Section <a href="#sect:exadc" data-reference-type="ref" data-reference="sect:exadc">[sect:exadc]</a>).</p></li>
<li><p>Restricted active space, spin-flip method for multi-configurational ground states and multi-electron excited states (P. M. Zimmerman, F. Bell, D. Casanova, and M. Head-Gordon; Section <a href="#sect:sfxcis" data-reference-type="ref" data-reference="sect:sfxcis">[sect:sfxcis]</a>).</p></li>
</ul></li>
<li><p>Post-Hartree–Fock methods for describing strong correlation:</p>
<ul>
<li><p>“Perfect quadruples” and “perfect hextuples” methods for strong correlation problems (J. A. Parkhill and M. Head-Gordon; Section <a href="#sect:mgc" data-reference-type="ref" data-reference="sect:mgc">[sect:mgc]</a>).</p></li>
<li><p>Coupled-cluster valence bond (CCVB) methods for multiple-bond breaking (D. W. Small, K. V. Lawler, and M. Head-Gordon; Section <a href="#sect:pairing" data-reference-type="ref" data-reference="sect:pairing">[sect:pairing]</a>).</p></li>
</ul></li>
<li><p>TDDFT for excited states:</p>
<ul>
<li><p>Nuclear gradients for TDDFT (Z. Gan, C.-P. Hsu, A. Dreuw, M. Head-Gordon, and J. Kong; Section <a href="#sect:tddftintro" data-reference-type="ref" data-reference="sect:tddftintro">[sect:tddftintro]</a>).</p></li>
<li><p>Direct coupling of charged states for study of charge transfer reactions (Z.-Q. You and C.-P. Hsu; Section <a href="#diabatic-state" data-reference-type="ref" data-reference="diabatic-state">[diabatic-state]</a>).</p></li>
<li><p>Analytical excited-state Hessian for TDDFT within the Tamm-Dancoff approximation (J. Liu and W. Liang; Section <a href="#subsec:tddft-hessian" data-reference-type="ref" data-reference="subsec:tddft-hessian">[subsec:tddft-hessian]</a>).</p></li>
<li><p>Self-consistent excited-states with the maximum overlap method (A. T. B. Gilbert, N. A. Besley, and P. M. W. Gill; Section <a href="#sect:mom" data-reference-type="ref" data-reference="sect:mom">[sect:mom]</a>).</p></li>
<li><p>Calculation of reactions via configuration interactions of charge-constrained states computed with constrained DFT (Q. Wu, B. Kaduk and T. Van Voorhis; Section <a href="#sect:CDFT" data-reference-type="ref" data-reference="sect:CDFT">[sect:CDFT]</a>).</p></li>
<li><p>Overlap analysis of the charge transfer in a TDDFT excited state (N. A. Besley; Section <a href="#subsec:TDDFT-reduced-space" data-reference-type="ref" data-reference="subsec:TDDFT-reduced-space">[subsec:TDDFT-reduced-space]</a>).</p></li>
<li><p>Localizing diabatic states with Boys or Edmiston-Ruedenberg localization, for charge or energy transfer (J. E Subotnik, R. P. Steele, N. Shenvi, and A. Sodt; Section <a href="#subsec:diabatic" data-reference-type="ref" data-reference="subsec:diabatic">[subsec:diabatic]</a>).</p></li>
<li><p>Non-collinear formalism for spin-flip TDDFT (Y. Shao, Y. A. Bernard, and A. I. Krylov; Section <a href="#sect:tddft" data-reference-type="ref" data-reference="sect:tddft">[sect:tddft]</a>)</p></li>
</ul></li>
<li><p>Solvation and condensed-phase modeling</p>
<ul>
<li><p>Smooth free energy surface for solvated molecules via SWIG-PCMs, for QM and QM/MM calculations, including a linear-scaling QM/MM/PCM algorithm (A. W. Lange and J. M. Herbert; Sections <a href="#subsec:PCM" data-reference-type="ref" data-reference="subsec:PCM">[subsec:PCM]</a> and <a href="#subsec:linear-PCM" data-reference-type="ref" data-reference="subsec:linear-PCM">[subsec:linear-PCM]</a>).</p></li>
<li><p>Klamt’s COSMO solvation model with DFT energy and gradient (Y. Shao; Section <a href="#subsec:COSMO-Klamt" data-reference-type="ref" data-reference="subsec:COSMO-Klamt">[subsec:COSMO-Klamt]</a>).</p></li>
<li><p>Polarizable explicit solvent via EFP, for ground- and excited-state calculations at the DFT/TDDFT and CCSD/EOM-CCSD levels, as well as CIS and CIS(D). A library of effective fragments for common solvents is also available, along with energy and gradient for EFP–EFP calculations (V. Vanovschi, D. Ghosh, I. Kaliman, D. Kosenkov, C. F. Williams, J. M. Herbert, M. S. Gordon, M. W. Schmidt, Y. Shao, L. V. Slipchenko, and A. I. Krylov; Section <a href="#sect:efp" data-reference-type="ref" data-reference="sect:efp">[sect:efp]</a>).</p></li>
</ul></li>
<li><p>Optimizations, vibrations, and dynamics:</p>
<ul>
<li><p>“Freezing” and “growing” string methods for efficient automated reaction-path finding (A. Behn, P. M. Zimmerman, A. T. Bell, and M. Head-Gordon; Section <a href="#sect:fstring" data-reference-type="ref" data-reference="sect:fstring">[sect:fstring]</a>).</p></li>
<li><p>Improved robustness of the intrinsic reaction coordinate (IRC)-following code (M. Head-Gordon).</p></li>
<li><p>Quantum-mechanical treatment of nuclear motion at equilibrium via path integrals (R. P. Steele; Section <a href="#sec:PIMC" data-reference-type="ref" data-reference="sec:PIMC">[sec:PIMC]</a>).</p></li>
<li><p>Calculation of local vibrational modes of interest with partial Hessian vibrational analysis (N. A. Besley; Section <a href="#partial-hess" data-reference-type="ref" data-reference="partial-hess">[partial-hess]</a>).</p></li>
<li><p>Accelerated <em>ab initio</em> molecular dynamics MP2 and/or dual-basis methods, based on <span class="math inline"><em>Z</em></span>-vector extrapolation (R. P. Steele; Section <a href="#subsec:dual-dyn" data-reference-type="ref" data-reference="subsec:dual-dyn">[subsec:dual-dyn]</a>).</p></li>
<li><p>Quasi-classical <em>ab initio</em> molecular dynamics (D. S. Lambrecht and M. Head-Gordon; Section <a href="#subsec:QCT" data-reference-type="ref" data-reference="subsec:QCT">[subsec:QCT]</a>).</p></li>
</ul></li>
<li><p>Fragment-based methods:</p>
<ul>
<li><p>Symmetry-adapted perturbation theory (SAPT) for computing and analyzing dimer interaction energies (L. D. Jacobson, M. A. Rohrdanz, and J. M. Herbert; Section <a href="#sec:sapt" data-reference-type="ref" data-reference="sec:sapt">13.11</a>).</p></li>
<li><p>Many-body generalization of SAPT (“XSAPT”), with empirical dispersion corrections for high accuracy and low cost in large clusters (L. D. Jacobson, K. U. Lao, and J. M. Herbert; Section <a href="#sect:XSAPT" data-reference-type="ref" data-reference="sect:XSAPT">13.12</a>).</p></li>
<li><p>Methods based on a truncated many-body expansion, including the fragment molecular orbital (FMO) method (K. U. Lao and J. M. Herbert; Section <a href="#sec:mbe" data-reference-type="ref" data-reference="sec:mbe">13.14</a>).</p></li>
</ul></li>
<li><p>Properties and wave function analysis:</p>
<ul>
<li><p>Analysis of metal oxidation states via localized orbital bonding analysis (A. J. W. Thom, E. J. Sundstrom, and M. Head-Gordon; Section <a href="#sect:LOBA" data-reference-type="ref" data-reference="sect:LOBA">[sect:LOBA]</a>).</p></li>
<li><p>Hirshfeld population analysis (S. Yeganeh; Section <a href="#subsec:pop_anal" data-reference-type="ref" data-reference="subsec:pop_anal">[subsec:pop_anal]</a>).</p></li>
<li><p>Visualization of non-covalent bonding using Johnson and Yang’s NCI algorithm (Y. Shao; Section <a href="#NCI-plots" data-reference-type="ref" data-reference="NCI-plots">[NCI-plots]</a>).</p></li>
<li><p>Electrostatic potential on a grid for transition densities (Y. Shao; Section <a href="#sec:ESP" data-reference-type="ref" data-reference="sec:ESP">[sec:ESP]</a>).</p></li>
</ul></li>
<li><p>Support for modern computing platforms</p>
<ul>
<li><p>Efficient multi-threaded parallel performance for CC, EOM, and ADC methods.</p></li>
<li><p>Better performance for multi-core systems with shared-memory parallel DFT and Hartree-Fock (Z. Gan, Y. Shao, and J. Kong) and RI-MP2 (M. Goldey and M. Head-Gordon; Section <a href="#CCparallel" data-reference-type="ref" data-reference="CCparallel">[CCparallel]</a>).</p></li>
<li><p>Accelerated RI-MP2 calculation on GPUs (R. Olivares-Amaya, M. Watson, R. Edgar, L. Vogt, Y. Shao, and A. Aspuru-Guzik; Section <a href="#sect:gpu-rimp2" data-reference-type="ref" data-reference="sect:gpu-rimp2">[sect:gpu-rimp2]</a>).</p></li>
</ul></li>
<li><p>Graphical user interfaces:</p>
<ul>
<li><p>Input file generation, <span><span class="smallcaps">Q-Chem</span></span> job submission, and visualization is supported by <span><span class="smallcaps">IQmol</span></span>, a fully integrated GUI developed by Andrew Gilbert. <span><span class="smallcaps">IQmol</span></span> is a free software and does not require purchasing a <span><span class="smallcaps">Q-Chem</span></span> license. See <a href="www.iqmol.org" class="uri">www.iqmol.org</a> for details and installation instructions.</p></li>
<li><p>Other graphical interfaces are also available, including <span><span class="smallcaps">MolDen</span></span>, <span><span class="smallcaps">MacMolPlt</span></span>, and <span><span class="smallcaps">Avogadro</span></span> (Chapter <a href="#chap:properties" data-reference-type="ref" data-reference="chap:properties">11</a> and elsewhere).</p></li>
</ul></li>
</ul>
<h3 id="summary-of-features-in-q-chem-versions-3.x">Summary of Features in <span><span class="smallcaps">Q-Chem</span></span> versions 3.<span class="math inline"> </span><em>x</em></h3>
<ul>
<li><p>DFT functionals and algorithms:</p>
<ul>
<li><p>Long-ranged corrected (LRC) functionals, also known as range-separated hybrid functionals (M. A. Rohrdanz and J. M. Herbert)</p></li>
<li><p>Constrained DFT (Q. Wu and T. Van Voorhis)</p></li>
<li><p>Grimme’s “DFT-D” empirical dispersion corrections (C.-D. Sherrill)</p></li>
<li><p>“Incremental” DFT method that significantly accelerates exchange-correlation quadrature in later SCF cycles (S. T. Brown)</p></li>
<li><p>Efficient SG-0 quadrature grid with approximately half the number of grid points relative to SG-1 (S.-H. Chien)</p></li>
</ul></li>
<li><p>Solvation models:</p>
<ul>
<li><p>SM8 model (A. V. Marenich, R. M. Olson, C. P. Kelly, C. J. Cramer, and D. G. Truhlar)</p></li>
<li><p>Onsager reaction-field model (C.-L. Cheng, T. Van Voorhis, K. Thanthiriwatte, and S. R. Gwaltney)</p></li>
<li><p>Chipman’s SS(V)PE model (S. T. Brown)</p></li>
</ul></li>
<li><p>Second-order perturbation theory algorithms for ground and excited states:</p>
<ul>
<li><p>Dual-basis RIMP2 energy and analytical gradient (R. P. Steele, R. A. DiStasio Jr., and M. Head-Gordon)</p></li>
<li><p>O2 energy and gradient (R. C. Lochan and M. Head-Gordon)</p></li>
<li><p>SOS-CIS(D), SOS-CIS(D<span class="math inline"><em></em><sub>0</sub></span>), and RI-CIS(D) for excited states (D. Casanova, Y. M. Rhee, and M. Head-Gordon)</p></li>
<li><p>Efficient resolution-of-identity (RI) implementations of MP2 and SOS-MP2 (including both energies and gradients), and of RI-TRIM and RI-CIS(D) energies (Y. Jung, R. A. DiStasio, Jr., R. C. Lochan, and Y. M. Rhee)</p></li>
</ul></li>
<li><p>Coupled-cluster methods (P. A. Pieniazek, E. Epifanovsky, A. I. Krylov):</p>
<ul>
<li><p>IP-CISD and EOM-IP-CCSD energy and gradient</p></li>
<li><p>Multi-threaded (OpenMP) parallel coupled-cluster calculations</p></li>
<li><p>Potential energy surface crossing minimization with CCSD and EOM-CCSD methods (E. Epifanovsky)</p></li>
<li><p>Dyson orbitals for ionization from the ground and excited states within CCSD and EOM-CCSD methods (M. Oana)</p></li>
</ul></li>
<li><p>QM/MM methods (H. L. Woodcock, A. Ghysels, Y. Shao, J. Kong, and H. B. Brooks)</p>
<ul>
<li><p><span><span class="smallcaps">Q-Chem</span></span>/<span><span class="smallcaps">Charmm</span></span> interface (H. L. Woodcock)</p></li>
<li><p>Full QM/MM Hessian evaluation and approximate mobile-block-Hessian evaluation</p></li>
<li><p>Two-layer ONIOM model (Y. Shao).</p></li>
<li><p>Integration with the <span><span class="smallcaps">Molaris</span></span> simulation package (E. Rosta).</p></li>
</ul></li>
<li><p>Improved two-electron integrals package</p>
<ul>
<li><p>Rewrite of the Head-Gordon–Pople algorithm for modern computer architectures (Y. Shao)</p></li>
<li><p>Fourier Transform Coulomb method for linear-scaling construction of the Coulomb matrix, even for basis sets with high angular moment and diffuse functions (L. Fusti-Molnar)</p></li>
</ul></li>
<li><p>Dual basis self-consistent field calculations, offering an order-of-magnitude reduction in the cost of large-basis DFT calculations (J. Kong and R. P. Steele)</p></li>
<li><p>Enhancements to the correlation package including:</p>
<ul>
<li><p>Most extensive range of EOM-CCSD methods available including EOM-SF-CCSD, EOM-EE-CCSD, EOM-DIP-CCSD, EOM-IP/EA-CCSD (A. I. Krylov).</p></li>
<li><p>Available for RHF, UHF, and ROHF references.</p></li>
<li><p>Analytic gradients and properties calculations (permanent and transition dipoles <em>etc.</em>.).</p></li>
<li><p>Full use of Abelian point-group symmetry.</p></li>
</ul></li>
<li><p>Coupled-cluster perfect-paring methods applicable to systems with <span class="math inline"> &gt; 100</span> active electrons (M. Head-Gordon)</p></li>
<li><p>Transition structure search using the “growing string” algorithm (A. Heyden and B. Peters):</p></li>
<li><p><em>Ab initio</em> molecular dynamics (J. M. Herbert)</p></li>
<li><p>Linear scaling properties for large systems (J. Kussmann, C. Ochsenfeld):</p>
<ul>
<li><p>NMR chemical shifts</p></li>
<li><p>Static and dynamic polarizabilities</p></li>
<li><p>Static hyper-polarizabilities, optical rectification, and electro-optical Pockels effect</p></li>
</ul></li>
<li><p>Anharmonic frequencies (C. Y. Lin)</p></li>
<li><p>Wave function analysis tools:</p>
<ul>
<li><p>Analysis of intermolecular interactions with ALMO-EDA (R. Z. Khaliullin and M. Head-Gordon)</p></li>
<li><p>Electron transfer analysis (Z.-Q. You and C.-P. Hsu)</p></li>
<li><p>Spin densities at the nuclei (V. A. Rassolov)</p></li>
<li><p>Position, momentum, and Wigner intracules (N. A. Besley and D. P. O’Neill)</p></li>
</ul></li>
<li><p>Graphical user interface options:</p>
<ul>
<li><p><span><span class="smallcaps">IQmol</span></span>, a fully integrated GUI. <span><span class="smallcaps">IQmol</span></span> includes input file generator and contextual help, molecular builder, job submission tool, and visualization kit (molecular orbital and density viewer, frequencies, etc). For the latest version and download/installation instructions, please see the <span><span class="smallcaps">IQmol</span></span> homepage (<a href="www.iqmol.org" class="uri">www.iqmol.org</a>).</p></li>
<li><p>Seamless integration with the <span><span class="smallcaps">Spartan</span></span> package (see <a href="www.wavefun.com" class="uri">www.wavefun.com</a>).</p></li>
<li><p>Support for several other public-domain visualization programs:</p>
<ul>
<li><p><span><span class="smallcaps">WebMO</span></span><br />
<a href="www.webmo.net" class="uri">www.webmo.net</a></p></li>
<li><p><span><span class="smallcaps">Avogadro</span></span><br />
<a href="https://avogadro.cc" class="uri">https://avogadro.cc</a></p></li>
<li><p><span><span class="smallcaps">MolDen</span></span><br />
<a href="http://www.cmbi.ru.nl/molden" class="uri">http://www.cmbi.ru.nl/molden</a></p></li>
<li><p><span><span class="smallcaps">MacMolPlt</span></span> (via a <span><span class="smallcaps">MolDen</span></span>-formatted input file)<br />
<a href="https://brettbode.github.io/wxmacmolplt" class="uri">https://brettbode.github.io/wxmacmolplt</a></p></li>
<li><p><span><span class="smallcaps">JMol</span></span><br />
<a href="www.sourceforge.net/project/showfiles.php?group_id= 23629&amp;release_id=66897" class="uri">www.sourceforge.net/project/showfiles.php?group_id= 23629&amp;release_id=66897</a></p></li>
</ul></li>
</ul></li>
</ul>
<h3 id="new-features-in-q-chem-3.2">New Features in <span><span class="smallcaps">Q-Chem</span></span> 3.2</h3>
<ul>
<li><p>Several new DFT options:</p>
<ul>
<li><p>Long-ranged corrected (LRC) functionals, also known as range-separated hybrid functionals (M. A. Rohrdanz and J. M. Herbert)</p></li>
<li><p>Baer-Neuhauser-Livshits (BNL) functional (R. Baer, D. Neuhauser, and E. Livshits)</p></li>
<li><p>Variations of <span class="math inline"><em>ω</em></span>B97 Functional (J.-D. Chai and M. Head-Gordon)</p></li>
<li><p>Constrained DFT (Q. Wu and T. Van Voorhis)</p></li>
<li><p>Grimme’s empirical dispersion (“DFT-D” corrections (C.-D. Sherrill)</p></li>
</ul></li>
<li><p>For DFT calculations, the default XC quadrature grid has been changed from SG-0 to SG-1.</p></li>
<li><p>Solvation models:</p>
<ul>
<li><p>SM8 model (energy and analytical gradient) for water and organic solvents (A. V. Marenich, R. M. Olson, C. P. Kelly, C. J. Cramer, and D. G. Truhlar)</p></li>
<li><p>Updates to the Onsager reaction-field model (C.-L. Cheng, T. Van Voorhis, K. Thanthiriwatte, and S. R. Gwaltney)</p></li>
</ul></li>
<li><p>Analysis of intermolecular interactions (R. Z. Khaliullin and M. Head-Gordon):</p>
<ul>
<li><p>SCF with absolutely localized molecular orbitals for molecule interaction (SCF-MI)</p></li>
<li><p>Roothaan-step (RS) correction following SCF-MI</p></li>
<li><p>Energy decomposition analysis (EDA)</p></li>
<li><p>Complimentary occupied-virtual pair (COVP) analysis for charge transfer</p></li>
<li><p>Automated counterpoise correction for basis-set superposition error</p></li>
</ul></li>
<li><p>Electron transfer analysis (Z.-Q. You, C.-P. Hsu)</p></li>
<li><p>Relaxed constraint algorithm (RCA) for converging SCF (C.-L. Cheng, T. Van Voorhis)</p></li>
<li><p>G3Large basis set for transition metals (V. A. Rassolov)</p></li>
<li><p>New MP2 options:</p>
<ul>
<li><p>Dual-basis RIMP2 energy and analytical gradient (R. P. Steele, R. A. DiStasio Jr., and M. Head-Gordon)</p></li>
<li><p>O2 energy and gradient (R. C. Lochan and M. Head-Gordon)</p></li>
</ul></li>
<li><p>SOS-CIS(D) and SOS-CIS(D<span class="math inline"><em></em><sub>0</sub></span>) and energy and gradient for excited states (D. Casanova, Y. M. Rhee, and M. Head-Gordon)</p></li>
<li><p>Coupled-cluster methods (P. A. Pieniazek, E. Epifanovsky, A. I. Krylov):</p>
<ul>
<li><p>IP-CISD and EOM-IP-CCSD energy and gradient</p></li>
<li><p>Multi-threaded (OpenMP) parallel coupled-cluster calculations</p></li>
</ul></li>
<li><p>QM/MM methods (H. L. Woodcock, A. Ghysels, Y. Shao, J. Kong, and H. B. Brooks)</p>
<ul>
<li><p>Full QM/MM Hessian evaluation</p></li>
<li><p>QM/MM mobile-block-Hessian evaluation</p></li>
<li><p>Gaussian-delocalized charges for MM link atoms</p></li>
</ul></li>
<li><p>Partial Hessian method for vibrational analysis (N. A. Besley)</p></li>
<li><p>Wave function analysis tools:</p>
<ul>
<li><p>Improved algorithms for computing localized orbitals (J. E. Subotnik, Y. M. Rhee, A. J. W. Thom, W. Kurlancheek, and M. Head-Gordon)</p></li>
<li><p>Distributed multipole analysis (V. Vanovschi, A. I. Krylov, C. F. Williams, and J. M. Herbert)</p></li>
<li><p>Analytical Wigner intracule (D. L. Crittenden and P. M. W. Gill)</p></li>
</ul></li>
</ul>
<h3 id="new-features-in-q-chem-3.1">New Features in <span><span class="smallcaps">Q-Chem</span></span> 3.1</h3>
<ul>
<li><p>New exchange-correlation functionals:</p>
<ul>
<li><p>The non-empirical GGA functional PBE (from the open DF Repository distributed by the QCG CCLRC Daresbury Lab., implemented in <span><span class="smallcaps">Q-Chem</span></span> 3.1 by E. I. Proynov).</p></li>
<li><p>M05 and M06 suites of meta-GGA functionals for more accurate predictions of various types of reactions and systems (Y. Zhao, N. E. Schultz, and D. G. Truhlar).</p></li>
</ul></li>
<li><p>A faster correlated excited state method: RI-CIS(D) (Y. M. Rhee)</p></li>
<li><p>Potential energy surface crossing minimization with CCSD and EOM-CCSD methods (E. Epifanovsky).</p></li>
<li><p>Dyson orbitals for ionization from the ground and excited states within CCSD and EOM-CCSD methods (M. Oana).</p></li>
</ul>
<h3 id="new-features-in-q-chem-3.0">New Features in <span><span class="smallcaps">Q-Chem</span></span> 3.0</h3>
<ul>
<li><p>Improved two-electron integrals package (Y. Shao):</p>
<ul>
<li><p>Code for the Head-Gordon–Pople algorithm rewritten to avoid cache misses and to take advantage of modern computer architectures.</p></li>
<li><p>Overall increased in performance, especially for computing derivatives.</p></li>
</ul></li>
<li><p>Fourier Transform Coulomb method (L. Fusti-Molnar):</p>
<ul>
<li><p>Highly efficient implementation for the calculation of Coulomb matrices and forces for DFT calculations.</p></li>
<li><p>Linear scaling regime is attained earlier than previous linear algorithms.</p></li>
<li><p>Present implementation works well for basis sets with high angular momentum and diffuse functions.</p></li>
</ul></li>
<li><p>Improved DFT quadrature evaluation:</p>
<ul>
<li><p>Incremental DFT method avoids calculating negligible contributions from grid points in later SCF cycles (S. T. Brown).</p></li>
<li><p>Highly efficient SG-0 quadrature grid with approximately half the accuracy and number of grid points as the SG-1 grid (S.-H. Chien).</p></li>
</ul></li>
<li><p>Dual basis self-consistent field calculations (J. Kong and R. P. Steele):</p>
<ul>
<li><p>Two-stage SCF calculations can reduce computational cost by an order of magnitude.</p></li>
<li><p>Customized basis subsets designed for optimal projection into larger bases.</p></li>
</ul></li>
<li><p>Auxiliary basis expansions for MP2 calculations:</p>
<ul>
<li><p>RI-MP2 and SOS-MP2 energies (Y. Jung) and gradients (R. A. DiStasio Jr.).</p></li>
<li><p>RI-TRIM MP2 energies (R. A. DiStasio Jr.).</p></li>
<li><p>Scaled opposite spin energies and gradients (R. C. Lochan).</p></li>
</ul></li>
<li><p>Enhancements to the correlation package including:</p>
<ul>
<li><p>Most extensive range of EOM-CCSD methods available including EOM-SF-CCSD, EOM-EE-CCSD, EOM-DIP-CCSD, EOM-IP/EA-CCSD (A. I. Krylov).</p></li>
<li><p>Available for RHF, UHF, and ROHF references.</p></li>
<li><p>Analytic gradients and properties calculations (permanent and transition dipoles <em>etc.</em>.).</p></li>
<li><p>Full use of Abelian point-group symmetry.</p></li>
<li><p>Singlet strongly orthogonal geminal (SSG) methods (V. A. Rassolov).</p></li>
</ul></li>
<li><p>Coupled-cluster perfect-paring methods (M. Head-Gordon):</p>
<ul>
<li><p>Perfect pairing (PP), imperfect pairing (IP) and restricted pairing (RP) models.</p></li>
<li><p>PP(2) Corrects for some of the worst failures of MP2 theory.</p></li>
<li><p>Useful in the study of singlet molecules with diradical character.</p></li>
<li><p>Applicable to systems with more than 100 active electrons.</p></li>
</ul></li>
<li><p>Hybrid quantum mechanics/molecular mechanics (QM/MM) methods:</p>
<ul>
<li><p>Fixed point-charge model based on the <span><span class="smallcaps">Amber</span></span> force field.</p></li>
<li><p>Two-layer ONIOM model (Y. Shao).</p></li>
<li><p>Integration with the <span><span class="smallcaps">Molaris</span></span> simulation package (E. Rosta).</p></li>
<li><p><span><span class="smallcaps">Q-Chem</span></span>/<span><span class="smallcaps">Charmm</span></span> interface (H. L. Woodcock)</p></li>
</ul></li>
<li><p>New continuum solvation models (S. T. Brown):</p>
<ul>
<li><p>Chipman’s Surface and Simulation of Volume Polarization for Electrostatics [SS(V)PE] model.</p></li>
<li><p>Available for HF and DFT calculations.</p></li>
</ul></li>
<li><p>New transition structure search algorithms (A. Heyden and B. Peters):</p>
<ul>
<li><p>Growing string method for finding transition states.</p></li>
<li><p>Dimer method that does not use the Hessian, and is therefore useful for large systems.</p></li>
</ul></li>
<li><p><em>Ab initio</em> molecular dynamics (J. M. Herbert):</p>
<ul>
<li><p>Available for SCF wave functions (HF, DFT).</p></li>
<li><p>Direct Born-Oppenheimer molecular dynamics (BOMD).</p></li>
<li><p>Extended Lagrangian <em>ab initio</em> molecular dynamics (ELMD).</p></li>
</ul></li>
<li><p>Linear scaling properties for large systems (J. Kussmann, C. Ochsenfeld):</p>
<ul>
<li><p>NMR chemical shifts.</p></li>
<li><p>Static and dynamic polarizabilities.</p></li>
<li><p>Static hyper-polarizabilities, optical rectification and electro-optical Pockels effect.</p></li>
</ul></li>
<li><p>Anharmonic frequencies (C. Y. Lin):</p>
<ul>
<li><p>Efficient implementation of high-order derivatives</p></li>
<li><p>Corrections via perturbation theory (VPT) or configuration interaction (VCI).</p></li>
<li><p>New transition optimized shifted Hermite (TOSH) method.</p></li>
</ul></li>
<li><p>Wave function analysis tools:</p>
<ul>
<li><p>Spin densities at the nuclei (V. A. Rassolov).</p></li>
<li><p>Efficient calculation of localized orbitals.</p></li>
<li><p>Optimal atomic point-charge models for densities (A. C. Simmonett).</p></li>
<li><p>Calculation of position, momentum and Wigner intracules (N. A. Besley, D. P. O’Neill).</p></li>
</ul></li>
<li><p>Graphical user interface options:</p>
<ul>
<li><p><span><span class="smallcaps">IQmol</span></span>, a fully integrated GUI. <span><span class="smallcaps">IQmol</span></span> includes input file generator and contextual help, molecular builder, job submission tool, and visualization kit (molecular orbital and density viewer, frequencies, etc). For the latest version and download/installation instructions, please see the <span><span class="smallcaps">IQmol</span></span> homepage (<a href="www.iqmol.org" class="uri">www.iqmol.org</a>).</p></li>
<li><p>Seamless integration with the <span><span class="smallcaps">Spartan</span></span> package (see <a href="www.wavefun.com" class="uri">www.wavefun.com</a>).</p></li>
<li><p>Support for several other public-domain visualization programs:</p>
<ul>
<li><p><span><span class="smallcaps">WebMO</span></span><br />
<a href="www.webmo.net" class="uri">www.webmo.net</a></p></li>
<li><p><span><span class="smallcaps">Avogadro</span></span><br />
<a href="https://avogadro.cc" class="uri">https://avogadro.cc</a></p></li>
<li><p><span><span class="smallcaps">MolDen</span></span><br />
<a href="http://www.cmbi.ru.nl/molden" class="uri">http://www.cmbi.ru.nl/molden</a></p></li>
<li><p><span><span class="smallcaps">MacMolPlt</span></span> (via a <span><span class="smallcaps">MolDen</span></span>-formatted input file)<br />
<a href="https://brettbode.github.io/wxmacmolplt" class="uri">https://brettbode.github.io/wxmacmolplt</a></p></li>
<li><p><span><span class="smallcaps">JMol</span></span><br />
<a href="www.sourceforge.net/project/showfiles.php?group_id= 23629&amp;release_id=66897" class="uri">www.sourceforge.net/project/showfiles.php?group_id= 23629&amp;release_id=66897</a></p></li>
</ul></li>
</ul></li>
</ul>
<h3 id="summary-of-features-prior-to-q-chem-3.0">Summary of Features Prior to <span><span class="smallcaps">Q-Chem</span></span> 3.0</h3>
<ul>
<li><p>Efficient algorithms for large-molecule density functional calculations:</p>
<ul>
<li><p>CFMM for linear scaling Coulomb interactions (energies and gradients) (C. A. White).</p></li>
<li><p>Second-generation J-engine and J-force engine (Y. Shao).</p></li>
<li><p>LinK for exchange energies and forces (C. Ochsenfeld and C. A. White).</p></li>
<li><p>Linear scaling DFT exchange-correlation quadrature.</p></li>
</ul></li>
<li><p>Local, gradient-corrected, and hybrid DFT functionals:</p>
<ul>
<li><p>Slater, Becke, GGA91 and Gill ‘96 exchange functionals.</p></li>
<li><p>VWN, PZ81, Wigner, Perdew86, LYP and GGA91 correlation functionals.</p></li>
<li><p>EDF1 exchange-correlation functional (R. Adamson).</p></li>
<li><p>B3LYP, B3P and user-definable hybrid functionals.</p></li>
<li><p>Analytical gradients and analytical frequencies.</p></li>
<li><p>SG-0 standard quadrature grid (S.-H. Chien).</p></li>
<li><p>Lebedev grids up to 5294 points (S. T. Brown).</p></li>
</ul></li>
<li><p>High level wave function-based electron correlation methods</p>
<ul>
<li><p>Efficient semi-direct MP2 energies and gradients.</p></li>
<li><p>MP3, MP4, QCISD, CCSD energies.</p></li>
<li><p>OD and QCCD energies and analytical gradients.</p></li>
<li><p>Triples corrections (QCISD(T), CCSD(T) and OD(T) energies).</p></li>
<li><p>CCSD(2) and OD(2) energies.</p></li>
<li><p>Active space coupled cluster methods: VOD, VQCCD, VOD(2).</p></li>
<li><p>Local second order M<span>ø</span>ller-Plesset (MP2) methods (DIM and TRIM).</p></li>
<li><p>Improved definitions of core electrons for post-HF correlation (V. A. Rassolov).</p></li>
</ul></li>
<li><p>Extensive excited state capabilities:</p>
<ul>
<li><p>CIS energies, analytical gradients and analytical frequencies.</p></li>
<li><p>CIS(D) energies.</p></li>
<li><p>Time-dependent density functional theory energies (TDDFT).</p></li>
<li><p>Coupled cluster excited state energies, OD and VOD (A. I. Krylov).</p></li>
<li><p>Coupled-cluster excited-state geometry optimizations.</p></li>
<li><p>Coupled-cluster property calculations (dipoles, transition dipoles).</p></li>
<li><p>Spin-flip calculations for CCSD and TDDFT excited states (A. I. Krylov and Y. Shao).</p></li>
</ul></li>
<li><p>High performance geometry and transition structure optimization (J. Baker):</p>
<ul>
<li><p>Optimizes in Cartesian, <em>Z</em>-matrix or delocalized internal coordinates.</p></li>
<li><p>Impose bond angle, dihedral angle (torsion) or out-of-plane bend constraints.</p></li>
<li><p>Freezes atoms in Cartesian coordinates.</p></li>
<li><p>Constraints do not need to be satisfied in the starting structure.</p></li>
<li><p>Geometry optimization in the presence of fixed point charges.</p></li>
<li><p>Intrinsic reaction coordinate (IRC) following code.</p></li>
</ul></li>
<li><p>Evaluation and visualization of molecular properties</p>
<ul>
<li><p>Onsager, SS(V)PE and Langevin dipoles solvation models.</p></li>
<li><p>Evaluate densities, electrostatic potentials, orbitals over cubes for plotting.</p></li>
<li><p>Natural Bond Orbital (NBO) analysis.</p></li>
<li><p>Attachment/detachment densities for excited states via CIS, TDDFT.</p></li>
<li><p>Vibrational analysis after evaluation of the nuclear coordinate Hessian.</p></li>
<li><p>Isotopic substitution for frequency calculations (R. Doerksen).</p></li>
<li><p>NMR chemical shifts (J. Kussmann).</p></li>
<li><p>Atoms in Molecules (AIMPAC) support (J. Ritchie).</p></li>
<li><p>Stability analysis of SCF wave functions (Y. Shao).</p></li>
<li><p>Calculation of position and momentum molecular intracules A. Lee, N. A. Besley, and D. P. O’Neill).</p></li>
</ul></li>
<li><p>Flexible basis set and effective core potential (ECP) functionality: (Ross Adamson and Peter Gill)</p>
<ul>
<li><p>Wide range of built-in basis sets and ECPs.</p></li>
<li><p>Basis set superposition error correction.</p></li>
<li><p>Support for mixed and user-defined basis sets.</p></li>
<li><p>Effective core potentials for energies and gradients.</p></li>
<li><p>Highly efficient PRISM-based algorithms to evaluate ECP matrix elements.</p></li>
<li><p>Faster and more accurate ECP second derivatives for frequencies.</p></li>
</ul></li>
</ul>
<h2 id="sec:cite">Citing <span><span class="smallcaps">Q-Chem</span></span></h2>
<p>Users who publish papers based on <span><span class="smallcaps">Q-Chem</span></span> calculations are asked to cite the official peer-reviewed literature citation for the software. For versions corresponding to 4.0 and later, this is:</p>
<ul>
<li><p>Y. Shao, Z. Gan, E. Epifanovsky, A. T. B. Gilbert, M. Wormit, J. Kussmann, A. W. Lange, A. Behn, J. Deng, X. Feng, D. Ghosh, M. Goldey P. R. Horn, L. D. Jacobson, I. Kaliman, R. Z. Khaliullin, T. Kús, A. Landau, J. Liu, E. I. Proynov, Y. M. Rhee, R. M. Richard, M. A. Rohrdanz, R. P. Steele, E. J. Sundstrom, H. L. Woodcock III, P. M. Zimmerman, D. Zuev, B. Albrecht, E. Alguire, B. Austin, G. J. O. Beran, Y. A. Bernard, E. Berquist, K. Brandhorst, K. B. Bravaya, S. T. Brown, D. Casanova, C.-M. Chang, Y. Chen, S. H. Chien, K. D. Closser, D. L. Crittenden, M. Diedenhofen, R. A. DiStasio Jr., H. Dop, A. D. Dutoi, R. G. Edgar, S. Fatehi, L. Fusti-Molnar, A. Ghysels, A. Golubeva-Zadorozhnaya, J. Gomes, M. W. D. Hanson-Heine, P. H. P. Harbach, A. W. Hauser, E. G. Hohenstein, Z. C. Holden, T.-C. Jagau, H. Ji, B. Kaduk, K. Khistyaev, J. Kim, J. Kim, R. A. King, P. Klunzinger, D. Kosenkov, T. Kowalczyk, C. M. Krauter, K. U. Lao, A. Laurent, K. V. Lawler, S. V. Levchenko, C. Y. Lin, F. Liu, E. Livshits, R. C. Lochan, A. Luenser, P. Manohar, S. F. Manzer, S.-P. Mao, N. Mardirossian, A. V. Marenich, S. A. Maurer, N. J. Mayhall, C. M. Oana, R. Olivares-Amaya, D. P. O’Neill, J. A. Parkhill, T. M. Perrine, R. Peverati, P. A. Pieniazek, A. Prociuk, D. R. Rehn, E. Rosta, N. J. Russ, N. Sergueev, S. M. Sharada, S. Sharmaa, D. W. Small, A. Sodt, T. Stein, D. Stück, Y.-C. Su, A. J. W. Thom, T. Tsuchimochi, L. Vogt, O. Vydrov, T. Wang, M. A. Watson, J. Wenzel, A. White, C. F. Williams, V. Vanovschi, S. Yeganeh, S. R. Yost, Z.-Q. You, I. Y. Zhang, X. Zhang, Y. Zhou, B. R. Brooks, G. K. L. Chan, D. M. Chipman, C. J. Cramer, W. A. Goddard III, M. S. Gordon, W. J. Hehre, A. Klamt, H. F. Schaefer III, M. W. Schmidt, C. D. Sherrill, D. G. Truhlar, A. Warshel, X. Xua, A. Aspuru-Guzik, R. Baer, A. T. Bell, N. A. Besley, J.-D. Chai, A. Dreuw, B. D. Dunietz, T. R. Furlani, S. R. Gwaltney, C.-P. Hsu, Y. Jung, J. Kong, D. S. Lambrecht, W. Liang, C. Ochsenfeld, V. A. Rassolov, L. V. Slipchenko, J. E. Subotnik, T. Van Voorhis, J. M. Herbert, A. I. Krylov, P. M. W. Gill, and M. Head-Gordon. <em>Advances in molecular quantum chemistry contained in the Q-Chem 4 program package.</em> <a href="http://www.tandfonline.com/doi/full/10.1080/00268976.2014.952696#.VTDQZBcarRB">[<em>Mol. Phys</em>. <strong>113</strong>, 184–215 (2015)]</a></p></li>
</ul>
<p>Literature citations for <span><span class="smallcaps">Q-Chem</span></span> v. 2.0<span class="citation" data-cites="QCHEM2"></span> and v. 3.0<span class="citation" data-cites="QCHEM3"></span> are also available, and the most current list of <span><span class="smallcaps">Q-Chem</span></span> authors can always be found on the website, <a href="http://www.q-chem.com">www.q-chem.com</a>. The primary literature is extensively referenced throughout this manual, and users are urged to cite the original literature for particular theoretical methods. This is how our large community of academic developers gets credit for its effort.</p>







<h1 id="chap:install">Installation, Customization, and Execution</h1>
<h2 id="installation-requirements">Installation Requirements</h2>
<h3 id="execution-environment">Execution Environment</h3>
<p><span><span class="smallcaps">Q-Chem</span></span> is shipped as a single executable along with several scripts. No compilation is required. Once the package is installed it is ready to run. Please refer to the installation notes for your particular platform, which are distributed with the software. The system software required to run <span><span class="smallcaps">Q-Chem</span></span> on your platform is minimal, and includes:</p>
<ul>
<li><p>A suitable operating system.</p></li>
<li><p>Run-time libraries (usually provided with your operating system).</p></li>
<li><p>Perl, version 5 or greater.</p></li>
<li><p>Vendor implementation of MPI or MPICH libraries (for the MPI-based parallel version only).</p></li>
</ul>
<p>Please check the <span><span class="smallcaps">Q-Chem</span></span> web site (<a href="http://www.q-chem.com">www.q-chem.com</a>), or contact <span><span class="smallcaps">Q-Chem</span></span> support (<a href="mailto:support@q-chem.com">support@q-chem.com</a>) if further details are required.</p>
<h3 id="hardware-platforms-and-operating-systems">Hardware Platforms and Operating Systems</h3>
<p><span><span class="smallcaps">Q-Chem</span></span> runs on a wide varieties of computer systems, ranging from Intel and AMD microprocessor-based PCs and workstations, to high-performance server nodes used in clusters and supercomputers. <span><span class="smallcaps">Q-Chem</span></span> supports the Linux, Mac, and Windows operating systems. To determine the availability of a specific platform or operating system, please contact <a href="mailto:support@q-chem.com">support@q-chem.com</a>.</p>
<h3 id="memory-and-disk-requirements">Memory and Disk Requirements</h3>
<p><strong><em>Memory</em></strong><br />
<span><span class="smallcaps">Q-Chem</span></span>, Inc. has endeavored to minimize memory requirements and maximize the efficiency of its use. Still, the larger the structure or the higher the level of theory, the more memory is needed. Although <span><span class="smallcaps">Q-Chem</span></span> can be run successfully in very small-memory environments, this is seldom an issue nowadays and we recommend 1 Gb as a minimum. <span><span class="smallcaps">Q-Chem</span></span> also offers the ability for user control of important, memory-intensive aspects of the program. In general, the more memory your system has, the larger the calculation you will be able to perform.</p>
<p><span><span class="smallcaps">Q-Chem</span></span> uses two types of memory: a chunk of static memory that is used by multiple data sets and managed by the code, and dynamic memory which is allocated using system calls. The size of the static memory is specified by the user through the <em>$rem</em> variable and has a default value of 64 Mb.</p>
<p>The <em>$rem</em> word specifies the limit of the total memory the user’s job can use. The default value is sufficiently large that on most machines it will allow <span><span class="smallcaps">Q-Chem</span></span> to use all the available memory. This value should be reduced on machines where this is undesirable (for example if the machine is used by multiple users). The limit for the dynamic memory allocation is given by ( <span class="math inline">−</span> ). The amount of needed depends on the size of the user’s particular job. Please note that one should not specify an excessively large value for , otherwise it will reduce the available memory for dynamic allocation. Memory settings in CC, EOM, and ADC calculations are described in Section <a href="#CCparallel" data-reference-type="ref" data-reference="CCparallel">[CCparallel]</a>. The use of <em>$rem</em> variables will be discussed in the next Chapter.<br />
<strong><em>Disk</em></strong><br />
The <span><span class="smallcaps">Q-Chem</span></span> executables, shell scripts, auxiliary files, samples and documentation require between 360–400 Mb of disk space, depending on the platform. The default <span><span class="smallcaps">Q-Chem</span></span> output, which is printed to the designated output file, is usually only a few kilobytes. This will be exceeded, of course, in difficult geometry optimizations, QM/MM and QM/EFP jobs, as well as in cases where users invoke non-default print options. In order to maximize the capabilities of your copy of <span><span class="smallcaps">Q-Chem</span></span>, additional disk space is required for scratch files created during execution, and these are automatically deleted on normal termination of a job. The amount of disk space required for scratch files depends critically on the type of job, the size of the molecule and the basis set chosen.</p>
<p><span><span class="smallcaps">Q-Chem</span></span> uses direct methods for Hartree-Fock and density functional theory calculations, which do not require large amount of scratch disk space. Wave function-based correlation methods, such as MP2 and coupled-cluster theory require substantial amounts of temporary (scratch) disk storage, and the faster the access speeds, the better these jobs will perform. With the low cost of disk drives, it is feasible to have between 100 and 1000 Gb of scratch space available as a dedicated file system for these large temporary job files. The more you have available, the larger the jobs that will be feasible and in the case of some jobs, like MP2, the jobs will also run faster as two-electron integrals are computed less often.</p>
<p>Although the size of any one of the <span><span class="smallcaps">Q-Chem</span></span> temporary files will not exceed 2 Gb, a user’s job will not be limited by this. <span><span class="smallcaps">Q-Chem</span></span> writes large temporary data sets to multiple files so that it is not bounded by the 2 Gb file size limitation on some operating systems.</p>
<h2 id="installing-q-chem">Installing <span><span class="smallcaps">Q-Chem</span></span></h2>
<p>Users are referred to the detailed installation instructions distributed with your copy of <span><span class="smallcaps">Q-Chem</span></span>.</p>
<p>An encrypted license file, <em>qchem.license.dat</em>, must be obtained from your vendor before you will be able to use <span><span class="smallcaps">Q-Chem</span></span>. This file should be placed in the directory <em>$QCAUX/license</em> and must be able to be read by all users of the software. This file is node-locked, <em>i.e.</em>, it will only operate correctly on the machine for which it was generated. Further details about obtaining this file, can be found in the installation instructions.</p>
<p><strong>Do not alter the license file unless directed by <span><span class="smallcaps">Q-Chem</span></span>, Inc.</strong></p>
<h2 id="q-chem-auxiliary-files-qcaux"><span><span class="smallcaps">Q-Chem</span></span> Auxiliary files (<em>$QCAUX</em>)</h2>
<p>The <em>$QCAUX</em> environment variable determines the directory where <span><span class="smallcaps">Q-Chem</span></span> searches for auxiliary files and the machine license. If not set explicitly, it defaults to <em>$QC/qcaux</em>.</p>
<p>The <em>$QCAUX</em> directory contains files required to run <span><span class="smallcaps">Q-Chem</span></span> calculations, including basis set and ECP specifications, SAD guesses (see Chapter <a href="#chap:scf" data-reference-type="ref" data-reference="chap:scf">4</a>), library of standard effective fragments (see Section <a href="#sect:efp" data-reference-type="ref" data-reference="sect:efp">[sect:efp]</a>), and instructions for the <span><span class="smallcaps">AOInts</span></span> package for generating two-electron integrals efficiently.</p>
<h2 id="q-chem-run-time-environment-variables"><span><span class="smallcaps">Q-Chem</span></span> Run-time Environment Variables</h2>
<p><span><span class="smallcaps">Q-Chem</span></span> requires the following shell environment variables setup prior to running any calculations:</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;">QC</td>
<td style="text-align: left;">Defines the location of the <span><span class="smallcaps">Q-Chem</span></span> directory structure. The <em>qchem.install</em> shell script determines this automatically.</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">QCAUX</td>
<td style="text-align: left;">Defines the location of the auxiliary information required by <span><span class="smallcaps">Q-Chem</span></span>, which includes the license required to run <span><span class="smallcaps">Q-Chem</span></span>. If not explicitly set by the user, this defaults to <em>$QC/qcaux.</em></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">QCSCRATCH</td>
<td style="text-align: left;"><p>Defines the directory in which <span><span class="smallcaps">Q-Chem</span></span> will store temporary files. <span><span class="smallcaps">Q-Chem</span></span> will usually remove these files on successful completion of the job, but they can be saved, if so wished. Therefore, <em>$QCSCRATCH</em> should not reside in a directory that will be automatically removed at the end of a job, if the files are to be kept for further calculations.</p>
<p>Note that many of these files can be very large, and it should be ensured that the volume that contains this directory has sufficient disk space available. The <em>$QCSCRATCH</em> directory should be periodically checked for scratch files remaining from abnormally terminated jobs. <em>$QCSCRATCH</em> defaults to the working directory if not explicitly set. Please see section <a href="#sect:running" data-reference-type="ref" data-reference="sect:running">2.7</a> for details on saving temporary files and consult your systems administrator.</p></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">QCLOCALSCR</td>
<td style="text-align: left;">On certain platforms, such as Linux clusters, it is sometimes preferable to write the temporary files to a disk local to the node. <em>$QCLOCALSCR</em> specifies this directory. The temporary files will be copied to <em>$QCSCRATCH</em> at the end of the job, unless the job is terminated abnormally. In such cases <span><span class="smallcaps">Q-Chem</span></span> will attempt to remove the files in <em>$QCLOCALSCR</em>, but may not be able to due to access restrictions. Please specify this variable only if required.</td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h2 id="user-account-adjustments">User Account Adjustments</h2>
<p>In order for individual users to run <span><span class="smallcaps">Q-Chem</span></span>, User file access permissions must be set correctly so that the user can read, write and execute the necessary <span><span class="smallcaps">Q-Chem</span></span> files. It may be advantageous to create a <em>qchem</em> user group on your machine and recursively change the group ownership of the <span><span class="smallcaps">Q-Chem</span></span> directory to <em>qchem</em> group.</p>
<p>The <span><span class="smallcaps">Q-Chem</span></span> run-time environment need to be initiated prior to running any <span><span class="smallcaps">Q-Chem</span></span> calculations, which is done by sourcing the environment setup script qcenv.sh (for bash) or qcenv.csh (for csh and tcsh) placed in your <span><span class="smallcaps">Q-Chem</span></span> top directory after a successful installation. It might be more convenient for user to include the <span><span class="smallcaps">Q-Chem</span></span> environment setup in their shell startup script, <em>e.g.</em>, <em>.cshrc/.tcshrc</em> for csh/tcsh or <em>.bashrc</em> for bash.</p>
<p>If using the csh or tcsh shell, add the following lines to the <em>.cshrc</em> file in the user’s home directory:</p>
<blockquote>
<pre><code>#
setenv  QC         qchem_root_directory_name
setenv  QCSCRATCH  scratch_directory_name
source  $QC/qcenv.csh
#</code></pre>
</blockquote>
<p>If using the Bourne-again shell (bash), add the following lines to the <em>.bashrc</em> file in the user’s home directory:</p>
<blockquote>
<pre><code># 
export QC=qchem_root_directory_name
export QCSCRATCH=scratch_directory_name
. $QC/qcenv.sh
#</code></pre>
</blockquote>
<h2 id="sect:customization">Further Customization: <em>.qchemrc</em> and <em>preferences</em> Files</h2>
<p><span><span class="smallcaps">Q-Chem</span></span> has developed a simple mechanism for users to set user-defined long-term defaults to override the built-in program defaults. Such defaults may be most suited to machine specific features such as memory allocation, as the total available memory will vary from machine to machine depending on specific hardware and accounting configurations. However, users may identify other important uses for this customization feature. <span><span class="smallcaps">Q-Chem</span></span> obtains input initialization variables from four sources:</p>
<ol>
<li><p>User input file</p></li>
<li><p><em>$HOME/.qchemrc</em> file</p></li>
<li><p><em>$QC/config/preferences</em> file</p></li>
<li><p>“Factory installed” program defaults</p></li>
</ol>
<p>Input mechanisms higher in this list override those that are lower. Mechanisms #2 and #3 allow the user to specify alternative default settings for certain variables that will override the <span><span class="smallcaps">Q-Chem</span></span> “factory-installed” defaults. This can be done by a system administrator via a <em>preferences</em> file added to the <em>$QC/config</em> directory, or by an individual user by means of a <em>.qchemrc</em> file in her home directory.</p>
<p><strong>Note: </strong></p>
<p>The <em>.qchemrc</em> and <em>preferences</em> files are not requisites for running<span><span class="smallcaps">Q-Chem</span></span> and currently only support keywords in the <em>$rem</em> input section.</p>
<p>The format of the <em>.qchemrc</em> and <em>preferences</em> files consists of a <em>$rem</em> keyword section, as in the <span><span class="smallcaps">Q-Chem</span></span> input file, terminated with the usual <em>$end</em> keyword. Any other <em>$whatever</em> section will be ignored. To aid in reproducibility, a copy of the <em>.qchemrc</em> file (if present) is included near the top of the job’s output file. (The <em>.qchemrc</em> and <em>preferences</em> files must have file permissions such that they are readable by the user invoking <span><span class="smallcaps">Q-Chem</span></span>.) The format of both of these files is as follows:</p>
<blockquote>
<pre><code>$rem
   rem_variable   option   comment
   rem_variable   option   comment
   ...
$end</code></pre>
</blockquote>

<p><strong>Example .</strong>  An example of a <em>.qchemrc</em> file to override default <em>$rem</em> settings for all of the user’s<span><span class="smallcaps">Q-Chem</span></span> jobs.</p>
<blockquote>
<pre><code>$rem
   INCORE_INTS_BUFFER   4000000   More integrals in memory
   DIIS_SUBSPACE_SIZE   5         Modify max DIIS subspace size
   THRESH               10        10**(-10) threshold
   MAX_SCF_CYCLES       100       More than the default of 50
$end</code></pre>
</blockquote>
<p>The following <em>$rem</em> variables are specifically recommended as those that a user might want to customize:</p>
<ul>
<li></li>
<li></li>
<li></li>
<li></li>
<li></li>
<li></li>
<li></li>
</ul>
<h2 id="sect:running">Running <span><span class="smallcaps">Q-Chem</span></span></h2>
<p>Once installation is complete, and any necessary adjustments are made to the user account, the user is now able to run <span><span class="smallcaps">Q-Chem</span></span>. There are several ways to invoke <span><span class="smallcaps">Q-Chem</span></span>:</p>
<ol>
<li><p><span><span class="smallcaps">IQmol</span></span> offers a fully integrated graphical interface for the <span><span class="smallcaps">Q-Chem</span></span> package and includes a sophisticated input generator with contextual help which is able to guide you through the many <span><span class="smallcaps">Q-Chem</span></span> options available. It also provides a molecular builder, job submission and monitoring tools, and is able to visualize molecular orbitals, densities and vibrational frequencies. For the latest version and download/installation instructions, please see the <span><span class="smallcaps">IQmol</span></span> homepage (<a href="www.iqmol.org" class="uri">www.iqmol.org</a>).</p></li>
<li><p><em>qchem</em> command line shell script. The simple format for command line execution is given below. The remainder of this manual covers the creation of input files in detail.</p></li>
<li><p>Via a third-party GUI. The two most popular ones are:</p>
<ul>
<li><p>A general web-based interface for electronic structure software, <span><span class="smallcaps">WebMO</span></span><br />
(<a href="www.webmo.net" class="uri">www.webmo.net</a>).</p></li>
<li><p>Wavefunction’s <span><span class="smallcaps">Spartan</span></span> user interface on some platforms. Contact Wavefunction, Inc. <br />
(<a href="www.wavefun.com" class="uri">www.wavefun.com</a>) or <span><span class="smallcaps">Q-Chem</span></span> for full details of current availability.</p></li>
</ul></li>
</ol>
<p>Using the <span><span class="smallcaps">Q-Chem</span></span> command line shell script (<em>qchem</em>) is straightforward provided <span><span class="smallcaps">Q-Chem</span></span> has been correctly installed on your machine and the necessary environment variables have been set in your <em>.cshrc</em>, <em>.profile</em>, or equivalent login file. If done correctly, the necessary changes will have been made to the <em>$PATH</em> variable automatically on login so that <span><span class="smallcaps">Q-Chem</span></span> can be invoked from your working directory.<br />
The <em>qchem</em> shell script can be used in either of the following ways:</p>
<blockquote>
<pre><code>qchem infile outfile
qchem infile outfile savename
qchem -save infile outfile savename</code></pre>
</blockquote>
<p>where <em>infile</em> is the name of a suitably formatted <span><span class="smallcaps">Q-Chem</span></span> input file (detailed in Chapter <a href="#chap:inputs" data-reference-type="ref" data-reference="chap:inputs">3</a>, and the remainder of this manual), and the <em>outfile</em> is the name of the file to which <span><span class="smallcaps">Q-Chem</span></span> will place the job output information.</p>
<p><strong>Note: </strong></p>
<p>If the <em>outfile</em> already exists in the working directory, it will be overwritten.</p>
<p>The use of the <em>savename</em> command line variable allows the saving of a few key scratch files between runs, and is necessary when instructing <span><span class="smallcaps">Q-Chem</span></span> to read information from previous jobs. If the <em>savename</em> argument is not given, <span><span class="smallcaps">Q-Chem</span></span> deletes all temporary scratch files at the end of a run. The saved files are in <em><span>$</span>QCSCRATCH/savename/</em>, and include files with the current molecular geometry, the current molecular orbitals and density matrix and the current force constants (if available). The <em>–save</em> option in conjunction with <em>savename</em> means that all temporary files are saved, rather than just the few essential files described above. Normally this is not required. When <em>$QCLOCALSCR</em> has been specified, the temporary files will be stored there and copied to <em>$QCSCRATCH/savename/</em> at the end of normal termination.</p>
<p>The name of the input parameters <em>infile</em>, <em>outfile</em> and <em>save</em> can be chosen at the discretion of the user (usual UNIX file and directory name restrictions apply). It maybe helpful to use the same job name for <em>infile</em> and <em>outfile</em>, but with varying suffixes. For example:</p>
<blockquote>
<pre><code>localhost-1&gt; qchem water.in water.out &amp;</code></pre>
</blockquote>
<p>invokes <span><span class="smallcaps">Q-Chem</span></span> where the input is taken from <em>water.in</em> and the output is placed into <em>water.out</em>. The <code>&amp;</code> places the job into the background so that you may continue to work in the current shell.</p>
<blockquote>
<pre><code>localhost-2&gt; qchem water.com water.log water &amp;</code></pre>
</blockquote>
<p>invokes <span><span class="smallcaps">Q-Chem</span></span> where the input is assumed to reside in <em>water.com</em>, the output is placed into <em>water.log</em> and the key scratch files are saved in a directory <em><span>$</span>QCSCRATCH/water/</em>.</p>
<p><strong>Note: </strong></p>
<p>A checkpoint file can be requested by setting in the <em>$rem</em> section of the input. The checkpoint file name is determined by the <em><span>$</span>GUIFILE</em> environment variable which by default is set to <em>${input}.fchk</em></p>
<h2 id="sec:parallel">Parallel <span><span class="smallcaps">Q-Chem</span></span> Jobs</h2>
<p>Parallel execution of <span><span class="smallcaps">Q-Chem</span></span> can be threaded across multiple processors on a single node, using the OpenMP protocol, or using the message-passing interface (MPI) protocol to parallelize over multiple processor cores and/or multiple compute nodes. A hybrid MPI + OpenMP scheme is also available for certain calculations, in which each MPI process spawns several OpenMP threads. In this hybrid scheme, cross-node communication is handled by the MPI protocol and intra-node communication is done implicitly using OpenMP threading for efficient utilization of shared-memory parallel (SMP) systems. This parallelization strategy reflects current trends towards multi-core architectures in cluster computing.</p>
<p>As of the v. 4.2 release, the OpenMP parallelization is fully supported by HF/DFT, RIMP2, CC, EOM-CC, and ADC methods. The MPI parallel capability is available for SCF, DFT, CIS, and TDDFT methods. The hybrid MPI+OpenMP parallelization is introduced in v. 4.2 for HF/DFT energy and gradient calculations only. Distributed memory MPI+OpenMP parallelization of CC and EOM-CC methods was added in <span><span class="smallcaps">Q-Chem</span></span> v. 4.3. Table <a href="#table:parallel_capabilities" data-reference-type="ref" data-reference="table:parallel_capabilities">[table:parallel_capabilities]</a> summarizes the parallel capabilities of <span><span class="smallcaps">Q-Chem</span></span> v. 5.0.</p>
<table>
<caption>Parallel capabilities of <span><span class="smallcaps">Q-Chem</span></span> v. 5.0</caption>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Method</strong></th>
<th style="text-align: center;">OpenMP</th>
<th style="text-align: center;">MPI</th>
<th style="text-align: center;">MPI+OpenMP</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">HF energy &amp; gradient</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">yes</td>
</tr>
<tr class="even">
<td style="text-align: left;">DFT energy &amp; gradient</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">yes</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CDFT/CDFT-CI</td>
<td style="text-align: center;">no</td>
<td style="text-align: center;">no</td>
<td style="text-align: center;">no</td>
</tr>
<tr class="even">
<td style="text-align: left;">RI-MP2 energy</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">no</td>
<td style="text-align: center;">no</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Attenuated RI-MP2 energy</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">no</td>
<td style="text-align: center;">no</td>
</tr>
<tr class="even">
<td style="text-align: left;">Integral transformation</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">no</td>
<td style="text-align: center;">no</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CCMAN &amp; CCMAN2 methods</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">yes</td>
</tr>
<tr class="even">
<td style="text-align: left;">ADC methods</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">no</td>
<td style="text-align: center;">no</td>
</tr>
<tr class="odd">
<td style="text-align: left;">CIS energy &amp; gradient</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">no</td>
</tr>
<tr class="even">
<td style="text-align: left;">TDDFT energy &amp; gradient</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">no</td>
</tr>
<tr class="odd">
<td style="text-align: left;">HF &amp; DFT analytical Hessian</td>
<td style="text-align: center;">no</td>
<td style="text-align: center;">yes</td>
<td style="text-align: center;">no</td>
</tr>
</tbody>
</table>
<p><span id="table:parallel_capabilities" label="table:parallel_capabilities">[table:parallel_capabilities]</span></p>
<p>To run <span><span class="smallcaps">Q-Chem</span></span> calculation with OpenMP threads specify the number of threads (<em>nthreads</em>) using <em>qchem</em> command option <em>-nt</em>. Since each thread uses one CPU core, you should not specify more threads than the total number of available CPU cores for performance reason. When unspecified, the number of threads defaults to 1 (serial calculation).</p>
<blockquote>
<pre><code>qchem -nt nthreads infile outfile
qchem -nt nthreads infile outfile save
qchem -save -nt nthreads infile outfile save</code></pre>
</blockquote>
<p>Similarly, to run parallel calculations with MPI use the option <em>-np</em> to specify the number of MPI processes to be spawned.</p>
<blockquote>
<pre><code>qchem -np n infile outfile
qchem -np n infile outfile savename
qchem -save -np n infile outfile savename</code></pre>
</blockquote>
<p>where <span class="math inline"><em>n</em></span> is the number of processors to use. If the <em>-np</em> switch is not given, <span><span class="smallcaps">Q-Chem</span></span> will default to running locally on a single node.</p>
<p>To run hybrid MPI+OpenMP HF/DFT calculations use combined options <em>-np</em> and <em>-nt</em> together, where <em>-np</em> followed by the number of MPI processes to be spawned and <em>-nt</em> followed by the number of OpenMP threads used in each MPI process.</p>
<blockquote>
<pre><code>qchem -np n -nt nthreads infile outfile
qchem -np n -nt nthreads infile outfile savename
qchem -save -np n -nt nthreads infile outfile savename</code></pre>
</blockquote>
<p>When the additional argument <em>savename</em> is specified, the temporary files for MPI-parallel <span><span class="smallcaps">Q-Chem</span></span> are stored in <em>$QCSCRATCH/savename.0</em> At the start of a job, any existing files will be copied into this directory, and on successful completion of the job, be copied to <em>$QCSCRATCH/savename/</em> for future use. If the job terminates abnormally, the files will not be copied.</p>
<p>To run parallel <span><span class="smallcaps">Q-Chem</span></span> using a batch scheduler such as PBS, users may need to set QCMPIRUN environment variable to point to the <em>mpirun</em> command used in the system. For further details users should read the <em><span>$</span>QC/README.Parallel</em> file, and contact <span><span class="smallcaps">Q-Chem</span></span> if any problems are encountered (<a href="mailto:support@q-chem.com">support@q-chem.com</a>).</p>
<h2 id="sect:iqmolinstall"><span><span class="smallcaps">IQmol</span></span> Installation Requirements</h2>
<p><span><span class="smallcaps">IQmol</span></span> provides a fully integrated molecular builder and viewer for the <span><span class="smallcaps">Q-Chem</span></span> package. It is available for the Windows, Linux, and Mac OS X platforms and instructions for downloading and installing the latest version can be found at <a href="www.iqmol.org/downloads.html" class="uri">www.iqmol.org/downloads.html</a>.</p>
<p><span><span class="smallcaps">IQmol</span></span> can be run as a stand-alone package which is able to open existing <span><span class="smallcaps">Q-Chem</span></span> input/output files, but it can also be used as a fully functional front end which is able to submit and monitor <span><span class="smallcaps">Q-Chem</span></span> jobs, and to analyze the resulting output. By default, <span><span class="smallcaps">IQmol</span></span> submits <span><span class="smallcaps">Q-Chem</span></span> jobs to a server that is owned by <span><span class="smallcaps">Q-Chem</span></span>, Inc., which provides prospective users with the opportunity to run short <span><span class="smallcaps">Q-Chem</span></span> demonstration jobs for free simply by downloading <span><span class="smallcaps">IQmol</span></span>, without the need to install <span><span class="smallcaps">Q-Chem</span></span>.</p>
<p>For customers who own <span><span class="smallcaps">Q-Chem</span></span>, it is necessary to configure <span><span class="smallcaps">IQmol</span></span> to submit jobs to an appropriate server. To do this, first ensure <span><span class="smallcaps">Q-Chem</span></span> has been correctly installed on the target machine and can be run from the command line. Second, open <span><span class="smallcaps">IQmol</span></span> and carry out the following steps:</p>
<ol>
<li><p>Select the Calculation<span class="math inline">→</span>Edit Servers menu option. A dialog will appear with a list of configured servers (which will initially be empty).</p></li>
<li><p>Click the Add New Server button with the ‘+’ icon. This opens a dialog which allows the new server to be configured. The server is the machine which has your <span><span class="smallcaps">Q-Chem</span></span> installation.</p></li>
<li><p>Give the server a name (this is simply used to identify the current server configuration and does not have to match the actual machine name) and select if the machine is local (<em>i.e.</em> the same machine as <span><span class="smallcaps">IQmol</span></span> is running on) or remote.</p></li>
<li><p>If there is PBS software running on the server, select the PBS ‘Type’ option, otherwise in most cases the Basic option should be sufficient. Please note that the server must be Linux based and cannot be a Windows server.</p></li>
<li><p>If required, the server can be further configured using the Configure button. Details on this can be found in the embedded <span><span class="smallcaps">IQmol</span></span> help which can be accessed via the Help<span class="math inline">→</span>Show Help menu option.</p></li>
<li><p>For non-PBS servers the number of concurrent <span><span class="smallcaps">Q-Chem</span></span> jobs can be limited using a simple inbuilt queuing system. The maximum number of jobs is set by the Job Limit control. If the Job Limit is set to zero the queue is disabled and any number of jobs can be run concurrently. Please note that this limit applies to the current <span><span class="smallcaps">IQmol</span></span> session and does not account for jobs submitted by other users or by other <span><span class="smallcaps">IQmol</span></span> sessions.</p></li>
<li><p>The $QC environment variable should be entered in the given box.</p></li>
<li><p>For remote servers the address of the machine and your user name are also required. <span><span class="smallcaps">IQmol</span></span> uses SSH2 to connect to remote machines and the most convenient way to set this up is by using authorized keys (<a href="www.debian.org/devel/passwordlessssh">)</a> for details on how these can be set up). <span><span class="smallcaps">IQmol</span></span> can then connect via the SSH Agent and will not have to prompt you for your password. If you are not able to use an SSH Agent, several other authentication methods are offered:</p>
<ul>
<li><p><span><strong>Public Key</strong></span> This requires you to enter your SSH passphrase (if any) to unlock your private key file. The passphrase is stored in memory, not disk, so you will need to re-enter this each time <span><span class="smallcaps">IQmol</span></span> is run.</p></li>
<li><p><span><strong>Password Prompt</strong></span> This requires each server password to be entered each time <span><span class="smallcaps">IQmol</span></span> is run. Once the connection has been established the memory used to hold the password is overwritten to reduce the risk of recovery from a core dump.</p></li>
</ul>
<p>Further configuration of SSH options should not be required unless your public/private keys are stored in a non-standard location.</p></li>
</ol>
<p>It is recommended that you test the server configuration to ensure everything is working before attempting to submit a job. Multiple servers can be configured if you have access to more than one copy of <span><span class="smallcaps">Q-Chem</span></span> or have different account configurations. In this case the default server is the first on the list and if you want to change this you should use the arrow buttons in the Server List dialog. The list of configured servers will be displayed when submitting <span><span class="smallcaps">Q-Chem</span></span> jobs and you will be able to select the desired server for each job.</p>
<p>Please note that while <span><span class="smallcaps">Q-Chem</span></span> is file-based, as of version 2.1 <span><span class="smallcaps">IQmol</span></span> uses a directory to keep the various files from a calculation. More details can be found in the <span><span class="smallcaps">IQmol</span></span> <a href="http://www.iqmol.org/downloads/IQmolUserGuide.pdf">user manual</a>.</p>
<h2 id="testing-and-exploring-q-chem">Testing and Exploring <span><span class="smallcaps">Q-Chem</span></span></h2>
<p><span><span class="smallcaps">Q-Chem</span></span> is shipped with a small number of test jobs which are located in the <em>$QC/samples</em> directory. If you wish to test your version of <span><span class="smallcaps">Q-Chem</span></span>, run the test jobs in the samples directory and compare the output files with the reference files (suffixed <em>.out</em>) of the same name.</p>
<p>These test jobs are not an exhaustive quality control test (a small subset of the test suite used at <span><span class="smallcaps">Q-Chem</span></span>, Inc.), but they should all run correctly on your platform. If any fault is identified in these, or any output files created by your version, do not hesitate to contact customer service immediately.</p>
<p>These jobs are also an excellent way to begin learning about <span><span class="smallcaps">Q-Chem</span></span>’s text-based input and output formats in detail. In many cases you can use these inputs as starting points for building your own input files, if you wish to avoid reading the rest of this manual!</p>
<p>Please check the <span><span class="smallcaps">Q-Chem</span></span> web page (<a href="http://www.q-chem.com">www.q-chem.com</a>) and the README files in the <em>$QC/bin</em> directory for updated information.</p>
<h1 id="chap:inputs"><span><span class="smallcaps">Q-Chem</span></span> Inputs</h1>
<h2 id="iqmol"><span><span class="smallcaps">IQmol</span></span></h2>
<p>The easiest way to run <span><span class="smallcaps">Q-Chem</span></span> is by using the <span><span class="smallcaps">IQmol</span></span> interface which can be downloaded for free from <a href="www.iqmol.org" class="uri">www.iqmol.org</a>. Before submitting a <span><span class="smallcaps">Q-Chem</span></span> job from you will need to configure a <span><span class="smallcaps">Q-Chem</span></span> server and details on how to do this are given in Section <a href="#sect:iqmolinstall" data-reference-type="ref" data-reference="sect:iqmolinstall">2.9</a> of this manual.</p>
<p><span><span class="smallcaps">IQmol</span></span> provides a free-form molecular builder and a comprehensive interface for setting up the input for <span><span class="smallcaps">Q-Chem</span></span> jobs. Additionally calculations can be submitted to either the local or a remote machine and monitored using the built in job monitor. The output can also be analyzed allowing visualization of molecular orbitals and densities, and animation of vibrational modes and reaction pathways. A more complete list of features can be found at <a href="www.iqmol.org/features.html" class="uri">www.iqmol.org/features.html</a>.</p>
<p>The <span><span class="smallcaps">IQmol</span></span> program comes with a built-in help system that details how to set up and submit <span><span class="smallcaps">Q-Chem</span></span> calculations. This help can be accessed via the Help<span class="math inline">→</span>Show Help menu option.</p>
<h2 id="general-form">General Form</h2>
<p><span><span class="smallcaps">IQmol</span></span> (or another graphical interface) is the simplest way to control <span><span class="smallcaps">Q-Chem</span></span>. However, the low level command line interface is available to enable maximum customization and allow the user to exploit all <span><span class="smallcaps">Q-Chem</span></span>’s features. The command line interface requires a <span><span class="smallcaps">Q-Chem</span></span> input file which is simply an ASCII text file. This input file can be created using your favorite editor (<em>e.g.</em>, vi, emacs, jot, <em>etc.</em>) following the basic steps outlined in the next few chapters.<br />
<span><span class="smallcaps">Q-Chem</span></span>’s input mechanism uses a series of <strong>keywords</strong> to signal user input sections of the input file. As required, the <span><span class="smallcaps">Q-Chem</span></span> program searches the input file for supported keywords. When <span><span class="smallcaps">Q-Chem</span></span> finds a keyword, it then reads the section of the input file beginning at the keyword until that keyword section is terminated the <em>$end</em> keyword. A short description of all <span><span class="smallcaps">Q-Chem</span></span> keywords is provided in Table <a href="#table:input_sections" data-reference-type="ref" data-reference="table:input_sections">[table:input_sections]</a> and the following sections. The user <strong>must</strong> understand the function and format of the <em>$molecule</em> (Section <a href="#sect:molinput" data-reference-type="ref" data-reference="sect:molinput">3.3</a>) and <em>$rem</em> (Section <a href="#sect:rem" data-reference-type="ref" data-reference="sect:rem">3.4</a>) keywords, as these keyword sections are where the user places the molecular geometry information and job specification details.</p>
<table>
<caption> A list of <span><span class="smallcaps">Q-Chem</span></span> input sections; the first two (<em>$molecule</em> and <em>$rem</em>) are required for all jobs, whereas the rest are required only for certain job types, or else are optional places to specify additional job-control variables. Each input section (“<em>$section</em>”) should be terminated with <em>$end</em>. See the <em>$QC/samples</em> directory that is included with your release for specific examples of <span><span class="smallcaps">Q-Chem</span></span> input files using these keywords. </caption>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Section Name</strong></th>
<th style="text-align: left;"><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><em>$molecule</em></td>
<td style="text-align: left;">Contains the molecular coordinate input (input file requisite).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$rem</em></td>
<td style="text-align: left;">Job specification and customization parameters (input file requisite).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$basis</em></td>
<td style="text-align: left;">User-defined basis set information (Chapter <a href="#chap:basis" data-reference-type="ref" data-reference="chap:basis">8</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$cdft</em></td>
<td style="text-align: left;">Options for the constrained DFT method (Section <a href="#sect:CDFT" data-reference-type="ref" data-reference="sect:CDFT">[sect:CDFT]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">Job control for the <span><span class="smallcaps">Q-Chem</span></span>/<span><span class="smallcaps">ChemSol</span></span> interface (Langevin dipoles</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">model; Section <a href="#subsec:LD" data-reference-type="ref" data-reference="subsec:LD">[subsec:LD]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$comment</em></td>
<td style="text-align: left;">User comments for inclusion into output file.</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">Contains parameters for complex-scaled and CAP-augmented EOM-CC</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">calculations (Chapter <a href="#sect:exeom" data-reference-type="ref" data-reference="sect:exeom">[sect:exeom]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$ecp</em></td>
<td style="text-align: left;">User-defined effective core potentials (Chapter <a href="#chap:ecp" data-reference-type="ref" data-reference="chap:ecp">9</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$efei</em></td>
<td style="text-align: left;">Application of external forces in a geometry optimization (Section <a href="#subsec:EFEI" data-reference-type="ref" data-reference="subsec:EFEI">[subsec:EFEI]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$efp_fragments</em></td>
<td style="text-align: left;">Specifies labels and positions of EFP fragments (Section <a href="#sect:efp" data-reference-type="ref" data-reference="sect:efp">[sect:efp]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$efp_params</em></td>
<td style="text-align: left;">Contains user-defined parameters for effective fragments (Section <a href="#sect:efp" data-reference-type="ref" data-reference="sect:efp">[sect:efp]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">User-defined van der Waals parameters for DFT dispersion correction</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">(Section <a href="#subsec:DFT-D" data-reference-type="ref" data-reference="subsec:DFT-D">[subsec:DFT-D]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$eom_user_guess</em></td>
<td style="text-align: left;">User-defined guess for EOM-CC calculations (Chapter <a href="#sect:exeom" data-reference-type="ref" data-reference="sect:exeom">[sect:exeom]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$external_charges</em></td>
<td style="text-align: left;">Specifies external point charges and their positions.</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$force_field_params</em></td>
<td style="text-align: left;">Force-field parameters for QM/MM calculations (Section <a href="#sec:QMMM" data-reference-type="ref" data-reference="sec:QMMM">[sec:QMMM]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$intracule</em></td>
<td style="text-align: left;">Intracule parameters (Section <a href="#sect:intracules" data-reference-type="ref" data-reference="sect:intracules">[sect:intracules]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$isotopes</em></td>
<td style="text-align: left;">Isotopic substitutions for vibrational calculations (Section <a href="#sect:isotopes" data-reference-type="ref" data-reference="sect:isotopes">[sect:isotopes]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">Information for mixing together multiple adiabatic states into diabatic</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">states (Chapter <a href="#chap:properties" data-reference-type="ref" data-reference="chap:properties">11</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$magnet</em></td>
<td style="text-align: left;">Job control for magnetic field-related response properties (Section <a href="#sec:magnetman" data-reference-type="ref" data-reference="sec:magnetman">[sec:magnetman]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$multipole_field</em></td>
<td style="text-align: left;">Details of an external multipole field (Section <a href="#sec:multipolefield" data-reference-type="ref" data-reference="sec:multipolefield">3.5.7</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$nbo</em></td>
<td style="text-align: left;">Options for the Natural Bond Orbital package (Section <a href="#sect:NBO" data-reference-type="ref" data-reference="sect:NBO">[sect:NBO]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$occupied</em></td>
<td style="text-align: left;">Guess orbitals to be occupied (Section <a href="#sect:occupied" data-reference-type="ref" data-reference="sect:occupied">[sect:occupied]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$opt</em></td>
<td style="text-align: left;">Constraint definitions for geometry optimizations (Section <a href="#sec:constrained_opt" data-reference-type="ref" data-reference="sec:constrained_opt">[sec:constrained_opt]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$pcm</em></td>
<td style="text-align: left;">Job control for polarizable continuum models (Section <a href="#subsec:PCM_job_control" data-reference-type="ref" data-reference="subsec:PCM_job_control">[subsec:PCM_job_control]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$plots</em></td>
<td style="text-align: left;">Generate plotting information over a grid of points (Section <a href="#sect:plots" data-reference-type="ref" data-reference="sect:plots">[sect:plots]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$qct_active_modes</em></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$qct_vib_distribution</em></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$qct_vib_phase</em></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$qm_atoms</em></td>
<td style="text-align: left;">Specify the QM region for QM/MM calculations (Section <a href="#sec:QMMM" data-reference-type="ref" data-reference="sec:QMMM">[sec:QMMM]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$response</em></td>
<td style="text-align: left;">Job control for the generalized response solver (Section <a href="#sec:GeneralResponseTheory" data-reference-type="ref" data-reference="sec:GeneralResponseTheory">[sec:GeneralResponseTheory]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">Additional parameters and variables for implicit solvent models</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">(Section <a href="#sect:solvent" data-reference-type="ref" data-reference="sect:solvent">[sect:solvent]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$smx</em></td>
<td style="text-align: left;">Job control for SM<span class="math inline"><em>x</em></span> implicit solvent models (Section <a href="#subsec:SMx" data-reference-type="ref" data-reference="subsec:SMx">[subsec:SMx]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$swap_occupied_virtual</em></td>
<td style="text-align: left;">Guess orbitals to be swapped (Section <a href="#sect:occupied" data-reference-type="ref" data-reference="sect:occupied">[sect:occupied]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$svp</em></td>
<td style="text-align: left;">Special parameters for the iso-density SS(V)PE module (Section <a href="#subsec:SS(V)PE" data-reference-type="ref" data-reference="subsec:SS(V)PE">[subsec:SS(V)PE]</a>).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>$svpirf</em></td>
<td style="text-align: left;">Initial guess for the iso-density SS(V)PE module (Section <a href="#subsec:SS(V)PE" data-reference-type="ref" data-reference="subsec:SS(V)PE">[subsec:SS(V)PE]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">Additional parameters for two-photon absorption calculations</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">(Section <a href="#sec:2PA" data-reference-type="ref" data-reference="sec:2PA">[sec:2PA]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">User-defined atomic radii for Langevin dipoles solvation (Section <a href="#subsec:LD" data-reference-type="ref" data-reference="subsec:LD">[subsec:LD]</a>)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">and PCMs (Section <a href="#subsec:PCM" data-reference-type="ref" data-reference="subsec:PCM">[subsec:PCM]</a>).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>$xc_functional</em></td>
<td style="text-align: left;">User-defined DFT exchange-correlation functional (Section <a href="#sect:userxc" data-reference-type="ref" data-reference="sect:userxc">[sect:userxc]</a>).</td>
</tr>
</tbody>
</table>
<p><span id="table:input_sections" label="table:input_sections">[table:input_sections]</span></p>
<p><strong>The keywords <em>$rem</em> and <em>$molecule</em> are required in any <span><span class="smallcaps">Q-Chem</span></span> input file</strong></p>
<p>As each keyword has a different function, the format required for specific keywords varies somewhat, to account for these differences (format requirements are summarized in Appendix <a href="#app:quick_ref" data-reference-type="ref" data-reference="app:quick_ref">16</a>). However, because each keyword in the input file is sought out independently by the program, the overall format requirements of <span><span class="smallcaps">Q-Chem</span></span> input files are much less stringent. For example, the <em>$molecule</em> section does not have to occur at the very beginning of the input file.</p>
<blockquote>
<p><strong>Note: </strong></p>
<p>(1) Users are able to enter keyword sections in any order.<br />
(2) Each keyword section must be terminated with the <em>$end</em> keyword.<br />
(3) The <em>$rem</em> and <em>$molecule</em> sections must be included.<br />
(4) It is not necessary to have all keywords in an input file.<br />
(5) Each keyword section is described in Appendix <a href="#app:quick_ref" data-reference-type="ref" data-reference="app:quick_ref">16</a>.<br />
(6) The entire<span><span class="smallcaps">Q-Chem</span></span> input is case-insensitive.</p>
</blockquote>
<p>The second general aspect of <span><span class="smallcaps">Q-Chem</span></span> input is that there are effectively four input sources:</p>
<ul>
<li><p>User input file (required)</p></li>
<li><p><em>.qchemrc</em> file in <em>$HOME</em> (optional)</p></li>
<li><p><em>preferences</em> file in <em>$QC/config</em> (optional)</p></li>
<li><p>Internal program defaults and calculation results (built-in)</p></li>
</ul>
<p>The order of preference is as shown, <em>i.e.</em>, the input mechanism offers a program default override for <em>all</em> users, default override for <em>individual</em> users and, of course, the input file provided by the user overrides all defaults. Refer to Section <a href="#sect:customization" data-reference-type="ref" data-reference="sect:customization">2.6</a> for details of <em>.qchemrc</em> and <em>preferences</em>. Currently, <span><span class="smallcaps">Q-Chem</span></span> only supports the <em>$rem</em> keyword in <em>.qchemrc</em> and <em>preferences</em> files.</p>
<p>In general, users will need to enter variables for the <em>$molecule</em> and <em>$rem</em> keyword section and are encouraged to add a <em>$comment</em> for future reference. The necessity of other keyword input will become apparent throughout the manual.</p>
<h2 id="sect:molinput">Molecular Coordinate Input (<em>$molecule</em>)</h2>
<p>The <em>$molecule</em> section communicates to the program the charge, spin multiplicity, and geometry of the molecule being considered. The molecular coordinates input begins with two integers: the net charge and the spin multiplicity of the molecule. The net charge must be between <span class="math inline">−</span>50 and 50, inclusive (0 for neutral molecules, 1 for cations, <span class="math inline">−</span>1 for anions, <em>etc.</em>). The multiplicity must be between 1 and 10, inclusive (1 for a singlet, 2 for a doublet, 3 for a triplet, <em>etc.</em>). Each subsequent line of the molecular coordinate input corresponds to a single atom in the molecule (or dummy atom), regardless of whether using <em>Z</em>-matrix internal coordinates or Cartesian coordinates.</p>
<p><strong>Note: </strong></p>
<p>The coordinate system used for declaring an initial molecular geometry by default does not affect that used in a geometry optimization procedure. See Appendix <a href="#app:geom_opt" data-reference-type="ref" data-reference="app:geom_opt">14</a> which discusses the<span><span class="smallcaps">Optimize</span></span> package in further detail.</p>
<p><span><span class="smallcaps">Q-Chem</span></span> begins all calculations by rotating and translating the user-defined molecular geometry into a Standard Nuclear Orientation whereby the center of nuclear charge is placed at the origin. This is a standard feature of most quantum chemistry programs. This action can be turned off by using .</p>
<p><strong>Note: </strong></p>
<p>= will also turn off determining and using of the point group symmetry.</p>
<p><strong>Note: </strong></p>
<p><span><span class="smallcaps">Q-Chem</span></span> ignores commas and equal signs, and requires all distances, positions and angles to be entered as<span>Å</span>ngstroms and degrees unless the <em>$rem</em> variable is set to , in which case all lengths are assumed to be in bohr.</p>
<h3 id="specifying-the-molecular-coordinates-manually">Specifying the Molecular Coordinates Manually</h3>
<h4 id="cartesian-coordinates">Cartesian Coordinates</h4>
<p><span><span class="smallcaps">Q-Chem</span></span> can accept a list of <span class="math inline"><em>N</em></span> atoms and their <span class="math inline">3<em>N</em></span> Cartesian coordinates. The atoms can be entered either as atomic numbers or atomic symbols where each line corresponds to a single atom. The <span><span class="smallcaps">Q-Chem</span></span> format for declaring a molecular geometry using Cartesian coordinates (in <span>Å</span>ngstroms) is:</p>
<blockquote>
<pre><code>atom  x-coordinate  y-coordinate  z-coordinate</code></pre>
</blockquote>
<p><strong>Note: </strong></p>
<p>The geometry can by specified in bohr by setting the <em>$rem</em> variable equal to .</p>

<p><strong>Example .</strong>  Atomic number Cartesian coordinate input for H<span class="math inline"><em></em><sub>2</sub></span>O. The first line species the molecular charge and multiplicity, respectively.</p>
<blockquote>
<pre><code>$molecule
   0 1
   8   0.000000   0.000000  -0.212195
   1   1.370265   0.000000   0.848778
   1  -1.370265   0.000000   0.848778
$end</code></pre>
</blockquote>

<p><strong>Example .</strong>  Atomic symbol Cartesian coordinate input for H<span class="math inline"><em></em><sub>2</sub></span>O.</p>
<blockquote>
<pre><code>$molecule
   0 1
   O   0.000000   0.000000  -0.212195
   H   1.370265   0.000000   0.848778
   H  -1.370265   0.000000   0.848778
$end</code></pre>
</blockquote>
<p><strong>Note: </strong></p>
<p>(1) Atoms can be declared by either atomic number or symbol.<br />
(2) Coordinates can be entered either as variables/parameters or real numbers.<br />
(3) Variables/parameters can be declared in any order.<br />
(4) A single blank line separates parameters from the atom declaration.</p>
<p><br />
Once all the molecular Cartesian coordinates have been entered, terminate the molecular coordinate input with the <em>$end</em> keyword.</p>
<h4 id="z-matrix-coordinates"><em>Z</em>-matrix Coordinates</h4>
<p>For small molecules, <em>Z</em>-matrix notation is a common input format. The <em>Z</em>-matrix defines the positions of atoms relative to previously defined atoms using a length, an angle and a dihedral angle. Again, note that all bond lengths and angles must be in <span>Å</span>ngstroms and degrees, unless is set to , in which case bond lengths are specified in bohr.</p>
<p><strong>Note: </strong></p>
<p>As with the Cartesian coordinate input method,<span><span class="smallcaps">Q-Chem</span></span> begins a calculation by taking the user-defined coordinates and translating and rotating them into a Standard Nuclear Orientation.</p>
<p><br />
The first three atom entries of a <em>Z</em>-matrix are different from the subsequent entries. The first <em>Z</em>-matrix line declares a single atom. The second line of the <em>Z</em>-matrix input declares a second atom, refers to the first atom and gives the distance between them. The third line declares the third atom, refers to either the first or second atom, gives the distance between them, refers to the remaining atom and gives the angle between them. All subsequent entries begin with an atom declaration, a reference atom and a distance, a second reference atom and an angle, a third reference atom and a dihedral angle. This can be summarized as:</p>
<ol>
<li><p>First atom.</p></li>
<li><p>Second atom, reference atom, distance.</p></li>
<li><p>Third atom, reference atom A, distance between A and the third atom, reference atom B, angle defined by atoms A, B and the third atom.</p></li>
<li><p>Fourth atom, reference atom A, distance, reference atom B, angle, reference atom C, dihedral angle (A, B, C and the fourth atom).</p></li>
<li><p>All subsequent atoms follow the same basic form as (4)</p></li>
</ol>

<p><strong>Example .</strong>  <em>Z</em>-matrix for hydrogen peroxide</p>
<blockquote>
<pre><code>  O1
  O2   O1   oo
  H1   O1   ho   O2   hoo
  H2   O2   ho   O1   hoo   H1   hooh</code></pre>
</blockquote>
<p>Line 1 declares an oxygen atom (O1). Line 2 declares the second oxygen atom (O2), followed by a reference to the first atom (O1) and a distance between them denoted <code>oo</code>. Line 3 declares the first hydrogen atom (H1), indicates it is separated from the first oxygen atom (O1) by a distance HO and makes an angle with the second oxygen atom (O2) of <code>hoo</code>. Line 4 declares the fourth atom and the second hydrogen atom (H2), indicates it is separated from the second oxygen atom (O2) by a distance HO and makes an angle with the first oxygen atom (O1) of <code>hoo</code> and makes a dihedral angle with the first hydrogen atom (H1) of <code>hooh</code>.</p>
<p>Some further points to note are:</p>
<ul>
<li><p>Atoms can be declared by either atomic number or symbol.</p>
<ul>
<li><p>If declared by atomic number, connectivity needs to be indicated by <em>Z</em>-matrix line number.</p></li>
<li><p>If declared by atomic symbol either number similar atoms (<em>e.g.</em>, H1, H2, O1, O2 <em>etc.</em>) and refer connectivity using this symbol, or indicate connectivity by the line number of the referred atom.</p></li>
</ul></li>
<li><p>Bond lengths and angles can be entered either as variables/parameters or real numbers.</p>
<ul>
<li><p>Variables/parameters can be declared in any order.</p></li>
<li><p>A single blank line separates parameters from the <em>Z</em>-matrix.</p></li>
</ul></li>
</ul>
<p>All the following examples are equivalent in the information forwarded to the <span><span class="smallcaps">Q-Chem</span></span> program.</p>

<p><strong>Example .</strong>  Using parameters to define bond lengths and angles, and using numbered symbols to define atoms and indicate connectivity.</p>
<blockquote>
<pre><code>$molecule
   0 1
   O1
   O2  O1  oo
   H1  O1  ho  O2  hoo
   H2  O2  ho  O1  hoo  H1  hooh

   oo   =   1.5
   oh   =   1.0
   hoo  = 120.0
   hooh = 180.0
$end</code></pre>
</blockquote>

<p><strong>Example .</strong>  Not using parameters to define bond lengths and angles, and using numbered symbols to define atoms and indicate connectivity.</p>
<blockquote>
<pre><code>$molecule
   0 1
   O1
   O2  O1  1.5
   H1  O1  1.0  O2  120.0
   H2  O2  1.0  O1  120.0  H1  180.0
$end</code></pre>
</blockquote>

<p><strong>Example .</strong>  Using parameters to define bond lengths and angles, and referring to atom connectivities by line number.</p>
<blockquote>
<pre><code>$molecule
   0 1
   8
   8  1  oo
   1  1  ho  2  hoo
   1  2  ho  1  hoo  3  hooh

   oo   =   1.5
   oh   =   1.0
   hoo  = 120.0
   hooh = 180.0
$end</code></pre>
</blockquote>

<p><strong>Example .</strong>  Referring to atom connectivities by line number, and entering bond length and angles directly.</p>
<blockquote>
<pre><code>$molecule
  0 1
  8
  8  1  1.5
  1  1  1.0  2  120.0
  1  2  1.0  1  120.0  3  180.0
$end</code></pre>
</blockquote>
<p>Obviously, a number of the formats outlined above are less appealing to the eye and more difficult for us to interpret than the others, but each communicates <em>exactly</em> the same <em>Z</em>-matrix to the <span><span class="smallcaps">Q-Chem</span></span> program.</p>
<h4 id="dummy-atoms">Dummy Atoms</h4>
<p>Dummy atoms are indicated by the identifier <span class="math inline"><em>X</em></span> and followed, if necessary, by an integer. (<em>e.g.</em>, <span class="math inline"><em>X</em>1</span>, <span class="math inline"><em>X</em>2</span>. Dummy atoms are often useful for molecules where symmetry axes and planes are not centered on a real atom, and have also been useful in the past for choosing variables for structure optimization and introducing symmetry constraints.</p>
<p><strong>Note: </strong></p>
<p>Dummy atoms play no role in the quantum mechanical calculation, and are used merely for convenience in specifying other atomic positions or geometric variables.</p>
<h3 id="reading-molecular-coordinates-from-a-previous-job-or-file">Reading Molecular Coordinates from a Previous Job or File</h3>
<p>Often users wish to perform several calculations in sequence, where the later calculations rely on results obtained from the previous ones. For example, a geometry optimization at a low level of theory, followed by a vibrational analysis and then, perhaps, single-point energy at a higher level. Rather than having the user manually transfer the coordinates from the output of the optimization to the input file of a vibrational analysis or single point energy calculation, <span><span class="smallcaps">Q-Chem</span></span> can transfer them directly from job to job.</p>
<p>To achieve this requires that:</p>
<ul>
<li><p>The <em>READ</em> variable is entered into the molecular coordinate input</p></li>
<li><p>Scratch files from a previous calculation have been saved. These may be obtained <em>explicitly</em> by using the save option across multiple job runs as described below and in Chapter <a href="#chap:install" data-reference-type="ref" data-reference="chap:install">2</a>, or <em>implicitly</em> when running multiple calculations in one input file, as described in Section <a href="#sec:multiple_jobs" data-reference-type="ref" data-reference="sec:multiple_jobs">3.6</a>.</p></li>
</ul>

<p><strong>Example .</strong>  Reading a geometry from a prior calculation.</p>
<blockquote>
<pre><code>$molecule
   READ
$end</code></pre>
</blockquote>
<blockquote>
<pre><code>localhost-1&gt; qchem job1.in job1.out job1
localhost-2&gt; qchem job2.in job2.out job1</code></pre>
</blockquote>
<p>In this example, the <em>job1</em> scratch files are saved in a directory <em>$QCSCRATCH/job1</em> and are then made available to the <em>job2</em> calculation.</p>
<p><strong>Note: </strong></p>
<p>The program must be instructed to read specific scratch files by the input of <em>job2</em>.</p>
<p>The <em>READ</em> function can also be used to read molecular coordinates from a second input file. The format for the coordinates in the second file follows that for standard <span><span class="smallcaps">Q-Chem</span></span> input, and must be delimited with the <em>$molecule</em> and <em>$end</em> keywords.</p>

<p><strong>Example .</strong>  Reading molecular coordinates from another file. <em>filename</em> may be given either as the full file path, or path relative to the working directory.</p>
<blockquote>
<pre><code>$molecule
   READ filename
$end</code></pre>
</blockquote>
<h2 id="sect:rem">Job Specification: The <em>$rem</em> Input Section</h2>
<p>The <em>$rem</em> section in the input file is the means by which users specify the type of calculation that they wish to perform (<em>i.e.</em>, level of theory, basis set, convergence criteria, additional special features, <em>etc.</em>). The keyword <em>$rem</em> signals the beginning of the overall job specification. Within the <em>$rem</em> section the user inserts <span><strong><em>$rem</em> variables</strong></span> (one per line) which define the essential details of the calculation. The allowed format is either</p>
<pre><code>REM_VARIABLE     VALUE    [ comment ]</code></pre>
<p>or alternatively</p>
<pre><code>REM_VARIABLE   =  VALUE   [ comment ]</code></pre>
<p>The “=” sign is automatically discarded and only the first two remaining arguments are read, so that all remaining text is ignored and can be used to place comments in the input file. Thus the <em>$rem</em> section that provides <span><span class="smallcaps">Q-Chem</span></span> job control takes the form shown in the following example.</p>

<p><strong>Example .</strong>  General format of the <em>$rem</em> section of the text input file.</p>
<blockquote>
<pre><code>$rem
  REM_VARIABLE  value  [ comment ]
  REM_VARIABLE  value  [ comment ]
  ...
$end</code></pre>
</blockquote>
<p><strong>Note: </strong></p>
<p>(1) Tab stops can be used to format input.<br />
(2) A line prefixed with an exclamation mark ‘!’ is treated as a comment and will be ignored by the program.<br />
(3) <em>$rem</em> variables are case-insensitive (as is the whole<span><span class="smallcaps">Q-Chem</span></span> input file).<br />
(4) Depending on the particular <em>$rem</em> variable, “<span>value</span>” may be a keyword (string), an integer, or a logical value (true or false).<br />
(5) A complete list of <em>$rem</em> variables can be found in Appendix <a href="#app:quick_ref" data-reference-type="ref" data-reference="app:quick_ref">16</a>.<br />
</p>
<p>In this manual, <em>$rem</em> variables will be described using the following format:</p>
<blockquote>

</blockquote>
<p>If a default setting is indicated for a particular <em>$rem</em> variable, then it is not necessary to declare that variable in order for the default setting to be used. For example, the default value for the variable is , indicating a single-point energy calculation, so to perform such a calculation the user does <strong>not</strong> need to set the variable. To perform a geometry optimization, however, it is necessary to override this default by setting = . System administrator preferences for default <em>$rem</em> settings can be specified in the <em>$QC/config/preferences</em> file, and user preferences in a <em>$HOME/.qchemrc</em> file, both of which are described in Section <a href="#sect:customization" data-reference-type="ref" data-reference="sect:customization">2.6</a>.</p>
<p><span><span class="smallcaps">Q-Chem</span></span> provides defaults for most <em>$rem</em> variables, but the user will always have to stipulate a few others. In a single point energy calculation, for example, the minimum requirements will be (defining the basis set) and (defining the level of theory for correlation and exchange). For example, = invokes a Hartree-Fock calculation, whereas = specifies a CIS excited-state calculation.</p>

<p><strong>Example .</strong>  Example of minimal <em>$rem</em> requirements to run an MP2/6-31G* single-point energy calculation.</p>
<blockquote>
<pre><code>$rem
   BASIS         6-31G*   Just a small basis set
   METHOD        mp2      MP2 
$end</code></pre>
</blockquote>
<p>The level of theory can alternatively be specified by setting values for two other <em>$rem</em> variables, (defining the level of theory to treat exchange) and (defining the level of theory to treat electron correlation, if required). For excited states computed using equation-of-motion (EOM) methods (Chapter <a href="#chap:excited" data-reference-type="ref" data-reference="chap:excited">7</a>), there is a third <em>$rem</em> variable, , which specifies the level of correlation for the target states.</p>
<p>For DFT calculations, specifies an exchange-correlation functional; see Section <a href="#sect:DFT_METHOD" data-reference-type="ref" data-reference="sect:DFT_METHOD">[sect:DFT_METHOD]</a> for a list of supported functionals. For wave function approaches, supported values of can be found in Section <a href="#sect:c5Intro" data-reference-type="ref" data-reference="sect:c5Intro">[sect:c5Intro]</a> for ground-state methods and in Section <a href="#sect:ExStatesSummary" data-reference-type="ref" data-reference="sect:ExStatesSummary">[sect:ExStatesSummary]</a> for excited-state methods. If a wave function-based correlation treatment such as MP2 or CC is requested using the keyword, then is taken as the default for .</p>
<h2 id="additional-input-sections">Additional Input Sections</h2>
<p>The <em>$molecule</em> and <em>$rem</em> sections are required for all <span><span class="smallcaps">Q-Chem</span></span> jobs, but depending on the details of the job a number of other input sections may be required. These are summarized briefly below, with references to more detailed descriptions to be found later in this manual.</p>
<h3 id="comments-comment">Comments (<em>$comment</em>)</h3>
<p>Users are able to add comments to the input file outside keyword input sections, which will be ignored by the program. This can be useful as reminders to the user, or perhaps, when teaching another user to set up inputs. Comments can also be provided in a <em>$comment</em> block, which is actually redundant given that the entire input deck is copied to the output file.</p>
<h3 id="user-defined-basis-sets-basis-and-aux_basis">User-Defined Basis Sets (<em>$basis</em> and <em>$aux_basis</em>)</h3>
<p>By setting the <em>$rem</em> keyword = , the user indicates that the basis set will be user defined. In that case, the <em>$basis</em> input section is used to specify the basis set. Similarly, if = then the <em>$aux_basis</em> input section is used to specify the auxiliary basis set. See Chapter <a href="#chap:basis" data-reference-type="ref" data-reference="chap:basis">8</a> for details on how to input a user-defined basis set.</p>
<h3 id="user-defined-effective-core-potential-ecp">User-Defined Effective Core Potential (<em>$ecp</em>)</h3>
<p>By setting = , the user indicates that the effective core potentials (pseudopotentials, which replace explicit core electrons) to be used will be defined by the user. In that case, the <em>$ecp</em> section is used to specify these pseudopotentials. See Chapter <a href="#chap:ecp" data-reference-type="ref" data-reference="chap:ecp">9</a> for further details.</p>
<h3 id="user-defined-exchange-correlation-density-functionals-xc_functional">User-Defined Exchange-Correlation Density Functionals<br />
(<em>$xc_functional</em>)</h3>
<p>If the keyword = then a DFT calculation will be performed using a user-specified combination of exchange and correlation functional(s), as described in Chapter <a href="#chap:scf" data-reference-type="ref" data-reference="chap:scf">4</a>. Custom functionals of this sort can be constructed as any linear combination of exchange and/or correlation functionals that are supported by <span><span class="smallcaps">Q-Chem</span></span>; see Section <a href="#sect:DFT_METHOD" data-reference-type="ref" data-reference="sect:DFT_METHOD">[sect:DFT_METHOD]</a> for a list of supported functionals. The format for the <em>$xc_functional</em> input section is the following:</p>

<blockquote>
<pre><code>$xc_functional
   X  exchange_symbol  coefficient
   X  exchange_symbol  coefficient
   ...
   C  correlation_symbol  coefficient
   C  correlation_symbol  coefficient
   ...
   K  coefficient
$end</code></pre>
</blockquote>
<p><strong>Note: </strong></p>
<p>The coefficients must be real numbers.</p>
<h3 id="user-defined-parameters-for-dft-dispersion-correction-empirical_dispersion">User-defined Parameters for DFT Dispersion Correction<br />
(<em>$empirical_dispersion</em>)</h3>
<p>If a user wants to change from the default values recommended by Grimme, then user-defined parameters can be specified using the <em>$empirical_dispersion</em> input section. See Section <a href="#subsec:DFT-D" data-reference-type="ref" data-reference="subsec:DFT-D">[subsec:DFT-D]</a> for details.</p>
<h3 id="addition-of-external-point-charges-external_charges">Addition of External Point Charges (<em>$external_charges</em>)</h3>
<p>If the <em>$external_charges</em> keyword is present, <span><span class="smallcaps">Q-Chem</span></span> scans for a set of external charges to be incorporated into a calculation. The format is shown below and consists of Cartesian coordinates and the value of the point charge, with one charge per line. The charge is in atomic units and the coordinates are in <span>Å</span>ngstroms, unless bohrs are selected by setting the <em>$rem</em> keyword to . The external charges are rotated with the molecule into the standard nuclear orientation.</p>

<p><strong>Example .</strong>  General format for incorporating a set of external charges.</p>
<blockquote>
<pre><code>$external_charges
   x-coord1  y-coord1  z-coord1  charge1
   x-coord2  y-coord2  z-coord2  charge2
   x-coord3  y-coord3  z-coord3  charge3
$end</code></pre>
</blockquote>
<p>In addition, the user can request to add a charged cage around the molecule (for so-called “charge stabilization” calculations) using the keyword . See Section <a href="#sec:CAGE" data-reference-type="ref" data-reference="sec:CAGE">[sec:CAGE]</a> for details.</p>
<h3 id="sec:multipolefield">Applying a Multipole Field (<em>$multipole_field</em>)</h3>
<p>A multipole field can be applied to the molecule under investigation by specifying the <em>$multipole_field</em> input section. Each line in this section consists of a single component of the applied field, in the following format.</p>

<p><strong>Example .</strong>  General format for imposing a multipole field.</p>
<blockquote>
<pre><code>$multipole_field
   field_component_1   value_1
   field_component_2   value_2
$end</code></pre>
</blockquote>
<p>Each <em>field<span>_</span>component</em> is stipulated using the Cartesian representation <em>e.g.</em>, X, Y, and/or Z, (dipole field components); XX, XY, and/or YY (quadrupole field components); XXX, XXY, <em>etc.</em>. The value (magnitude) of each field component should be provided in atomic units.</p>
<h3 id="user-defined-occupied-guess-orbitals-occupied-and-swap_occupied_virtual">User-Defined Occupied Guess Orbitals (<em>$occupied</em> and<br />
<em>$swap_occupied_virtual</em>)</h3>
<p>It is sometimes useful for the occupied guess orbitals to be different from the lowest <span class="math inline"><em>N</em><sub><em>α</em></sub></span> (or <span class="math inline"><em>N</em><sub><em>α</em></sub> + <em>N</em><sub><em>β</em></sub></span>) orbitals. <span><span class="smallcaps">Q-Chem</span></span> allows the occupied guess orbitals to be defined using the <em>$occupied</em> keyword. Using the <em>$occupied</em> input section, the user can choose which orbitals (by number) to occupy by specifying the <span class="math inline"><em>α</em></span>-spin orbitals on the first line of the <em>$occupied</em> section and the <span class="math inline"><em>β</em></span>-spin orbitals on the second line. For large molecules where only a few occupied <span class="math inline">→</span> virtual promotions are desired, it is simpler to use the <em>$swap_occupied_virtual</em> input section. Details can be found in Section <a href="#sect:occupied" data-reference-type="ref" data-reference="sect:occupied">[sect:occupied]</a>.</p>
<h3 id="polarizable-continuum-solvation-models-pcm">Polarizable Continuum Solvation Models (<em>$pcm</em>)</h3>
<p>The <em>$pcm</em> section provides fine-tuning of the job control for polarizable continuum models (PCMs), which are requested by setting the <em>$rem</em> keyword equal to . Supported PCMs include C-PCM, IEF-PCM, and SS(V)PE, which share a common set of job-control variables. Details are provided in Section <a href="#subsec:PCM" data-reference-type="ref" data-reference="subsec:PCM">[subsec:PCM]</a>.</p>
<h3 id="ssvpe-solvation-modeling-svp-and-svpirf">SS(V)PE Solvation Modeling (<em>$svp</em> and <em>$svpirf</em>)</h3>
<p>The <em>$svp</em> section is available to specify special parameters to the solvation module such as cavity grid parameters and modifications to the numerical integration procedure. The <em>$svpirf</em> section allows the user to specify an initial guess for the solution of the cavity charges. As discussed in section <a href="#subsec:SS(V)PE" data-reference-type="ref" data-reference="subsec:SS(V)PE">[subsec:SS(V)PE]</a>, the <em>$svp</em> and <em>$svpirf</em> input sections are used to specify parameters for the iso-density implementation of SS(V)PE. An alternative implementation of the SS(V)PE mode, based on a more empirical definition of the solute cavity, is available in the PCM (see Section <a href="#subsec:PCM" data-reference-type="ref" data-reference="subsec:PCM">[subsec:PCM]</a>) and controlled from within the <em>$pcm</em> input section.</p>
<h3 id="user-defined-van-der-waals-radii-van_der_waals">User-Defined van der Waals Radii (<em>$van_der_waals</em>)</h3>
<p>The <em>$van_der_waals</em> section of the input enables the user to customize the van der Waals radii that are important parameters in the Langevin dipoles solvation model; see Section <a href="#sect:solvent" data-reference-type="ref" data-reference="sect:solvent">[sect:solvent]</a>.</p>
<h3 id="effective-fragment-potential-calculations-efp_fragments-and-efp_params">Effective Fragment Potential Calculations (<em>$efp_fragments</em> and <em>$efp_params</em>)</h3>
<p>These keywords are used to specify positions and parameters for effective fragments in EFP calculations. Details are provided in Section <a href="#sect:efp" data-reference-type="ref" data-reference="sect:efp">[sect:efp]</a>.</p>
<h3 id="natural-bond-orbital-package-nbo">Natural Bond Orbital Package (<em>$nbo</em>)</h3>
<p>When is set to in the <em>$rem</em> section, a natural bond orbital (NBO) calculation is performed, using the <span><span class="smallcaps">Q-Chem</span></span> interface to the NBO 5.0 and NBO 6.0 packages. In such cases, the <em>$nbo</em> section may contain standard parameters and keywords for the NBO program.</p>
<h3 id="orbitals-densities-and-electrostatic-potentials-on-a-mesh-plots">Orbitals, Densities and Electrostatic Potentials on a Mesh (<em>$plots</em>)</h3>
<p>The <em>$plots</em> part of the input permits the evaluation of molecular orbitals, densities, electrostatic potentials, transition densities, electron attachment and detachment densities on a user-defined mesh of points. <span><span class="smallcaps">Q-Chem</span></span> will print out the raw data, but can also format these data into the form of a “cube” file that is a standard input format for volumetric data that can be read various visualization programs. See Section <a href="#sect:plots" data-reference-type="ref" data-reference="sect:plots">[sect:plots]</a> for details.</p>
<h3 id="intracules-intracule">Intracules (<em>$intracule</em>)</h3>
<p>Setting the <em>$rem</em> keyword = requests a molecular intracule calculation, in which case additional customization is possible using the <em>$intracule</em> input section. See Section <a href="#sect:intracules" data-reference-type="ref" data-reference="sect:intracules">[sect:intracules]</a>.</p>
<h3 id="geometry-optimization-with-constraints-opt">Geometry Optimization with Constraints (<em>$opt</em>)</h3>
<p>For = , <span><span class="smallcaps">Q-Chem</span></span> scans the input file for the <em>$opt</em> section. Here, the user may specify distance, angle, dihedral and out-of-plane bend constraints to be imposed on the optimization procedure, as described in Chapter <a href="#chap:opt" data-reference-type="ref" data-reference="chap:opt">10</a>.</p>
<h3 id="isotopic-substitutions-isotopes">Isotopic Substitutions (<em>$isotopes</em>)</h3>
<p>For vibrational frequency calculations ( = ), nuclear masses are set by default to be those corresponding to the most abundant naturally-occurring isotopes. Alternative masses for one or more nuclei can be requested by setting = in the <em>$rem</em> section, in which case the <em>$isotopes</em> section is used to specify the desired masses as described in Section <a href="#sect:isotopes" data-reference-type="ref" data-reference="sect:isotopes">[sect:isotopes]</a>. Isotopic substitutions incur negligible additional cost in a frequency calculation.</p>
<h2 id="sec:multiple_jobs">Multiple Jobs in a Single File: <span><span class="smallcaps">Q-Chem</span></span> Batch Jobs</h2>
<p>It is sometimes useful to place a sequence of jobs into a single <span><span class="smallcaps">Q-Chem</span></span> input file, where the individual inputs should be separated from one another by a line consisting of the string <code>@@@</code>. The output from these jobs is then appended sequentially to a single output file. This is useful to (a) use information obtained in a prior job (<em>i.e.</em>, an optimized geometry) in a subsequent job; or (b) keep related calculations together in a single output file.</p>
<p>Some limitations should be kept in mind:</p>
<ul>
<li><p>The first job will overwrite any existing output file of the same name in the working directory. Restarting the job will also overwrite any existing file.</p></li>
<li><p><span><span class="smallcaps">Q-Chem</span></span> reads all the jobs from the input file immediately and stores them. Therefore no changes can be made to the details of subsequent jobs following command-line initiation of <span><span class="smallcaps">Q-Chem</span></span>, even if these subsequent jobs have not yet run.</p></li>
<li><p>If any single job fails, <span><span class="smallcaps">Q-Chem</span></span> proceeds to the next job in the batch file, for good or ill.</p></li>
<li><p>No check is made to ensure that dependencies are satisfied, or that information is consistent. For example, in a geometry optimization followed by a frequency calculation, no attempt is made by the latter to check that the optimization was successful. When reading MO coefficients from a previous job, it is the user’s responsibility to ensure that the basis set is the same in both calculations, as this is assumed by the program.</p></li>
<li><p>Scratch files are saved from one job to the next in a batch job, so that information from previous jobs can be shared with subsequent ones, but are deleted upon completion of the entire batch job unless the <em>–save</em> command-line argument is supplied, as discussed in Chapter <a href="#chap:install" data-reference-type="ref" data-reference="chap:install">2</a>.</p></li>
</ul>
<p>The following example requests a batch job consisting of (<span class="math inline"><em>i</em></span>) a HF/6-31G* geometry optimization; followed by (<span class="math inline"><em>i</em><em>i</em></span>) a frequency calculation at the same level of theory that uses the previously-optimized geometry (and also reads in the final MOs from the optimization job); and finally (<span class="math inline"><em>i</em><em>i</em><em>i</em></span>) a single-point calculation at the same geometry but at a higher level of theory, MP2/6-311G(d,p).</p>

<p><strong>Example .</strong>  Example of using information from previous jobs in a single input file.</p>
<blockquote>
<pre><code>$comment
   Optimize H-H at HF/6-31G*
$end

$molecule
   0 1
   H
   H  1  r

   r = 1.1
$end

$rem
   JOBTYPE       opt    Optimize the bond length
   METHOD        hf
   BASIS         6-31G*
$end

@@@

$comment
   Now calculate the frequency of H-H at the same level of theory.
$end

$molecule
   read
$end

$rem
   JOBTYPE       freq    Calculate vibrational frequency
   METHOD        hf
   BASIS         6-31G*
   SCF_GUESS     read    Read the MOs from disk
$end

@@@

$comment
  Now a single point calculation at at MP2/6-311G(d,p)//HF/6-31G*
$end

$molecule
   read
$end

$rem
   METHOD        mp2
   BASIS         6-311G(d,p)
$end</code></pre>
</blockquote>
<h2 id="q-chem-output-file"><span><span class="smallcaps">Q-Chem</span></span> Output File</h2>
<p>When <span><span class="smallcaps">Q-Chem</span></span> is invoked using</p>
<blockquote>
<pre><code>	qchem infile outfile</code></pre>
</blockquote>
<p>the output file <em>outfile</em> contains a variety of information, depending on the type of job(s), but in general consists of the following.</p>
<ul>
<li><p><span><span class="smallcaps">Q-Chem</span></span> citation</p></li>
<li><p>User input (for record-keeping purposes)</p></li>
<li><p>Molecular geometry in Cartesian coordinates</p></li>
<li><p>Molecular point group, nuclear repulsion energy, number of <span class="math inline"><em>α</em></span>- and <span class="math inline"><em>β</em></span>-spin electrons</p></li>
<li><p>Basis set information (number of functions, shells and function pairs)</p></li>
<li><p>SCF details (method, guess, and convergence procedure)</p></li>
<li><p>Energy and DIIS error for each SCF iteration</p></li>
<li><p>Results of any post-SCF calculation that is requested</p></li>
<li><p>Results of any excited-state calculation that is requested</p></li>
<li><p>Molecular orbital symmetries and energies</p></li>
<li><p>Wave function analysis</p></li>
<li><p>Message signaling successful job completion</p></li>
</ul>
<p><strong>Note: </strong></p>
<p>If <em>outfile</em> above already exists when the job is started, then the existing file is overwritten with the results of the new calculation.</p>
<h1 id="chap:scf">Self-Consistent Field Ground-State Methods</h1>







<h1 id="chap:DFT">Density Functional Theory</h1>
<h1 id="chap:correlation">Wave Function-Based Correlation Methods</h1>







<h1 id="chap:excited">Open-Shell and Excited-State Methods</h1>







<h1 id="chap:basis">Basis Sets</h1>







<h1 id="chap:ecp">Effective Core Potentials</h1>







<h1 id="chap:opt">Exploring Potential Energy Surfaces: Searches for Critical Points and Molecular Dynamics</h1>







<h1 id="chap:properties">Molecular Properties and Analysis</h1>
<h1 id="molecules-in-complex-environments-solvent-models-qmmm-and-qmefp-features-density-embedding">Molecules in Complex Environments: Solvent Models, QM/MM and QM/EFP Features, Density Embedding</h1>

<p><span id="chap:env" label="chap:env">[chap:env]</span></p>







<h1 id="chap:frag">Fragment-Based Methods</h1>
<h2 id="introduction">Introduction</h2>
<p>Molecular complexes and molecular clusters represent a broad class of systems with interesting chemical and physical properties. Such systems can be naturally partitioned into fragments each representing a molecule or several molecules. <span><span class="smallcaps">Q-Chem</span></span> contains a set of methods designed to use such partitioning either for physical or computational advantage. Some of these methods (<em>e.g.</em> the ALMO-EDA method and its most recent updates/extensions) were developed and implemented by Dr. Rustam Z. Khaliullin, Dr. Paul R. Horn, Yuezhi Mao, Dr. Jonathan Thirman, Dr. Daniel S. Levine, and Qinghui Ge working with Prof. Martin Head-Gordon at the University of California–Berkeley. Other methods [<em>e.g.</em>, the XSAPT family of methods and TDDFT(MI)] were developed by Drs. Leif Jacobson, Ka Un Lao, and Jie Liu working with Prof. John Herbert at Ohio State University.</p>
<p>The list of methods that use partitioning includes:</p>
<ul>
<li><p>Initial guess at the MOs as a superposition of the converged MOs on the isolated fragments (FRAGMO guess).<span class="citation" data-cites="Khaliullin:2006"></span></p></li>
<li><p>Constrained (locally-projected) SCF methods for molecular interactions (SCF MI methods) between both closed-shell<span class="citation" data-cites="Khaliullin:2006"></span> and open-shell<span class="citation" data-cites="Horn:2013"></span> fragments.</p></li>
<li><p>Single Roothaan-step (RS) correction methods that improve FRAGMO and SCF MI description of molecular systems.<span class="citation" data-cites="Khaliullin:2006 Horn:2013"></span></p></li>
<li><p>Automated calculation of the BSSE with counterpoise correction method (full SCF and RS implementation).</p></li>
<li><p>The original version the ALMO-EDA method (energy decomposition analysis based on absolutely localized molecular orbitals), including the associated charge transfer analysis,<span class="citation" data-cites="Khaliullin:2007 Khaliullin:2008 Horn:2013"></span> and the analysis of intermolecular bonding in terms of complementary occupied-virtual pairs (COVPs).<span class="citation" data-cites="Khaliullin:2008 Khaliullin:2009 Horn:2013"></span></p></li>
<li><p>An improved version (“second generation”) of the ALMO-EDA method,<span class="citation" data-cites="Horn:2015 Horn:2016a Horn:2016b Horn:2016c"></span>, including its extension to single-bond interactions.<span class="citation" data-cites="Levine:2016"></span></p></li>
<li><p>An extension of the ALMO-EDA to MP2.<span class="citation" data-cites="Thirman:2015 Thirman:2017"></span></p></li>
<li><p>The adiabatic ALMO-EDA method that bridges intermolecular interactions and molecular properties.<span class="citation" data-cites="Mao:2017"></span></p></li>
<li><p>The variational explicit polarization (XPol) method, a self-consistent, charge-embedded, monomer-based SCF calculation.<span class="citation" data-cites="Xie:2008 Jacobson:2011 Herbert:2012"></span></p></li>
<li><p>Symmetry-adapted perturbation theory (SAPT), a monomer-based method for computing intermolecular interaction energies and decomposing them into physically-meaningful components.<span class="citation" data-cites="Jeziorski:1994 Szalewicz:2012"></span></p></li>
<li><p>XPol+SAPT (XSAPT), which extends the SAPT methodology to systems consisting of more than two monomers.<span class="citation" data-cites="Jacobson:2011 Herbert:2012 Jacobson:2013"></span></p></li>
<li><p>Closed- and open-shell AO-XSAPT(KS)+D, a dispersion-corrected version of XSAPT in atomic orbital basis that affords accurate intermolecular interaction energies at very low cost.<span class="citation" data-cites="Lao:2012b Lao:2013 Lao:2015"></span></p></li>
<li><p>A stable and physically-motivated energy decomposition approach, SAPT/cDFT, in which cDFT is used to define the charge-transfer component of the interaction energy and SAPT defines the electrostatic, polarization, Pauli repulsion, and van der Waals contributions.<span class="citation" data-cites="Lao:2016b"></span></p></li>
<li><p>The electrostatically-embedded many-body expansion<span class="citation" data-cites="Dahlke:2007 Richard:2014a Richard:2014b Lao:2016a"></span> and the fragment molecular orbital method,<span class="citation" data-cites="Kitaura:1999 Fedorov:2007"></span> for decomposing large clusters into small numbers of monomers, facilitating larger calculations.</p></li>
<li><p>The <em>Ab Initio</em> Frenkel Davydov Model,<span class="citation" data-cites="Morrison:2014 Morrison:2015"></span> a low-order scaling, highly parallelizable approach to computing excited state properties of liquids, crystals, and aggregates.</p></li>
<li><p>TDDFT for molecular interactions [TDDFT(MI)], an excited-state extension of SCF MI that offers a reduced-cost way to compute excited states in molecular clusters, crystals, and aggregates.<span class="citation" data-cites="Liu:2015 Liu:2016 Herbert:2016b"></span></p></li>
</ul>
<p>Another fragment-based approach, the Effective Fragment Potential (EFP) method,<span class="citation" data-cites="Ghosh:2010"></span> was developed by Prof. Lyudmila Slipchenko at Purdue University and Prof. Anna Krylov at USC; this method is described in Section <a href="#sect:efp" data-reference-type="ref" data-reference="sect:efp">[sect:efp]</a>.</p>
<h2 id="specifying-fragments-in-the-molecule-section">Specifying Fragments in the <em>$molecule</em> Section</h2>
<p>To request any of the methods mentioned above one must specify how system is partitioned into fragments. All atoms and all electrons in the systems should be assigned to a fragment. Each fragment must contain an integer number of electrons. In the current implementation, both open and closed-shell fragments are allowed. In order to specify fragments, the fragment descriptors must be inserted into the <em>$molecule</em> section of the <span><span class="smallcaps">Q-Chem</span></span> input file. A fragment descriptor consists of two lines: the first line must start with two hyphens followed by optional comments, the second line must contain the charge and the multiplicity of the fragment. At least two fragments must be specified. Fragment descriptors in the <em>$molecule</em> section does not affect jobs that are not designed to use fragmentation.</p>

<p><strong>Example .</strong>  Fragment descriptors in the <em>$molecule</em> section.</p>
<blockquote>
<pre><code>$molecule
   0 1
-- water molecule - proton donor
   0 1
   O1
   H2 O1 0.96
   H3 O1 0.96 H2 105.4
-- water molecule - proton acceptor
   0 1
   O4 O1 ROO  H2 105.4 H3   0.0
   X5 O4 2.00 O1 120.0 H2 180.0
   H6 O4 0.96 X5  55.6 O1  90.0
   H7 O4 0.96 X5  55.6 01 -90.0

   ROO = 2.4
$end</code></pre>
</blockquote>
<p>Open shell systems must have a number of alpha electrons greater than the number of beta electrons. However, individual fragments in the system can be made to contain excess beta electrons by specifying a negative multiplicity. For instance, a multiplicity of <span class="math inline"> − 2</span> indicates one excess beta electron, as in the second fragment of the following example.</p>

<p><strong>Example .</strong>  <span id="example:osfrag" label="example:osfrag">[example:osfrag]</span> Open shell fragment descriptors in the <em>$molecule</em> section. The second fragment is made with a negative multiplicity, so that overall the number of alpha and beta electrons match, yielding an approximate singlet state.</p>
<blockquote>
<pre><code>$molecule
   0 1
--  An alpha spin H atom
   0 2
   H1
--  A beta spin H atom
   0 -2
   H2 H1  1.50
$end</code></pre>
</blockquote>
<h2 id="fragmo-initial-guess-for-scf-methods">FRAGMO Initial Guess for SCF Methods</h2>
<p><span id="sect:FRAGMOguess" label="sect:FRAGMOguess">[sect:FRAGMOguess]</span></p>
<p>An accurate initial guess can be generated for molecular systems by superimposing converged molecular orbitals on isolated fragments. This initial guess is requested by specifying option for keyword and can be used for both the conventional SCF methods and the locally-projected SCF methods. The number of SCF iterations can be greatly reduced when is used instead of . This can lead to significant time savings for jobs on multi-fragment systems with large basis sets.<span class="citation" data-cites="Khaliullin:2007"></span> Unlike the guess, the guess is idempotent.</p>
<p>To converge molecular orbitals on isolated fragments, a child <span><span class="smallcaps">Q-Chem</span></span> job is executed for each fragment. <em>$rem</em> variables of the child jobs are inherited from the <em>$rem</em> section of the parent job. If is set to the output of the child jobs is redirected to the output file of the parent job. Otherwise, the output is suppressed.</p>
<p>Additional keywords that control child <span><span class="smallcaps">Q-Chem</span></span> processes can be set in the <em>$rem_frgm</em> section of the parent input file. This section has the same structure as the <em>$rem</em> section. Options in the <em>$rem_frgm</em> section override options of the parent job. <em>$rem_frgm</em> is intended to specify keywords that control the SCF routine on isolated fragments. Please be careful with the keywords in <em>$rem_frgm</em> section. <em>$rem</em> variables , , , , , are not allowed in <em>$rem_frgm</em> and will be ignored. <em>$rem</em> variables , , , and are not inherited from the parent job.</p>

<p><strong>Example .</strong>  guess can be used with the conventional SCF calculations. <em>$rem_frgm</em> keywords in this example specify that the SCF on isolated fragments does not have to be converged tightly. See also Example <a href="#chap:env" data-reference-type="ref" data-reference="chap:env">[chap:env]</a>.<a href="#example:osfrag" data-reference-type="ref" data-reference="example:osfrag">[example:osfrag]</a> for an open-shell fragment example.</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
O          -0.106357    0.087598    0.127176
H           0.851108    0.072355    0.136719
H          -0.337031    1.005310    0.106947
--
0 1
O           2.701100   -0.077292   -0.273980
H           3.278147   -0.563291    0.297560
H           2.693451   -0.568936   -1.095771
--
0 1
O           2.271787   -1.668771   -2.587410
H           1.328156   -1.800266   -2.490761
H           2.384794   -1.339543   -3.467573
--
0 1
O          -0.518887   -1.685783   -2.053795
H          -0.969013   -2.442055   -1.705471
H          -0.524180   -1.044938   -1.342263
$end

$rem
   METHOD          EDF1
   BASIS           6-31(2+,2+)g(df,pd)
   SCF_GUESS       FRAGMO
   SCF_PRINT_FRGM  FALSE
$end

$rem_frgm
   SCF_CONVERGENCE 2
   THRESH          5
$end</code></pre>
</blockquote>
<p>The use of guess is also supported when = . It is extended to support more SCF orbital types (R/U/RO/G). Meanwhile, users are allowed to read in the previously generated FRAGMO guess instead of recalculating them if there is no difference between these jobs on the fragment level. This can be particularly useful for scenarios such as scanning a potential energy curve for an intermolecular complex, or for restarting an EDA job. This is controlled by the <em>$rem</em> variable .</p>
<blockquote>

</blockquote>

<p><strong>Example .</strong>  guess for ROSCF calculation in GEN_SCFMAN. The first fragment is RO and the second fragment is close-shell, while the super-system is computed with RO as well. The complex in the second job has a modified interfragment distance so it can make use of the guess generated by the first job. Note that = is needed to treat the fragments and the super-system consistently.</p>
<blockquote>
<pre><code>$molecule
1 2
--
0 2
C1 
H1 C1 1.09267
H2 C1 1.09267 H1 107.60335
H3 C1 1.09267 H2 107.60335 H1 115.692
--
1 1
Na C1 scan    H3 111.28008 H2 -122.154

scan = 2.0 
$end

$rem
   METHOD           b3lyp
   BASIS            6-31g
   GEN_SCFMAN       true
   UNRESTRICTED     false
   SCF_CONVERGENCE  8
   ROSCF            true
   SCF_GUESS        fragmo
   THRESH           14
   SYMMETRY         false
   SYM_IGNORE       true
$end

@@@

$molecule
1 2
--
0 2
C1
H1 C1 1.09267
H2 C1 1.09267 H1 107.60335
H3 C1 1.09267 H2 107.60335 H1 115.692
--
1 1
Na C1 scan    H3 111.28008 H2 -122.154

scan = 3.0
$end

$rem
   METHOD             b3lyp
   BASIS              6-31g
   GEN_SCFMAN         true
   UNRESTRICTED       false
   SCF_CONVERGENCE    8
   ROSCF              true
   SCF_GUESS          fragmo
   FRAGMO_GUESS_MODE  2 !read in the available fragment data
   THRESH             14
   SYMMETRY           false
   SYM_IGNORE         true
$end</code></pre>
</blockquote>
<h2 id="locally-projected-scf-methods">Locally-Projected SCF Methods</h2>
<p>Constrained locally-projected SCF is an efficient method for removing the SCF diagonalization bottleneck in calculations for systems of weakly interacting components such as molecular clusters and molecular complexes.<span class="citation" data-cites="Khaliullin:2006 Horn:2013"></span> The method is based on the equations of the locally-projected SCF for molecular interactions (SCF MI).<span class="citation" data-cites="Stoll:1980 Gianinetti:1996 Nagata:2001 Khaliullin:2006 Horn:2013"></span> In the SCF MI method, the occupied molecular orbitals on a fragment can be expanded only in terms of the atomic orbitals of the same fragment. Such constraints produce non-orthogonal MOs that are localized on fragments and are called absolutely-localized molecular orbitals (ALMOs). The ALMO approximation excludes charge-transfer from one fragment to another. It also prevents electrons on one fragment from borrowing the atomic orbitals of other fragments to compensate for incompleteness of their own AOs and, therefore, removes the BSSE from the interfragment binding energies. The locally-projected SCF methods perform an iterative minimization of the SCF energy with respect to the ALMOs coefficients. The convergence of the algorithm is accelerated with the locally-projected modification of the DIIS extrapolation method.<span class="citation" data-cites="Khaliullin:2006"></span></p>
<p>The ALMO approximation significantly reduces the number of variational degrees of freedom of the wave function. The computational advantage of the locally-projected SCF methods over the conventional SCF method grows with both basis set size and number of fragments. Although still cubic scaling, SCF MI effectively removes the diagonalization step as a bottleneck in these calculations, because it contains such a small prefactor. In the current implementation, the SCF MI methods do not speed up the evaluation of the Fock matrix and, therefore, do not perform significantly better than the conventional SCF in the calculations dominated by the Fock build.</p>
<p>Two locally-projected schemes are implemented. One is based on the locally-projected equations of Stoll <em>et al.</em>,<span class="citation" data-cites="Stoll:1980"></span> the other uses the locally-projected equations of Gianinetti <em>et al.</em>.<span class="citation" data-cites="Gianinetti:1996"></span> These methods have comparable performance. The Stoll iteration is only slightly faster than the Gianinetti iteration but the Stoll equations might be a little bit harder to converge. The Stoll equations also produce ALMOs that are orthogonal within a fragment. The type of the locally-projected SCF calculations is requested by specifying either or for the keyword.</p>

<p><strong>Example .</strong>  Locally-projected SCF method of Stoll</p>
<blockquote>
<pre><code>$molecule
0 1
--
-1 1
B           0.068635    0.164710    0.123580
F          -1.197609    0.568437   -0.412655
F           0.139421   -1.260255   -0.022586
F           1.118151    0.800969   -0.486494
F           0.017532    0.431309    1.531508
--
+1 1
N          -2.132381   -1.230625    1.436633
H          -1.523820   -1.918931    0.977471
H          -2.381590   -0.543695    0.713005
H          -1.541511   -0.726505    2.109346
H          -2.948798   -1.657993    1.873482
$end

$rem
   METHOD          BP86
   BASIS           6-31(+,+)G(d,p)
   FRGM_METHOD     STOLL
$end

$rem_frgm
   SCF_CONVERGENCE 2
   THRESH          5
$end</code></pre>
</blockquote>
<h3 id="locally-projected-scf-methods-with-single-roothaan-step-correction">Locally-Projected SCF Methods with Single Roothaan-Step Correction</h3>
<p>Locally-projected SCF cannot quantitatively reproduce the full SCF intermolecular interaction energies for systems with significant charge-transfer between the fragments (<em>e.g.</em>, hydrogen bonding energies in water clusters). Good accuracy in the intermolecular binding energies can be achieved if the locally-projected SCF MI iteration scheme is combined with a charge-transfer perturbative correction.<span class="citation" data-cites="Khaliullin:2006"></span> To account for charge-transfer, one diagonalization of the full Fock matrix is performed after the locally-projected SCF equations are converged and the final energy is calculated as infinite-order perturbative correction to the locally-projected SCF energy. This procedure is known as single Roothaan-step (RS) correction.<span class="citation" data-cites="Khaliullin:2006 Liang:2004a Liang:2004b"></span> It is performed if is set to . To speed up evaluation of the charge-transfer correction, second-order perturbative correction to the energy can be evaluated by solving the linearized single-excitation amplitude equations. This algorithm is called the approximate Roothaan-step correction and can be requested by setting to .</p>
<p>Both and corrected energies are very close to the full SCF energy for systems of weakly interacting fragments but are less computationally expensive than the full SCF calculations. To test the accuracy of the ARS and RS methods, the full SCF calculation can be done in the same job with the perturbative correction by setting to or to . It is also possible to evaluate only the full SCF correction by setting to .</p>
<p>The iterative solution of the linear single-excitation amplitude equations in the method is controlled by a set of keywords described below.</p>
<p><em>Restrictions.</em> Only single point HF and DFT energies can be evaluated with the locally-projected methods. Geometry optimization can be performed using numerical gradients. Wave function correlation methods (MP2, CC, <em>etc.</em>.) are not implemented for the absolutely-localized molecular orbitals. cannot be set to anything but , however, all SCF convergence algorithms can be used on isolated fragments (set in the <em>$rem_frgm</em> section).</p>

<p><strong>Example .</strong>  Comparison between the RS corrected energies and the conventional SCF energies can be made by calculating both energies in a single run.</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
O      -1.56875     0.11876     0.00000
H      -1.90909    -0.78106     0.00000
H      -0.60363     0.02937     0.00000
--
0 1
O       1.33393    -0.05433     0.00000
H       1.77383     0.32710    -0.76814
H       1.77383     0.32710     0.76814
$end

$rem
   METHOD          HF
   BASIS           AUG-CC-PVTZ
   FRGM_METHOD     GIA
   FRGM_LPCORR     RS_EXACT_SCF
$end

$rem_frgm
   SCF_CONVERGENCE 2
   THRESH          5
$end</code></pre>
</blockquote>
<h3 id="roothaan-step-corrections-to-the-initial-guess">Roothaan-Step Corrections to the Initial Guess</h3>
<p>For some systems good accuracy for the intermolecular interaction energies can be achieved without converging SCF MI calculations and applying either the RS or ARS charge-transfer correction directly to the initial guess. Set to or to request the single Roothaan correction or approximate Roothaan correction, respectively. To get a somewhat better energy estimate set to and . In the case of , the same steps as in the method are performed followed by one more Fock build and calculation of the proper SCF energy. In the case of the double Roothaan-step correction, , the same steps as in are performed followed by one more diagonalization. The final energy in the method is evaluated as a perturbative correction, similar to the single Roothaan-step correction.</p>
<p>Charge-transfer corrections applied directly to the guess are included in <span><span class="smallcaps">Q-Chem</span></span> to test accuracy and performance of the locally-projected SCF methods. However, for some systems they give a reasonable estimate of the binding energies at a cost of one (or two) SCF step(s).</p>
<h3 id="sect:ALMO-BSSE">Automated Evaluation of the Basis-Set Superposition Error</h3>
<p>Evaluation of the basis-set superposition error (BSSE) is automated in <span><span class="smallcaps">Q-Chem</span></span>. To calculate BSSE-corrected binding energies, specify fragments in the <em>$molecule</em> section and set to . The BSSE jobs are not limited to the SCF energies and can be evaluated for multi-fragment systems at any level of theory. <span><span class="smallcaps">Q-Chem</span></span> separates the system into fragments as specified in the <em>$molecule</em> section and performs a series of jobs on (a) each fragment, (b) each fragment with the remaining atoms in the system replaced by the ghost atoms, and (c) on the entire system. <span><span class="smallcaps">Q-Chem</span></span> saves all calculated energies and prints out the uncorrected and the BSSE corrected binding energies. The <em>$rem_frgm</em> section can be used to control calculations on fragments, however, make sure that the fragments and the entire system are treated equally. It means that all numerical methods and convergence thresholds that affect the final energies (such as , , , ) should be the same for the fragments and for the entire system. Avoid using <em>$rem_frgm</em> in the BSSE jobs unless absolutely necessary.</p>
<p><em>Important</em>. It is recommended to include keyword in all BSSE jobs. basis cannot be used for the BSSE calculations in the current implementation. Use basis instead.</p>

<p><strong>Example .</strong>  Evaluation of the BSSE corrected intermolecular interaction energy</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
O          -0.089523    0.063946    0.086866         
H           0.864783    0.058339    0.103755       
H          -0.329829    0.979459    0.078369       
--
0 1
O           2.632273   -0.313504   -0.750376
H           3.268182   -0.937310   -0.431464
H           2.184198   -0.753305   -1.469059
--
0 1
O           0.475471   -1.428200   -2.307836
H          -0.011373   -0.970411   -1.626285       
H           0.151826   -2.317118   -2.289289
$end

$rem
   JOBTYPE         BSSE
   METHOD          MP2
   BASIS           6-31(+,+)G(d,p)
   GEN_SCFMAN      FALSE
$end</code></pre>
</blockquote>
<h2 id="sect:EDA-CTA">The First-Generation ALMO-EDA and Charge-Transfer Analysis (CTA)</h2>
<h3 id="energy-decomposition-analysis-based-on-absolutely-localized-molecular-orbitals">Energy Decomposition Analysis Based on Absolutely Localized Molecular Orbitals</h3>
<p>The strength of intermolecular binding is inextricably connected to the fundamental nature of interactions between the molecules. Intermolecular complexes can be stabilized through weak dispersive forces, electrostatic effects (<em>e.g.</em>, charge–charge, charge–dipole, and charge–induced dipole interactions) and donor-acceptor type orbital interactions such as forward and back-donation of electron density between the molecules. Depending on the extent of these interactions, the intermolecular binding could vary in strength from just several kJ/mol (van der Waals complexes) to several hundred kJ/mol (metal–ligand bonds in metal complexes). Understanding the contributions of various interaction modes enables one to tune the strength of the intermolecular binding to the ideal range by designing materials that promote desirable effects. One of the most powerful techniques that modern first principles electronic structure methods provide to study and analyze the nature of intermolecular interactions is the decomposition of the total molecular binding energy into the physically meaningful components such as dispersion, electrostatic, polarization, charge transfer, and geometry relaxation terms.</p>
<p>Energy decomposition analysis based on absolutely-localized molecular orbitals (ALMO-EDA) is implemented in <span><span class="smallcaps">Q-Chem</span></span>,<span class="citation" data-cites="Khaliullin:2007"></span> including the open shell generalization.<span class="citation" data-cites="Horn:2013"></span> In ALMO-EDA, the total intermolecular binding energy is decomposed into the “frozen density” component (FRZ), the polarization (POL) term, and the charge-transfer (CT) term. The “frozen density” term is defined as the energy change that corresponds to bringing infinitely separated fragments together without <em>any</em> relaxation of their MOs. The FRZ term is calculated as a difference between the guess energy and the sum of the converged SCF energies on isolated fragments. The polarization (POL) energy term is defined as the energy lowering due to the <em>intrafragment</em> relaxation of the frozen occupied MOs on the fragments. The POL term is calculated as a difference between the converged SCF MI energy and the guess energy. Finally, the charge-transfer (CT) energy term is due to further <em>interfragment</em> relaxation of the MOs. It is calculated as a difference between the fully converged SCF energy and the converged SCF MI energy.</p>
<p>The total charge-transfer term includes the energy lowering due to electron transfer from the occupied orbitals on one molecule (more precisely, occupied in the converged SCF MI state) to the virtual orbitals of another molecule as well as the further energy change caused by induction that accompanies such an occupied/virtual mixing. The energy lowering of the occupied-virtual electron transfer can be described with a single non-iterative Roothaan-step correction starting from the converged SCF MI solution. Most importantly, the mathematical form of the SCF MI(RS) energy expression allows one to decompose the occupied-virtual mixing term into bonding and back-bonding components for each pair of molecules in the complex. The remaining charge-transfer energy term (<em>i.e.</em>, the difference between SCF MI(RS) energy and the full SCF energy) includes all induction effects that accompany occupied-virtual charge transfer and is generally small. This last term is called higher order (HO) relaxation. Unlike the RS contribution, the higher order term cannot be divided naturally into forward and back-donation terms. The BSSE associated with each charge-transfer term (forward donation, back-bonding, and higher order effects) can be corrected individually.</p>
<p>To perform energy decomposition analysis, specify fragments in the <em>$molecule</em> section and set to . For a complete job, <span><span class="smallcaps">Q-Chem</span></span></p>
<ul>
<li><p>performs the SCF on isolated fragments (use the <em>$rem_frgm</em> section if convergence issues arise but make sure that keywords in this section do not affect the final energies of the fragments),</p></li>
<li><p>generates the guess to obtain the FRZ term,</p></li>
<li><p>converges the SCF MI equations to evaluate the POL term,</p></li>
<li><p>performs evaluation of the perturbative (RS or ARS) variational correction to calculate the forward donation and back-bonding components of the CT term for each pair of molecules in the system,</p></li>
<li><p>converges the full SCF procedure to evaluate the higher order relaxation component of the CT term.</p></li>
</ul>
<p>The keyword controls evaluation of the CT term in an EDA job. To evaluate all of the CT components mentioned above set this keyword to or . If the HO term in not important then the final step (<em>i.e.</em>, the SCF calculation) can be skipped by setting to or . If only the total CT term is required then set to .</p>
<p>ALMO charge transfer analysis (ALMO-CTA) is performed together with ALMO EDA.<span class="citation" data-cites="Khaliullin:2008"></span> The ALMO charge transfer scale, <span class="math inline"><em>Δ</em><em>Q</em></span>, provides a measure of the distortion of the electronic clouds upon formation of an intermolecular bond and is such that all CT terms (<em>i.e.</em>, forward-donation, back-donation, and higher order relaxation) have well defined energetic effects (<em>i.e.</em>, ALMO-CTA is consistent with ALMO-EDA).</p>
<p>To remove the BSSE from the CT term (both on the energy and charge scales), set to . <span><span class="smallcaps">Q-Chem</span></span> generates an input file for each fragment with basis set to perform the BSSE correction. As with all jobs with basis set and d or higher angular momentum basis functions on atoms, the keyword needs to be initiated. If = basis sets cannot be used in the current implementation.</p>
<p>Please note that the energy of the geometric distortion of the fragments is not included into the total binding energy calculated in an EDA job. The geometry optimization of isolated fragments must be performed to account for this term.</p>

<p><strong>Example .</strong>  Energy decomposition analysis of the binding energy between the water molecules in a tetramer. ALMO-CTA results are also printed out.</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
O          -0.106357    0.087598    0.127176
H           0.851108    0.072355    0.136719
H          -0.337031    1.005310    0.106947
--
0 1
O           2.701100   -0.077292   -0.273980
H           3.278147   -0.563291    0.297560
H           2.693451   -0.568936   -1.095771
--
0 1
O           2.271787   -1.668771   -2.587410
H           1.328156   -1.800266   -2.490761
H           2.384794   -1.339543   -3.467573
--
0 1
O          -0.518887   -1.685783   -2.053795
H          -0.969013   -2.442055   -1.705471
H          -0.524180   -1.044938   -1.342263
$end

$rem
   JOBTYPE         EDA
   METHOD          EDF1
   BASIS           6-31(+,+)g(d,p)
   PURECART        1112
   FRGM_METHOD     GIA
   FRGM_LPCORR     RS_EXACT_SCF
   EDA_BSSE        TRUE
$end</code></pre>
</blockquote>

<p><strong>Example .</strong>  An open shell EDA example of Na<span class="math inline"><em></em><sup>+</sup></span> interacting with the methyl radical.</p>
<blockquote>
<pre><code>$molecule
   1 2
--
   0 2
   C       -1.447596  -0.000023   0.000019
   H       -1.562749   0.330361  -1.023835
   H       -1.561982   0.721445   0.798205
   H       -1.561187  -1.052067   0.225866
--
   1 1
   Na       1.215591   0.000036  -0.000032
$end

$rem
   JOBTYPE              EDA
   METHOD               B3LYP
   BASIS                6-31G*
   UNRESTRICTED         TRUE
   SCF_GUESS            FRAGMO
   FRGM_METHOD          STOLL
   FRGM_LPCORR          RS_EXACT_SCF
   EDA_BSSE             TRUE
   DIIS_SEPARATE_ERRVEC 1
$end</code></pre>
</blockquote>
<h3 id="analysis-of-charge-transfer-based-on-complementary-occupiedvirtual-pairs">Analysis of Charge-Transfer Based on Complementary Occupied/Virtual Pairs</h3>
<p>In addition to quantifying the amount and energetics of intermolecular charge transfer, it is often useful to have a simple description of orbital interactions in intermolecular complexes. The polarized ALMOs obtained from the SCF MI procedure and used as a reference basis set in the decomposition analysis do not directly show which occupied-virtual orbital pairs are of most importance in forming intermolecular bonds. By performing rotations of the polarized ALMOs within a molecule, it is possible to find a “chemist’s basis set” that represents bonding between molecules in terms of just a few localized orbitals called complementary occupied-virtual pairs (COVPs). This orbital interaction model validates existing conceptual descriptions of intermolecular bonding. For example, in the modified ALMO basis, hydrogen bonding in water dimer is represented as an electron pair localized on an oxygen atom donating electrons to the O–H <span class="math inline"><em>σ</em></span>-antibonding orbital on the other molecule,<span class="citation" data-cites="Khaliullin:2009"></span> and the description of synergic bonding in metal complexes agrees well with simple Dewar-Chatt-Duncanson model.<span class="citation" data-cites="Cobar:2007 Khaliullin:2008 Lochan:2008"></span></p>
<p>Set to to perform the COVP analysis of the CT term in an job. COVP analysis is currently implemented only for systems of two fragments. Set to to print out localized orbitals that form occupied-virtual pairs. In this case, MOs obtained in the end of the run (SCF MI orbitals, SCF MI(RA) orbitals, converged SCF orbitals) are replaced by the orbitals of COVPs. Each orbital is printed with the corresponding CT energy term in kJ/mol (instead of the energy eigenvalues in hartrees). These energy labels make it easy to find correspondence between an occupied orbital on one molecule and the virtual orbital on the other molecule. The examples below show how to print COVP orbitals. One way is to set <em>$rem</em> variable , the other is to set to 200 and use the <em>$plots</em> section in the <span><span class="smallcaps">Q-Chem</span></span> input. In the first case the orbitals can be visualized using MOLDEN (set to ), in the second case use VMD or a similar third party program capable of making 3D plots.</p>

<p><strong>Example .</strong>  COVP analysis of the CT term. The COVP orbitals are printed in the<span><span class="smallcaps">Q-Chem</span></span> and MOLDEN formats.</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
O          -1.521720    0.129941    0.000000
H          -1.924536   -0.737533    0.000000
H          -0.571766   -0.039961    0.000000
--
0 1
O           1.362840   -0.099704    0.000000
H           1.727645    0.357101   -0.759281
H           1.727645    0.357101    0.759281
$end

$rem
   JOBTYPE         EDA
   BASIS           6-31G
   PURECART        1112
   METHOD          B3LYP
   FRGM_METHOD     GIA
   FRGM_LPCORR     RS_EXACT_SCF
   EDA_COVP        TRUE
   EDA_PRINT_COVP  TRUE
   PRINT_ORBITALS  16
   MOLDEN_FORMAT   TRUE
$end</code></pre>
</blockquote>


<p><strong>Example .</strong>  COVP analysis of the CT term. Note that it is not necessary to run a full EDA job. It is suffice to set to or and to to perform the COVP analysis. The orbitals of the most significant occupied-virtual pair are printed into an ASCII file called <em>plot.mo</em> which can be converted into a cube file and visualized in VMD.</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
O          -1.521720    0.129941    0.000000
H          -1.924536   -0.737533    0.000000
H          -0.571766   -0.039961    0.000000
--
0 1
O           1.362840   -0.099704    0.000000
H           1.727645    0.357101   -0.759281
H           1.727645    0.357101    0.759281
$end

$rem
   BASIS           6-31G
   PURECART        1112
   METHOD          B3LYP
   FRGM_METHOD     GIA
   FRGM_LPCORR     RS
   IANLTY          200
   EDA_COVP        TRUE
   EDA_PRINT_COVP  TRUE
$end

$plots
MOs
   80  -4.0  4.0
   60  -3.0  3.0
   60  -3.0  3.0
   2   0  0  0
   6  11
$end</code></pre>
</blockquote>
<h2 id="sect:old_almo_job">Job Control for Locally-Projected SCF Methods</h2>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<h2 id="the-second-generation-almo-eda-method">The Second-Generation ALMO-EDA Method</h2>
<p><span id="sect:eda2" label="sect:eda2">[sect:eda2]</span></p>
<p>The ALMO-EDA method introduced in Section <a href="#sect:EDA-CTA" data-reference-type="ref" data-reference="sect:EDA-CTA">13.5</a> is a very useful tool for unraveling the nature of intermolecular interactions. Nevertheless, it has two major shortcomings: (i) Although the polarization (POL) energy is variationally evaluated, it does not have a meaningful basis set limit. As the employed basis set becomes larger, the POL term starts to be contaminated by charge transfer (CT) and loses its intended meaning. (ii) The frozen (FRZ) interaction is a monolithic term in the original ALMO-EDA scheme. In practice, further decomposition of the FRZ term is often desired. For example, if one wants to use ALMO-EDA as a tool for the development of empirical force fields, the separation of the FRZ term into contributions from permanent electrostatics, Pauli repulsion and dispersion will be helpful since they are usually modeled by distinct functional forms in classical force fields. These drawbacks have been addressed recently, defining the second generation of the ALMO-EDA method (also referred to as “EDA2&quot; in the text below).<span class="citation" data-cites="Horn:2015 Horn:2016b Horn:2016c"></span></p>
<h3 id="sect:gen_scfmi">Generalized SCFMI Calculations and Additional Features</h3>
<p>The original definition of the ALMOs used in SCFMI calculations is based on the fragment-blocking structure of the AO-to-MO transformation matrix, <em>i.e.</em>, for a given fragment, the associated MOs can only be expanded by the AO basis functions centered on the atoms that belong to the same fragment. Here we propose a generalized definition for SCFMI calculations: given a set of basis vectors (<span class="math inline"><strong>G</strong></span>) in which each of them is tagged to a fragment but is allowed to be spanned by any AO basis function, it defines the working basis of the SCFMI problem. Then, within this basis, the locally projected SCF equations can be solved in a similar way, with the constraint that the MO coefficient matrix in the working basis (<span class="math inline"><strong>G</strong></span>) is fragment-block-diagonal, while the MO coefficient matrix in the AO basis does not necessarily retain the blocking structure. The basis vectors in <span class="math inline"><strong>G</strong></span> can be either non-orthogonal or orthogonal between fragments. More details on the generalized SCFMI equations are available in ref. </p>
<p><span class="citation" data-cites="Horn:2015"></span>.</p>
<p>This generalized SCFMI scheme is implemented in GEN_SCFMAN (the original AO-block based scheme is available in GEN_SCFMAN as well). It is used for the variational optimization of the polarized (but CT-forbidden) intermediate state in “EDA2&quot; (see Section <a href="#sect:ferf" data-reference-type="ref" data-reference="sect:ferf">13.7.2</a>). Another preferable feature of this generalized scheme is that the interfragment linear dependency in <span class="math inline"><strong>G</strong></span> can be properly handled. Therefore, this scheme can be used to replace the original AO-block based SCFMI without becoming ill-defined when interfragment linear dependency occurs. In contrast, the original ALMO-EDA method that employs the AO-block based approach fails when the sum of the number of orbitals on each fragment is not equal to the number of orbitals for the super-system (the latter is determined by the total number of AO basis functions and ), which often happens when substantially large basis sets are used or when the super-system comprises a large number of fragments.</p>
<p>SCFMI calculations based on the GEN_SCFMAN implementation are triggered by setting = and = or (the other options of are not allowed). A subset of supported algorithms in GEN_SCFMAN are available for restricted (R) and unrestricted (U) SCFMI, including DIIS, GDM_LS, and NEWTON_CG. While the DIIS algorithm iteratively solves for the locally-projected SCF equations, the latter two methods use the energy derivatives with respect to the on-fragment orbital rotations to minimize the energy until the gradient reaches zero. As for standard calculations using GEN_SCFMAN, internal stability analysis is also available for R- and U-SCFMI, and one can set to if the analytical Hessian is not available for the employed density functional.</p>
<p>As in the original implementation, perturbative corrections can be applied on top of the SCFMI solution to approach the full SCF result, and this is still controlled by . Note that among the options introduced in Section <a href="#sect:old_almo_job" data-reference-type="ref" data-reference="sect:old_almo_job">13.6</a>, only and are allowed here since the exact SCF calculation is actually beyond the scope of SCFMI.</p>
<p>In addition, with this more general implementation users are allowed to specify some fragments to be frozen during the SCFMI calculation, <em>i.e.</em>, intrafragment relaxation does not occur on these fragments. This is achieved by specifying the <em>$rem</em> variable . Such a calculation can be interpreted as an active fragment being embedded in a frozen environment where the interaction between them is treated quantum mechanically.</p>
<blockquote>

</blockquote>
<blockquote>

</blockquote>

<p><strong>Example .</strong>  Generalized SCFMI calculation for the water dimer with single Roothaan-step perturbative correction. For this specific case, the result is identical to that given by AO-block based SCFMI ( = 0).</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
O  -1.551007  -0.114520   0.000000
H  -1.934259   0.762503   0.000000
H  -0.599677   0.040712   0.000000
--
0 1
O   1.350625   0.111469   0.000000
H   1.680398  -0.373741  -0.758561
H   1.680398  -0.373741   0.758561
$end

$rem
   JOBTYPE          sp
   METHOD           b3lyp
   GEN_SCFMAN       true
   BASIS            6-31+G(d)
   GEN_SCFMAN       true
   SCF_CONVERGENCE  8
   THRESH           14
   SYMMETRY         false
   SYM_IGNORE       true
   FRGM_METHOD      stoll
   FRGM_LPCORR      rs
   SCFMI_MODE       1 !gen scfmi (non-orthogonal)
$end</code></pre>
</blockquote>
<h3 id="sect:ferf">Polarization Energy with a Well-defined Basis Set Limit</h3>
<p>The definition of polarization energy lowering in the original ALMO-EDA used the full AO space of each fragment as the variational degrees of freedom. This is based on the assumption that the AO basis functions are fragment-ascribable based on their atomic centers. However, this assumption becomes inappropriate when very large basis sets are used, especially those with diffuse functions (<em>e.g.</em> def2-QZVPPD). In such scenarios, basis functions on a given fragment tend to describe other fragments so that the “absolute localization&quot; constraint becomes weaker and finally gets effectively removed. This is why the original ALMO-EDA scheme does not have a well-defined basis set limit for its polarization energy.</p>
<p>To overcome this problem, Horn and Head-Gordon proposed a new definition for the POL term in the ALMO-EDA method based on fragment electrical response functions (FERFs).<span class="citation" data-cites="Horn:2015"></span> FERFs on a given fragment are prepared by solving CPSCF equations after its SCF solution is found: <br /><span class="math display"><em>H</em><sub><em>a</em><em>i</em>, <em>b</em><em>j</em></sub>(<em>Δ</em><sub><em>μ</em></sub>)<sub><em>b</em><em>j</em></sub> = (<em>M</em><sub><em>μ</em></sub>)<sub><em>a</em><em>i</em></sub>,</span><br /> where <span class="math inline"><strong>H</strong></span> is SCF orbital Hessian and <span class="math inline"><strong>M</strong><sub><em>μ</em></sub></span> is a component (<span class="math inline"><em>μ</em></span>) of a multipole matrix with a certain order. The resulting fragment response matrices (<span class="math inline">{<em>Δ</em><sub><em>μ</em></sub>}</span>) are a set of <span class="math inline"><em>n</em><sub><em>v</em></sub> × <em>n</em><sub><em>o</em></sub></span> matrices. Then, a singular value decomposition (SVD) is performed on <span class="math inline"><em>Δ</em><sub><em>μ</em></sub></span>: <br /><span class="math display">(<em>Δ</em><sub><em>μ</em></sub>)<sub><em>a</em><em>i</em></sub> = (<em>L</em><sub><em>μ</em></sub>)<sub><em>a</em><em>b</em></sub>(<em>d</em><sub><em>μ</em></sub>)<sub><em>b</em><em>j</em></sub>(<em>R</em><sub><em>μ</em></sub><sup><em>T</em></sup>)<sub><em>j</em><em>i</em></sub>,</span><br /> and the left vectors (not including the null vectors) will be used to construct a truncated virtual space, which is used to define the variational degrees of freedom for the SCFMI problem: <br /><span class="math display"><strong>V</strong><sub><em>μ</em></sub> = <strong>C</strong><sub><em>v</em><em>i</em><em>r</em></sub><strong>L</strong><sub><em>μ</em></sub>,</span><br /> where <span class="math inline"><strong>C</strong><sub><em>v</em><em>i</em><em>r</em></sub></span> denotes the original virtual orbitals of the given fragment.</p>
<p>The basic spirit of using FERFs is to obtain a subset of virtuals that is most pertinent to the electrical polarization of a given fragment, while the redundant variational degrees of freedom (which might be CT-like) are excluded. This scheme is shown to give a well-defined basis set limit for the polarization energy that relies on the SCFMI calculation. The multipole orders (dipole (D), quadrupole (Q), and octopole (O)) included on the RHS of eq. <a href="#eq:ferf_cpscf" data-reference-type="ref" data-reference="eq:ferf_cpscf">[eq:ferf_cpscf]</a> decide the span of FERFs on each fragment. Numerical experiments suggest that the inclusion of dipole- and quadrupole-type responses is able to long-range induced electrostatics correctly and also gives a well-defined basis set limit, which is thus recommended as the working basis of the SCFMI problem. The full span of the polarization subspace of fragment <span class="math inline"><em>A</em></span> is thus: <br /><span class="math display"><strong>O</strong><sub><em>A</em></sub> ⊕ <em>s</em><em>p</em><em>a</em><em>n</em>{<strong>V</strong><sub><em>μ</em><em>x</em></sub>, <strong>V</strong><sub><em>μ</em><em>y</em></sub>, <strong>V</strong><sub><em>μ</em><em>z</em></sub>} ⊕ <em>s</em><em>p</em><em>a</em><em>n</em>{<strong>V</strong><sub><em>Q</em>2,  − 2</sub>, <strong>V</strong><sub><em>Q</em>2,  − 1</sub>, <strong>V</strong><sub><em>Q</em>2, 0</sub>, <strong>V</strong><sub><em>Q</em>2, 1</sub>, <strong>V</strong><sub><em>Q</em>2, 2</sub>}.</span><br /> Therefore, each occupied orbital will be paired with eight virtual orbitals (if the employed AO basis is large enough).</p>
<p>The polarization subspaces constructed as in eq. <a href="#eq:pol_ss" data-reference-type="ref" data-reference="eq:pol_ss">[eq:pol_ss]</a> are non-orthogonal between fragments. Therefore, it is named as the “nDQ&quot; model for polarization. There is another version of this method which enforces interfragment orthogonality between the polarization subspaces and it is correspondingly termed as “oDQ&quot; (or with other multipole orders). The preparation of orthogonal FERFs is more complicated (see ref. </p>
<p><span class="citation" data-cites="Horn:2015"></span> for the details) and usually gives less favorable polarization energies. For most general cases, we recommend the use of the “nDQ&quot; model. Calculations using FERFs are performed using the generalized SCFMI procedure introduced in Section <a href="#sect:gen_scfmi" data-reference-type="ref" data-reference="sect:gen_scfmi">13.7.1</a>.</p>
<blockquote>

</blockquote>
<blockquote>

</blockquote>

<p><strong>Example .</strong>  Generalized SCFMI calculation for the water dimer using nDQ FERFs.</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
O  -1.551007  -0.114520   0.000000
H  -1.934259   0.762503   0.000000
H  -0.599677   0.040712   0.000000
--
0 1
O   1.350625   0.111469   0.000000
H   1.680398  -0.373741  -0.758561
H   1.680398  -0.373741   0.758561
$end

$rem
   JOBTYPE           sp
   METHOD            wb97x-v
   GEN_SCFMAN        true
   BASIS             6-31+G(d)
   GEN_SCFMAN        true
   SCF_ALGORITHM     diis
   SCF_CONVERGENCE   8
   THRESH            14
   SYMMETRY          false
   SYM_IGNORE        true
   SCF_FINAL_PRINT   1
   FRGM_METHOD       stoll
   SCFMI_MODE        1 !nonortho gen scfmi
   CHILD_MP          true
   CHILD_MP_ORDERS   232 !DQ 
   FD_MAT_VEC_PROD   false
$end</code></pre>
</blockquote>
<h3 id="sect:frz_decomp">Further Decomposition of the Frozen Interaction Energy</h3>
<p>The frozen interaction energy in ALMO-EDA is defined as the energy difference between the unrelaxed frozen (Heitler-London) wave function and the isolated fragments. In other literature (<em>e.g.</em> Ref. </p>
<p><span class="citation" data-cites="Hopffgarten:2012"></span>), this interaction is often decomposed in a classical fashion: <br /><span class="math display"><em>Δ</em><em>E</em><sub><em>f</em><em>r</em><em>z</em></sub> = <em>Δ</em><em>E</em><sub><em>e</em><em>l</em><em>e</em><em>c</em></sub><sup><em>c</em><em>l</em><em>s</em></sup> + <em>Δ</em><em>E</em><sub><em>p</em><em>a</em><em>u</em><em>l</em><em>i</em></sub><sup><em>c</em><em>l</em><em>s</em></sup>,</span><br /> where the contribution from permanent electrostatics is defined as the Coulomb interaction between isolated fragment charge distributions: <br /><span class="math display">$$\label{eq:cls_elec}
\Delta E_{\mathrm{elec}}^{\mathrm{cls}} = \sum_{A &lt; B} \int_{r_1} \int_{r_2} \rho_{A}^{\mathrm{tot}}({\textbf r_1}) 
\frac{1}{{\textbf r_{12}}} \rho_{B}^{\mathrm{tot}}({\textbf r_2}) \mathrm{d}\mathbf{r}_1 \mathrm{d}\mathbf{r}_2$$</span><br /> and the remainder constitutes the Pauli (or exchange) term. Such a decomposition (referred to as the classical decomposition below) is associated with two issues: (i) the evaluation of permanent electrostatics makes use of the “promolecule&quot; state (whose density is the simple sum of monomer densities) rather than a properly anti-symmetrized wave function; (ii) when dispersion-corrected density functionals are used, the Pauli term contains dispersion interaction and thus loses its original meaning.</p>
<p>Horn <em>et al.</em> proposed a new scheme to further decompose the frozen term into contributions from permanent electrostatics (ELEC), Pauli repulsion (PAULI) and dispersion (DISP):<span class="citation" data-cites="Horn:2016b"></span> <br /><span class="math display"><em>Δ</em><em>E</em><sub><em>f</em><em>r</em><em>z</em></sub> = <em>Δ</em><em>E</em><sub><em>e</em><em>l</em><em>e</em><em>c</em></sub> + <em>Δ</em><em>E</em><sub><em>p</em><em>a</em><em>u</em><em>l</em><em>i</em></sub> + <em>Δ</em><em>E</em><sub><em>d</em><em>i</em><em>s</em><em>p</em></sub></span><br /></p>
<p>This approach is compatible with the use of all kinds of density functionals except double-hybrids, and all three components of the FRZ term are computed with the antisymmetrized frozen wave function. The key step of this method is the orthogonal decomposition of the 1PDM associated with the frozen wave function into contributions from individual fragments: <span class="math inline">$\mathbf{P}_{\mathrm{frz}} = \sum_A \tilde{\mathbf{P}}_A$</span>. This is achieved by minimizing an objective function as follows: <br /><span class="math display">$$\label{eq:kep_obj}
\Omega = \sum_{A} E_{A}[\tilde{\mathbf{P}}_A] - E_{A}[\mathbf{P}_A]$$</span><br /> while interfragment orthogonality is enforced between <span class="math inline">$\tilde{\mathbf{P}}_A$</span>’s. The readers are referred to Ref. </p>
<p><span class="citation" data-cites="Horn:2016b"></span> for more details about the orthogonal decomposition.</p>
<p>The ELEC term is then defined as the Coulomb interaction between distorted fragment densities (<span class="math inline"><em>ρ̃</em><sub><em>A</em></sub>(<strong>r</strong>)</span>): <br /><span class="math display">$$\label{eq:eda_elec}
\Delta E_{\mathrm{elec}} = \sum_{A &lt; B} \int_{r_1} \int_{r_2} \tilde{\rho}_{A}^{\mathrm{tot}}({\textbf r_1}) \frac{1}{{\textbf r_{12}}} 
\tilde{\rho}_{B}^{\mathrm{tot}}({\textbf r_2}) \mathrm{d}\mathbf{r}_1 \mathrm{d}\mathbf{r}_2.$$</span><br /></p>
<p>The DISP term is evaluated by subtracting the dispersion-free part of the total exchange-correlation (XC) interaction, where an auxiliary “dispersion-free&quot; (DF) XC functional is used in company with the primary XC functional: <br /><span class="math display">$$\label{eq:eda_disp}
\Delta E_{\mathrm{disp}} = \left(E_{\mathrm{xc}}[\mathbf{P}_{\mathrm{frz}}] - \sum_{A} E_{\mathrm{xc}}[\tilde{\mathbf{P}}_A]\right) - 
\left(E_{\mathrm{xc}}^{\mathrm{DF}}[\mathbf{P}_{\mathrm{frz}}] - \sum_{A} E_{\mathrm{xc}}^{\mathrm{DF}}[\tilde{\mathbf{P}}_A]\right).$$</span><br /> It is suggested that HF is an appropriate DFXC to be used for dispersion-corrected hybrid functionals (<em>e.g.</em> <span class="math inline"><em>ω</em></span>B97M-V, B3LYP-D3), while revPBE is appropriate for semi-local functionals (<em>e.g.</em> B97M-V).</p>
<p>The remainder of the frozen interaction goes into the PAULI term, which includes the net repulsive interaction given by eq. <a href="#eq:kep_obj" data-reference-type="ref" data-reference="eq:kep_obj">[eq:kep_obj]</a> and the “dispersion-free&quot; part of the XC interaction: <br /><span class="math display">$$\label{eq:eda_pauli}
\Delta E_{\mathrm{pauli}} = \sum_{A} (E_{A}[\tilde{\mathbf{P}}_A] - E_{A}[\mathbf{P}_A])
+ \left(E_{\mathrm{xc}}^{\mathrm{DF}}[\mathbf{P}_{\mathrm{frz}}] - \sum_{A} E_{\mathrm{xc}}^{\mathrm{DF}}[\tilde{\mathbf{P}}_A]\right).$$</span><br /> The PAULI term and the ELEC term can also be combined together and reported as the dispersion-free frozen (DFFRZ) term if desired.</p>
<p>In <span><span class="smallcaps">Q-Chem</span></span>’s implementation of “EDA2&quot;, the classical frozen decomposition and the new scheme defined by eqs. <a href="#eq:eda_elec" data-reference-type="ref" data-reference="eq:eda_elec">[eq:eda_elec]</a>–<a href="#eq:eda_pauli" data-reference-type="ref" data-reference="eq:eda_pauli">[eq:eda_pauli]</a> are both calculated by default. The classical ELEC term only depends on monomer properties and the distances between fragments, therefore, it can be particularly useful for scenarios such as force field development (as the reference for permanent electrostatics). When the DISP term calculated by the new scheme is available, a modified classical Pauli term<span class="citation" data-cites="Mao:2016"></span> is also reported, which is simply defined as <br /><span class="math display"><em>Δ</em><em>E</em><sub><em>p</em><em>a</em><em>u</em><em>l</em><em>i</em></sub><sup><em>m</em><em>o</em><em>d</em></sup> = <em>Δ</em><em>E</em><sub><em>p</em><em>a</em><em>u</em><em>l</em><em>i</em></sub><sup><em>c</em><em>l</em><em>s</em></sup> − <em>Δ</em><em>E</em><sub><em>d</em><em>i</em><em>s</em><em>p</em></sub>,</span><br /> <em>i.e.</em>, the dispersion contribution is removed from the classical Pauli term computed using its original definition. Alternatively, this can also be achieved without performing the orthogonal decomposition, by setting to . This also evaluates the DISP term via eq. <a href="#eq:eda_disp" data-reference-type="ref" data-reference="eq:eda_disp">[eq:eda_disp]</a> except that undistorted monomer densities (<span class="math inline"><strong>P</strong><sub><em>A</em></sub></span>’s) are used instead of their distorted counterparts (<span class="math inline">$\tilde{\mathbf{P}}_A$</span>’s).</p>
<blockquote>

</blockquote>
<p><strong>Note: </strong></p>
<p>Users are allowed to turn off the orthogonal decomposition by setting to -1. Also, for calculations that involve ECPs, it is automatically set to since unreasonable results will be produced otherwise.</p>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<h3 id="sect:eda2_job">Job Control for EDA2</h3>
<p>The use of the FERF model for the evaluation of polarization energy and the further decomposition of the frozen term define the second generation of the ALMO-EDA method. Meanwhile, under the same code structure, the original AO-block based ALMO model and other related methods (such as the constrained relaxation of the frozen wave function<span class="citation" data-cites="Horn:2016a"></span> which renders the frozen energy variationally computed, and the polMO model<span class="citation" data-cites="Azar:2013"></span> that arguably gives a lower limit to the polarization contribution) are also available. This entire set of methods implemented in <span><span class="smallcaps">Q-Chem</span></span> based on GEN_SCFMAN (see Section <a href="#sect:gen_scfman" data-reference-type="ref" data-reference="sect:gen_scfman">[sect:gen_scfman]</a>) is referred to as “EDA2&quot;.</p>
<p>The job control for EDA2 is largely simplified by a series of preset options provided by the developers. The option number is set through the <em>$rem</em> variable (introduced below). Besides that, for the sake of flexibility, users are allowed to overwrite the values of part of the preset <em>$rem</em> variables:</p>
<ul>
<li><p>Related to the isolated fragment calculations:</p>
<ul>
<li><p>: use the super-system basis for fragment calculations (default: ).</p></li>
<li><p>: as introduced in Section <a href="#sect:FRAGMOguess" data-reference-type="ref" data-reference="sect:FRAGMOguess">[sect:FRAGMOguess]</a> (default: 0).</p></li>
</ul></li>
<li><p>Related to the decomposition of the FRZ term:</p>
<ul>
<li><p>: it can be turned off by setting its value to -1 in the <em>$rem</em> section<br />
(default: ).</p></li>
<li><p>: as introduced in Section <a href="#sect:frz_decomp" data-reference-type="ref" data-reference="sect:frz_decomp">13.7.3</a> (default: 6).</p></li>
<li><p>: as introduced in Section <a href="#sect:frz_decomp" data-reference-type="ref" data-reference="sect:frz_decomp">13.7.3</a> (default: ).</p></li>
<li><p>: as introduced in Section <a href="#sect:frz_decomp" data-reference-type="ref" data-reference="sect:frz_decomp">13.7.3</a> (default: ).</p></li>
<li><p>: as introduced in Section <a href="#sect:frz_decomp" data-reference-type="ref" data-reference="sect:frz_decomp">13.7.3</a> (default: ).</p></li>
</ul></li>
<li><p>Related to the evaluation of POL:</p>
<ul>
<li><p>: as introduced in Section <a href="#sect:ferf" data-reference-type="ref" data-reference="sect:ferf">13.7.2</a> (default: 122 (DQ)).</p></li>
<li><p>: as introduced in Section <a href="#sect:gen_scfmi" data-reference-type="ref" data-reference="sect:gen_scfmi">13.7.1</a> (default: 0).</p></li>
</ul></li>
<li><p>Related to the evaluation of CT and BSSE:</p>
<ul>
<li><p>: skip the evaluation of the CT term in the EDA procedure<br />
(default: (automatically turned on when = )).</p></li>
<li><p>: use counterpoise-corrected monomer calculations to evaluate the BSSE<br />
(default: ).</p></li>
<li><p>: turn on perturbative charge transfer analysis (Roothaan step based).</p></li>
<li><p>: perform COVP analysis for charge transfer (see Section <a href="#sect:EDA-CTA" data-reference-type="ref" data-reference="sect:EDA-CTA">13.5</a>).</p></li>
<li><p>: dump COVPs to the MO coefficient file (see Section <a href="#sect:EDA-CTA" data-reference-type="ref" data-reference="sect:EDA-CTA">13.5</a>). EDA2 can automatically generate the cubes for the dominant occupied-virtual pair for each pair of donor and acceptor when is greater than 1.</p></li>
</ul></li>
</ul>
<blockquote>

</blockquote>

<p><strong>Example .</strong>  Energy decomposition analysis for the ammonia-borane complex. The FERF-nDQ model is used for the POL term (as very large basis set is employed here), and decomposition of the frozen interaction energy is performed (Hartree-Fock is employed as the DFXC functional by default).</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
N         0.000000    0.000000   -0.727325
H         0.947371    0.000000   -1.091577
H        -0.473685   -0.820448   -1.091577
H        -0.473685    0.820448   -1.091577
--
0 1
B         0.000000    0.000000    0.930725
H        -1.165774    0.000000    1.243063
H         0.582887   -1.009590    1.243063
H         0.582887    1.009590    1.243063
$end

$rem
   JOBTYPE          eda 
   EDA2             1
   METHOD           wB97M-V
   BASIS            def2-QZVPPD
   SYMMETRY         false
   MEM_TOTAL        4000
   MEM_STATIC       1000
   THRESH           14  
   SCF_CONVERGENCE  8
   XC_GRID          000099000590
   NL_GRID          1   
   FD_MAT_VEC_PROD  false
$end</code></pre>
</blockquote>


<p><strong>Example .</strong>  Energy decomposition analysis of the water dimer with a low-cost model chemistry. The original ALMO model is used for the evaluation of polarization energy, and revPBE is chosen as the DFXC functional. Counterpoise correction for the BSSE is applied.</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
H1
O1 H1 0.95641
H2 O1 0.96500  H1 104.77306
--
0 1
O2 H2 dist     O1 171.85474 H1 180.000
H3 O2 0.95822  H2 111.79807 O1 -58.587
H4 O2 0.95822  H2 111.79807 O1 58.587

dist = 2.0 
$end

$rem
   JOBTYPE          eda 
   EDA2             2
   METHOD           b97m-v
   BASIS            def2-svpd
   SCF_CONVERGENCE  8
   THRESH           14  
   SYMMETRY         false
   DISP_FREE_X      revPBE
   DISP_FREE_C      PBE 
   EDA_BSSE         true
$end</code></pre>
</blockquote>
<h2 id="the-mp2-almo-eda-method">The MP2 ALMO-EDA Method</h2>
<p><span id="sect:mp2_eda" label="sect:mp2_eda">[sect:mp2_eda]</span></p>
<p>The previously described EDA methods are limited to SCF methods such as HF and DFT. However, for many systems, it is preferable to use a wave function based correlation method. For this reason, the ALMO-EDA has been extended to MP2.<span class="citation" data-cites="Thirman:2015 Thirman:2017"></span> The MP2 ALMO-EDA is based on the first-generation ALMO-EDA. It provides an MP2 correction to the FRZ, POL, and CT terms defined by the ALMO-EDA and also adds in a term corresponding to the London dispersion force. This is done by defining a constrained intermediate MP2 wave function corresponding to each HF intermediate.</p>
<p>The current implementation is limited to only RI-MP2 rather than full MP2, and only works in the closed shell, spin restricted case. Frozen core and spin scaling are also not yet supported. Attempting to use the EDA with a correlation method other then RI-MP2 or with unrestricted orbitals will result in a crash. Frozen core and spin scaling settings will be ignored with a warning by the EDA, but not by the final energy, leading to inconsistent results.</p>
<p>Though the MP2 EDA is based on the first generation ALMO-EDA, the code path and REM settings are shared with the second generation ALMO-EDA. The MP2 ALMO-EDA does not define any new REM variables of its own. Rather, running an EDA job with EDA2 and GEN_SCFMAN will trigger an MP2 ALMO-EDA when the correlation method is RI-MP2. The correlation setting causes the SCF portion of the EDA to be switched back to the original scheme and will also decompose the correlation energy. Most settings intended for the second generation ALMO-EDA are not supported, but and are. An example appears below.</p>

<p><strong>Example .</strong>  MP2 energy decomposition analysis of the water dimer.</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
O  -0.031783  -0.057754   0.000000
H  -0.415035   0.819269   0.000000
H   0.919546   0.097478   0.000000
--
0 1
O   2.960796   0.171800   0.000000
H   3.290569  -0.313410  -0.758561
H   3.290569  -0.313410   0.758561
$end

$rem
JOBTYPE         EDA
GEN_SCFMAN      TRUE
EDA2            TRUE
FRGM_METHOD     STOLL
EXCHANGE        HF
CORRELATION     RIMP2
SYMMETRY        FALSE
BASIS           aug-cc-pVTZ
AUX_BASIS       rimp2-aug-cc-pVTZ
THRESH          14
SCF_CONVERGENCE 10
N_FROZEN_CORE   0
EDA_BSSE        TRUE
$end</code></pre>
</blockquote>
<h2 id="the-adiabatic-almo-eda-method">The Adiabatic ALMO-EDA Method</h2>
<p><span id="sect:adiabatic_eda" label="sect:adiabatic_eda">[sect:adiabatic_eda]</span></p>
<p>Despite the huge success and usefulness of today’s most popular EDA methods, they still face some limitations in their capabilities. For instance, EDAs are usually performed at complex geometries that are obtained from unconstrained electronic structure calculations (<em>e.g.</em>, optimized equilibrium geometries). For strongly interacting systems, close intermolecular contacts driven by POL and particularly CT often result in largely unfavorable FRZ interaction, which offers little physical insights besides indicating obviously substantial intermolecular overlap. Another limitation is that the conventional EDA methods often partitions a “single-point&quot; interaction energy evaluated at a given geometry. Therefore, the influence of FRZ, POL and CT on the structural and vibrational properties of an intermolecular complex cannot be directly characterized.</p>
<p>Recently Mao <em>et al.</em> reformulated the original ALMO-EDA method in an adiabatic picture,<span class="citation" data-cites="Mao:2017"></span> where the term “adiabatic&quot; is borrowed from spectroscopy and indicates that energy differences are evaluated at relaxed geometry on each potential energy surface (PES). In this scheme, the total binding energy (including monomer geometry distortions) is repartitioned into adiabatic FRZ, POL and CT terms: <br /><span class="math display"><em>Δ</em><em>E</em><sub><em>b</em><em>i</em><em>n</em><em>d</em></sub> = <em>Δ</em><em>E</em><sub><em>f</em><em>r</em><em>z</em></sub><sup>(<em>a</em><em>d</em>)</sup> + <em>Δ</em><em>E</em><sub><em>p</em><em>o</em><em>l</em></sub><sup>(<em>a</em><em>d</em>)</sup> + <em>Δ</em><em>E</em><sub><em>c</em><em>t</em></sub><sup>(<em>a</em><em>d</em>)</sup>.</span><br /> The adiabatic frozen interaction energy is given by the difference between the energy minimum of the frozen PES (on which the energy of each point is computed using the corresponding frozen wave function) and the sum of fully relaxed, non-interacting fragment energies: <br /><span class="math display"><em>Δ</em><em>E</em><sub><em>f</em><em>r</em><em>z</em></sub><sup>(<em>a</em><em>d</em>)</sup> = <em>E</em>[<strong>P</strong><sub><em>f</em><em>r</em><em>z</em></sub><sup>(<em>f</em><em>r</em><em>z</em>)</sup>] − ∑<sub><em>A</em></sub><em>E</em><sub><em>A</em></sub><sup>(0)</sup>.</span><br /> Similarly, the adiabatic POL and CT terms can be obtained by performing geometry optimizations on the polarized (SCFMI) and fully relaxed (unconstrained SCF) PESs: <br /><span class="math display">$$\begin{aligned}
\Delta E^{(\mathrm{ad})}_{\mathrm{pol}} &amp;= E[\mathbf{P}_{\mathrm{pol}}^{(\mathrm{pol})}] - E[\mathbf{P}_{\mathrm{frz}}^{(\mathrm{frz})}], \label{eq:ad_POL} \\
\Delta E^{(\mathrm{ad})}_{\mathrm{ct}} &amp;= E[\mathbf{P}^{(\mathrm{full})}_{\mathrm{full}}] - E[\mathbf{P}_{\mathrm{pol}}^{(\mathrm{pol})}]. \label{eq:ad_CT}\end{aligned}$$</span><br /></p>
<p>With this method, the changes in monomer structures and intermolecular coordinates due to FRZ, POL and CT and the accompanied energetics are provided. Moreover, at the energy minimum (or other stationary points) on each PES, the other properties such as multipole points, vibrational frequencies and intensities can also be computed, therefore the effect of different intermolecular interaction components on them can also be characterized.</p>
<p>The geometry optimization on the frozen PES is facilitated by the analytical gradient of the frozen wave function energy implemented in <span><span class="smallcaps">Q-Chem</span></span>. As for the geometry optimization on the polarized PES, the nuclear gradient of the SCFMI energy has the same form as that of the full SCF energy if the original ALMO model is used. These analytical gradients can also be used for finite difference calculations of harmonic frequencies by setting = 1. We note that the analytical gradients of SCFMI calculations that use FERFs are not available yet, and = 0 is required for computing the forces on the frozen and polarized PESs. Also, the current implementation of this method requires users to perform geometry optimization on the three PESs separately (see the example below) and evaluate the energy components by taking several <span><span class="smallcaps">Q-Chem</span></span> outputs (including geometry optimizations for the monomers) together, which is probably not so convenient. We look forward to extending the functionality of this method and improving its implementation in the near future.</p>
<blockquote>

</blockquote>
<blockquote>

</blockquote>

<p><strong>Example .</strong>  Geometry optimization of the ammonia-borane complex on the fully relaxed, polarized, and frozen potential energy surfaces successively.</p>
<blockquote>
<pre><code>$molecule
0 1
--
0 1
H       0.000000     0.000000     0.000000
H       0.000000     0.000000     1.629090
H       1.417687     0.000000     0.814543
N       0.473683    -0.370067     0.814542
--
0 1
N       3.494032    -1.531250     0.814538
H       3.967715    -1.901317    -0.000008
H       2.550028    -1.901319     0.814537
H       3.967715    -1.901317     1.629083
$end

$rem
   JOBTYPE          opt  !optimization on the fully relaxed PES
   GEN_SCFMAN       true
   METHOD           wb97x-d
   BASIS            6-31+g*
   XC_GRID          1
   THRESH           14  
   SCF_CONVERGENCE  9
   SCF_GUESS        fragmo
   SYMMETRY         false
   SYM_IGNORE       true
$end

@@@

$molecule
   read
$end

$rem
   JOBTYPE          opt 
   POL_GEOM         true !optimization on the polarized PES
   GEN_SCFMAN       true
   METHOD           wb97x-d
   BASIS            6-31+g*
   XC_GRID          1
   THRESH           14  
   SCF_CONVERGENCE  9
   SYMMETRY         false
   SYM_IGNORE       true
   SCFMI_MODE       0
$end

@@@

$molecule
   read
$end

$rem
   JOBTYPE          opt 
   FRZ_GEOM         true !optimization on the frozen PES
   GEN_SCFMAN       true
   METHOD           wb97x-d
   BASIS            6-31+g*
   XC_GRID          1
   THRESH           14  
   SCF_CONVERGENCE  9
   SYMMETRY         false
   SYM_IGNORE       true
   SCFMI_MODE       0
$end</code></pre>
</blockquote>

<p><strong>Example .</strong>  Geometry optimization of the CF<span class="math inline"><em></em><sub>3</sub></span>H<span class="math inline">⋯</span>Cl<span class="math inline"><em></em><sup>−</sup></span> complex on the frozen PES followed by a calculation for harmonic frequencies at the equilibrium geometry.</p>
<blockquote>
<pre><code>$molecule
-1 1
--
0 1
C         0.000000    0.000000    0.905509
H         0.000000    0.000000   -0.179815
F         1.250779    0.000000    1.406437
F        -0.625389   -1.083207    1.406437
F        -0.625389    1.083207    1.406437
--
-1 1
Cl        0.000000    0.000000   -2.664391
$end

$rem
   JOBTYPE                     opt 
   FRZ_GEOM                    true
   GEN_SCFMAN                  true
   METHOD                      wb97X-V
   BASIS                       def2-svpd
   XC_GRID                     000075000302
   THRESH                      14  
   SCF_CONVERGENCE             9
   SYMMETRY                    false
   SYM_IGNORE                  false
   SCFMI_MODE                  0
   GEOM_OPT_TOL_GRADIENT       100 
   GEOM_OPT_TOL_DISPLACEMENT   100 
   GEOM_OPT_TOL_ENERGY         10
   FD_MAT_VEC_PROD             false
$end

@@@

$molecule
   read
$end

$rem
   JOBTYPE           freq
   FRZ_GEOM          true
   GEN_SCFMAN        true
   METHOD            wb97X-V
   BASIS             def2-svpd
   XC_GRID           000075000302
   THRESH            14  
   SCF_CONVERGENCE   9
   SYMMETRY          false
   SYM_IGNORE        false
   IDERIV            1   
   SCFMI_MODE        0
   FD_MAT_VEC_PROD   false
$end</code></pre>
</blockquote>
<h2 id="the-explicit-polarization-xpol-method">The Explicit Polarization (XPol) Method</h2>
<p><span id="sect:xpol" label="sect:xpol">[sect:xpol]</span></p>
<h3 id="theory">Theory</h3>
<p>XPol is an approximate, fragment-based molecular orbital method that was developed as a “next-generation” force field.<span class="citation" data-cites="Gao:1998 Xie:2007 Xie:2008 Xie:2009"></span> The basic idea of the method is to treat a molecular liquid, solid, or cluster as a collection of fragments, where each fragment is a molecule. Intra-molecular interactions are treated with a self-consistent field method (Hartree-Fock or DFT), but each fragment is embedded in a field of point charges that represent electrostatic interactions with the other fragments. These charges are updated self-consistently by collapsing each fragment’s electron density onto a set of atom-centered point charges, using charge analysis procedures (Mulliken, Löwdin, or ChElPG, for example; see Section <a href="#subsec:pop_anal" data-reference-type="ref" data-reference="subsec:pop_anal">[subsec:pop_anal]</a>). This approach incorporates many-body polarization, at a cost that scales linearly with the number of fragments, but neglects the anti-symmetry requirement of the total electronic wave function. As a result, intermolecular exchange-repulsion is neglected, as is dispersion since the latter is an electron correlation effect. As such, the XPol treatment of polarization must be augmented with empirical, Lennard–Jones-type intermolecular potentials in order to obtain meaningful optimized geometries, vibrational frequencies or dynamics.</p>
<p>The XPol method is based upon an <em>ansatz</em> in which the super-system wave function is written as a direct product of fragment wave functions, <br /><span class="math display">$$\label{eq:hartree_pdt}
   |\Psi\rangle = \prod_A^{N_\ensuremath{\mathrm{frag}}} | \Psi_{\!A} \rangle  ,$$</span><br /> where <span class="math inline"><em>N</em><sub><em>f</em><em>r</em><em>a</em><em>g</em></sub></span> is the number of fragments. We assume here that the fragments are molecules and that covalent bonds remain intact. The fragment wave functions are anti-symmetric with respect to exchange of electrons within a fragment, but not to exchange between fragments. For closed-shell fragments described by Hartree-Fock theory, the XPol total energy is<span class="citation" data-cites="Xie:2008 Jacobson:2011"></span> <br /><span class="math display">$$\label{eq:E_xpol}
   E_{\rm XPol} = \sum_A \left[ 2 \sum_{a} \mathbf{c}_a^\dagger \left( \mathbf{h}^A+ \mathbf{J}^A 
   -  \tfrac{1}{2}\mathbf{K}^A 	\right) \mathbf{c}_a^{} +E_\ensuremath{\mathrm{nuc}}^A \right] 
   + E_\ensuremath{\mathrm{embed}}  .$$</span><br /> The term in square brackets is the ordinary Hartree-Fock energy expression for fragment <span class="math inline"><em>A</em></span>. Thus, <span class="math inline"><strong>c</strong><sub><em>a</em></sub><sup></sup></span> is a vector of occupied MO expansion coefficients (in the AO basis) for the occupied MO <span class="math inline"><em>a</em> ∈ <em>A</em></span>; <span class="math inline"><strong>h</strong><sup><em>A</em></sup></span> consists of the one-electron integrals; and <span class="math inline"><strong>J</strong><sup><em>A</em></sup></span> and <span class="math inline"><strong>K</strong><sup><em>A</em></sup></span> are the Coulomb and exchange matrices, respectively, constructed from the density matrix for fragment <span class="math inline"><em>A</em></span>. The additional terms in Eq. <a href="#eq:E_xpol" data-reference-type="eqref" data-reference="eq:E_xpol">[eq:E_xpol]</a>, <br /><span class="math display">$$\label{eq:E_QMMM}
   E_\ensuremath{\mathrm{embed}} = 
   \tfrac{1}{2} \sum_A \sum_{B\neq A} \sum_{J\in B} \left( -2 \sum_a \mathbf{c}_a^\dagger 
   \mathbf{I}_J \mathbf{c}_a + \sum_{I \in A} L_{IJ} \right) q_J^{}  ,$$</span><br /> arise from the electrostatic embedding. The matrix <span class="math inline"><strong>I</strong><sub><em>J</em></sub></span> is defined by its AO matrix elements, <br /><span class="math display">$$\label{eq:I}
   \left( \mathbf{I}_J \right)_{\mu \nu} = 
   \left\langle \mu \left| \frac{1}{\bigl|\vec{r}-\vec{R}_J \bigr|} \right| \nu \right\rangle  ,$$</span><br /> and <span class="math inline"><em>L</em><sub><em>I</em><em>J</em></sub></span> is given by <br /><span class="math display">$$\label{eq:L}
   L_{IJ} = \frac{Z_I}{\bigl|\vec{R}_I-\vec{R}_J\bigr|}  .$$</span><br /> According to Eqs. <a href="#eq:E_xpol" data-reference-type="eqref" data-reference="eq:E_xpol">[eq:E_xpol]</a> and <a href="#eq:E_QMMM" data-reference-type="eqref" data-reference="eq:E_QMMM">[eq:E_QMMM]</a>, each fragment is embedded in the electrostatic potential arising from a set of point charges, <span class="math inline">{<em>q</em><sub><em>J</em></sub><sup></sup>}</span>, on all of the other fragments; the factor of <span class="math inline">1/2</span> in Eq. <a href="#eq:E_QMMM" data-reference-type="eqref" data-reference="eq:E_QMMM">[eq:E_QMMM]</a> avoids double-counting. Exchange interactions between fragments are ignored, and the electrostatic interactions between fragments are approximated by interactions between the charge density of one fragment and point charges on the other fragments.</p>
<p>Crucially, the vectors <span class="math inline"><strong>c</strong><sub><em>a</em></sub></span> are constructed within the ALMO <em>ansatz</em>,<span class="citation" data-cites="Khaliullin:2006"></span> so that MOs for each fragment are represented in terms of only those AOs that are centered on atoms in the same fragment. This choice affords a method whose cost grows linearly with respect to <span class="math inline"><em>N</em><sub><em>f</em><em>r</em><em>a</em><em>g</em></sub></span>, and where basis set superposition error is excluded by construction. In compact basis sets, the ALMO <em>ansatz</em> excludes inter- fragment charge transfer as well.</p>
<p>The original XPol method of Xie <em>et al.</em><span class="citation" data-cites="Xie:2007 Xie:2008 Xie:2009"></span> uses Mulliken charges for the embedding charges <span class="math inline"><em>q</em><sub><em>J</em></sub><sup></sup></span> in Eq. <a href="#eq:E_QMMM" data-reference-type="eqref" data-reference="eq:E_QMMM">[eq:E_QMMM]</a>, though other charge schemes could be envisaged. In non-minimal basis sets, the use of Mulliken charges is beset by severe convergence problems,<span class="citation" data-cites="Jacobson:2011"></span> and <span><span class="smallcaps">Q-Chem</span></span>’s implementation of XPol offers the alternative of using either Löwdin charges or ChElPG charges,<span class="citation" data-cites="Breneman:1990"></span> the latter being derived from the electrostatic potential as discussed in Section <a href="#subsec:pop_anal" data-reference-type="ref" data-reference="subsec:pop_anal">[subsec:pop_anal]</a>. The ChElPG charges are found to be stable and robust, albeit with a somewhat larger computational cost as compared to Mulliken or Löwdin charges.<span class="citation" data-cites="Jacobson:2011 Herbert:2012"></span> An algorithm to compute ChElPG charges using atom-centered Lebedev grids rather than traditional Cartesian grids is available (see Section <a href="#subsec:pop_anal" data-reference-type="ref" data-reference="subsec:pop_anal">[subsec:pop_anal]</a>),<span class="citation" data-cites="Holden:2013"></span> which uses far fewer grid points and thus can significantly improve the performance for the XPol/ChElPG method, where these charges must be iteratively updated.</p>
<p>Researchers who use <span><span class="smallcaps">Q-Chem</span></span>’s XPol code are asked to cite Refs. </p>
<p><span class="citation" data-cites="Jacobson:2011"></span>, and</p>
<p><span class="citation" data-cites="Herbert:2012"></span>.</p>
<h3 id="supplementing-xpol-with-empirical-potentials">Supplementing XPol with Empirical Potentials</h3>
<p>In order to obtain physical results, one must either supplement the XPol energy expression with either empirical intermolecular potentials or else with an <em>ab initio</em> treatment of intermolecular interactions. The latter approach is described in Section <a href="#sect:XSAPT" data-reference-type="ref" data-reference="sect:XSAPT">13.12</a>. Here, we describe how to add Lennard-Jones or Buckingham potentials to the XPol energy, using the <em>$xpol_mm</em> and <em>$xpol_params</em> sections described below.</p>
<p>The Lennard-Jones potential is <br /><span class="math display">$$\label{eq:Lennard-Jones}
 	V_{\rm LJ}^{}(R_{ij}) = 4  \epsilon^{}_{ij} \left[ 
		\left(\frac{\sigma^{}_{ij}}{R_{ij}}\right)^{\!12} 
		- \left(\frac{\sigma^{}_{ij}}{R_{ij}}\right)^{\!6} 
	\right] \; ,$$</span><br /> where <span class="math inline"><em>R</em><sub><em>i</em><em>j</em></sub></span> represents the distance between atoms <span class="math inline"><em>i</em></span> and <span class="math inline"><em>j</em></span>. This potential is characterized by two parameters, a well depth <span class="math inline"><em>ϵ</em><sub><em>i</em><em>j</em></sub><sup></sup></span> and a length scale <span class="math inline"><em>σ</em><sub><em>i</em><em>j</em></sub><sup></sup></span>. Although quite common, the <span class="math inline"><em>R</em><sup> − 12</sup></span> repulsion is unrealistically steep. The Buckingham potential replaces this with an exponential function, <br /><span class="math display">$$\label{eq:Buckingham}
	V_{\rm Buck}^{}(R_{ij}) =  \epsilon^{}_{ij} \left[ 
		A e^{-B \frac{R_{ij}}{\sigma^{}_{ij}}} 
		- C \left(\frac{\sigma^{}_{ij}}{R_{ij}}\right)^{\!6} 
	\right] \; ,$$</span><br /> Here, <span class="math inline"><em>A</em></span>, <span class="math inline"><em>B</em></span>, and <span class="math inline"><em>C</em></span> are additional (dimensionless) constants, independent of atom type. In both Eq. <a href="#eq:Lennard-Jones" data-reference-type="eqref" data-reference="eq:Lennard-Jones">[eq:Lennard-Jones]</a> and Eq. <a href="#eq:Buckingham" data-reference-type="eqref" data-reference="eq:Buckingham">[eq:Buckingham]</a>, the parameters <span class="math inline"><em>ϵ</em><sub><em>i</em><em>j</em></sub><sup></sup></span> and <span class="math inline"><em>σ</em><sub><em>i</em><em>j</em></sub><sup></sup></span> are determined using the geometric mean of atomic well-depth and length-scale parameters. For example, <br /><span class="math display">$$\sigma^{}_{ij} = \sqrt{\sigma^{}_i \sigma_j^{}} \; .$$</span><br /></p>
<p>The atomic parameters <span class="math inline"><em>σ</em><sub><em>i</em></sub><sup></sup></span> and <span class="math inline"><em>ϵ</em><sub><em>i</em></sub><sup></sup></span> must be specified using a <em>$xpol_mm</em> section in the <span><span class="smallcaps">Q-Chem</span></span> input file. The format is a molecular mechanics-like specification of atom types and connectivities. All atoms specified in the <em>$molecule</em> section must also be specified in the <em>$xpol_mm</em> section. Each line must contain an atom number, atomic symbol, Cartesian coordinates, integer atom type, and any connectivity data. The <em>$xpol_params</em> section specifies, for each atom type, a value for <span class="math inline"><em>ϵ</em></span> in kcal/mol and a value for <span class="math inline"><em>σ</em></span> in <span>Å</span>ngstroms. A Lennard-Jones potential is used by default; if a Buckingham potential is desired, then the first line of the <em>$xpol_params</em> section should contain the string followed by values for the <span class="math inline"><em>A</em></span>, <span class="math inline"><em>B</em></span>, and <span class="math inline"><em>C</em></span> parameters.</p>
<h3 id="job-control-variables-for-xpol">Job Control Variables for XPol</h3>
<p>XPol calculations are enabled by setting the <em>$rem</em> variable to TRUE. These calculations can be used in combination with Hartree-Fock theory and with most density functionals, a notable exception being that XPol is not yet implemented for meta-GGA functionals (Section <a href="#sect:XCfuncs" data-reference-type="ref" data-reference="sect:XCfuncs">[sect:XCfuncs]</a>). Combining XPol with solvation models (Section <a href="#sect:solvent" data-reference-type="ref" data-reference="sect:solvent">[sect:solvent]</a>) or external charges (<em>$external_charges</em>) is also not available. Analytic gradients are available when Mulliken or Löwdin embedding charges are used, but not yet available for ChElPG embedding charges.</p>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<h3 id="examples">Examples</h3>
<p>XPol on its own is not a useful method (as it neglects all intermolecular interactions except for polarization), so the two examples below demonstrate the use of XPol in conjunction with a Lennard-Jones and a Buckingham potential, respectively.</p>

<p><strong>Example .</strong>  An XPol single point calculation on the water dimer using a Lennard-Jones potential.</p>
<blockquote>
<pre><code>$molecule
0 1
-- water 1
0 1
O          -1.364553     .041159     .045709
H          -1.822645     .429753    -.713256
H          -1.841519    -.786474     .202107
-- water 2
0 1
O           1.540999     .024567     .107209
H            .566343     .040845     .096235
H           1.761811    -.542709    -.641786
$end

$rem
   METHOD           HF
   BASIS            3-21G
   XPOL             TRUE
   XPOL_CHARGE_TYPE QLOWDIN
$end

$xpol_mm
   1 O          -1.364553     .041159     .045709 1 2 3
   2 H          -1.822645     .429753    -.713256 2 1
   3 H          -1.841519    -.786474     .202107 2 1
   4 O           1.540999     .024567     .107209 1 5 6
   5 H            .566343     .040845     .096235 2 4
   6 H           1.761811    -.542709    -.641786 2 4
$end

$xpol_params
   1            0.16     3.16
   2            0.00     0.00
$end</code></pre>
</blockquote>


<p><strong>Example .</strong>  An XPol single point calculation on the water dimer using a Buckingham potential.</p>
<blockquote>
<pre><code>$molecule
0 1
-- water 1
0 1
O          -1.364553     .041159     .045709
H          -1.822645     .429753    -.713256
H          -1.841519    -.786474     .202107
-- water 2
0 1
O           1.540999     .024567     .107209
H            .566343     .040845     .096235
H           1.761811    -.542709    -.641786
$end

$rem
   METHOD           HF
   BASIS            3-21G
   XPOL             TRUE
   XPOL_CHARGE_TYPE QLOWDIN
$end

$xpol_mm
   1 O          -1.364553     .041159     .045709 1 2 3
   2 H          -1.822645     .429753    -.713256 2 1
   3 H          -1.841519    -.786474     .202107 2 1
   4 O           1.540999     .024567     .107209 1 5 6
   5 H            .566343     .040845     .096235 2 4
   6 H           1.761811    -.542709    -.641786 2 4
$end

$xpol_params
   BUCKINGHAM   500000.0 12.5 2.25
   1            0.16     3.16
   2            0.00     0.00
$end</code></pre>
</blockquote>
<h2 id="sec:sapt">Symmetry-Adapted Perturbation Theory (SAPT)</h2>
<h3 id="theory-1">Theory</h3>
<p>Symmetry-adapted perturbation theory (SAPT) is a theory of intermolecular interactions. When computing intermolecular interaction energies one typically computes the energy of two molecules infinitely separated and in contact, then computes the interaction energy by subtraction. SAPT, in contrast, is a perturbative expression for the interaction energy itself. The various terms in the perturbation series are physically meaningful, and this decomposition of the interaction energy can aid in the interpretation of the results. A brief overview of the theory is given below; for additional technical details, the reader is referred to Jeziorski <em>et al.</em>.<span class="citation" data-cites="Jeziorski:1993 Jeziorski:1994"></span> Additional context can be found in a pair of more recent review articles.<span class="citation" data-cites="Szalewicz:2012 Hohenstein:2012"></span></p>
<p>In SAPT, the Hamiltonian for the <span class="math inline">$\rm A\cdots B$</span> dimer is written as <br /><span class="math display">$$\label{eq:H_SAPT}
   \Hat{H} = \Hat{F}^A + \Hat{F}^B + \xi \Hat{W}^A + \eta \Hat{W}^B + \zeta \Hat{V} \; ,$$</span><br /> where <span class="math inline">$\Hat{W}^A$</span> and <span class="math inline">$\Hat{W}^B$</span> are M<span>ø</span>ller-Plesset fluctuation operators for fragments <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span>, whereas <span class="math inline"><em>V̂</em></span> consists of the intermolecular Coulomb operators. This part of the perturbation is conveniently expressed as <br /><span class="math display">$$\hat{V} = \sum_{i\in A} \sum_{j\in B} \Hat{v}(ij)$$</span><br /> with <br /><span class="math display">$$\label{eq:pairwise_perturbation}
   \Hat{v}(ij) = \frac{1}{\bigl|\vec{r}_i-\vec{r}_j\bigr|} 
   + \frac{\hat{v}_A(j)}{N_A} + \frac{\hat{v}_B(i)}{N_B}
   + \frac{V_0}{N_A N_B} \; .$$</span><br /> The quantity <span class="math inline"><em>V</em><sub>0</sub></span> is the nuclear interaction energy between the two fragments and <br /><span class="math display">$$\label{va_sapt}
   \Hat{v}_A(j) = -\sum_{I\in A} \frac{Z_I}{\bigl|\vec{r}_j - \vec{R}_I\bigr|}$$</span><br /> describes the interaction of electron <span class="math inline"><em>j</em> ∈ <em>B</em></span> with nucleus <span class="math inline"><em>I</em> ∈ <em>A</em></span>.</p>
<p>Starting from a zeroth-order Hamiltonian <span class="math inline">$\Hat{H}_0 = \Hat{F}^A + \Hat{F}^B$</span> and zeroth-order wave functions that are direct products of monomer wave functions, <span class="math inline">|<em>Ψ</em><sub>0</sub>⟩ = |<em>Ψ</em><sub><em>A</em></sub>⟩|<em>Ψ</em><sub><em>B</em></sub>⟩</span>, the SAPT approach is based on a symmetrized Rayleigh-Schrödinger perturbation expansion<span class="citation" data-cites="Jeziorski:1993 Jeziorski:1994"></span> with respect to the perturbation parameters <span class="math inline"><em>ξ</em></span>, <span class="math inline"><em>η</em></span>, and <span class="math inline"><em>ζ</em></span> in Eq. <a href="#eq:H_SAPT" data-reference-type="eqref" data-reference="eq:H_SAPT">[eq:H_SAPT]</a>. The resulting interaction energy can be expressed as<span class="citation" data-cites="Jeziorski:1993 Jeziorski:1994"></span> <br /><span class="math display">$$\label{eq:triple_pert}
	E_\ensuremath{\mathrm{int}} = \sum_{i=1}^\infty\sum_{j=0}^\infty
	\left(E^{(ij)}_\ensuremath{\mathrm{pol}} + E^{(ij)}_\ensuremath{\mathrm{exch}}\right) \; .$$</span><br /> Because it makes no sense to treat <span class="math inline">$\Hat{W}^A$</span> and <span class="math inline">$\Hat{W}^B$</span> at different orders of perturbation theory, there are only two indices in this expansion: <span class="math inline"><em>j</em></span> for the monomer fluctuations potentials and <span class="math inline"><em>i</em></span> for the intermolecular perturbation. The terms <span class="math inline"><em>E</em><sub><em>p</em><em>o</em><em>l</em></sub><sup>(<em>i</em><em>j</em>)</sup></span> are known collectively as the <em>polarization expansion</em>, and these are precisely the same terms that would appear in ordinary Rayleigh-Schrödinger perturbation theory, which is valid when the monomers are well-separated. The polarization expansion contains electrostatic, induction and dispersion interactions, but in the <em>symmetrized</em> Rayleigh-Schrödinger expansion, each term <span class="math inline"><em>E</em><sub><em>p</em><em>o</em><em>l</em></sub><sup>(<em>i</em><em>j</em>)</sup></span> has a corresponding exchange term, <span class="math inline"><em>E</em><sub><em>e</em><em>x</em><em>c</em><em>h</em></sub><sup>(<em>i</em><em>j</em>)</sup></span>, that arises from an anti-symmetrizer <span class="math inline">$\Hat{\cal A}_{AB}$</span> that is introduced in order to project away the Pauli-forbidden components of the interaction energy that would otherwise appear.<span class="citation" data-cites="Jeziorski:1994"></span></p>
<p>The version of SAPT that is implemented in <span><span class="smallcaps">Q-Chem</span></span> assumes that <span class="math inline"><em>ξ</em> = <em>η</em> = 0</span>, an approach that is usually called SAPT0.<span class="citation" data-cites="Hohenstein:2012"></span> Within the SAPT0 formalism, the interaction energy is formally expressed by the following symmetrized Rayleigh-Schrödinger expansion:<span class="citation" data-cites="Jeziorski:1993 Jeziorski:1994"></span> <br /><span class="math display">$$\label{eq:E(SRS)}
	E_\ensuremath{\mathrm{int}}(\zeta) =
	\frac{
		\langle\Psi_0|\zeta\Hat{V}\Hat{\cal A}_{AB}|\Psi(\zeta)\rangle
	}{
		\langle\Psi_0|\Hat{\cal A}_{AB}|\Psi(\zeta)\rangle
	} \; ,$$</span><br /> The anti-symmetrizer <span class="math inline">$\Hat{\cal A}_{AB}$</span> in this expression can be written as <br /><span class="math display">$$\label{eq:AB_antisymmetrizer}
	\Hat{\cal A}_{AB} = \frac{N_A! N_B!}{(N_A+N_B)!} \;
	\Hat{\cal A}_A \Hat{\cal A}_B\bigl(
		\Hat{1} + \Hat{\cal P}^{AB} + \Hat{\cal P}'
	\bigr) \; ,$$</span><br /> where <span class="math inline">$\Hat{\cal A}_A$</span> and <span class="math inline">$\Hat{\cal A}_B$</span> are anti-symmetrizers for the two monomers and <span class="math inline">$\Hat{\cal P}^{AB}$</span> is a sum of all one-electron exchange operators between the two monomers. The operator <span class="math inline">$\Hat{\cal P}'$</span> in Eq. <a href="#eq:AB_antisymmetrizer" data-reference-type="eqref" data-reference="eq:AB_antisymmetrizer">[eq:AB_antisymmetrizer]</a> denotes all of the three-electron and higher-order exchanges. This operator is neglected in what is known as the “single-exchange” approximation,<span class="citation" data-cites="Jeziorski:1993 Jeziorski:1994"></span> which is expected to be quite accurate at typical van der Waals and larger intermolecular separations, but sometimes breaks down at smaller intermolecular separations.<span class="citation" data-cites="Lao:2012a"></span></p>
<p>Only terms up to <span class="math inline"><em>ζ</em> = 2</span> in Eq. <a href="#eq:E(SRS)" data-reference-type="eqref" data-reference="eq:E(SRS)">[eq:E(SRS)]</a>—that is, second order in the intermolecular interaction—have been implemented in <span><span class="smallcaps">Q-Chem</span></span>. It is common to relabel these low-order terms in the following way [<span><em>cf</em></span>. Eq. <a href="#eq:triple_pert" data-reference-type="eqref" data-reference="eq:triple_pert">[eq:triple_pert]</a>]: <br /><span class="math display">$$\label{eq:SAPT_interaction}
    E_\ensuremath{\mathrm{int}}^{\rm SAPT0} = E^{(1)}_\ensuremath{\mathrm{elst}} + E^{(1)}_\ensuremath{\mathrm{exch}} + E_\ensuremath{\mathrm{pol}}^{(2)} + E_\ensuremath{\mathrm{exch}}^{(2)} \; .$$</span><br /> The electrostatic part of the first-order energy correction is denoted <span class="math inline"><em>E</em><sub><em>e</em><em>l</em><em>s</em><em>t</em></sub><sup>(1)</sup></span> and represents the Coulomb interaction between the two monomer electron densities.<span class="citation" data-cites="Jeziorski:1994"></span> The quantity <span class="math inline"><em>E</em><sub><em>e</em><em>x</em><em>c</em><em>h</em></sub><sup>(1)</sup></span> is the corresponding first-order (<em>i.e.</em>, Hartree-Fock) exchange correction. Explicit formulas for these corrections can be found in Ref. </p>
<p><span class="citation" data-cites="Jeziorski:1993"></span>. The second-order term from the polarization expansion, denoted <span class="math inline"><em>E</em><sub><em>p</em><em>o</em><em>l</em></sub><sup>(2)</sup></span> in Eq. <a href="#eq:SAPT_interaction" data-reference-type="eqref" data-reference="eq:SAPT_interaction">[eq:SAPT_interaction]</a>, consists of a dispersion contribution (which arises for the first time at second order) as well as a second-order correction for induction. The latter can be written <br /><span class="math display"><em>E</em><sub><em>i</em><em>n</em><em>d</em></sub><sup>(2)</sup> = <em>E</em><sub><em>i</em><em>n</em><em>d</em></sub><sup>(2)</sup>(<em>A</em> ← <em>B</em>) + <em>E</em><sub><em>i</em><em>n</em><em>d</em></sub><sup>(2)</sup>(<em>B</em> ← <em>A</em>) ,</span><br /> where the notation <span class="math inline"><em>A</em> ← <em>B</em></span>, for example, indicates that the frozen charge density of <span class="math inline"><em>B</em></span> polarizes the density of <span class="math inline"><em>A</em></span>. In detail, <br /><span class="math display"><em>E</em><sub><em>i</em><em>n</em><em>d</em></sub><sup>(2)</sup>(<em>A</em> ← <em>B</em>) = 2∑<sub><em>a</em><em>r</em></sub><em>t</em><sub><em>a</em><em>r</em></sub>(<em>w</em><sub><em>B</em></sub><sup></sup>)<sub><em>r</em><em>a</em></sub></span><br /> where <br /><span class="math display">(<em>w</em><sub><em>B</em></sub><sup></sup>)<sub><em>a</em><em>r</em></sub> = (<em>v̂</em><sub><em>B</em></sub><sup></sup>)<sub><em>a</em><em>r</em></sub> + ∑<sub><em>b</em></sub>(<em>a</em><em>r</em>|<em>b</em><em>b</em>)</span><br /> and <span class="math inline"><em>t</em><sub><em>a</em><em>r</em></sub> = (<em>w</em><sub><em>B</em></sub><sup></sup>)<sub><em>a</em><em>r</em></sub>/(<em>ϵ</em><sub><em>a</em></sub> − <em>ϵ</em><sub><em>r</em></sub>)</span>. The second term in Eq. <a href="#eq:E(2)_ind" data-reference-type="eqref" data-reference="eq:E(2)_ind">[eq:E(2)_ind]</a>, in which <span class="math inline"><em>A</em></span> polarizes <span class="math inline"><em>B</em></span>, is obtained by interchanging labels.<span class="citation" data-cites="Jacobson:2011"></span> The second-order dispersion correction has a form reminiscent of the MP2 correlation energy: <br /><span class="math display">$$\label{eq:E(2)_disp}
    E_\ensuremath{\mathrm{disp}}^{(2)} = 4 \sum_{abrs} \frac{  (ar|bs) (ra|sb)}
    {\epsilon_a + \epsilon_b - \epsilon_r - \epsilon_s} \; .$$</span><br /> The induction and dispersion corrections both have accompanying exchange corrections (exchange-induction and exchange-dispersion).<span class="citation" data-cites="Jeziorski:1993 Jeziorski:1994"></span></p>
<p>The similarity between Eq. <a href="#eq:E(2)_disp" data-reference-type="eqref" data-reference="eq:E(2)_disp">[eq:E(2)_disp]</a> and the MP2 correlation energy means that SAPT jobs, like MP2 calculations, can be greatly accelerated using resolution-of-identity (RI) techniques, and an RI version of SAPT is available in <span><span class="smallcaps">Q-Chem</span></span>. To use it, one must specify an auxiliary basis set. The same ones used for RI-MP2 work equally well for RI-SAPT, but one should always select the auxiliary basis set that is tailored for use with the primary basis of interest, as in the RI-MP2 examples in Section <a href="#sect:rimp2" data-reference-type="ref" data-reference="sect:rimp2">[sect:rimp2]</a>.</p>
<p>It is common to replace <span class="math inline"><em>E</em><sub><em>i</em><em>n</em><em>d</em></sub><sup>(2)</sup></span> and <span class="math inline"><em>E</em><sub><em>e</em><em>x</em><em>c</em><em>h</em> − <em>i</em><em>n</em><em>d</em></sub><sup>(2)</sup></span> in Eq. <a href="#eq:SAPT_interaction" data-reference-type="eqref" data-reference="eq:SAPT_interaction">[eq:SAPT_interaction]</a> with their “response” (<span class="math inline"><em>r</em><em>e</em><em>s</em><em>p</em></span>) analogues, which are the infinite-order correction for polarization arising from a frozen partner density.<span class="citation" data-cites="Jeziorski:1993 Jeziorski:1994"></span> Operationally, this substitution involves replacing the second-order induction amplitudes, <span class="math inline"><em>t</em><sub><em>a</em><em>r</em></sub></span> in Eq. <a href="#eq:E(2)_ind_A&lt;-B" data-reference-type="eqref" data-reference="eq:E(2)_ind_A&lt;-B">[eq:E(2)_ind_A&lt;-B]</a>, with amplitudes obtained from solution of the coupled-perturbed Hartree-Fock equations.<span class="citation" data-cites="Pople:1979"></span> (The perturbation is simply the electrostatic potential of the other monomer.) In addition, it is common to correct the SAPT0 binding energy for higher-order polarization effects by adding a correction term of the form<span class="citation" data-cites="Jeziorski:1994 Hohenstein:2012"></span> <br /><span class="math display">$$\label{eq:delta_hf}
    \delta E_\ensuremath{\mathrm{int}}^{\rm HF} = E_\ensuremath{\mathrm{int}}^{\rm HF} - \Bigl( E_\ensuremath{\mathrm{elst}}^{(1)} + E_\ensuremath{\mathrm{exch}}^{(1)}
    + E_\ensuremath{\mathrm{ind,resp}}^{(2)} 
    + E_\ensuremath{\mathrm{exch-ind,resp}}^{(2)}\Bigr)$$</span><br /> to the interaction energy. Here, <span class="math inline">$E_{int}^{\rm HF}$</span> is the counterpoise-corrected Hartree-Fock binding energy for <span class="math inline">$\rm A\cdots B$</span>. Both the response corrections and the <span class="math inline">$\delta E^{\rm HF}_{\rm int}$</span> correction are optionally available in <span><span class="smallcaps">Q-Chem</span></span>’s implementation of SAPT.</p>
<p>It is tempting to replace Hartree-Fock MOs and eigenvalues in the SAPT0 formulas with their Kohn-Sham counterparts, as a low-cost means of introducing monomer electron correlation. The resulting procedure is known as SAPT(KS),<span class="citation" data-cites="Williams:2001"></span> and does offer an improvement on SAPT0 for some strongly hydrogen-bonded systems.<span class="citation" data-cites="Herbert:2012"></span> Unfortunately, SAPT(KS) results are generally in poor agreement with benchmark dispersion energies,<span class="citation" data-cites="Herbert:2012"></span> owing to incorrect asymptotic behavior of approximate exchange-correlation potentials.<span class="citation" data-cites="Misquitta:2002"></span> The dispersion energies can be greatly improved through the use of long-range corrected (LRC) functionals in which the range-separation parameter, <span class="math inline"><em>ω</em></span>, is “tuned” so as to satisfy the condition <span class="math inline">$\epsilon_{\rm HOMO}^{} =
-\mbox{IP}$</span>, where <span class="math inline">$\epsilon_{\rm HOMO}^{}$</span> is the HOMO energy and “IP&quot; represents the ionization potential.<span class="citation" data-cites="Lao:2014"></span> Monomer-specific values of <span class="math inline"><em>ω</em></span>, tuned using the individual monomer IPs, substantially improve SAPT(KS) dispersion energies, though the results are still not of benchmark quality.<span class="citation" data-cites="Lao:2014"></span> Other components of the interaction energy, however, can be described quite accurately SAPT(KS) in conjunction with a tuned version of LRC-<span class="math inline"><em>ω</em></span>PBE.<span class="citation" data-cites="Lao:2014"></span> Use of monomer-specific <span class="math inline"><em>ω</em></span> values is controlled by the variable in the <em>$rem</em> section, and individual values are entered via an <em>$lrc_omega</em> input section. The omega values of monomers at geometries optimized at RIMP2/aug-cc-pVDZ using LRC-<span class="math inline"><em>ω</em></span>PBE functional based on <span class="math inline">$\epsilon_{\rm HOMO}^{} = -\mbox{IP}$</span> condition are listed in Table <a href="#table:omega-tuning" data-reference-type="ref" data-reference="table:omega-tuning">[table:omega-tuning]</a>.</p>
<table>
<caption> Tuned values of the range separation parameter, <span class="math inline"><em>ω</em></span>. <span label="table:omega-tuning"></span></caption>
<thead>
<tr class="header">
<th style="text-align: left;">Monomer</th>
<th style="text-align: center;"><span class="math inline">$\omega^{}_{\rm IP}$</span> / <span class="math inline"><em>a</em><sub>0</sub><sup> − 1</sup></span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">adenine</td>
<td style="text-align: center;">0.271</td>
</tr>
<tr class="even">
<td style="text-align: left;">2-aminopyridine</td>
<td style="text-align: center;">0.293</td>
</tr>
<tr class="odd">
<td style="text-align: left;">benzene</td>
<td style="text-align: center;">0.280</td>
</tr>
<tr class="even">
<td style="text-align: left;">ethyne</td>
<td style="text-align: center;">0.397</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ethene</td>
<td style="text-align: center;">0.359</td>
</tr>
<tr class="even">
<td style="text-align: left;">methane</td>
<td style="text-align: center;">0.454</td>
</tr>
<tr class="odd">
<td style="text-align: left;">formamide</td>
<td style="text-align: center;">0.460</td>
</tr>
<tr class="even">
<td style="text-align: left;">formic acid</td>
<td style="text-align: center;">0.412</td>
</tr>
<tr class="odd">
<td style="text-align: left;">water</td>
<td style="text-align: center;">0.502</td>
</tr>
<tr class="even">
<td style="text-align: left;">HCN</td>
<td style="text-align: center;">0.452</td>
</tr>
<tr class="odd">
<td style="text-align: left;">indole</td>
<td style="text-align: center;">0.267</td>
</tr>
<tr class="even">
<td style="text-align: left;">ammonia</td>
<td style="text-align: center;">0.440</td>
</tr>
<tr class="odd">
<td style="text-align: left;">phenol</td>
<td style="text-align: center;">0.292</td>
</tr>
<tr class="even">
<td style="text-align: left;">pyrazine</td>
<td style="text-align: center;">0.367</td>
</tr>
<tr class="odd">
<td style="text-align: left;">2-pyridoxine</td>
<td style="text-align: center;">0.294</td>
</tr>
<tr class="even">
<td style="text-align: left;">thymine</td>
<td style="text-align: center;">0.284</td>
</tr>
<tr class="odd">
<td style="text-align: left;">uracil</td>
<td style="text-align: center;">0.295</td>
</tr>
<tr class="even">
<td style="text-align: left;"><span class="math inline">$\rm{MeNH_2}$</span></td>
<td style="text-align: center;">0.397</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><span class="math inline">$\rm{MeOH}$</span></td>
<td style="text-align: center;">0.438</td>
</tr>
<tr class="even">
<td style="text-align: left;"><span class="math inline">$\rm{AcNH_2}$</span></td>
<td style="text-align: center;">0.453</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><span class="math inline">$\rm{AcOH}$</span></td>
<td style="text-align: center;">0.381</td>
</tr>
<tr class="even">
<td style="text-align: left;">cyclopentane</td>
<td style="text-align: center;">0.420</td>
</tr>
<tr class="odd">
<td style="text-align: left;">neopentane</td>
<td style="text-align: center;">0.287</td>
</tr>
<tr class="even">
<td style="text-align: left;">pentane</td>
<td style="text-align: center;">0.365</td>
</tr>
<tr class="odd">
<td style="text-align: left;">peptide</td>
<td style="text-align: center;">0.341</td>
</tr>
<tr class="even">
<td style="text-align: left;">pyridine</td>
<td style="text-align: center;">0.316</td>
</tr>
<tr class="odd">
<td style="text-align: left;">F<span class="math inline"><em></em><sup>−</sup></span></td>
<td style="text-align: center;">0.480</td>
</tr>
<tr class="even">
<td style="text-align: left;">Cl<span class="math inline"><em></em><sup>−</sup></span></td>
<td style="text-align: center;">0.372</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SO<span class="math inline"><em></em><sub>4</sub><sup>2−</sup></span></td>
<td style="text-align: center;">0.344</td>
</tr>
<tr class="even">
<td style="text-align: left;">Li<span class="math inline"><em></em><sup>+</sup></span></td>
<td style="text-align: center;">2.006</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Na<span class="math inline"><em></em><sup>+</sup></span></td>
<td style="text-align: center;">1.049</td>
</tr>
<tr class="even">
<td style="text-align: left;">K<span class="math inline"><em></em><sup>+</sup></span></td>
<td style="text-align: center;">0.755</td>
</tr>
</tbody>
</table>
<p>Finally, some discussion of basis sets is warranted. Typically, SAPT calculations are performed in the so-called dimer-centered basis set (DCBS),<span class="citation" data-cites="Williams:1995"></span> which means that the combined <span class="math inline"><em>A</em> + <em>B</em></span> basis set is used to calculate the zeroth-order wave functions for both <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span>. This leads to the unusual situation that there are more MOs than basis functions: one set of occupied and virtual MOs for each monomer, both expanded in the same (dimer) AO basis. As an alternative to the DCBS, one might calculate <span class="math inline">|<em>Ψ</em><sub>​<em>A</em></sub>⟩</span> using only <span class="math inline"><em>A</em></span>’s basis functions (similarly for <span class="math inline"><em>B</em></span>), in which case the SAPT calculation is said to employ the monomer-centered basis set (MCBS).<span class="citation" data-cites="Williams:1995"></span> However, MCBS results are generally of poorer quality. As an efficient alternative to the DCBS, Jacobson and Herbert<span class="citation" data-cites="Jacobson:2011"></span> introduced a projected (“proj”) or “pseudocanonicalized” basis set, borrowing an idea from dual-basis MP2 calculations.<span class="citation" data-cites="Steele:2006a"></span> In this approach, the SCF iterations are performed in the MCBS but then Fock matrices for fragments <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span> are constructed in the dimer (<span class="math inline"><em>A</em> + <em>B</em></span>) basis set and then pseudocanonicalized, meaning that the occupied-occupied and virtual-virtual blocks of these matrices are diagonalized. This procedure does not mix occupied and virtual orbitals, and thus leaves the fragment densities and zeroth-order fragment energies unchanged. However, it does provide a larger set of virtual orbitals that extend over the partner fragment. This larger virtual space is then used to evaluate the perturbative corrections. All three of these basis options (MCBS, DCBS, and projected basis) are available in <span><span class="smallcaps">Q-Chem</span></span>.</p>
<h3 id="job-control-for-sapt-calculations">Job Control for SAPT Calculations</h3>
<p><span><span class="smallcaps">Q-Chem</span></span>’s implementation of SAPT0 was designed from the start as a correction for XPol calculations, a functionality that is described in Section <a href="#sect:XSAPT" data-reference-type="ref" data-reference="sect:XSAPT">13.12</a>. As such, a SAPT calculation is requested by setting both of the <em>$rem</em> variable and to . (Alternatively, one may set = to use the RI version of SAPT.) If one wishes to perform a traditional SAPT calculation based on gas-phase SCF monomer wave functions rather than XPol monomer wave functions, then the <em>$rem</em> variable should be set to .</p>
<p>SAPT energy components are printed separately at the end of a SAPT job. If = , then the calculation corresponds to SAPT0, whereas a SAPT(KS) calculation is requested by specifying the desired density functional.</p>
<p><strong>Note: </strong></p>
<p>(1) Meta-GGAs are not yet available for SAPT(KS) calculations when the projected (pseudocanonicalized) basis set is used. SAPT(KS) calculations can be performed with meta-GGAs using the monomer or dimer basis sets.<br />
(2) Both closed- and open-shell (unrestricted) SAPT(KS) calculations are available.<br />
(3) Frozen orbitals are <em>not</em> available for use with SAPT(KS).</p>
<p>Researchers who use <span><span class="smallcaps">Q-Chem</span></span>’s SAPT code are asked to cite Refs. </p>
<p><span class="citation" data-cites="Jacobson:2011"></span> and</p>
<p><span class="citation" data-cites="Herbert:2012"></span>.</p>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>

<p><strong>Example .</strong>  Example showing a SAPT0 calculation using the RI approximation in a DCBS.</p>
<blockquote>
<pre><code>$rem
   BASIS            AUG-CC-PVDZ
   AUX_BASIS        RIMP2-AUG-CC-PVDZ
   METHOD           HF
   RISAPT           TRUE
   XPOL             TRUE
   XPOL_MPOL_ORDER  GAS   ! gas-phase monomer wave functions
   SAPT_BASIS       DIMER
   SYM_IGNORE       TRUE
$end

$molecule
0 1
-- formamide
   0 1
   C -2.018649  0.052883  0.000000
   O -1.452200  1.143634  0.000000
   N -1.407770 -1.142484  0.000000
   H -1.964596 -1.977036  0.000000
   H -0.387244 -1.207782  0.000000
   H -3.117061 -0.013701  0.000000
-- formamide
   0 1
   C  2.018649 -0.052883  0.000000
   O  1.452200 -1.143634  0.000000
   N  1.407770  1.142484  0.000000
   H  1.964596  1.977036  0.000000
   H  0.387244  1.207782  0.000000
   H  3.117061  0.013701  0.000000
$end</code></pre>
</blockquote>
<h2 id="sect:XSAPT">The XPol+SAPT (XSAPT) Method</h2>
<h3 id="theory-2">Theory</h3>
<p>XPol+SAPT, or “XSAPT”, was introduced by Jacobson and Herbert<span class="citation" data-cites="Jacobson:2011 Herbert:2012"></span> (later significantly extended by Lao and Herbert<span class="citation" data-cites="Lao:2012b Lao:2013 Lao:2015 Lao:2018"></span>) as a low-scaling, systematically-improvable method for intermolecular interactions that could be applicable to large systems. The original idea was to replace the need for empirical parameters in the XPol method with on-the-fly evaluation of exchange-repulsion and dispersion interactions via pairwise-additive SAPT. Stated differently, XSAPT uses XPol to evaluate many-body (non-pairwise-additive) polarization effects, but then assumes that dispersion and exchange-repulsion interactions <span><em>are</em></span> pairwise additive, and evaluates them via pairwise SAPT0 or SAPT(KS) calculations. An overview of XSAPT-based methods can be found in Ref. </p>
<p><span class="citation" data-cites="Lao:2015"></span>.</p>
<p>The zeroth-order Hamiltonian for XSAPT is taken by the sum of fragment Fock operators defined by the XPol procedure, and the perturbation is the usual SAPT intermolecular perturbation [Eq. <a href="#eq:pairwise_perturbation" data-reference-type="eqref" data-reference="eq:pairwise_perturbation">[eq:pairwise_perturbation]</a>] less the intermolecular interactions contained in the XPol fragment Fock operators. A standard SAPT0 correction is then computed for each pair of monomers, using Eq. <a href="#eq:SAPT_interaction" data-reference-type="eqref" data-reference="eq:SAPT_interaction">[eq:SAPT_interaction]</a> in conjunction with the modified perturbation, to obtain dimer interaction energy <span class="math inline"><em>E</em><sub><em>i</em><em>n</em><em>t</em></sub><sup><em>A</em><em>B</em></sup></span>. The total XSAPT energy is then <br /><span class="math display">$$\label{eq:XSAPT_total_E}
  E^{}_{\rm XSAPT} =  \sum_A \left(   
  	\sum_a \Bigl[ 2\,\epsilon_a^A - \mathbf{c}_a^\dagger 
		( \mathbf{J}^A - \tfrac{1}{2}\mathbf{K}^A)\mathbf{c}_a\Bigr] 
               +  E_\ensuremath{\mathrm{nuc}}^A  + \sum_{B&gt;A} E_\ensuremath{\mathrm{int}}^{AB} \right) \; .$$</span><br /> In this expression, we have removed the over-counting of two-electron interactions present in Hartree-Fock theory, effectively taking the <em>intra</em>fragment perturbation to first order. The generalization to a Kohn-Sham description of the monomers is straightforward, and is available in <span><span class="smallcaps">Q-Chem</span></span>.</p>
<p>The inclusion of many-body polarization within the zeroth-order Hamiltonian makes the subsequent SAPT corrections less meaningful in terms of energy decomposition analysis. For instance, the first-order electrostatic correction in XSAPT is <span><em>not</em></span> the total electrostatic energy, since the former corrects for errors in the approximate electrostatic treatment at zeroth order (<em>i.e.</em>, the electrostatic embedding). The dispersion correction may be less contaminated, since all of the XSAPT modifications to the traditional SAPT perturbation are one-electron operators and therefore the pairwise dispersion correction differs from its traditional SAPT analogue only insofar as the MOs are perturbed by the electrostatic embedding. This should be kept in mind when interpreting the output of an XSAPT calculation, although Lao and Herbert<span class="citation" data-cites="Lao:2013 Lao:2015"></span> later proposed a many-body energy decomposition scheme for XSAPT that extends traditional SAPT energy decomposition to systems containing more than two monomers. (The aforementioned contamination problems are avoid through pairwise <span class="math inline">$\delta_\ensuremath{\mathrm{int}}^{\rm HF}$</span> corrections, comparing XSAPT results to traditional SAPT based on gas-phase monomers.) An XSAPT calculation is requested by setting the <em>$rem</em> variables and equal to and also setting = .</p>
<p>Researchers who use <span><span class="smallcaps">Q-Chem</span></span>’s XPol+SAPT code are asked to cite Refs. </p>
<p><span class="citation" data-cites="Jacobson:2011"></span> and</p>
<p><span class="citation" data-cites="Herbert:2012"></span>. The latter contains a thorough discussion of the theory; a briefer summary can be found in Ref. </p>
<p><span class="citation" data-cites="Jacobson:2013"></span>.</p>

<p><strong>Example .</strong>  Example showing an XPol+SAPT0 calculation using ChElPG charges and CPHF.</p>
<blockquote>
<pre><code>$rem
   BASIS            CC-PVDZ
   METHOD           HF
   XPOL             TRUE
   XPOL_MPOL_ORDER  CHARGES
   XPOL_CHARGE_TYPE QCHELPG
   SAPT             TRUE
   SAPT_CPHF        TRUE
   SYM_IGNORE       TRUE
$end

$molecule
0 1
-- formic acid
   0 1
   C  -1.888896 -0.179692  0.000000
   O  -1.493280  1.073689  0.000000
   O  -1.170435 -1.166590  0.000000
   H  -2.979488 -0.258829  0.000000
   H  -0.498833  1.107195  0.000000
-- formic acid
   0  1
   C  1.888896  0.179692  0.000000
   O  1.493280 -1.073689  0.000000
   O  1.170435  1.166590  0.000000
   H  2.979488  0.258829  0.000000
   H  0.498833 -1.107195  0.000000
$end</code></pre>
</blockquote>
<h3 id="subsec:AO-XSAPT(KS)+D">AO-XSAPT(KS)+<em>ai</em>D</h3>
<p>As mentioned above, the dispersion components of the SAPT(KS) or XSAPT(KS) interaction energy are not of benchmark quality, even when tuned LRC functionals are employed.<span class="citation" data-cites="Lao:2014"></span> It happens that the dispersion and exchange-dispersion terms are also the most expensive part of a SAPT0 or SAPT(KS) calculation, scaling as the fourth and fifth powers of monomer size, respectively, whereas other terms are cubic scaling at worst. Both the efficiency and the accuracy of XSAPT(KS) calculations is thus improved if second-order dispersion (<span class="math inline"><em>E</em><sub><em>d</em><em>i</em><em>s</em><em>p</em></sub><sup>(2)</sup> + <em>E</em><sub><em>e</em><em>x</em><em>c</em><em>h</em> − <em>d</em><em>i</em><em>s</em><em>p</em></sub><sup>(2)</sup></span>) is replaced by an empirical atom–atom dispersion potential. Because the dispersion energy is well-defined (and separable) within the SAPT formalism, it can be replaced by atom–atom dispersion potentials (of the <span class="math inline"> − <em>C</em><sub>6</sub>/<em>R</em><sup>6</sup> − <em>C</em><sub>8</sub>/<em>R</em><sup>8</sup> − ⋯</span> variety) without any double-counting problem (as there is in empirical dispersion corrections for DFT). Moreover, these dispersion potentials can be fit directly to <em>ab initio</em> dispersion energies from high-level SAPT calculations [SAPT(DFT) and SAPT2+(3)], such that the dispersion potential, while it is classical in its form and does contain fitting parameters, can reasonably said to be an <em>ab initio</em> dispersion potential. To distinguish this approach from DFT-D (Section <a href="#subsec:DFT-D" data-reference-type="ref" data-reference="subsec:DFT-D">[subsec:DFT-D]</a>), which is more empirical and does have a potential double-counting problem, we now refer to this <em>ab initio</em> dispersion correction as “+<em>ai</em>D”,<span class="citation" data-cites="Lao:2018"></span> although it was called simply “+D” in earlier work.<span class="citation" data-cites="Lao:2012b Lao:2013 Lao:2015"></span> The composite method is called XSAPT(KS)+<em>ai</em>D; see Ref. </p>
<p><span class="citation" data-cites="Lao:2013"></span> for an overview. A more efficient, atomic orbital (AO) implementation was reported quite recently,<span class="citation" data-cites="Lao:2018"></span> extending the method to supramolecular complexes containing large monomers.</p>
<p>An XSAPT(KS)+<em>ai</em>D calculation is requested by setting and equal to . There are three versions of the <em>ab initio</em> the dispersion potential: “first generation&quot; (+<em>ai</em>D1),<span class="citation" data-cites="Lao:2012b"></span> second-generation (+<em>ai</em>D2),<span class="citation" data-cites="Lao:2013"></span> and third-generation (+<em>ai</em>D3) version.<span class="citation" data-cites="Lao:2015"></span> The user can select amongst these using the <em>$rem</em> variable. Although all three versions exhibit similar performance for total interaction energies,<span class="citation" data-cites="Lao:2015"></span> only the +<em>ai</em>D2 and +<em>ai</em>D3 potentials were fit directly to <em>ab initio</em> dispersion data (rather than being fit to reproduce interaction energies themselves), and they do a much better job of reproducing individual energy components, as compared to +<em>ai</em>D1.<span class="citation" data-cites="Lao:2013 Lao:2015"></span> The latter succeeds by error cancellation amongst the energy components and is not recommended. The difference between +<em>ai</em>D2 and +<em>ai</em>D3 is a larger training set for the latter, which was designed to afford better coverage of <span class="math inline"><em>π</em></span>-stacked systems. The +<em>ai</em>D3 correction is the recommended one.</p>
<p>As with XPol, the XSAPT and XSAPT(KS)+<em>ai</em>D methods do not function with a solvation model or with external changes. Only single-point energies are available, and frozen orbitals orbitals are not allowed. Both restricted and unrestricted versions are available. Researchers who use XSAPT(KS)+<em>ai</em>D are asked to cite Ref. </p>
<p><span class="citation" data-cites="Lao:2012b"></span> for +<em>ai</em>D1, Ref. </p>
<p><span class="citation" data-cites="Lao:2013"></span> for +<em>ai</em>D2, or Ref. </p>
<p><span class="citation" data-cites="Lao:2015"></span> for +<em>ai</em>D3. The AO implementation of XSAPT is described in Ref. </p>
<p><span class="citation" data-cites="Lao:2018"></span>.</p>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>

<p><strong>Example .</strong>  AO-XSAPT(KS)+D3 calculation of water-water interaction.</p>
<blockquote>
<pre><code>$rem
   SYM_IGNORE         true
   EXCHANGE           gen
   BASIS              aug-cc-pVTZ
   XPOL               true ! must be set to true for sapt jobs too
   XPOL_MPOL_ORDER    charges 
   XPOL_CHARGE_TYPE   qchelpg  
   XPOL_OMEGA         true
   XPOL_PRINT         3
   SAPT_PRINT         3
   SAPT               true
   SAPT_AO            true
   SAPT_ORDER         2          ! can be set to 1, ELST or RSPT
   SAPT_BASIS         projected ! monomer, dimer (if only 2 monomers), or projected
   SAPT_DISP_CORR     true
   SAPT_DISP_VERSION  3
   MEM_TOTAL          46000
   MEM_STATIC         4000
   AO2MO_DISK         35000
   THRESH             12
   SCF_CONVERGENCE    7
   LRC_DFT            true
   CHELPG             true
   CHELPG_DX          5
   CHELPG_HEAD        30
   CHELPG_H           110
   CHELPG_HA          590
$end

$xc_functional
   x   wPBE   1.0
   c   PBE   1.0
$end

$lrc_omega
   502
   502
$end

$molecule 
0 1
--
   0 1
   O  -1.551007  -0.114520   0.000000
   H  -1.934259   0.762503   0.000000
   H  -0.599677   0.040712   0.000000
--
   0 1
   O   1.350625   0.111469   0.000000
   H   1.680398  -0.373741  -0.758561
   H   1.680398  -0.373741   0.758561
$end</code></pre>
</blockquote>
<h2 id="energy-decomposition-analysis-based-on-saptcdft">Energy Decomposition Analysis based on SAPT/cDFT</h2>
<p><span id="sect:SAPTEDA" label="sect:SAPTEDA">[sect:SAPTEDA]</span></p>
<p>Many schemes for decomposing quantum chemical calculations of intermolecular interaction energies into physically meaningful components can be found in the literature, but the definition of the charge-transfer (CT) contribution has proven particularly vexing to define in a satisfactory way and typically depends strongly on the choice of basis set,<span class="citation" data-cites="Ronca:2014 Rezac:2015 Lao:2016b"></span> because as virtual orbitals on monomer <span class="math inline"><em>A</em></span> start to extend significantly over monomer <span class="math inline"><em>B</em></span> as the basis set approaches completeness, the distinction between polarization (excitations localized on <span class="math inline"><em>A</em></span>, introduced by the perturbing influence of <span class="math inline"><em>B</em></span>) and CT (excitations from <span class="math inline"><em>A</em></span> to <span class="math inline"><em>B</em></span>) becomes blurred.<span class="citation" data-cites="Lao:2016b"></span> This ambiguity renders orbital-dependent definitions of CT highly dependent on the choice of atomic orbital basis set. On the other hand, constrained density functional theory (cDFT, Section <a href="#sect:CDFT" data-reference-type="ref" data-reference="sect:CDFT">[sect:CDFT]</a>), where a CT-free reference state can be defined based on “promolecule” densities, affords a definition of CT that is scarcely dependent on the basis set and is in accord with chemical intuition in simple cases.<span class="citation" data-cites="Lao:2016b"></span></p>
<p>For intermolecular interactions, the cDFT definition of CT can be combined with a definition of the remaining components of the interaction energy (electrostatics, induction, Pauli repulsion, and van der Waals interactions) based on symmetry-adapted perturbation theory (SAPT, Section <a href="#sec:sapt" data-reference-type="ref" data-reference="sec:sapt">13.11</a>). In traditional SAPT, the CT interaction energy resides within the induction energy (also known as the polarization energy), which is therefore itself highly dependent upon the basis set. However, using cDFT to define the CT component and subtracting this out of the SAPT induction energy, both the CT and the remaining induction energies are largely independent of basis set.<span class="citation" data-cites="Lao:2016b"></span> SAPT/cDFT therefore provides a stable and physically-motivated energy decomposition, which can be invoked by setting the <em>$rem</em> variable = in a SAPT calculation. A <em>$cdft</em> section must be set to specify the monomer charges and spins for the cDFT calculation.</p>
<p>While the cDFT definition of CT exhibits only a very mild basis-set dependence, its quantitative details do depend upon how the charge constraints in cDFT are defined relative to fragment populations (Section <a href="#sect:CDFT" data-reference-type="ref" data-reference="sect:CDFT">[sect:CDFT]</a>). For SAPT/cDFT, both atomic Becke<span class="citation" data-cites="Becke:1988a"></span> and fragment-based Hirshfeld<span class="citation" data-cites="Rezac:2015"></span> (FBH) charge partitioning methods are available. The former involves construction of atomic cell functions that amount to smoothed Voronoi polyhedra centered about each atom. A switching function defines the atomic cell of atom <span class="math inline"><em>a</em></span>, and falls rapidly from <span class="math inline"> ≈ 1</span> near the nucleus for atom <span class="math inline"><em>a</em></span>, to <span class="math inline"> ≈ 0</span> near any other nucleus. Becke<span class="citation" data-cites="Becke:1988a"></span> defined atomic cell functions <span class="math inline"><em>P</em><sub><em>a</em></sub>(<strong>r</strong>)</span> that are products of switching functions and that can be used to define the cDFT integration weight for monomer <span class="math inline"><em>A</em></span> by summing over atoms <span class="math inline"><em>a</em> ∈ <em>A</em></span>: <br /><span class="math display">$$\label{eq:becky_weights}
	w^{\rm Becke}_{A}(\textbf{r}) = 
	\frac{
           \sum_{a\in A} P_{a}(\textbf{r}) 
        }{
           \sum_b P_{b}(\textbf{r}) 
        } \; .$$</span><br /> The sum in the denominator runs over all atoms in both monomers, <span class="math inline"><em>A</em></span> and <span class="math inline"><em>B</em></span>. Becke populations, however, are rooted in a somewhat arbitrarily-defined topology, based in part on assumed atomic radii, whereas FBH partitioning derives physical significance from isolated monomer densities <span class="math inline"><em>ρ̃</em><sub><em>A</em></sub><sup></sup>(<strong>r</strong>)</span> and <span class="math inline"><em>ρ̃</em><sub><em>B</em></sub><sup></sup>(<strong>r</strong>)</span>. The cDFT weight function for monomer <span class="math inline"><em>A</em></span> is<span class="citation" data-cites="Rezac:2015"></span> <br /><span class="math display">$$\label{eq:fbh_weights}
	w_A^{\rm FBH}(\mathbf{r}) = \frac{
		\widetilde{\rho}^{}_A(\mathbf{r})
	}{
		\widetilde{\rho}^{}_A(\mathbf{r}) + \widetilde{\rho}^{}_B(\mathbf{r})
	} \; ,
	%w_{A}(\textbf{r}) = \sum\limits_{A}^{N_{frag}}\frac{\rho_{A}(\textbf{r})\delta_{AB}}{\widetilde{\rho}(\textbf{r})}$$</span><br /> which is the same “stockholder” scheme used to define atomic Hirshfeld populations (Section <a href="#subsec:pop_anal" data-reference-type="ref" data-reference="subsec:pop_anal">[subsec:pop_anal]</a>), but applied here to the entire monomer. In the language of cDFT, the denominator in this expression would be called the <em>promolecule</em> density for the dimer <span class="math inline"><em>A</em> + <em>B</em></span>. In order to set a molecular fragment constraint, simply retain the existing syntax in the <em>$cdft</em> input section (as described in Section <a href="#sect:CDFT" data-reference-type="ref" data-reference="sect:CDFT">[sect:CDFT]</a>) and specify all atoms within a given molecular fragment.</p>
<p>To perform SAPT/cDFT energy decomposition analysis, the user must request a normal SAPT calculation and in addition set = . Users of this method are asked to cite Ref. </p>
<p><span class="citation" data-cites="Lao:2016b"></span>.</p>
<blockquote>

</blockquote>
<blockquote>

</blockquote>

<p><strong>Example .</strong>  Energy decomposition analysis for the water dimer using AO-SAPT+<em>ai</em>D3/cDFT.</p>
<blockquote>
<pre><code>$comment
   a $cdft input section must be set to specify the monomer 
   charges and spins for the cDFT calculation.
$end

$rem
   SYM_IGNORE         true
   EXCHANGE           gen
   BASIS              aug-cc-pvdz
   XPOL               true ! must be set to true for sapt jobs too
   XPOL_MPOL_ORDER    gas ! gas or charges
   XPOL_OMEGA         true
   XPOL_PRINT         3
   SAPT_PRINT         3
   SAPT               true
   SAPT_AO            true
   SAPT_ORDER         2       ! can be set to 1, ELST or RSPT
   SAPT_BASIS         dimer ! monomer, dimer (if only 2 monomers), or projected
   SAPT_DISP_CORR     true
   SAPT_DISP_VERSION  3
   LRC_DFT            true
   SAPT_CDFT_EDA      true
   CDFT_POP           fbh ! Fragment-Based Hirshfeld (FBH) charge partitioning
$end

$xc_functional
   x   wPBE   1.0
   c    PBE   1.0
$end

$lrc_omega
   500
   500
$end

$cdft
   0
   1 1 3
   0
   1 1 3 s
$end

$molecule
0 1
--
   0 1
   O   -0.702196054  -0.056060256   0.009942262
   H   -1.022193224   0.846775782  -0.011488714
   H    0.257521062   0.042121496   0.005218999
--
0 1
   O    2.220871067   0.026716792   0.000620476
   H    2.597492682  -0.411663274   0.766744858
   H    2.593135384  -0.449496183  -0.744782026
$end</code></pre>
</blockquote>
<h2 id="sec:mbe">The Many-Body Expansion Method</h2>
<h3 id="theory-and-implementation-details">Theory and Implementation Details</h3>
<p>The many-body expansion (MBE) for a system of <span class="math inline"><em>N</em></span> monomers is given by <br /><span class="math display">$$\label{eq:mb}
	E = \sum_{I=1}^N\mbox{$E_{I}^{}$}+\sum_{I}^{N}\sum_{J&gt;I}^{N}\mbox{$\Delta E_{IJ}^{}$}+ \\
	\sum_{I}^{N}\sum_{J&gt;I}^{N}\sum_{K&gt;J}^{N}\mbox{$\Delta E_{IJK}^{}$} + \\
	 \cdots ,$$</span><br /> in which <span class="math inline"><em>E</em><sub><em>I</em></sub></span> represents the energy of monomer <span class="math inline"><em>I</em></span>, <span class="math inline"><em>Δ</em><em>E</em><sub><em>I</em><em>J</em></sub></span> = <span class="math inline"><em>E</em><sub><em>I</em><em>J</em></sub></span> <span class="math inline">−</span> <span class="math inline"><em>E</em><sub><em>I</em></sub></span> <span class="math inline">−</span> <span class="math inline"><em>E</em><sub><em>J</em></sub></span> is a two-body correction for dimer <span class="math inline"><em>I</em><em>J</em></span>, and <span class="math inline"><em>Δ</em><em>E</em><sub><em>I</em><em>J</em><em>K</em></sub></span> = <span class="math inline"><em>E</em><sub><em>I</em><em>J</em><em>K</em></sub></span> <span class="math inline">−</span> <span class="math inline"><em>Δ</em><em>E</em><sub><em>I</em><em>J</em></sub></span> <span class="math inline">−</span> <span class="math inline"><em>Δ</em><em>E</em><sub><em>I</em><em>K</em></sub></span> <span class="math inline">−</span> <span class="math inline"><em>Δ</em><em>E</em><sub><em>J</em><em>K</em></sub></span> <span class="math inline">−</span> <span class="math inline"><em>E</em><sub><em>I</em></sub></span> <span class="math inline">−</span> <span class="math inline"><em>E</em><sub><em>J</em></sub></span> <span class="math inline">−</span> <span class="math inline"><em>E</em><sub><em>k</em></sub></span> is a three-body correction for trimer <span class="math inline"><em>I</em><em>J</em><em>K</em></span>, etc. In a large system and/or a large basis set, truncation of this expression at the two- or three-body level may dramatically reduce the amount of computer time that is required to compute the energy. Convergence of the MBE can be accelerated by embedding the monomer (<span class="math inline"><em>E</em><sub><em>I</em></sub></span>), dimer (<span class="math inline"><em>E</em><sub><em>I</em><em>J</em></sub></span>), trimer (<span class="math inline"><em>E</em><sub><em>I</em><em>J</em><em>K</em></sub></span>), <span class="math inline">…</span> calculations in some representation of the electrostatic potential of the rest of the system. A simple means to do this is via atom-centered point charges that could be obtained when the <span class="math inline"><em>E</em><sub><em>I</em></sub></span> terms are calculated; this is the so-called electrostatically-embedded many-body expansion (EE-MBE).<span class="citation" data-cites="Dahlke:2007 Richard:2014a Richard:2014b Lao:2016a"></span> Alternatively, since the monomer electron densities are available from the <span class="math inline"><em>E</em><sub><em>I</em></sub></span> terms as well, one could use these densities to compute the actual monomer–monomer Coulomb interactions, and this forms the basis of the fragment molecular orbital (FMO) method.<span class="citation" data-cites="Kitaura:1999 Fedorov:2007"></span> In particular, the “bodies” (fragments) cannot be covalently bonded to one another, and therefore this is a method appropriate for non-covalent clusters of molecules. Moreover, individual subsystem calculations have been parallelized across processors using MPI, hence the present implementation is using the real power of the MBE. Analytic gradients are also available for MBE and EE-MBE, but not FMO.</p>
<p>It is well known that the interaction energies of non-covalent clusters are usually overestimated (often substantially) owing to basis-set superposition error (BSSE), which disappears only very slowly as the basis sets approach completeness. The widely used Boys–Bernardi counterpoise (CP) procedure corrects for this by computing all energies (including cluster and monomers) using the cluster basis set. (Note, however, that basis-set extrapolation is still necessary for high-quality binding energies; in <span class="math inline">$(\rm H_2O)_6$</span>, for example, a CP-corrected MP2/aug-cc-pVQZ calculation is still <span class="math inline"> ≈ 1</span> kcal/mol from the MP2 basis-set limit.<span class="citation" data-cites="Richard:2013a"></span> Fortunately, the MBE allows for use of large basis sets in order to perform basis-set extrapolations in sizable clusters.<span class="citation" data-cites="Richard:2013a Richard:2013b"></span>) Two low-cost CP corrections that are consistent with an <span class="math inline"><em>n</em></span>-body expansion have been proposed: the many-body CP correction, MBCP(<span class="math inline"><em>n</em></span>),<span class="citation" data-cites="Richard:2013a Richard:2013b"></span> and the <span class="math inline"><em>n</em></span>-body Valiron-Mayer function counterpoise correction, VMFC(<span class="math inline"><em>n</em></span>).<span class="citation" data-cites="Kamiya:2008"></span> The two approaches are equivalent for <span class="math inline"><em>n</em> = 2</span> but the MBCP(<span class="math inline"><em>n</em></span>) method requires far fewer subsystem calculations starting at <span class="math inline"><em>n</em> = 3</span> and is thus significantly cheaper, while affording very similar results as compared to VMFC(<span class="math inline"><em>n</em></span>).<span class="citation" data-cites="Richard:2013a Richard:2013b"></span></p>
<p><span><span class="smallcaps">Q-Chem</span></span>’s implementation of the EE-MBE(<span class="math inline"><em>n</em></span>) approach (electrostatically-embedded <span class="math inline"><em>n</em></span>-body expansion) is designed to use <em>$mbe_charges</em> section to specify embedding charges. As such, a MBE calculation is requested by setting = . The variable = sets the MBE calculations with or without using embedding charges. The variable sets the truncation order, <span class="math inline"><em>n</em></span>. As an alternative to point charges, density embedding is also available, in which Coulomb interactions are computed between proper monomer electron densities. The MBE with density embedding is equivalent to the original version of the fragment molecular orbital (FMO).<span class="citation" data-cites="Kitaura:1999"></span> (Many subsequent modifications to the FMO algorithm have been introduced<span class="citation" data-cites="Fedorov:2007 Fedorov:2009"></span> but are not yet available in <span><span class="smallcaps">Q-Chem</span></span>. These include, in particular, the option to use point charges or approximate electron repulsion integrals to compute the Coulomb interactions between distant monomers,<span class="citation" data-cites="Nakano:2002 Fedorov:2010"></span> which actually makes FMO more like EE-MBE at long range.) This density-embedded version of FMO is available in <span><span class="smallcaps">Q-Chem</span></span> by setting = , = , and = . The variable sets the truncation order, <span class="math inline"><em>n</em></span>. The MBE version of BSSE is requested by setting = or . The variable sets the truncation order, <span class="math inline"><em>n</em></span>.</p>
<p>Researchers who use <span><span class="smallcaps">Q-Chem</span></span>’s MBE code are asked to cite Ref. </p>
<p><span class="citation" data-cites="Richard:2014a Lao:2016a"></span> and—if the MBCP(<span class="math inline"><em>n</em></span>) method is used—to cite Ref. </p>
<p><span class="citation" data-cites="Richard:2013a"></span> as well.</p>
<h3 id="job-control-and-examples">Job Control and Examples</h3>
<p>The following <em>$rem</em> variables control MBE jobs.</p>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>

<p><strong>Example .</strong>  Example showing a 3-body EE-MBE calculation using TIP3P charges.</p>
<blockquote>
<pre><code>$rem
   SYM_IGNORE        true
   METHOD            B3LYP
   BASIS             cc-pVDZ
   MANY_BODY_INT     true
   MBE_ORDER         3
   MBE_EMBEDDING     charges
   THRESH            14
   SCF_CONVERGENCE   7
$end

$molecule
0 1
--
   0 1
   O -1.126149 -1.748387 -0.423240
   H -0.234788 -1.493897 -0.661862
   H -1.062789 -2.681331 -0.218819
--
   0 1
   O -0.254210 1.611495 -1.293845
   H -1.001520 1.163510 -1.690129
   H -0.153399 2.411746 -1.809248
--
   0 1
   O 1.694541 -0.226287 1.705739
   H 0.785920 0.073487 1.677909
   H 2.047134 0.150917 2.511706
--
   0 1
   O -0.864533 0.522472 1.218817
   H -0.694120 1.093542 0.469789
   H -1.131418 -0.310426 0.829702
$end

$mbe_charges
  -0.834
   0.417
   0.417
  -0.834
   0.417
   0.417
  -0.834
   0.417
   0.417
  -0.834
   0.417
   0.417
$end</code></pre>
</blockquote>


<p><strong>Example .</strong>  Example showing a 3-body MBCP calculation.</p>
<blockquote>
<pre><code>$rem
   SYM_IGNORE        true
   METHOD            B3LYP
   BASIS             cc-pVDZ
   MANY_BODY_BSSE    mbcp
   MBE_BSSE_ORDER    3
   THRESH            14
   SCF_CONVERGENCE   7
$end

$molecule
0 1
--
   0 1
   O -1.126149 -1.748387 -0.423240
   H -0.234788 -1.493897 -0.661862
   H -1.062789 -2.681331 -0.218819
--
   0 1
   O -0.254210 1.611495 -1.293845
   H -1.001520 1.163510 -1.690129
   H -0.153399 2.411746 -1.809248
--
   0 1
   O 1.694541 -0.226287 1.705739
   H 0.785920 0.073487 1.677909
   H 2.047134 0.150917 2.511706
--
   0 1
   O -0.864533 0.522472 1.218817
   H -0.694120 1.093542 0.469789
   H -1.131418 -0.310426 0.829702
$end</code></pre>
</blockquote>



<p><strong>Example .</strong>  Example showing a 3-body FMO calculation.</p>
<blockquote>
<pre><code>$rem
   SYM_IGNORE        true
   METHOD            B3LYP
   BASIS             cc-pVDZ
   XPOL              true 
   XPOL_MPOL_ORDER   density
   FRAG_MOL_ORB      true
   FMO_ORDER         3
   THRESH            14
   SCF_CONVERGENCE   7
$end

$molecule
0 1
--
   0 1
   O -1.126149 -1.748387 -0.423240
   H -0.234788 -1.493897 -0.661862
   H -1.062789 -2.681331 -0.218819
--
   0 1
   O -0.254210 1.611495 -1.293845
   H -1.001520 1.163510 -1.690129
   H -0.153399 2.411746 -1.809248
--
   0 1
   O 1.694541 -0.226287 1.705739
   H 0.785920 0.073487 1.677909
   H 2.047134 0.150917 2.511706
--
   0 1
   O -0.864533 0.522472 1.218817
   H -0.694120 1.093542 0.469789
   H -1.131418 -0.310426 0.829702
$end</code></pre>
</blockquote>
<h2 id="sect:AIFDEM"><em>Ab Initio</em> Frenkel Davydov Exciton Model (AIFDEM)</h2>
<h3 id="theory-3">Theory</h3>
<p>A fairly old idea for describing the (potentially delocalized) excited states of molecular crystals, or more generally noncovalent assemblies or aggregates of (weakly) electronically-coupled chromophores, is the so called <em>Frenkel-Davydov Exciton model</em>. In such a model, a collective excitation of the entire aggregate is expressed as a linear combination of excitations that are localized on molecular sites. The <span class="math inline"><em>I</em></span>th excited state, <span class="math inline">|<em>Ξ</em><sub><em>I</em></sub>⟩</span>, is thus written <br /><span class="math display">$$\label{eq:AIFDEM_direct_pdt}
	|\Xi_I\rangle=\sum_n^{\rm sites} \;\sum_i^{\rm states} K_n^{Ii} |\Psi_n^i\rangle \prod_{m\neq n}|\Psi_m\rangle \;,$$</span><br /> where <span class="math inline">|<em>Ψ</em><sub><em>n</em></sub><sup><em>i</em></sup>⟩</span> is the <span class="math inline"><em>i</em></span>th excited state of the <span class="math inline"><em>n</em></span>th molecular fragment and <span class="math inline">|<em>Ψ</em><sub><em>m</em></sub>⟩</span> is the ground-state wave function of the <span class="math inline"><em>m</em></span>th fragment. Eigenstates and energies are found by constructing and diagonalizing the electronic Hamiltonian matrix in this direct product, “exciton-site basis”.</p>
<p>In the <em>ab initio</em> Frenkel-Davydov exciton model (AIFDEM) developed by Morrison and Herbert,<span class="citation" data-cites="Morrison:2014 Morrison:2015"></span> the ground-state wave functions in Eq. <a href="#eq:AIFDEM_direct_pdt" data-reference-type="eqref" data-reference="eq:AIFDEM_direct_pdt">[eq:AIFDEM_direct_pdt]</a> are single Slater determinants (obtained from SCF calculations on isolated fragments), and the fragment excited-state wave functions are linear combinations of singly-excited determinants: <br /><span class="math display">|<em>Ψ</em><sub>​<em>A</em></sub><sup>*</sup>⟩ = ∑<sub><em>i</em><em>a</em></sub><em>C</em><sup><em>i</em><em>a</em></sup>|<em>Φ</em><sub>​<em>A</em></sub><sup><em>i</em><em>a</em></sup>⟩ .</span><br /> The AIFDEM approach computes elements of the exact Hamiltonian, <br /><span class="math display">⟨<em>Ψ</em><sub>​<em>A</em></sub><sup>*</sup><em>Ψ</em><sub>​<em>B</em></sub><em>Ψ</em><sub>​<em>C</em></sub>…|<em>Ĥ</em>|<em>Ψ</em><sub>​<em>A</em></sub><em>Ψ</em><sub>​<em>B</em></sub><sup>*</sup><em>Ψ</em><sub>​<em>C</em></sub>…⟩ = ∑<sub><em>i</em><em>a</em><em>σ</em></sub>∑<sub><em>k</em><em>b</em><em>τ</em></sub><em>C</em><sub><em>σ</em></sub><sup><em>i</em><em>a</em></sup><em>C</em><sub><em>τ</em></sub><sup><em>k</em><em>b</em></sup>⟨<em>Φ</em><sub>​<em>A</em></sub><sup><em>i</em><em>a</em></sup><em>Φ</em><sub>​<em>B</em></sub><em>Φ</em><sub>​<em>C</em></sub>…|<em>Ĥ</em>|<em>Φ</em><sub>​<em>A</em></sub><em>Φ</em><sub>​<em>B</em></sub><sup><em>k</em><em>b</em></sup><em>Φ</em><sub>​<em>C</em></sub>…⟩ .</span><br /> In particular, no dipole-coupling approximation is made (as is often invoked in simple exciton models). Such an approximation may be valid for well-separated chromophores but likely less so for tightly-packed chromophores in a molecular crystal. Overlap matrices <br /><span class="math display">⟨<em>Ψ</em><sub>​<em>A</em></sub><sup>*</sup><em>Ψ</em><sub>​<em>B</em></sub><em>Ψ</em><sub>​<em>C</em></sub>…|<em>Ψ</em><sub>​<em>A</em></sub><em>Ψ</em><sub>​<em>B</em></sub><sup>*</sup><em>Ψ</em><sub>​<em>C</em></sub>…⟩ = ∑<sub><em>i</em><em>a</em><em>σ</em></sub>∑<sub><em>k</em><em>b</em><em>τ</em></sub><em>C</em><sub><em>σ</em></sub><sup><em>i</em><em>a</em></sup><em>C</em><sub><em>τ</em></sub><sup><em>k</em><em>b</em></sup>⟨<em>Φ</em><sub>​<em>A</em></sub><sup><em>i</em><em>a</em></sup><em>Φ</em><sub>​<em>B</em></sub><em>Φ</em><sub>​<em>C</em></sub>…|<em>Φ</em><sub>​<em>A</em></sub><em>Φ</em><sub>​<em>B</em></sub><sup><em>k</em><em>b</em></sup><em>Φ</em><sub>​<em>C</em></sub>…⟩</span><br /> are also required because molecular orbitals located on different fragments are not orthogonal to one another. In order to reduce the number of terms in Eqs. <a href="#eq:AIFDEM_H" data-reference-type="eqref" data-reference="eq:AIFDEM_H">[eq:AIFDEM_H]</a> and <a href="#eq:AIFDEM_S" data-reference-type="eqref" data-reference="eq:AIFDEM_S">[eq:AIFDEM_S]</a>, the fragment excited states are transformed into the natural transition orbital (NTO) basis (see Section <a href="#sect:NTOs" data-reference-type="ref" data-reference="sect:NTOs">[sect:NTOs]</a>) and then the corresponding orbitals transformation (Section <a href="#subsubsect:COT" data-reference-type="ref" data-reference="subsubsect:COT">[subsubsect:COT]</a>) is used to compute matrix elements between non-orthogonal Slater determinants. The size of the exciton-site basis is sufficiently small such that eigenvectors and energies of the exciton Hamiltonian can be printed and saved to scratch files. Transition dipole moments between the ground state and the first ten excited states of the exciton Hamiltonian are also computed.</p>
<p>The cost to compute each matrix element scales with the size of the supersystem (somewhere between quadratic and quartic with monomer size), since all fragments must be included in the direct products. To reduce this scaling, a physically-motivated charge embedding scheme was introduced<span class="citation" data-cites="Morrison:2015"></span> that only treats the excited fragments, and neighbors within a user-specified distance threshold, with full QM calculation, while the other ground state fragment interactions are approximated by point charges. (The nature of the point charges is controlled by the <em>$rem</em> variable .) In general, inclusion of neighboring fragments in the QM part of the matrix element evaluation does not seem to significantly improve the accuracy and diminishes the cost savings of the charge-embedding procedure. Therefore, the minimal “ 0 Å” threshold, where only the excited fragments are described at a QM level, can be considered optimal. Charge embedding with the minimal threshold affords an algorithm that scales as <span class="math inline"><em>F</em><sup>2</sup> × 𝒪(<em>n</em><sub>pair</sub><sup>2 − 4</sup>)</span>, where <span class="math inline"><em>F</em></span> is the number of fragments and <span class="math inline"><em>n</em><sub>pair</sub></span> is the size of a pair of fragments.</p>
<p>The exciton-site basis can be expanded to include higher-lying fragment excited states which affords the wave function increased variational flexibility, and can significantly improve the accuracy for polar systems and delocalized excited states. The number of fragment excited states included in the basis is specified by the keyword, which must be <span class="math inline"> ≥ 1</span>. A cost effective means of including polarization effects is to use the XPol method to compute fragment ground states, see Section <a href="#sect:xpol" data-reference-type="ref" data-reference="sect:xpol">[sect:xpol]</a> and associated job controls. Fragment excited states are then computed using the XPol-polarized MOs. Charge transfer-type states of the form <span class="math inline">|<em>Φ</em><sub>​<em>A</em></sub><sup>+</sup><em>Φ</em><sub>​<em>B</em></sub><sup>−</sup><em>Φ</em><sub>​<em>C</em></sub>…⟩</span>, where <span class="math inline"><em>Φ</em><sub>​<em>A</em></sub><sup>±</sup></span> are cationic or anionic determinants from unrestricted SCF calculations on the isolated fragments, can also be included in the basis.</p>
<p>The exciton-site basis states are spin-adapted to form proper <span class="math inline">$\Hat{S}^2$</span> eigenstates. Their multiplicity determines that of the target excited state and this must be specified by setting or to . The number of terms included in Eqs. <a href="#eq:AIFDEM_H" data-reference-type="eqref" data-reference="eq:AIFDEM_H">[eq:AIFDEM_H]</a> and <a href="#eq:AIFDEM_S" data-reference-type="eqref" data-reference="eq:AIFDEM_S">[eq:AIFDEM_S]</a> can be rationally truncated at some fraction of the norm of the fragment NTO amplitudes, in order to reduce cost at the expense of accuracy, although the approximation is controllable by means of the truncation threshold. Computation time scales approximately quadratically with the number of terms and a threshold of about 85% has been found to maintain acceptable accuracy for organic molecules with reasonable cost. The fragment orbitals and excited states may be computed with any SCF and single-excitation theory, including DFT and TDDFT, however the coupling matrix elements are always computed with a CIS-like Hamiltonian with no DFT exchange-correlation. Both OpenMP and MPI parallel implementations are available; the former distributes two-electron integral computation across cores in a node as in a traditional excited-state calculation, the latter can distribute matrix element evaluations across hundreds of cores with minimal overhead.</p>
<p><strong>Note: </strong></p>
<p>Symmetry should be turned off for AIFDEM calculations.</p>
<h3 id="job-control">Job Control</h3>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>

<p><strong>Example .</strong>  Example showing singlet excited state calculation, on (H<span class="math inline"><em></em><sub>2</sub></span>O)<span class="math inline"><em></em><sub>4</sub></span>. XPol is used to generate monomer wave functions with ChElPG charges. Minimal QM charge embedding is used for the exciton model with three excited states per fragment.</p>
<blockquote>
<pre><code>$rem
   BASIS                aug-cc-pvdz
   EXCHANGE             HF
   CIS_N_ROOTS          3
   CIS_TRIPLETS         FALSE
   XPOL                 TRUE
   XPOL_CHARGE_TYPE     QCHELPG
   AIFDEM               TRUE
   AIFDEM_EMBED_RANGE   0
   AIFDEM_NTOTHRESH     90
   NTO_PAIRS            1
$end

$molecule
0 1
--H2O 0
   0 1
   O          1.74078        1.59716       -1.49814
   H          2.22908        2.18316       -2.08914
   H          0.88038        2.04726       -1.32684
--H2O 1
   0 1
   O          1.31998       -1.18934       -1.91734
   H          1.49988       -0.22974       -1.89044
   H          1.69058       -1.52594       -1.07704
--H2O 2
   0 1
   O         -0.68982        2.59476       -0.72224
   H         -1.14372        3.37086       -1.07364
   H         -1.35592        1.84986       -0.78334
--H2O 3
   0 1
   O         -1.27512       -1.77394       -1.69524
   H         -0.32252       -1.52884       -1.85604
   H         -1.53992       -2.30454       -2.45644
$end</code></pre>
</blockquote>
<h3 id="subsec:AIFDEM-derivs">Derivative Couplings</h3>
<p>There are many chemical processes of interest where motion of nuclei induces electronic transitions, in a breakdown of the Born-Oppenheimer approximation. In order to investigate such processes it is useful to calculate some quantity that codifies the coupling of adiabatic electronic states dues to nuclear motion. In molecular electronic structure theory this quantity is the nonadiabatic coupling (or “derivative coupling”) vector <span class="math inline">⟨<em>Ψ</em><sub><em>I</em></sub>|(∂/∂<em>x</em>)|<em>Ψ</em><sub><em>J</em></sub>⟩</span>, which describes how the nuclear position derivative <span class="math inline">∂/∂<em>x</em></span> couples adiabatic (Born-Oppenheimer) electronic states <span class="math inline"><em>Ψ</em><sub><em>I</em></sub></span> and <span class="math inline"><em>Ψ</em><sub><em>J</em></sub></span>. In solid-state physics these ideas are typically not discussed in terms of the Born-Oppenheimer approximation but rather in terms of the so-called “Holstein” and “Peierls” exciton/phonon coupling constants (see below). Within the framework of the AIFDEM, each of these quantities can be computed from a common intermediate <span class="math inline"><strong>H</strong><sup>[<em>x</em>]</sup></span>, which is the derivative of the AIFDEM Hamiltonian matrix with respect to some nuclear coordinate <span class="math inline"><em>x</em></span>.<span class="citation" data-cites="Morrison:2017a Morrison:2017b"></span> Diagonal matrix elements <span class="math inline"><em>H</em><sub><em>A</em><em>A</em></sub><sup>[<em>x</em>]</sup> ≡ ∂<em>H</em><sub><em>A</em><em>A</em></sub>/∂<em>x</em></span> describe how the exciton-site energies are modulated by nuclear motion and off-diagonal matrix elements <span class="math inline"><em>H</em><sub><em>A</em><em>B</em></sub><sup>[<em>x</em>]</sup> ≡ ∂<em>H</em><sub><em>A</em><em>B</em></sub>/∂<em>x</em></span> describe how nuclear motion modifies the energy-transfer couplings.</p>
<p>Once <span class="math inline"><strong>H</strong><sup>[<em>x</em>]</sup></span> has been constructed, then the nonadiabatic coupling vector between eigenstates <span class="math inline"><em>Ψ</em><sub><em>I</em></sub></span> and <span class="math inline"><em>Ψ</em><sub><em>J</em></sub></span> of the exciton Hamiltonian is simply <br /><span class="math display"><strong>h</strong><sup><em>I</em><em>J</em></sup> = <strong>K</strong><sub><em>I</em></sub><sup>†</sup><strong>H</strong><sup>[<em>x</em>]</sup><strong>K</strong><sub><em>J</em></sub> .</span><br /> The Holstein (<span class="math inline"><em>A</em> = <em>B</em></span>) and Peierls (<span class="math inline"><em>A</em> ≠ <em>B</em></span>) coupling constants <span class="math inline"><em>g</em><sub><em>A</em><em>B</em><em>θ</em></sub><sup></sup></span>, expressed in dimensionless normal mode coordinates, are<span class="citation" data-cites="Morrison:2017b"></span> <br /><span class="math display"><em>g</em><sub><em>A</em><em>B</em><em>θ</em></sub><sup></sup> = (2<em>μ</em><sub><em>θ</em></sub><em>ω</em><sub><em>θ</em></sub>)<sup> − 1/2</sup>∑<sub><em>x</em></sub><em>H̃</em><sub><em>A</em><em>B</em></sub><sup>[<em>x</em>]</sup><em>L</em><sub><em>x</em><em>θ</em></sub></span><br /> where the matrix <span class="math inline"><strong>L</strong></span> is the transformation between Cartesians coordinates and normal (phonon) modes. The columns of <span class="math inline"><strong>L</strong></span> contain the normalized Cartesian displacements of normal mode <span class="math inline"><em>θ</em></span>, whose frequency and effective mass are <span class="math inline"><em>ω</em><sub><em>θ</em></sub></span> and <span class="math inline"><em>μ</em><sub><em>θ</em></sub></span>, respectively. The tilde on, <span class="math inline"><em>H̃</em><sub><em>A</em><em>B</em></sub><sup>[<em>x</em>]</sup></span> indicates that the matrix element derivatives have been orthogonalized (including the derivative of the orthogonalization transformation).<span class="citation" data-cites="Morrison:2017b"></span></p>
<h3 id="job-control-for-aifdem-derivative-couplings">Job Control for AIFDEM Derivative Couplings</h3>
<p>Analytic evaluation of <span class="math inline"><strong>H</strong><sup>[<em>x</em>]</sup></span> and the analogous overlap derivative <span class="math inline"><strong>S</strong><sup>[<em>x</em>]</sup></span> have been implemented in <span><span class="smallcaps">Q-Chem</span></span>.<span class="citation" data-cites="Morrison:2017b"></span> To compute AIFDEM derivatives, request a standard AIFDEM job and set = 1. Currently, the AIFDEM derivatives do not support charge embedding so the keyword must be omitted from these jobs; this precludes the use of XPol wavefunctions for the fragments. Furthermore, only one excited state per fragment is supported so the keyword = 1. is required.</p>
<p>The derivatives of the AIFDEM Hamiltonian matrix and overlap matrix are printed in the output file in sets of the three Cartesian coordinates that belong to a single atom. For convenience, the orthogonalized AIFDEM Hamiltonian matrix elements are saved in the scratch directory, <em>$QCSCRATCH/aifdem_deriv</em>. These are organized such that the derivatives for each unique matrix element are stored in individual files in the order of the atomic Cartesian coordinates. These files can facilitate external calculation of exciton/phonon coupling constants.</p>

<p><strong>Example .</strong> <br />
</p>
<blockquote>
<p>A basic AIFDEM derivative calculation on a chain of helium atoms.</p>
<pre><code>$rem
basis = cc-pvdz
exchange = hf
aifdem = true
cis_n_roots = 1
cis_singlets = true
cis_triplets = false
cis_state_deriv = 1
nto_pairs = 1
mem_total = 1000
mem_static = 1000
max_cis_cycles = 200
max_scf_cycles = 200
thresh = 10
aifdem_ntothresh = 100
$end

$molecule
0 1
--frgm 0
0 1
He 0.0 0.0 0.0
He 0.0 0.0 1.4
--frgm 1
0 1
He 0.0 0.0 2.8
He 0.0 0.0 4.2
$end</code></pre>
</blockquote>
<h2 id="sec:TDDFT(MI)">TDDFT for Molecular Interactions</h2>
<h3 id="theory-4">Theory</h3>
<p>There exist a broad class of weakly interacting molecular complexes which give rise to interesting excited-state properties that are potentially very different from those of a single chromophore. The “TDDFT for molecular interactions&quot; or TDDFT(MI) method is designed for efficient excited-state calculations in such cases in (potentially large) systems composed of weakly-interacting but electronically-coupled monomers.<span class="citation" data-cites="Liu:2015 Herbert:2016b"></span> Such systems include molecular aggregates, chromophores in explicit solvent, and even proteins, for which the traditional TDDFT method become prohibitively expensive. TDDFT(MI) starts from a ground-state SCF MI calculation, and the use of ALMOs is central to its efficiency. In addition, the excitations are confined within monomer units and the explicit charge-transfer excitations are ignored, significantly reduced the two-electron integrals cost. The method works by coupling together excitations computed individually on different molecular fragments, and the number of excited states per fragment can be increased (at very low cost) in order to increase the variational flexibility of this exciton-type basis. Thus, despite the localized nature of the basis states, TDDFT(MI) is capable of describing collective excitations that are delocalized over multiple monomer units, as for example in the case of organic semiconductors. In general, TDDFT(MI) reproduces full super-system TDDFT excitation energies to within <span class="math inline">∼</span>0.2 eV, but with an order or magnitude reduction in total CPU time.<span class="citation" data-cites="Liu:2015"></span> Formally, the cost of the method scales as <span class="math inline">${\cal
O}(N_{\rm fragment}^2 N_{\rm roots}^2 N_{\rm sub\mbox{-}AO}^x)$</span> where <span class="math inline">$N_{\rm
fragment}$</span> is the number of monomers, <span class="math inline">$N_{\rm roots}$</span> is the number of excited states per monomer, and <span class="math inline">$N_{\rm sub\mbox{-}AO}$</span> is the number of AOs on a dimer subsystem. The exponent <span class="math inline"><em>x</em></span> (with <span class="math inline">2 ≤ <em>x</em> ≤ 4</span>) reflects the cost of forming the Fock-like matrices of a traditional TDDFT calculation.</p>
<p>An especially promising application of the TDDFT(MI) method is to study excitation energies of a single chromophore in solution using a large number of explicit, quantum-mechanical solvent molecules. In such cases, the excitations are localized on the single chromophore and we can introduce a local excitation approximation (LEA) to TDDFT(MI) in which all of the Coulomb and exchange couplings between the solvent molecules and the chromophore are neglected.<span class="citation" data-cites="Liu:2016"></span> Following the ground-state SCF(MI) calculation, the cost of the TDDFT part of the calculation becomes essentially the same as the cost of a TDDFT calculation on the gas-phase chromophore. In addition, this approach avoids the appearance of, and mixing with, spurious charge-transfer-to-solvent states,<span class="citation" data-cites="Liu:2016 Herbert:2016b"></span> of the sort that are known to arise in TDDFT calculations with explicit solvent.<span class="citation" data-cites="Lange:2007 Isborn:2013"></span> Three versions of LEA-TDDFT(MI), named LEA0, LEA-Q and LEAc, have been implemented in <span><span class="smallcaps">Q-Chem</span></span>.<span class="citation" data-cites="Liu:2016"></span> In the LEA0 method, ALMOs from the ground state SCF(MI) calculation are used to perform the TDDFT calculation. In LEAc, a sub-block of the TDDFT(MI) working equation localized on chromophore is extracted to calculate the excitation energies. Finally, LEA-Q is almost the same as LEAc except for some transformations to eliminate the overlap matrices. These approaches have been applied to converge solvatochromatic shifts for several aqueous chromophores.<span class="citation" data-cites="Liu:2016"></span></p>
<h3 id="job-control-1">Job Control</h3>
<p>In addition to the normal TDDFT job controls variables described in Section <a href="#subsec:TDDFT_Job_Control" data-reference-type="ref" data-reference="subsec:TDDFT_Job_Control">[subsec:TDDFT_Job_Control]</a>, there are several others to request TDDFT(MI). Note that only single-point energies (not gradients) are available for this method.</p>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<h2 id="sec:ALMO-CIS">The ALMO-CIS and ALMO-CIS+CT methods</h2>
<h3 id="theory-5">Theory</h3>
<p>The ALMO-CIS<span class="citation" data-cites="Closser:2015"></span> and ALMO-CIS+CT<span class="citation" data-cites="Ge:2017"></span> methods are local variants of configuration interaction singles (CIS), and are formulated through the use of absolutely localized molecular orbitals (ALMOs). They share the same spirit with the TDDFT(MI) method (Section <a href="#sec:TDDFT(MI)" data-reference-type="ref" data-reference="sec:TDDFT(MI)">13.16</a>), but were originally designed to target large numbers of excited states in atomic/molecular clusters, such as the entire <span class="math inline"><em>n</em> = 2</span> band in helium clusters that contain hundreds of atoms.</p>
<p>In ALMO-CIS and ALMO-CIS+CT, we solve a truncated non-orthogonal CIS eigen-equation: <br /><span class="math display"><em>A</em><sub><em>i</em><em>a</em>, <em>j</em><em>b</em></sub><em>t</em><sup><em>j</em><em>b</em></sup> = <em>ω</em><em>S</em><sub><em>i</em><em>a</em>, <em>j</em><em>b</em></sub><em>t</em><sup><em>j</em><em>b</em></sup></span><br /> The use of ALMOs allows associating each MO index (<span class="math inline"><em>i</em></span>, <span class="math inline"><em>a</em></span>, <span class="math inline"><em>j</em></span> or <span class="math inline"><em>b</em></span>) to a fragment. In ALMO-CIS, approximation is made such that only the amplitudes corresponding to intrafragment transitions are non-zero, i.e., <span class="math inline"><em>t</em><sup><em>j</em><em>b</em></sup> = 0</span> if the occupied orbital <span class="math inline"><em>j</em></span> and the virtual orbital <span class="math inline"><em>b</em></span> reside on two different fragments. The Hamiltonian and overlap matrix are also truncated, with <span class="math inline"><em>i</em></span>(<span class="math inline"><em>j</em></span>) and <span class="math inline"><em>a</em></span>(<span class="math inline"><em>b</em></span>) belonging to the same fragment. This approximation excludes interfragment charge transfer (CT) excitations entirely and sometimes may lead to insufficient accuracy. In ALMO-CIS+CT, the CT effect is reintroduced by setting a distance cutoff <span class="math inline"><em>r</em><sub><em>c</em><em>u</em><em>t</em></sub></span>, so that transitions between neighboring fragments within a distance smaller than <span class="math inline"><em>r</em><sub><em>c</em><em>u</em><em>t</em></sub></span> are allowed (<span class="math inline"><em>i</em></span>(<span class="math inline"><em>j</em></span>) and <span class="math inline"><em>a</em></span>(<span class="math inline"><em>b</em></span>) belonging to such a pair of fragments are also included in the eigen-equation). In both ALMO-CIS and ALMO-CIS+CT, dimension of the eigenvalue problem scales linearly with respect to system size, instead of having a quadratic scaling as in standard CIS. Because of the reduction of matrix size, it is computationally feasible to explicitly build the Hamiltonian and diagonalize it to get a full band of eigenstates for relatively large systems, and the overall scaling of ALMO-CIS/ALMO-CIS+CT is cubic, in contrast with the sixth order scaling of standard CIS for a full-spectrum calculation.</p>
<p>Besides the full-spectrum calculations described above, use of the Davidson algorithm is also available for ALMO-CIS and ALMO-CIS+CT, which targets a few lowest excited states. This implementation, unlike the original full-spectrum version, also supports the local variants of TDDFT/TDA calculations that share the same working equation (eq. <a href="#eq:almo-cis" data-reference-type="ref" data-reference="eq:almo-cis">[eq:almo-cis]</a>).</p>
<h3 id="job-control-2">Job Control</h3>
<p>In addition to the standard CIS job controls variables described in Section <a href="#sec:CIS-jobcontrol" data-reference-type="ref" data-reference="sec:CIS-jobcontrol">[sec:CIS-jobcontrol]</a>, there are several additional <em>$rem</em> variables to specify for an ALMO-CIS/ALMO-CIS+CT calculation.</p>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>

<p><strong>Example .</strong>  ALMO-CIS+CT calculation (<span class="math inline"><em>r</em><sub><em>c</em><em>u</em><em>t</em></sub> = 10</span> a<span class="math inline"><em></em><sub>0</sub></span>) for all the <span class="math inline"><em>n</em> = 2</span> states of a helium dimer.</p>
<blockquote>
<pre><code>$molecule
0 1
--
   0 1
   He      2.8    0.     0.  
--
   0 1
   He      0.     0.     0.  
$end

$rem
   basis         gen 
   aux_basis     rimp2-cc-pvdz
   purecart      1111
   method        hf
   sym_ignore    true
   symmetry      false
   frgm_method   stoll
   cis_n_roots   8
   cis_triplets  false
   local_cis     2 ! use RI for ALMO-CIS  
   nn_thresh     10
$end

$rem_frgm
   cis_n_roots 0
$end

$basis
****
HE   0
S    3    1.000000
3.84216340D+01    2.37660000D-02
5.77803000D+00    1.54679000D-01
1.24177400D+00    4.69630000D-01
S    1    1.000000
2.97964000D-01    1.00000000D+00
SP   1    1.000000
4.80000000D-02    1.00000000D+00   1.00000000D+00
****
$end</code></pre>
</blockquote>










<h1 id="app:geom_opt">Geometry Optimization with <span><span class="smallcaps">Q-Chem</span></span></h1>
<h2 id="introduction-1">Introduction</h2>
<p>Geometry optimization refers to the determination of stationary points, principally minima and transition states, on molecular potential energy surfaces. It is an iterative process, requiring the repeated calculation of energies, gradients and (possibly) Hessians at each optimization cycle until convergence is attained. The optimization step involves modifying the current geometry, utilizing current and previous energy, gradient and Hessian information to produce a revised geometry which is closer to the target stationary point than its predecessor was. The art of geometry optimization lies in calculating the step <span class="math inline"><strong>h</strong></span>, the displacement from the starting geometry on that cycle, so as to converge in as few cycles as possible.</p>
<p>There are four main factors that influence the rate of convergence. These are:</p>
<ul>
<li><p>Initial starting geometry.</p></li>
<li><p>Algorithm used to determine the step <span class="math inline"><strong>h</strong></span>.</p></li>
<li><p>Quality of the Hessian (second derivative) matrix.</p></li>
<li><p>Coordinate system chosen.</p></li>
</ul>
<p>The first of these factors is obvious: the closer the initial geometry is to the final converged geometry the fewer optimization cycles it will take to reach it. The second factor is again obvious: if a poor step <span class="math inline"><strong>h</strong></span> is predicted, this will obviously slow down the rate of convergence. The third factor is related to the second: the best algorithms make use of second derivative (curvature) information in calculating <span class="math inline"><strong>h</strong></span>, and the better this information is, the better will be the predicted step. The importance of the fourth factor (the coordinate system) has been generally appreciated later on: a good choice of coordinates can enhance the convergence rate by an order of magnitude (a factor of 10) or more, depending on the molecule being optimized.</p>
<p><span><span class="smallcaps">Q-Chem</span></span> includes a powerful suite of algorithms for geometry optimization written by Jon Baker and known collectively as <span><span class="smallcaps">Optimize</span></span>. These algorithms have been developed and perfected over the past ten years and the code is robust and has been well tested. <span><span class="smallcaps">Optimize</span></span> is a general geometry optimization package for locating both minima and transition states. It can optimize using Cartesian, <em>Z</em>-matrix coordinates or delocalized internal coordinates. The last of these are generated automatically from the Cartesian coordinates and are often found to be particularly effective. It also handles fixed constraints on distances, angles, torsions and out-of-plane bends, between any atoms in the molecule, whether or not the desired constraint is satisfied in the starting geometry. Finally it can freeze atomic positions, or any <span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span>, <span class="math inline"><em>z</em></span> Cartesian atomic coordinates.</p>
<p><span><span class="smallcaps">Optimize</span></span> is designed to operate with minimal user input. All that is required is the initial guess geometry, either in Cartesian coordinates (<em>e.g.</em>, from a suitable model builder such as HyperChem) or as a <em>Z</em>-matrix, the type of stationary point being sought (minimum or transition state) and details of any imposed constraints. All decisions as to the optimization strategy (what algorithm to use, what coordinate system to choose, how to handle the constraints) are made by <span><span class="smallcaps">Optimize</span></span>.</p>
<p>Note particularly, that although the starting geometry is input in a particular coordinate system (as a <em>Z</em>-matrix, for example) these coordinates are not necessarily used during the actual optimization. The best coordinates for the majority of geometry optimizations are delocalized internals, and these will be tried first. Only if delocalized internals fail for some reason, or if conditions prevent them being used (<em>e.g.</em>, frozen atoms) will other coordinate systems be tried. If all else fails the default is to switch to Cartesian coordinates. Similar defaults hold for the optimization algorithm, maximum step size, convergence criteria, <em>etc.</em> You may of course override the default choices and force a particular optimization strategy, but it is not normally necessary to provide <span><span class="smallcaps">Optimize</span></span> with anything other than the minimal information outlined above.</p>
<p>The heart of the <span><span class="smallcaps">Optimize</span></span> package (for both minima and transition states) is Baker’s eigenvector-following (EF) algorithm.<span class="citation" data-cites="Baker:1986"></span> This was developed following the work of Cerjan and Miller,<span class="citation" data-cites="Cerjan:1981"></span> and of Simons and co-workers.<span class="citation" data-cites="Simons:1983 Banerjee:1985"></span> The Hessian mode-following option incorporated into this algorithm is capable of locating transition states by walking uphill from the associated minima. By following the lowest Hessian mode, the EF algorithm can locate transition states starting from any reasonable input geometry and Hessian.</p>
<p>An additional option available for minimization is Pulay’s GDIIS algorithm,<span class="citation" data-cites="Csaszar:1984"></span> which is based on the well known DIIS technique for accelerating SCF convergence.<span class="citation" data-cites="Pulay:1982"></span> GDIIS must be specifically requested, as the EF algorithm is the default.</p>
<p>Although optimizations can be carried out in Cartesian or <em>Z</em>-matrix coordinates, the best choice, as noted above, is usually delocalized internal coordinates. These coordinates were developed by Baker <em>et al.</em>,<span class="citation" data-cites="Baker:1996"></span> and can be considered as a further extension of the natural internal coordinates developed by Pulay <em>et al.</em><span class="citation" data-cites="Pulay:1979 Fogarasi:1992"></span> and the redundant optimization method of Pulay and Fogarasi.<span class="citation" data-cites="Pulay:1992"></span></p>
<p><span><span class="smallcaps">Optimize</span></span> incorporates a very accurate and efficient Lagrange multiplier algorithm for constrained optimization. This was originally developed for use with Cartesian coordinates<span class="citation" data-cites="Baker:1992 Baker:1993"></span> and can handle constraints that are not satisfied in the starting geometry. The Lagrange multiplier approach has been modified for use with delocalized internals;<span class="citation" data-cites="Baker:1997"></span> this is much more efficient and is now the default. The Lagrange multiplier code can locate constrained transition states as well as minima.</p>
<h2 id="theoretical-background">Theoretical Background</h2>
<p>Consider the energy, <span class="math inline"><em>E</em>(<strong>x</strong><sub>0</sub>)</span> at some point <span class="math inline"><strong>x</strong><sub>0</sub></span> on a potential energy surface. We can express the energy at a nearby point <span class="math inline"><strong>x</strong> = <strong>x</strong><sub>0</sub> + <strong>h</strong></span> by means of the Taylor series <br /><span class="math display">$$\label{eq:a1}
	E(\ensuremath{\mathbf{x}}_0 +\ensuremath{\mathbf{h}}) = E(\ensuremath{\mathbf{x}}_0) + \ensuremath{\mathbf{h}}^t \frac{d E(\ensuremath{\mathbf{x}}_0)}{d\ensuremath{\mathbf{x}}} + 
	\frac{1}{2} \ensuremath{\mathbf{h}}^t \frac{d^2E(\ensuremath{\mathbf{x}}_0) }{d\ensuremath{\mathbf{x}}_1 d\ensuremath{\mathbf{x}}_2} \ensuremath{\mathbf{h}} +\cdots$$</span><br /></p>
<p>If we knew the exact form of the energy functional <span class="math inline"><em>E</em>(<strong>x</strong>)</span> and all its derivatives, we could move from the current point <span class="math inline"><strong>x</strong><sub>0</sub></span> directly to a stationary point, (<em>i.e.</em>, we would know exactly what the step <span class="math inline"><strong>h</strong></span> ought to be). Since we typically know only the lower derivatives of <span class="math inline"><em>E</em>(<strong>x</strong>)</span> at best, then we can estimate the step <span class="math inline"><strong>h</strong></span> by differentiating the Taylor series with respect to <span class="math inline"><strong>h</strong></span>, keeping only the first few terms on the right hand side, and setting the left hand side, <span class="math inline"><em>d</em><em>E</em>(<strong>x</strong><sub>0</sub> + <strong>h</strong>)/<em>d</em><strong>h</strong></span>, to zero, which is the value it would have at a genuine stationary point. Thus <br /><span class="math display">$$\label{eq:a2}
	\frac{d E(\ensuremath{\mathbf{x}}_0 + \ensuremath{\mathbf{h}})}{d\ensuremath{\mathbf{h}}} =  \frac{dE(\ensuremath{\mathbf{x}}_0)}{d\ensuremath{\mathbf{x}}}
	+\frac{d^2E(\ensuremath{\mathbf{x}}_0)}{d\ensuremath{\mathbf{x}}_1 d\ensuremath{\mathbf{x}}_2} \ensuremath{\mathbf{h}}+\mbox{higher terms (ignored)}$$</span><br /> from which <br /><span class="math display"><strong>h</strong> =  − <strong>H</strong><sup> − 1</sup><strong>g</strong></span><br /> where <br /><span class="math display">$$\frac{dE}{d{\rm {\bf x}}}\equiv {\rm {\bf g}}\mbox{ (gradient vector),}
\qquad
	\frac{d^2E}{d{\rm {\bf x}}_{\rm {\bf 1}} d{\rm {\bf x}}_{\rm {\bf 2}} 
	}\equiv {\rm {\bf H}}\mbox{ (Hessian matrix)}$$</span><br /> Equation (<a href="#eq:a3" data-reference-type="ref" data-reference="eq:a3">[eq:a3]</a>) is known as the Newton-Raphson step. It is the major component of almost all geometry optimization algorithms in quantum chemistry.</p>
<p>The above derivation assumed exact first (gradient) and second (Hessian) derivative information. Analytical gradients are available for all methodologies supported in <span><span class="smallcaps">Q-Chem</span></span>; however analytical second derivatives are not. Furthermore, even if they were, it would not necessarily be advantageous to use them as their evaluation is usually computationally demanding, and, efficient optimizations can in fact be performed without an exact Hessian. An excellent compromise in practice is to begin with an approximate Hessian matrix, and update this using gradient and displacement information generated as the optimization progresses. In this way the starting Hessian can be “improved” at essentially no cost. Using Eq. (<a href="#eq:a3" data-reference-type="ref" data-reference="eq:a3">[eq:a3]</a>) with an approximate Hessian is called the quasi Newton-Raphson step.</p>
<p>The nature of the Hessian matrix (in particular its eigenvalue structure) plays a crucial role in a successful optimization. All stationary points on a potential energy surface have a zero gradient vector; however the character of the stationary point (<em>i.e.</em>, what type of structure it corresponds to) is determined by the Hessian. Diagonalization of the Hessian matrix can be considered to define a set of mutually orthogonal directions on the energy surface (the eigenvectors) together with the curvature along those directions (the eigenvalues). At a local minimum (corresponding to a well in the potential energy surface) the curvature along all of these directions must be positive, reflecting the fact that a small displacement along any of these directions causes the energy to rise. At a transition state, the curvature is negative (<em>i.e.</em>, the energy is a maximum) along one direction, but positive along all the others. Thus, for a stationary point to be a transition state the Hessian matrix at that point must have one and only one negative eigenvalue, while for a minimum the Hessian must have all positive eigenvalues. In the latter case the Hessian is called <em>positive definite</em>. If searching for a minimum it is important that the Hessian matrix be positive definite; in fact, unless the Hessian is positive definite there is no guarantee that the step predicted by Eq. (<a href="#eq:a3" data-reference-type="ref" data-reference="eq:a3">[eq:a3]</a>) is even a descent step (<em>i.e.</em>, a direction that will actually lower the energy). Similarly, for a transition state search, the Hessian must have one negative eigenvalue. Maintaining the Hessian eigenvalue structure is not difficult for minimization, but it can be a difficulty when trying to find a transition state.</p>
<p>In a diagonal Hessian representation the Newton-Raphson step can be written <br /><span class="math display">$$\label{eq:a4}
	\ensuremath{\mathbf{h}} = \sum \frac{-F_i}{b_i} {\ensuremath{\mathbf{u}}_i}$$</span><br /> where <span class="math inline"><strong>u</strong><sub><em>i</em></sub></span> and <span class="math inline"><em>b</em><sub><em>i</em></sub></span> are the eigenvectors and eigenvalues of the Hessian matrix <span class="math inline"><strong>H</strong></span> and <span class="math inline"><em>F</em><sub><em>i</em></sub> = <strong>u</strong><sub><em>i</em></sub><sup><em>t</em></sup><strong>g</strong></span> is the component of <span class="math inline"><strong>g</strong></span> along the local direction (eigenmode)<span class="math inline"><strong>u</strong><sub><em>i</em></sub></span>. As discussed by Simons <em>et al.</em>,<span class="citation" data-cites="Simons:1983"></span> the Newton-Raphson step can be considered as minimizing along directions <span class="math inline"><strong>u</strong><sub><em>i</em></sub></span> which have positive eigenvalues and maximizing along directions with negative eigenvalues. Thus, if the user is searching for a minimum and the Hessian matrix is positive definite, then the Newton-Raphson step is appropriate since it is attempting to minimize along all directions simultaneously. However, if the Hessian has one or more negative eigenvalues, then the basic Newton-Raphson step is not appropriate for a minimum search, since it will be maximizing and not minimizing along one or more directions. Exactly the same arguments apply during a transition state search except that the Hessian must have one negative eigenvalue, because the user has to maximize along one direction. However, there must be <em>only</em> one negative eigenvalue. A positive definite Hessian is a disaster for a transition state search because the Newton-Raphson step will then lead towards a minimum.</p>
<p>If firmly in a region of the potential energy surface with the right Hessian character, then a careful search (based on the Newton-Raphson step) will almost always lead to a stationary point of the correct type. However, this is only true if the Hessian is exact. If an approximate Hessian is being improved by updating, then there is no guarantee that the Hessian eigenvalue structure will be retained from one cycle to the next unless one is very careful during the update. Updating procedures that “guarantee” conservation of a positive definite Hessian do exist (or at least warn the user if the update is likely to introduce negative eigenvalues). This can be very useful during a minimum search; but there are no such guarantees for preserving the Hessian character (one and only one negative eigenvalue) required for a transition state.</p>
<p>In addition to the difficulties in retaining the correct Hessian character, there is the matter of obtaining a “correct” Hessian in the first instance. This is particularly acute for a transition state search. For a minimum search it is possible to “guess” a reasonable, positive-definite starting Hessian (for example, by carrying out a molecular mechanics minimization initially and using the mechanics Hessian to begin the <em>ab initio</em> optimization) but this option is usually not available for transition states. Even if the user calculates the Hessian exactly at the starting geometry, the guess for the structure may not be sufficiently accurate, and the expensive, exact Hessian may not have the desired eigenvalue structure.</p>
<p>Consequently, particularly for a transition state search, an alternative to the basic Newton-Raphson step is clearly needed, especially when the Hessian matrix is inappropriate for the stationary point being sought.</p>
<p>One of the first algorithms that was capable of taking corrective action during a transition state search if the Hessian had the wrong eigenvalue structure, was developed by Poppinger,<span class="citation" data-cites="Poppinger:1975"></span> who suggested that, instead of taking the Newton-Raphson step, if the Hessian had all positive eigenvalues, the lowest Hessian mode be followed uphill; whereas, if there were two or more negative eigenvalues, the mode corresponding to the least negative eigenvalue be followed downhill. While this step should lead the user back into the right region of the energy surface, it has the disadvantage that the user is maximizing or minimizing along one mode only, unlike the Newton-Raphson step which maximizes/minimizes along all modes simultaneously. Another drawback is that successive such steps tend to become linearly dependent, which degrades most of the commonly used Hessian updates.</p>
<h2 id="sect:ef">Eigenvector-Following (EF) Algorithm</h2>
<p>The work of Cerjan and Miller,<span class="citation" data-cites="Cerjan:1981"></span> and later Simons and co-workers,<span class="citation" data-cites="Simons:1983 Banerjee:1985"></span> showed that there was a better step than simply directly following one of the Hessian eigenvectors. A simple modification to the Newton-Raphson step is capable of guiding the search away from the current region towards a stationary point with the required characteristics. This is <br /><span class="math display">$$\label{eq:a5t}
	\ensuremath{\mathbf{h}} = \sum_i \frac{-F_i }{\left( b_i -\lambda  \right)} \ensuremath{\mathbf{u}}_i$$</span><br /> in which <span class="math inline"><em>λ</em></span> can be regarded as a shift parameter on the Hessian eigenvalue <span class="math inline"><em>b</em><sub><em>i</em></sub></span>. Scaling the Newton-Raphson step in this manner effectively directs the step to lie primarily, but not exclusively (unlike Poppinger’s algorithm<span class="citation" data-cites="Poppinger:1975"></span>), along one of the local eigenmodes, depending on the value chosen for <span class="math inline"><em>λ</em></span>. References </p>
<p><span class="citation" data-cites="Cerjan:1981 Simons:1983 Banerjee:1985"></span> all use the same basic approach of Eq. (<a href="#eq:a5t" data-reference-type="ref" data-reference="eq:a5t">[eq:a5t]</a>) but differ in the means of determining the value of <span class="math inline"><em>λ</em></span>.</p>
<p>The EF algorithm<span class="citation" data-cites="Baker:1986"></span> uses the rational function approach presented in Refs. </p>
<p><span class="citation" data-cites="Banerjee:1985"></span>, yielding an eigenvalue equation of the form <br /><span class="math display">$$\label{eq:a6}
\left( {{\begin{array}{*{20}c}
 {\rm {\bf H}} \hfill &amp; {\rm {\bf g}} \hfill \\
 {{\rm {\bf g}}^t} \hfill &amp; 0 \hfill \\
\end{array} }} \right)\left( {{\begin{array}{*{20}c}
 {\rm {\bf h}} \hfill \\
 1 \hfill \\
\end{array} }} \right)=\lambda \left( {{\begin{array}{*{20}c}
 {\rm {\bf h}} \hfill \\
 1 \hfill \\
\end{array} }} \right)$$</span><br /> from which a suitable <span class="math inline"><em>λ</em></span> can be obtained. Expanding Eq. (<a href="#eq:a6" data-reference-type="ref" data-reference="eq:a6">[eq:a6]</a>) yields <br /><span class="math display">(<strong>H</strong> − <em>λ</em>)<strong>h</strong> + <strong>g</strong> = 0</span><br /> and <br /><span class="math display"><strong>g</strong><sup><em>t</em></sup><strong>h</strong> = <em>λ</em></span><br /></p>
<p>In terms of a diagonal Hessian representation, Eq. (<a href="#eq:a7a" data-reference-type="ref" data-reference="eq:a7a">[eq:a7a]</a>) rearranges to Eq. (<a href="#eq:a5t" data-reference-type="ref" data-reference="eq:a5t">[eq:a5t]</a>), and substitution of Eq. (<a href="#eq:a5t" data-reference-type="ref" data-reference="eq:a5t">[eq:a5t]</a>) into the diagonal form of Eq. (<a href="#eq:a7b" data-reference-type="ref" data-reference="eq:a7b">[eq:a7b]</a>) gives <br /><span class="math display">$$\sum_i {\frac{-F_i^2 }{\left( {b_i -\lambda } \right)}} =\lambda$$</span><br /> which can be used to evaluate <span class="math inline"><em>λ</em></span> iteratively.</p>
<p>The eigenvalues, <span class="math inline"><em>λ</em></span>, of the RFO equation Eq. (<a href="#eq:a6" data-reference-type="ref" data-reference="eq:a6">[eq:a6]</a>) have the following important properties:<span class="citation" data-cites="Banerjee:1985"></span></p>
<ul>
<li><p>The <span class="math inline">(<em>n</em> + 1)</span> values of <span class="math inline"><em>λ</em></span> bracket the <span class="math inline"><em>n</em></span> eigenvalues of the Hessian matrix <span class="math inline"><em>λ</em><sub><em>i</em></sub> &lt; <em>b</em><sub><em>i</em></sub> &lt; <em>λ</em><sub><em>i</em> + 1</sub></span>.</p></li>
<li><p>At a stationary point, one of the eigenvalues, <span class="math inline"><em>λ</em></span>, of Eq. (<a href="#eq:a6" data-reference-type="ref" data-reference="eq:a6">[eq:a6]</a>) is zero and the other <span class="math inline"><em>n</em></span> eigenvalues are those of the Hessian at the stationary point.</p></li>
<li><p>For a saddle point of order <span class="math inline"><em>m</em></span>, the zero eigenvalue separates the <span class="math inline"><em>m</em></span> negative and the <span class="math inline">(<em>n</em> − <em>m</em>)</span> positive Hessian eigenvalues.</p></li>
</ul>
<p>This last property, the separability of the positive and negative Hessian eigenvalues, enables two shift parameters to be used, one for modes along which the energy is to be maximized and the other for which it is minimized. For a transition state (a first-order saddle point), in terms of the Hessian eigenmodes, we have the two matrix equations <br /><span class="math display">$$\label{eq:a9a}
\left( {{\begin{array}{*{20}c}
 {b_1 } \hfill &amp; {F_1 } \hfill \\
 {F_1 } \hfill &amp; 0 \hfill \\
\end{array} }} \right)\left( {{\begin{array}{*{20}c}
 {h_1 } \hfill \\
 1 \hfill \\
\end{array} }} \right)=\lambda _p \left( {{\begin{array}{*{20}c}
 {h_1 } \hfill \\
 1 \hfill \\
\end{array} }} \right)$$</span><br /> <br /><span class="math display">$$\label{eq:a9b}
\left( {{\begin{array}{*{20}c}
 {b_2 } \hfill &amp; \hfill &amp; \hfill &amp; {F_2 } \hfill \\
 \hfill &amp; \ddots \hfill &amp; {\rm {\bf 0}} \hfill &amp; \vdots \hfill \\
 \hfill &amp; {\rm {\bf 0}} \hfill &amp; {b_n } \hfill &amp; {F_n } \hfill \\
 {F_2 } \hfill &amp; \cdots \hfill &amp; {F_n } \hfill &amp; 0 \hfill \\
\end{array} }} \right)\left( {{\begin{array}{*{20}c}
 {h_2 } \hfill \\
 \vdots \hfill \\
 {h_n } \hfill \\
 1 \hfill \\
\end{array} }} \right)=\lambda _n \left( {{\begin{array}{*{20}c}
 {h_2 } \hfill \\
 \vdots \hfill \\
 {h_n } \hfill \\
 1 \hfill \\
\end{array} }} \right)$$</span><br /> where it is assumed that we are maximizing along the lowest Hessian mode <span class="math inline"><strong>u</strong><sub>1</sub></span>. Note that <span class="math inline"><em>λ</em><sub><em>p</em></sub></span> is the highest eigenvalue of Eq. (<a href="#eq:a9a" data-reference-type="ref" data-reference="eq:a9a">[eq:a9a]</a>), which is always positive and approaches zero at convergence, and <span class="math inline"><em>λ</em><sub><em>n</em></sub></span> is the lowest eigenvalue of Eq. (<a href="#eq:a9b" data-reference-type="ref" data-reference="eq:a9b">[eq:a9b]</a>), which it is always negative and again approaches zero at convergence.</p>
<p>Choosing these values of <span class="math inline"><em>λ</em></span> gives a step that attempts to maximize along the lowest Hessian mode, while at the same time minimizing along all the other modes. It does this regardless of the Hessian eigenvalue structure (unlike the Newton-Raphson step). The two shift parameters are then used in Eq. (<a href="#eq:a5t" data-reference-type="ref" data-reference="eq:a5t">[eq:a5t]</a>) to give the final step <br /><span class="math display">$$\ensuremath{\mathbf{h}} = \frac{-F_1 }{(b_1 -\lambda_p )} \ensuremath{\mathbf{u}}_1 
	-\sum\limits_{i=2}^n \frac{-F_i }{(b_i -\lambda _n )}\ensuremath{\mathbf{u}}_i$$</span><br /> If this step is greater than the maximum allowed, it is scaled down. For minimization only one shift parameter, <span class="math inline"><em>λ</em><sub><em>n</em></sub></span>, would be used which would act on all modes.</p>
<p>In Eq. (<a href="#eq:a9a" data-reference-type="ref" data-reference="eq:a9a">[eq:a9a]</a>) and Eq. (<a href="#eq:a9b" data-reference-type="ref" data-reference="eq:a9b">[eq:a9b]</a>) it was assumed that the step would maximize along the lowest Hessian mode, <span class="math inline"><em>b</em><sub>1</sub></span>, and minimize along all the higher modes. However, it is possible to maximize along modes other than the lowest, and in this way perhaps locate transition states for alternative rearrangements/dissociations from the same initial starting point. For maximization along the <span class="math inline"><em>k</em></span>th mode (instead of the lowest mode), Eq. (<a href="#eq:a9a" data-reference-type="ref" data-reference="eq:a9a">[eq:a9a]</a>) is replaced by <br /><span class="math display">$$\left( {{\begin{array}{*{20}c} {b_k } \hfill &amp; {F_k } \hfill \\
{F_k } \hfill &amp; 0 \hfill \\
\end{array} }} \right)\left( {{\begin{array}{*{20}c}
 {h_k } \hfill \\
 1 \hfill \\
\end{array} }} \right)=\lambda _p \left( {{\begin{array}{*{20}c}
 {h_k } \hfill \\
 1 \hfill \\
\end{array} }} \right)$$</span><br /> and Eq. (<a href="#eq:a9b" data-reference-type="ref" data-reference="eq:a9b">[eq:a9b]</a>) would now exclude the <span class="math inline"><em>k</em></span>th mode but include the lowest. Since what was originally the <span class="math inline"><em>k</em></span>th mode is the mode along which the negative eigenvalue is required, then this mode will eventually become the lowest mode at some stage of the optimization. To ensure that the original mode is being followed smoothly from one cycle to the next, the mode that is actually followed is the one with the greatest overlap with the mode followed on the previous cycle. This procedure is known as <em>mode following</em>. For more details and some examples, see Ref. </p>
<p><span class="citation" data-cites="Baker:1986"></span>.</p>
<h2 id="delocalized-internal-coordinates">Delocalized Internal Coordinates</h2>
<p>The choice of coordinate system can have a major influence on the rate of convergence during a geometry optimization. For complex potential energy surfaces with many stationary points, a different choice of coordinates can result in convergence to a different final structure.</p>
<p>The key attribute of a good set of coordinates for geometry optimization is the degree of coupling between the individual coordinates. In general, the less coupling the better, as variation of one particular coordinate will then have minimal impact on the other coordinates. Coupling manifests itself primarily as relatively large partial derivative terms between different coordinates. For example, a strong harmonic coupling between two different coordinates, <span class="math inline"><em>i</em></span> and <span class="math inline"><em>j</em></span>, results in a large off-diagonal element, <span class="math inline"><em>H</em><sub><em>i</em><em>j</em></sub></span>, in the Hessian (second derivative) matrix. Normally this is the only type of coupling that can be directly “observed” during an optimization, as third and higher derivatives are ignored in almost all optimization algorithms.</p>
<p>In the early days of computational quantum chemistry geometry optimizations were carried out in Cartesian coordinates. Cartesians are an obvious choice as they can be defined for all systems and gradients and second derivatives are calculated directly in Cartesian coordinates. Unfortunately, Cartesians normally make a poor coordinate set for optimization as they are heavily coupled. Cartesians have been returning to favor later on because of their very general nature, and because it has been clearly demonstrated that if reliable second derivative information is available (<em>i.e.</em>, a good starting Hessian) and the initial geometry is reasonable, then Cartesians can be as efficient as any other coordinate set for small to medium-sized molecules.<span class="citation" data-cites="Baker:1991 Baker:1993"></span> Without good Hessian data, however, Cartesians are inefficient, especially for long chain acyclic systems.</p>
<p>In the 1970s Cartesians were replaced by <em>Z</em>-matrix coordinates. Initially the <em>Z</em>-matrix was used simply as a means of geometry input; it is far easier to describe a molecule in terms of bond lengths, bond angles and dihedral angles (the natural way a chemist thinks of molecular structure) than to develop a suitable set of Cartesian coordinates. It was subsequently found that optimization was generally more efficient in <em>Z</em>-matrix coordinates than in Cartesians, especially for acyclic systems. This is not always the case, and care must be taken in constructing a suitable <em>Z</em>-matrix. A good general rule is ensure that each variable is defined in such a way that changing its value will not change the values of any of the other variables. A brief discussion concerning good <em>Z</em>-matrix construction strategy is given by Schlegel.<span class="citation" data-cites="Schlegel:1984"></span></p>
<p>In 1979 Pulay <em>et al.</em> published a key paper, introducing what were termed natural internal coordinates into geometry optimization.<span class="citation" data-cites="Pulay:1979"></span> These coordinates involve the use of individual bond displacements as stretching coordinates, but linear combinations of bond angles and torsions as deformational coordinates. Suitable linear combinations of bends and torsions (the two are considered separately) are selected using group theoretical arguments based on local pseudo-symmetry. For example, bond angles around an <span class="math inline"><em>s</em><em>p</em><sup>3</sup></span> hybridized carbon atom are all approximately tetrahedral, regardless of the groups attached, and idealized tetrahedral symmetry can be used to generate deformational coordinates around the central carbon atom.</p>
<p>The major advantage of natural internal coordinates in geometry optimization is their ability to significantly reduce the coupling, both harmonic and anharmonic, between the various coordinates. Compared to natural internals, <em>Z</em>-matrix coordinates arbitrarily omit some angles and torsions (to prevent redundancy), and this can induce strong anharmonic coupling between the coordinates, especially with a poorly constructed <em>Z</em>-matrix. Another advantage of the reduced coupling is that successful minimizations can be carried out in natural internals with only an approximate (<em>e.g.</em>, diagonal) Hessian provided at the starting geometry. A good starting Hessian is still needed for a transition state search.</p>
<p>Despite their clear advantages, natural internals have only become used widely at a later stage. This is because, when used in the early programs, it was necessary for the user to define them. This situation changed in 1992 with the development of computational algorithms capable of automatically generating natural internals from input Cartesians.<span class="citation" data-cites="Fogarasi:1992"></span> For minimization, natural internals have become the coordinates of first choice.<span class="citation" data-cites="Fogarasi:1992 Baker:1993"></span></p>
<p>There are some disadvantages to natural internal coordinates as they are commonly constructed and used:</p>
<ul>
<li><p>Algorithms for the automatic construction of natural internals are complicated. There are a large number of structural possibilities, and to adequately handle even the most common of them can take several thousand lines of code.</p></li>
<li><p>For the more complex molecular topologies, most assigning algorithms generate more natural internal coordinates than are required to characterize all possible motions of the system (<em>i.e.</em>, the generated coordinate set contains redundancies).</p></li>
<li><p>In cases with a very complex molecular topology (<em>e.g.</em>, multiply fused rings and cage compounds) the assigning algorithm may be unable to generate a suitable set of coordinates.</p></li>
</ul>
<p>The redundancy problem has been addressed in an excellent paper by Pulay and Fogarasi,<span class="citation" data-cites="Pulay:1992"></span> who have developed a scheme for carrying out geometry optimization directly in the redundant coordinate space.</p>
<p>Baker <em>et al.</em><span class="citation" data-cites="Baker:1996"></span> developed a set of delocalized internal coordinates that eliminate all of the above-mentioned difficulties. Building on some of the ideas in the redundant optimization scheme of Pulay and Fogarasi,<span class="citation" data-cites="Pulay:1992"></span> delocalized internals form a complete, non-redundant set of coordinates which are as good as, if not superior to, natural internals, and which can be generated in a simple and straightforward manner for essentially any molecular topology, no matter how complex.</p>
<p>Consider a set of <span class="math inline"><em>n</em></span> internal coordinates <span class="math inline"><strong>q</strong> = (<em>q</em><sub>1</sub>, <em>q</em><sub>2</sub>, …<em>q</em><sub><em>n</em></sub>)<sup><em>t</em></sup></span> Displacements <span class="math inline"><em>Δ</em><strong>q</strong></span> in <span class="math inline"><strong>q</strong></span> are related to the corresponding Cartesian displacements <span class="math inline"><em>Δ</em><strong>X</strong></span> by means of the usual <span class="math inline"><strong>B</strong></span>-matrix,<span class="citation" data-cites="Wilson:1955"></span> <br /><span class="math display"><em>Δ</em><strong>q</strong> = <strong>B</strong><em>Δ</em><strong>X</strong></span><br /> If any of the internal coordinates <span class="math inline"><strong>q</strong></span> are redundant, then the rows of the <span class="math inline"><strong>B</strong></span>-matrix will be linearly dependent.</p>
<p>Delocalized internal coordinates are obtained simply by constructing and diagonalizing the matrix <span class="math inline"><strong>G</strong> = <strong>B</strong><strong>B</strong><sup><em>t</em></sup></span>. Diagonalization of <span class="math inline"><strong>G</strong></span> results in two sets of eigenvectors; a set of <span class="math inline"><em>m</em></span> (typically <span class="math inline">3<em>N</em> − 6</span>, where <span class="math inline"><em>N</em></span> is the number of atoms) eigenvectors with eigenvalues <span class="math inline"><em>λ</em> &gt; 0</span>, and a set of <span class="math inline"><em>n</em><em>m</em></span> eigenvectors with eigenvalues <span class="math inline"><em>λ</em> = 0</span> (to numerical precision). In this way, any redundancies present in the original coordinate set <span class="math inline"><strong>q</strong></span> are isolated (they correspond to those eigenvectors with zero eigenvalues). The eigenvalue equation of <span class="math inline"><strong>G</strong></span> can thus be written <br /><span class="math display">$$\label{eq:a13}
{\rm {\bf G}}({\rm {\bf UR}})=({\rm {\bf UR}})\left( {{\begin{array}{*{20}c}
 \Lambda \hfill &amp; 0 \hfill \\
 0 \hfill &amp; 0 \hfill \\
\end{array} }} \right)$$</span><br /> where <span class="math inline"><strong>U</strong></span> is the set of non-redundant eigenvectors of <span class="math inline"><strong>G</strong></span> (those with <span class="math inline"><em>λ</em> &gt; 0</span>) and <span class="math inline"><strong>R</strong></span> is the corresponding redundant set.</p>
<p>The nature of the original set of coordinates <span class="math inline"><strong>q</strong></span> is unimportant, as long as it spans all the degrees of freedom of the system under consideration. We include in <span class="math inline"><strong>q</strong></span>, all bond stretches, all planar bends and all proper torsions that can be generated based on the atomic connectivity. These individual internal coordinates are termed <em>primitives</em>. This blanket approach generates far more primitives than are necessary, and the set <span class="math inline"><strong>q</strong></span> contains much redundancy. This is of little concern, as solution of Eq. (<a href="#eq:a13" data-reference-type="ref" data-reference="eq:a13">[eq:a13]</a>) takes care of all redundancies.</p>
<p>Note that eigenvectors in both <span class="math inline"><strong>U</strong></span> and <span class="math inline"><strong>R</strong></span> will each be linear combinations of potentially all the original primitives. Despite this apparent complexity, we take the set of non-redundant vectors <span class="math inline"><strong>U</strong></span> as our working coordinate set. Internal coordinates so defined are much more delocalized than natural internal coordinates (which are combinations of a relatively small number of bends or torsions) hence, the term delocalized internal coordinates.</p>
<p>It may appear that because delocalized internals are such a complicated mixing of the original primitive internals, they are a poor choice for use in an actual optimization. On the contrary, arguments can be made that delocalized internals are, in fact, the “best” possible choice, certainly at the starting geometry. The interested reader is referred to the original literature for more details.<span class="citation" data-cites="Baker:1996"></span></p>
<p>The situation for geometry optimization, comparing Cartesian, <em>Z</em>-matrix and delocalized internal coordinates, and assuming a “reasonable” starting geometry, is as follows:</p>
<ul>
<li><p>For small or very rigid medium-sized systems (up to about 15 atoms), optimizations in Cartesian and internal coordinates (“good” <em>Z</em>-matrix or delocalized internals) should perform similarly.</p></li>
<li><p>For medium-sized systems (say 15–30 atoms) optimizations in Cartesians should perform as well as optimizations in internal coordinates, provided a reliable starting Hessian is available.</p></li>
<li><p>For large systems (30+ atoms), unless these are very rigid, neither Cartesian nor <em>Z</em>-matrix coordinates can compete with delocalized internals, even with good quality Hessian information. As the system increases, and with less reliable starting geometries, the advantage of delocalized internals can only increase.</p></li>
</ul>
<p>There is one particular situation in which Cartesian coordinates may be the best choice. Natural internal coordinates (and by extension delocalized internals) show a tendency to converge to low energy structures.<span class="citation" data-cites="Baker:1993"></span> This is because steps taken in internal coordinate space tend to be much larger when translated into Cartesian space, and, as a result, higher energy local minima tend to be “jumped over”, especially if there is no reliable Hessian information available (which is generally not needed for a successful optimization). Consequently, if the user is looking for a local minimum (<em>i.e.</em>, a meta-stable structure) and has both a good starting geometry and a decent Hessian, the user should carry out the optimization in Cartesian coordinates.</p>
<h2 id="constrained-optimization">Constrained Optimization</h2>
<p>Constrained optimization refers to the optimization of molecular structures in which certain parameters (<em>e.g.</em>, bond lengths, bond angles or dihedral angles) are fixed. In quantum chemistry calculations, this has traditionally been accomplished using <em>Z</em>-matrix coordinates, with the desired parameter set in the <em>Z</em>-matrix and simply omitted from the optimization space. In 1992, Baker presented an algorithm for constrained optimization directly in Cartesian coordinates.<span class="citation" data-cites="Baker:1992"></span> Baker’s algorithm used both penalty functions and the classical method of Lagrange multipliers,<span class="citation" data-cites="Fletcher:1981"></span> and was developed in order to impose constraints on a molecule obtained from a graphical model builder as a set of Cartesian coordinates. Some improvements widening the range of constraints that could be handled were made in 1993.<span class="citation" data-cites="Baker:1993"></span> <span><span class="smallcaps">Q-Chem</span></span> includes the latest version of this algorithm, which has been modified to handle constraints directly in delocalized internal coordinates.<span class="citation" data-cites="Baker:1997"></span></p>
<p>The essential problem in constrained optimization is to minimize a function of, for example, <span class="math inline"><em>n</em></span> variables <span class="math inline"><em>F</em>(<strong>x</strong>)</span> subject to a series of m constraints of the form <span class="math inline"><em>C</em><sub><em>i</em></sub>(<strong>x</strong>) = 0</span>, <span class="math inline"><em>i</em> = <em>l</em></span>. <span class="math inline"><em>m</em></span>. Assuming <span class="math inline"><em>m</em> &lt; <em>n</em></span>, then perhaps the best way to proceed (if this were possible in practice) would be to use the <span class="math inline"><em>m</em></span> constraint equations to eliminate <span class="math inline"><em>m</em></span> of the variables, and then solve the resulting unconstrained problem in terms of the (<span class="math inline">(<em>n</em> − <em>m</em>)</span> independent variables. This is exactly what occurs in a <em>Z</em>-matrix optimization. Such an approach cannot be used in Cartesian coordinates as standard distance and angle constraints are non-linear functions of the appropriate coordinates. For example a distance constraint (between atoms <span class="math inline"><em>i</em></span> and <span class="math inline"><em>j</em></span> in a molecule) is given in Cartesians by <span class="math inline">(<em>R</em><sub><em>i</em><em>j</em></sub> − <em>R</em><sub>0</sub>) = 0</span>, with <br /><span class="math display">$$R_{ij} =\sqrt {\left( {x_i -x_j } \right)^2+\left( {y_i -y_j } 
	\right)^2+\left( {z_i -z_j } \right)^2}$$</span><br /> and <span class="math inline"><em>R</em><sub>0</sub></span> the constrained distance. This obviously cannot be satisfied by elimination. What can be eliminated in Cartesians are the individual <span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span> and <span class="math inline"><em>z</em></span> coordinates themselves and in this way individual atoms can be totally or partially frozen.</p>
<p>Internal constraints can be handled in Cartesian coordinates by introducing the Lagrangian function <br /><span class="math display">$$\label{eq:a15}
L(\ensuremath{\mathbf{x}},\lambda) = F(\ensuremath{\mathbf{x}})-\sum\limits_{i=1}^m {\lambda _i C_i(\ensuremath{\mathbf{x}})}$$</span><br /> which replaces the function <span class="math inline"><em>F</em>(<strong>x</strong>)</span> in the unconstrained case. Here, the <span class="math inline"><em>λ</em><sub><em>i</em></sub></span> are the so-called Lagrange multipliers, one for each constraint <span class="math inline"><em>C</em><sub><em>i</em></sub>(<strong>x</strong>)</span>. Differentiating Eq. (<a href="#eq:a15" data-reference-type="ref" data-reference="eq:a15">[eq:a15]</a>) with respect to <span class="math inline"><strong>x</strong></span> and <span class="math inline"><em>λ</em></span> affords <br /><span class="math display">$$\begin{aligned}
 \label{eq:a16a}
	\frac{dL(\ensuremath{\mathbf{x}},\lambda )}{dx_j} &amp; = &amp;  \frac{dF(\ensuremath{\mathbf{x}})}{dx_j}
	-\sum\limits_{i=1}^m \lambda_i \frac{dC_i(\ensuremath{\mathbf{x}})}{dx_j} \\
	\frac{dL(\ensuremath{\mathbf{x}},\lambda)}{d\lambda_i} &amp; = &amp; -C_i (\ensuremath{\mathbf{x}})\end{aligned}$$</span><br /></p>
<p>At a stationary point of the Lagrangian we have <span class="math inline">∇<strong>L</strong> = 0</span>, <em>i.e.</em>, all <span class="math inline"><em>d</em><em>L</em>/<em>d</em><em>x</em><sub><em>j</em></sub> = 0</span> and all <span class="math inline"><em>d</em><em>L</em>/<em>d</em><em>λ</em><sub><em>i</em></sub> = 0</span>. This latter condition means that all <span class="math inline"><em>C</em><sub><em>i</em></sub>(<strong>x</strong>) = 0</span> and thus all constraints are satisfied. Hence, finding a set of values (<span class="math inline"><strong>x</strong></span>, <span class="math inline"><em>λ</em></span>) for which <span class="math inline">∇<strong>L</strong> = 0</span> will give a possible solution to the constrained optimization problem in exactly the same way as finding an <span class="math inline"><strong>x</strong></span> for which <span class="math inline"><strong>g</strong> = ∇<strong>F</strong> = 0</span> gives a solution to the corresponding unconstrained problem.</p>
<p>The Lagrangian second derivative matrix, which is the analogue of the Hessian matrix in an unconstrained optimization, is given by <br /><span class="math display">$$\label{eq:a17}
\nabla^2 \ensuremath{\mathbf{L}} = \left( 
\begin{array}{cc}
\frac{d^2L(\ensuremath{\mathbf{x}},\lambda)}{dx_j dx_k}       &amp; \frac{d^2L(\ensuremath{\mathbf{x}},\lambda)}{dx_j d\lambda _i} \\
\frac{d^2L(\ensuremath{\mathbf{x}},\lambda)}{dx_j d\lambda_i} &amp; \frac{d^2L(\ensuremath{\mathbf{x}},\lambda)}{d\lambda_j d\lambda_i} 
\end{array} \right)$$</span><br /> where <br /><span class="math display">$$\begin{aligned}
% jmh - there are indices here that are clearly incorrect, but I don't have the time to figure out what it really means.
	\frac{d^2L(\ensuremath{\mathbf{x}},\lambda)}{dx_j dx_k} &amp;=&amp; \frac{d^2F(\ensuremath{\mathbf{x}})}{dx_j dx_k} 
	-\sum\limits_{i=1}^m \lambda _i \frac{d^2C_i(\ensuremath{\mathbf{x}})}{dx_j dx_k} \\
	\frac{d^2L(\ensuremath{\mathbf{x}},\lambda)}{dx_j d\lambda_i} &amp;=&amp; \frac{-dC_i(\ensuremath{\mathbf{x}})}{dx_j} \\
	\frac{d^2L(\ensuremath{\mathbf{x}},\lambda)}{d\lambda_j d\lambda_i} &amp;=&amp; 0\end{aligned}$$</span><br /></p>
<p>Thus, in addition to the standard gradient vector and Hessian matrix for the unconstrained function <span class="math inline"><em>F</em>(<strong>x</strong>)</span>, we need both the first and second derivatives (with respect to coordinate displacement) of the constraint functions. Once these quantities are available, the corresponding Lagrangian gradient, given by Eq. (<a href="#eq:a16a" data-reference-type="ref" data-reference="eq:a16a">[eq:a16a]</a>), and Lagrangian second derivative matrix, given by Eq. (<a href="#eq:a17" data-reference-type="ref" data-reference="eq:a17">[eq:a17]</a>), can be formed, and the optimization step calculated in a similar manner to that for a standard unconstrained optimization.<span class="citation" data-cites="Baker:1992"></span></p>
<p>In the Lagrange multiplier method, the unknown multipliers, <span class="math inline"><em>λ</em><sub><em>i</em></sub></span>, are an integral part of the parameter set. This means that the optimization space consists of all <span class="math inline"><em>n</em></span> variables <span class="math inline"><strong>x</strong></span> plus all <span class="math inline"><em>m</em></span> Lagrange multipliers <span class="math inline"><em>λ</em></span>, one for each constraint. The total dimension of the constrained optimization problem, <span class="math inline"><em>n</em><em>m</em></span>, has thus increased by <span class="math inline"><em>m</em></span> compared to the corresponding unconstrained case. The Lagrangian Hessian matrix, <span class="math inline">∇<sup>2</sup><strong>L</strong></span>, has <span class="math inline"><em>m</em></span> extra modes compared to the standard (unconstrained) Hessian matrix, <span class="math inline">∇<sup>2</sup><strong>F</strong></span>. What normally happens is that these additional modes are dominated by the constraints (<em>i.e.</em>, their largest components correspond to the constraint Lagrange multipliers) and they have negative curvature (a negative Hessian eigenvalue). This is perhaps not surprising when one realizes that any motion in the parameter space that breaks the constraints is likely to lower the energy.</p>
<p>Compared to a standard unconstrained minimization, where a stationary point is sought at which the Hessian matrix has all positive eigenvalues, in the constrained problem we are looking for a stationary point of the Lagrangian function at which the Lagrangian Hessian matrix has as many negative eigenvalues as there are constraints (<em>i.e.</em>, we are looking for an <span class="math inline"><em>m</em></span>th-order saddle point). For further details and practical applications of constrained optimization using Lagrange multipliers in Cartesian coordinates; see Ref. </p>
<p><span class="citation" data-cites="Baker:1992"></span></p>
<p>Eigenvector following can be implemented in a constrained optimization in a similar way to the unconstrained case. Considering a constrained minimization with <span class="math inline"><em>m</em></span> constraints, then Eq. <a href="#eq:a9a" data-reference-type="eqref" data-reference="eq:a9a">[eq:a9a]</a> is replaced by <br /><span class="math display">$$\label{eq:a18a}
\left( {{\begin{array}{*{20}c}
 {b_1 } \hfill &amp; \hfill &amp; \hfill &amp; {F_1 } \hfill \\
 \hfill &amp; \ddots \hfill &amp; {\rm {\bf 0}} \hfill &amp; \vdots \hfill \\
 \hfill &amp; {\rm {\bf 0}} \hfill &amp; {b_m } \hfill &amp; {F_m } \hfill \\
 {F_1 } \hfill &amp; \cdots \hfill &amp; {F_m } \hfill &amp; 0 \hfill \\
\end{array} }} \right)\left( {{\begin{array}{*{20}c}
 {h_1 } \hfill \\
 \vdots \hfill \\
 {h_m } \hfill \\
 1 \hfill \\
\end{array} }} \right)=\lambda _p \left( {{\begin{array}{*{20}c}
 {h_1 } \hfill \\
 \vdots \hfill \\
 {h_m } \hfill \\
 1 \hfill \\
\end{array} }} \right)$$</span><br /> and Eq. (<a href="#eq:a9b" data-reference-type="ref" data-reference="eq:a9b">[eq:a9b]</a>) by <br /><span class="math display">$$\label{eq:a18b}
\left( {{\begin{array}{*{20}c}
 {b_{m+1} } \hfill &amp; \hfill &amp; \hfill &amp; {F_{m+1} } \hfill \\
 \hfill &amp; \ddots \hfill &amp; {\rm {\bf 0}} \hfill &amp; \vdots \hfill \\
 \hfill &amp; {\rm {\bf 0}} \hfill &amp; {b_{m+n} } \hfill &amp; {F_{m+n} } \hfill \\
 {F_{m+1} } \hfill &amp; \cdots \hfill &amp; {F_{m+n} } \hfill &amp; 0 \hfill \\
\end{array} }} \right)\left( {{\begin{array}{*{20}c}
 {h_{m+1} } \hfill \\
 \vdots \hfill \\
 {h_{m+n} } \hfill \\
 1 \hfill \\
\end{array} }} \right)=\lambda _n \left( {{\begin{array}{*{20}c}
 {h_{m+1} } \hfill \\
 \vdots \hfill \\
 {h_{m+n} } \hfill \\
 1 \hfill \\
\end{array} }} \right)$$</span><br /> where now the <span class="math inline"><em>b</em><sub><em>i</em></sub></span> are the eigenvalues of <span class="math inline">∇<sup>2</sup><strong>L</strong></span>, with corresponding eigenvectors <span class="math inline"><strong>u</strong><sub><em>i</em></sub></span>, and <span class="math inline"><em>F</em><sub><em>i</em></sub> = <strong>u</strong><sub><em>i</em></sub><sup><em>t</em></sup>∇<strong>L</strong></span>. Here Eq. (<a href="#eq:a18a" data-reference-type="ref" data-reference="eq:a18a">[eq:a18a]</a>) includes the <span class="math inline"><em>m</em></span> constraint modes along which a negative Lagrangian Hessian eigenvalue is required, and Eq. (<a href="#eq:a18b" data-reference-type="ref" data-reference="eq:a18b">[eq:a18b]</a>) includes all the other modes.</p>
<p>Equations (<a href="#eq:a18a" data-reference-type="ref" data-reference="eq:a18a">[eq:a18a]</a>) and (<a href="#eq:a18b" data-reference-type="ref" data-reference="eq:a18b">[eq:a18b]</a>) implement eigenvector following for a constrained minimization. Constrained transition state searches can be carried out by selecting one extra mode to be maximized in addition to the <span class="math inline"><em>m</em></span> constraint modes, <em>i.e.</em>, by searching for a saddle point of the Lagrangian function of order <span class="math inline"><em>m</em> + <em>l</em></span>.</p>
<p>It should be realized that, in the Lagrange multiplier method, the desired constraints are only satisfied at convergence, and not necessarily at intermediate geometries. The Lagrange multipliers are part of the optimization space; they vary just as any other geometrical parameter and, consequently the degree to which the constraints are satisfied changes from cycle to cycle, approaching 100% satisfied near convergence. One advantage this brings is that, unlike in a standard <em>Z</em>-matrix approach, constraints do not have to be satisfied in the starting geometry.</p>
<p>Imposed constraints can normally be satisfied to very high accuracy, <span class="math inline">10<sup> − 6</sup></span> or better. However, problems can arise for both bond and dihedral angle constraints near <span class="math inline">0<sup>∘</sup></span> and <span class="math inline">180<sup>∘</sup></span> and, instead of attempting to impose a single constraint, it is better to split angle constraints near these limiting values into two by using a dummy atom,<span class="citation" data-cites="Baker:1993"></span> exactly analogous to splitting a <span class="math inline">180<sup>∘</sup></span> bond angle into two <span class="math inline">90<sup>∘</sup></span> angles in a <em>Z</em>-matrix.</p>
<p><strong>Note: </strong></p>
<p>Exact <span class="math inline">0<sup>∘</sup></span> and <span class="math inline">180<sup>∘</sup></span> single angle constraints cannot be imposed, as the corresponding constraint normals, <span class="math inline">∇<strong>C</strong><sub><em>i</em></sub></span>, are zero, and would result in rows and columns of zeros in the Lagrangian Hessian matrix.</p>

<h2 id="delocalized-internal-coordinates-1">Delocalized Internal Coordinates</h2>
<p>We do not give further details of the optimization algorithms available in <span><span class="smallcaps">Q-Chem</span></span> for imposing constraints in Cartesian coordinates, as it is far simpler and easier to do this directly in delocalized internal coordinates.</p>
<p>At first sight it does not seem particularly straightforward to impose any constraints at all in delocalized internals, given that each coordinate is potentially a linear combination of all possible primitives. However, this is deceptive, and in fact all standard constraints can be imposed by a relatively simple Schmidt orthogonalization procedure. In this instance consider a unit vector with unit component corresponding to the primitive internal (stretch, bend or torsion) that one wishes to keep constant. This vector is then projected on to the full set, <span class="math inline"><strong>U</strong></span>, of active delocalized coordinates, normalized, and then all <span class="math inline"><em>n</em></span>, for example, delocalized internals are Schmidt orthogonalized in turn to this normalized, projected constraint vector. The last coordinate taken in the active space should drop out (since it will be linearly dependent on the other vectors and the constraint vector) leaving (<span class="math inline"><em>n</em> − 1</span>) active vectors and one constraint vector.</p>
<p>In more detail, the procedure is as follows (taken directly from Ref. </p>
<p><span class="citation" data-cites="Baker:1996"></span>). The initial (usually unit) constraint vector <span class="math inline"><strong>C</strong></span> is projected on to the set <span class="math inline"><strong>U</strong></span> of delocalized internal coordinates according to <br /><span class="math display">$$\ensuremath{\mathbf{C}}^{\ensuremath{\mathrm{proj}}}=\sum {\left\langle {{\rm {\bf C}}} {\left| 
{{{\rm {\bf C}} {{\rm {\bf U}}_k }}} \right. 
\kern-\nulldelimiterspace} {{\rm {\bf U}}_k } \right\rangle {\rm {\bf U}}_k 
}$$</span><br /> where the summation is over all <span class="math inline"><em>n</em></span> active coordinates <span class="math inline"><strong>U</strong><sub><em>k</em></sub></span>. The projected vector <span class="math inline"><strong>C</strong><sup><em>p</em><em>r</em><em>o</em><em>j</em></sup></span> is then normalized and an (<span class="math inline"><em>n</em> + <em>l</em></span>) dimensional vector space <span class="math inline"><strong>V</strong></span> is formed, comprising the normalized, projected constraint vector together with all active delocalized coordinates <br /><span class="math display"><strong>V</strong> = {<strong>C</strong><sup><em>p</em><em>r</em><em>o</em><em>j</em></sup>,<strong>U</strong><sub><em>k</em></sub> <em>k</em>=1.<em>n</em>}</span><br /> This set of vectors is Schmidt orthogonalized according to the standard procedure, <br /><span class="math display">$$\label{eq:a21}
\tilde{\ensuremath{\mathbf{V}}}_k =\alpha _k \left( {{\rm {\bf V}}_k 
-\sum\limits_{l=1}^{k-1} {\left\langle {{\rm {\bf V}}_k } {\left| 
{{{{\rm {\bf V}}_k } {{\rm {\bf \tilde {V}}}_l }}} \right. 
\kern-\nulldelimiterspace} {{\rm {\bf \tilde {V}}}_l } \right\rangle {\rm 
{\bf \tilde {V}}}_l } } \right)$$</span><br /> where the first vector taken, <span class="math inline"><strong>V</strong><sub>1</sub></span>, is <span class="math inline"><strong>C</strong><sup><em>p</em><em>r</em><em>o</em><em>j</em></sup></span>. The <span class="math inline"><em>α</em><sub><em>k</em></sub></span> in Eq. (<a href="#eq:a21" data-reference-type="ref" data-reference="eq:a21">[eq:a21]</a>) is a normalization factor. As noted above, the last vector taken, <span class="math inline"><strong>V</strong><sub><em>n</em> + 1</sub> = <strong>U</strong><sub><em>k</em></sub></span>, will drop out, leaving a fully orthonormal set of (<span class="math inline"><em>n</em> − 1</span>) active vectors and one constraint vector.</p>
<p>After the Schmidt orthogonalization the constraint vector will contain all the weight in the active space of the primitive to be fixed, which will have a zero component in all of the other (<span class="math inline"><em>n</em> − 1</span>) vectors. The fixed primitive has thus been isolated entirely in the constraint vector which can now be removed from the active subspace for the geometry optimization step.</p>
<p>Extension of the above procedure to multiple constraints is straightforward. In addition to constraints on individual primitives, it is also possible to impose combinatorial constraints. For example, if, instead of a unit vector, one started the constraint procedure with a vector in which two components were set to unity, then this would impose a constraint in which the sum of the two relevant primitives were always constant. In theory any desired linear combination of any primitives could be constrained.</p>
<p>Note further that imposed constraints are not confined to those primitive internals generated from the initial atomic connectivity. If we wish to constrain a distance, angle or torsion between atoms that are not formally connected, then all we need to do is add that particular coordinate to our primitive set. It can then be isolated and constrained in exactly the same way as a formal connectivity constraint.</p>
<p>Everything discussed thus far regarding the imposition of constraints in delocalized internal coordinates has involved isolating each constraint in one vector which is then eliminated from the optimization space. This is very similar in effect to a <em>Z</em>-matrix optimization, in which constraints are imposed by elimination. This, of course, can only be done if the desired constraint is satisfied in the starting geometry. We have already seen that the Lagrange multiplier algorithm, used to impose distance, angle and torsion constraints in Cartesian coordinates, can be used even when the constraint is not satisfied initially. The Lagrange multiplier method can also be used with delocalized internals, and its implementation with internal coordinates brings several simplifications and advantages.</p>
<p>In Cartesians, as already noted, standard internal constraints (bond distances, angles and torsions) are somewhat complicated non-linear functions of the <span class="math inline"><em>x</em></span>, <span class="math inline"><em>y</em></span> and <span class="math inline"><em>z</em></span> coordinates of the atoms involved. A torsion, for example, which involves four atoms, is a function of twelve different coordinates. In internals, on the other hand, each constraint is a coordinate in its own right and is therefore a simple linear function of just one coordinate (itself).</p>
<p>If we denote a general internal coordinate by <span class="math inline"><em>R</em></span>, then the constraint function <span class="math inline"><em>C</em><sub><em>i</em></sub>(<strong>R</strong>)</span> is a function of one coordinate, <span class="math inline"><em>R</em><sub><em>i</em></sub></span>, and it and its derivatives can be written <br /><span class="math display"><em>C</em><sub><em>i</em></sub>(<em>R</em><sub><em>i</em></sub>) = <em>R</em><sub><em>i</em></sub> − <em>R</em><sub>0</sub></span><br /> <br /><span class="math display"><em>d</em><em>C</em><sub><em>i</em></sub>(<em>R</em><sub><em>i</em></sub>)/<em>d</em><em>R</em><sub><em>i</em></sub> = 1; <em>d</em><em>C</em><sub><em>i</em></sub>(<em>R</em><sub><em>i</em></sub>)/<em>d</em><em>R</em><sub><em>j</em></sub> = 0</span><br /> <br /><span class="math display"><em>d</em><sup>2</sup><em>C</em><sub><em>i</em></sub>(<em>R</em><sub><em>i</em></sub>)/<em>d</em><em>R</em><sub><em>i</em></sub><em>d</em><em>R</em><sub><em>j</em></sub> = 0</span><br /> where <span class="math inline"><em>R</em><sub>0</sub></span> is the desired value of the constrained coordinate, and <span class="math inline"><em>R</em><sub><em>i</em></sub></span> is its current value. From Eq. (<a href="#eq:a22b" data-reference-type="ref" data-reference="eq:a22b">[eq:a22b]</a>) we see that the constraint normals, <span class="math inline"><em>d</em><em>C</em><sub><em>i</em></sub>(<strong>R</strong>)/<em>d</em><em>R</em><sub><em>i</em></sub></span>, are simply unit vectors and the Lagrangian Hessian matrix, Eq. (<a href="#eq:a17" data-reference-type="ref" data-reference="eq:a17">[eq:a17]</a>), can be obtained from the normal Hessian matrix by adding <span class="math inline"><em>m</em></span> columns (and <span class="math inline"><em>m</em></span> rows) of, again, unit vectors.</p>
<p>A further advantage, in addition to the considerable simplification, is the handling of <span class="math inline">0<sup>∘</sup></span> and <span class="math inline">180<sup>∘</sup></span> dihedral angle constraints. In Cartesian coordinates it is not possible to formally constrain bond angles and torsions to exactly <span class="math inline">0<sup>∘</sup></span> or <span class="math inline">180<sup>∘</sup></span> because the corresponding constraint normal is a zero vector. Similar difficulties do not arise in internal coordinates, at least for torsions, because the constraint normals are unit vectors regardless of the value of the constraint; thus <span class="math inline">0<sup>∘</sup></span> and <span class="math inline">180<sup>∘</sup></span> dihedral angle constraints can be imposed just as easily as any other value. <span class="math inline">180<sup>∘</sup></span> bond angles still cause difficulties, but near-linear arrangements of atoms require special treatment even in unconstrained optimizations; a typical solution involves replacing a near <span class="math inline">180<sup>∘</sup></span> bond angle by two special linear co-planar and perpendicular bends,<span class="citation" data-cites="Califano:1976"></span> and modifying the torsions where necessary. A linear arrangement can be enforced by constraining the co-planar and perpendicular bends.</p>
<p>One other advantage over Cartesians is that in internals the constraint coordinate can be eliminated once the constraint is satisfied to the desired accuracy (the default tolerance is <span class="math inline">10<sup> − 6</sup></span> in atomic units: Bohrs and radians). This is not possible in Cartesians due to the functional form of the constraint. In Cartesians, therefore, the Lagrange multiplier algorithm must be used throughout the entire optimization, whereas in delocalized internal coordinates it need only be used until all desired constraints are satisfied; as constraints become satisfied they can simply be eliminated from the optimization space and once all constraint coordinates have been eliminated standard algorithms can be used in the space of the remaining unconstrained coordinates. Normally, unless the starting geometry is particularly poor in this regard, constraints are satisfied fairly early on in the optimization (and at more or less the same time for multiple constraints), and Lagrange multipliers only need to be used in the first half-dozen or so cycles of a constrained optimization in internal coordinates.</p>
<h2 id="gdiis">GDIIS</h2>
<p>Direct inversion in the iterative subspace (DIIS) was originally developed by Pulay for accelerating SCF convergence.<span class="citation" data-cites="Pulay:1982"></span> Subsequently, Csaszar and Pulay used a similar scheme for geometry optimization, which they termed GDIIS.<span class="citation" data-cites="Csaszar:1984"></span> The method is somewhat different from the usual quasi-Newton type approach and is included in <span><span class="smallcaps">Optimize</span></span> as an alternative to the EF algorithm. Tests indicate that its performance is similar to EF, at least for small systems; however there is rarely an advantage in using GDIIS in preference to EF.</p>
<p>In GDIIS, geometries <span class="math inline"><strong>x</strong><sub><em>i</em></sub></span> generated in previous optimization cycles are linearly combined to find the “best” geometry on the current cycle <br /><span class="math display">$$\label{eq:a23}
	\ensuremath{\mathbf{x}}_n = \sum\limits_{i=1}^m c_i \ensuremath{\mathbf{x}}_i$$</span><br /> where the problem is to find the best values for the coefficients <span class="math inline"><em>c</em><sub><em>i</em></sub></span>.</p>
<p>If we express each geometry, <span class="math inline"><strong>x</strong><sub><em>i</em></sub></span>, by its deviation from the sought-after final geometry, <span class="math inline"><strong>x</strong><sub><em>f</em></sub></span>, <em>i.e.</em>, <span class="math inline"><strong>x</strong><sub><em>f</em></sub> = <strong>x</strong><sub><em>i</em></sub> + <strong>e</strong><sub><em>i</em></sub></span>, where <span class="math inline"><strong>e</strong><sub><em>i</em></sub></span> is an error vector, then it is obvious that if the conditions <br /><span class="math display"><strong>r</strong> = ∑<em>c</em><sub><em>i</em></sub><strong>e</strong><sub><em>i</em></sub></span><br /> and <br /><span class="math display">∑<em>c</em><sub><em>i</em></sub> = 1</span><br /> are satisfied, then the relation <br /><span class="math display">∑<em>c</em><sub><em>i</em></sub><strong>x</strong><sub><em>i</em></sub> = <strong>x</strong><sub><em>f</em></sub></span><br /> also holds.</p>
<p>The true error vectors <span class="math inline"><strong>e</strong><sub><em>i</em></sub></span> are, of course, unknown. However, in the case of a nearly quadratic energy function they can be approximated by <br /><span class="math display"><strong>e</strong><sub><em>i</em></sub> =  − <strong>H</strong><sup> − 1</sup><strong>g</strong><sub><em>i</em></sub></span><br /> where <span class="math inline"><strong>g</strong><sub><em>i</em></sub></span> is the gradient vector corresponding to the geometry <span class="math inline"><strong>x</strong><sub><em>i</em></sub></span> and <span class="math inline"><strong>H</strong></span> is an approximation to the Hessian matrix. Minimization of the norm of the residuum vector <span class="math inline"><strong>r</strong></span>, Eq. (<a href="#eq:a24" data-reference-type="ref" data-reference="eq:a24">[eq:a24]</a>), together with the constraint equation, Eq. (<a href="#eq:a25" data-reference-type="ref" data-reference="eq:a25">[eq:a25]</a>), leads to a system of (<span class="math inline"><em>m</em> + <em>l</em></span>) linear equations <br /><span class="math display">$$\label{eq:a28}
\left( {{\begin{array}{*{20}c}
 {B_{11} } \hfill &amp; \cdots \hfill &amp; {B_{1m} } \hfill &amp; 1 \hfill \\
 \vdots \hfill &amp; \ddots \hfill &amp; \vdots \hfill &amp; \vdots \hfill \\
 {B_{m1} } \hfill &amp; \cdots \hfill &amp; {B_{mm} } \hfill &amp; 1 \hfill \\
 1 \hfill &amp; \cdots \hfill &amp; 1 \hfill &amp; 0 \hfill \\
\end{array} }} \right)\left( {{\begin{array}{*{20}c}
 {c_1 } \hfill \\
 \vdots \hfill \\
 {c_m } \hfill \\
 {-\lambda } \hfill \\
\end{array} }} \right)=\left( {{\begin{array}{*{20}c}
 0 \hfill \\
 \vdots \hfill \\
 0 \hfill \\
 1 \hfill \\
\end{array} }} \right)$$</span><br /> where <span class="math inline"><em>B</em><sub><em>i</em><em>j</em></sub> = ⟨<strong>e</strong><sub><em>i</em></sub>|<strong>e</strong><sub><em>j</em></sub>⟩</span> is the scalar product of the error vectors <span class="math inline"><strong>e</strong><sub><em>i</em></sub></span> and <span class="math inline"><strong>e</strong><sub><em>j</em></sub></span>, and <span class="math inline"><em>λ</em></span> is a Lagrange multiplier.</p>
<p>The coefficients <span class="math inline"><em>c</em><sub><em>i</em></sub></span> determined from Eq. (<a href="#eq:a28" data-reference-type="ref" data-reference="eq:a28">[eq:a28]</a>) are used to calculate an intermediate interpolated geometry <br /><span class="math display"><strong>x</strong><sub><em>m</em> + 1</sub><sup>′</sup> = ∑<em>c</em><sub><em>i</em></sub><strong>x</strong><sub><em>i</em></sub></span><br /> and its corresponding interpolated gradient <br /><span class="math display"><strong>g</strong><sub><em>m</em> + 1</sub><sup>′</sup> = ∑<em>c</em><sub><em>i</em></sub><strong>g</strong><sub><em>i</em></sub></span><br /> A new, independent geometry is generated from the interpolated geometry and gradient according to <br /><span class="math display"><strong>x</strong><sub><em>m</em> + 1</sub> = <strong>x</strong><sub><em>m</em> + 1</sub><sup>′</sup> − <strong>H</strong><sup> − 1</sup><strong>g</strong><sub><em>m</em> + 1</sub><sup>′</sup></span><br /></p>
<p><strong>Note: </strong></p>
<p>Convergence is theoretically guaranteed regardless of the quality of the Hessian matrix (as long as it is positive definite), and the original GDIIS algorithm used a static Hessian (<em>i.e.</em>, the original starting Hessian, often a simple unit matrix, remained unchanged during the entire optimization). However, updating the Hessian at each cycle generally results in more rapid convergence, and this is the default in<span><span class="smallcaps">Optimize</span></span>.</p>
<p>Other modifications to the original method include limiting the number of previous geometries used in Eq. <a href="#eq:a23" data-reference-type="eqref" data-reference="eq:a23">[eq:a23]</a> and, subsequently, by neglecting earlier geometries, and eliminating any geometries more than a certain distance from the current geometry (default = 0.3 a.u.).</p>






<h1 id="app:AOINTS"><span><span class="smallcaps">AOInts</span></span></h1>
<h2 id="introduction-2">Introduction</h2>
<p>Within the <span><span class="smallcaps">Q-Chem</span></span> program, an Atomic Orbital integrals (<span><span class="smallcaps">AOInts</span></span>) package has been developed which, while relatively invisible to the user, is one of the keys to the overall speed and efficiency of the <span><span class="smallcaps">Q-Chem</span></span> program.</p>
<p>“Ever since Boys’ introduction of Gaussian basis sets to quantum chemistry in 1950, the calculation and handling of the notorious two-electron repulsion integrals (ERIs) over Gaussian functions has been an important avenue of research for practicing computational chemists. Indeed, the emergence of practically useful computer programs has been fueled in no small part by the development of sophisticated algorithms to compute the very large number of ERIs that are involved in calculations on molecular systems of even modest size”.<span class="citation" data-cites="Gill:1990"></span></p>
<p>The ERI engine of any competitive quantum chemistry software package will be one of the most complicated aspects of the package as whole. Coupled with the importance of such an engine’s efficiency, a useful yardstick of a program’s anticipated performance can be quickly measured by considering the components of its ERI engine. In recent times, developers at <span><span class="smallcaps">Q-Chem</span></span>, Inc. have made significant contributions to the advancement of ERI algorithm technology (for example, see Refs. </p>
<p><span class="citation" data-cites="Gill:1990 Gill:1994a Adams:1997 Frisch:1993 Gill:1991a Gill:1991b Gill:1994b Head-Gordon:1988b Johnson:1993a Johnson:1993b"></span>), and it is not surprising that <span><span class="smallcaps">Q-Chem</span></span>’s <span><span class="smallcaps">AOInts</span></span> package is considered the most advanced of its kind.</p>
<h2 id="historical-perspective">Historical Perspective</h2>
<p>Prior to the 1950s, the most difficult step in the systematic application of Schrödinger wave mechanics to chemistry was the calculation of the notorious two-electron integrals that measure the repulsion between electrons. Boys<span class="citation" data-cites="Boys:1950"></span> showed that this step can be made easier (although still time consuming) if Gaussian, rather than Slater, orbitals are used in the basis set. Following the landmark paper of computational chemistry<span class="citation" data-cites="Boys:1956"></span> (again due to Boys) programs were constructed that could calculate all the ERIs that arise in the treatment of a general polyatomic molecule with <span class="math inline"><em>s</em></span> and <span class="math inline"><em>p</em></span> orbitals. However, the programs were painfully slow and could only be applied to the smallest of molecular systems.</p>
<p>In 1969, Pople constructed a breakthrough ERI algorithm, a hundred time faster than its predecessors. The algorithm remains the fastest available for its associated integral classes and is now referred to as the Pople-Hehre axis-switch method.<span class="citation" data-cites="Pople:1978"></span></p>
<p>Over the two decades following Pople’s initial development, an enormous amount of research effort into the construction of ERIs was documented, which built on Pople’s original success. Essentially, the advances of the newer algorithms could be identified as either better coping with angular momentum (<span class="math inline"><em>L</em></span>) or, contraction (<span class="math inline"><em>K</em></span>); each new method increasing the speed and application of quantum mechanics to solving real chemical problems.</p>
<p>By 1990, another barrier had been reached. The contemporary programs had become sophisticated and both academia and industry had begun to recognize and use the power of <em>ab initio</em> quantum chemistry, but the software was struggling with “dusty deck syndrome” and it had become increasingly difficult for it to keep up with the rapid advances in hardware development. Vector processors, parallel architectures and the advent of the graphical user interface were all demanding radically different approaches to programming and it had become clear that a fresh start, with a clean slate, was both inevitable and desirable. Furthermore, the integral bottleneck had re-emerged in a new guise and the standard programs were now hitting the <span class="math inline"><em>N</em><sup>2</sup></span> wall. Irrespective of the speed at which ERIs could be computed, the unforgiving fact remained that the number of ERIs required scaled quadratically with the size of the system.</p>
<p>The <span><span class="smallcaps">Q-Chem</span></span> project was established to tackle this problem and to seek new methods that circumvent the <span class="math inline"><em>N</em><sup>2</sup></span> wall. Fundamentally new approaches to integral theory were sought and the ongoing advances that have resulted<span class="citation" data-cites="White:1994a Adamson:1996 Dombroski:1996 Challacombe:1997 Schwegler:1996a"></span> have now placed <span><span class="smallcaps">Q-Chem</span></span> firmly at the vanguard of the field. It should be emphasized, however, that the methods that we have developed still require short-range ERIs to treat interactions between nearby electrons, thus the importance of contemporary ERI code remains.</p>
<p>The chronological development and evolution of integral methods can be summarized by considering a time line showing the years in which important new algorithms were first introduced. These are best discussed in terms of the type of ERI or matrix elements that the algorithm can compute efficiently.</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;">1950</td>
<td style="text-align: left;">Boys</td>
<td style="text-align: left;"><span class="citation" data-cites="Boys:1950"></span></td>
<td style="text-align: left;">ERIs with low <span class="math inline"><em>L</em></span> and low <span class="math inline"><em>K</em></span></td>
</tr>
<tr class="even">
<td style="text-align: left;">1969</td>
<td style="text-align: left;">Pople</td>
<td style="text-align: left;"><span class="citation" data-cites="Pople:1978"></span></td>
<td style="text-align: left;">ERIs with low <span class="math inline"><em>L</em></span> and high <span class="math inline"><em>K</em></span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">1976</td>
<td style="text-align: left;">Dupuis</td>
<td style="text-align: left;"><span class="citation" data-cites="Dupuis:1976"></span></td>
<td style="text-align: left;">Integrals with any <span class="math inline"><em>L</em></span> and low <span class="math inline"><em>K</em></span></td>
</tr>
<tr class="even">
<td style="text-align: left;">1978</td>
<td style="text-align: left;">McMurchie</td>
<td style="text-align: left;"><span class="citation" data-cites="McMurchie:1978"></span></td>
<td style="text-align: left;">Integrals with any <span class="math inline"><em>L</em></span> and low <span class="math inline"><em>K</em></span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">1982</td>
<td style="text-align: left;">Almlöf</td>
<td style="text-align: left;"><span class="citation" data-cites="Almlof:1982"></span></td>
<td style="text-align: left;">Introduction of the direct SCF approach</td>
</tr>
<tr class="even">
<td style="text-align: left;">1986</td>
<td style="text-align: left;">Obara</td>
<td style="text-align: left;"><span class="citation" data-cites="Obara:1986"></span></td>
<td style="text-align: left;">Integrals with any <span class="math inline"><em>L</em></span> and low <span class="math inline"><em>K</em></span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">1988</td>
<td style="text-align: left;">Head-Gordon</td>
<td style="text-align: left;"><span class="citation" data-cites="Head-Gordon:1988b"></span></td>
<td style="text-align: left;">Integrals with any <span class="math inline"><em>L</em></span> and low <span class="math inline"><em>K</em></span></td>
</tr>
<tr class="even">
<td style="text-align: left;">1991</td>
<td style="text-align: left;">Gill</td>
<td style="text-align: left;"><span class="citation" data-cites="Gill:1990 Gill:1991b"></span></td>
<td style="text-align: left;">Integrals with any <span class="math inline"><em>L</em></span> and any <span class="math inline"><em>K</em></span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">1994</td>
<td style="text-align: left;">White</td>
<td style="text-align: left;"><span class="citation" data-cites="White:1994a"></span></td>
<td style="text-align: left;">J matrix in linear work</td>
</tr>
<tr class="even">
<td style="text-align: left;">1996</td>
<td style="text-align: left;">Schwegler</td>
<td style="text-align: left;"><span class="citation" data-cites="Schwegler:1996a Schwegler:1996b"></span></td>
<td style="text-align: left;">HF exchange matrix in linear work</td>
</tr>
<tr class="odd">
<td style="text-align: left;">1997</td>
<td style="text-align: left;">Challacombe</td>
<td style="text-align: left;"><span class="citation" data-cites="Challacombe:1997"></span></td>
<td style="text-align: left;">Fock matrix in linear work</td>
</tr>
</tbody>
</table>
<h2 id="aoints-calculating-eris-with-q-chem"><span><span class="smallcaps">AOInts</span></span>: Calculating ERIs with <span><span class="smallcaps">Q-Chem</span></span></h2>
<p>The area of molecular integrals with respect to Gaussian basis functions has recently been reviewed<span class="citation" data-cites="Gill:1994a"></span> and the user is referred to this review for deeper discussions and further references to the general area. The purpose of this short account is to present the basic approach, and in particular, the implementation of ERI algorithms and aspects of interest to the user in the <span><span class="smallcaps">AOInts</span></span> package which underlies the <span><span class="smallcaps">Q-Chem</span></span> program.</p>
<p>We begin by observing that all of the integrals encountered in an <em>ab initio</em> calculation, of which overlap, kinetic energy, multipole moment, internuclear repulsion, nuclear-electron attraction and inter electron repulsion are the best known, can be written in the general form <br /><span class="math display">$$\label{eq:b1}
\left( {{\rm {\bf ab}}\vert {\rm {\bf cd}}} \right)=\int {\phi _{\rm {\bf 
a}} ({\rm {\bf r}}_{\rm {\bf 1}} )\phi _{\rm {\bf b}} ({\rm {\bf r}}_{\rm 
{\bf 1}} )\theta (r_{12} )\phi _{\rm {\bf c}} ({\rm {\bf r}}_{\rm {\bf 2}} 
)\phi _{\rm {\bf d}} ({\rm {\bf r}}_{\rm {\bf 2}} )d{\rm {\bf r}}_{\rm {\bf 
1}} d{\rm {\bf r}}_{\rm {\bf 2}} }$$</span><br /> where the basis functions are contracted Gaussians (CGTF) <br /><span class="math display">$$\phi _{\rm {\bf a}} ({\rm {\bf r}})=\left( {x-A_x } \right)^{a_x }\left( 
{y-A_y } \right)^{a_y }\left( {z-A_z } \right)^{a_z }\sum\limits_{i=1}^{K_a 
} {D_{ai} e^{-\alpha _i \left| {{\rm {\bf r}}-{\rm {\bf A}}} \right|^2}}$$</span><br /> and the operator <span class="math inline"><em>θ</em></span> is a two-electron operator. Of the two-electron operators (Coulomb, CASE, anti-Coulomb and delta-function) used in the <span><span class="smallcaps">Q-Chem</span></span> program, the most significant is the Coulomb, which leads us to the ERIs.</p>
<p>An ERI is the classical Coulomb interaction (<span class="math inline"><em>θ</em>(<em>x</em>) = 1/<em>x</em></span> in <a href="#eq:b1" data-reference-type="ref" data-reference="eq:b1">[eq:b1]</a>) between two charge distributions referred to as bras <span class="math inline">(<strong>a</strong><strong>b</strong>|</span> and kets <span class="math inline">|<strong>c</strong><strong>d</strong>)</span>.</p>
<h2 id="shell-pair-data">Shell-Pair Data</h2>
<p>It is common to characterize a bra, a ket and a bra-ket by their degree of contraction and angular momentum. In general, it is more convenient to compile data for shell-pairs rather than basis-function pairs. A shell is defined as that sharing common exponents and centers. For example, in the case of a number of Pople derived basis sets, four basis functions, encompassing a range of angular momentum types (<em>i.e.</em>, <span class="math inline"><em>s</em></span>, <span class="math inline"><em>p</em><sub><em>x</em></sub></span>, <span class="math inline"><em>p</em><sub><em>y</em></sub></span>, <span class="math inline"><em>p</em><sub><em>z</em></sub></span> on the same atomic center sharing the same exponents constitute a single shell.</p>
<p>The shell-pair data set is central to the success of any modern integral program for three main reasons. First, in the formation of shell-pairs, all pairs of shells in the basis set are considered and categorized as either significant or negligible. A shell-pair is considered negligible if the shells involved are so far apart, relative to their diffuseness, that their overlap is negligible. Given the rate of decay of Gaussian basis functions, it is not surprising that most of the shell-pairs in a large molecule are negligible, that is, the number of significant shell-pairs increases linearly with the size of the molecule. Second, a number of useful intermediates which are frequently required within ERI algorithms should be computed once in shell-pair formation and stored as part of the shell-pair information, particularly those which require costly divisions. This prevents re-evaluating simple quantities. Third, it is useful to sort the shell-pair information by type (<em>i.e.</em>, angular momentum and degree of contraction). The reasons for this are discussed below.</p>
<p><span><span class="smallcaps">Q-Chem</span></span>’s shell-pair formation offers the option of two basic integral shell-pair cutoff criteria; one based on the integral threshold (<em>$rem</em> variable ) and the other relative to machine precision.</p>
<p>Intelligent construction of shell-pair data scales linearly with the size of the basis set, requires a relative amount of CPU time which is almost entirely negligible for large direct SCF calculations, and for small jobs, constitutes approximately 10<span>%</span> of the job time.</p>
<h2 id="shell-quartets-and-integral-classes">Shell-Quartets and Integral Classes</h2>
<p>Given a sorted list of shell-pair data, it is possible to construct all potentially important shell-quartets by pairing of the shell-pairs with one another. Because the shell-pairs have been sorted, it is possible to deal with batches of integrals of the same type or class (<em>e.g.</em>, <span class="math inline">(<em>s</em><em>s</em>|<em>s</em><em>s</em>)</span>, <span class="math inline">(<em>s</em><em>p</em>|<em>s</em><em>p</em>)</span>, <span class="math inline">(<em>d</em><em>d</em>|<em>d</em><em>d</em>)</span>, <em>etc.</em>) where an integral class is characterized by both angular momentum (<span class="math inline"><em>L</em></span>) and degree of contraction (<span class="math inline"><em>K</em></span>). Such an approach is advantageous for vector processors and for semi-direct integral algorithms where the most expensive (high <span class="math inline"><em>K</em></span> or <span class="math inline"><em>L</em></span> integral classes can be computed once, stored in memory (or disk) and only less expensive classes rebuilt on each iteration.</p>
<p>While the shell-pairs may have been carefully screened, it is possible for a pair of significant shell-pairs to form a shell-quartet which need not be computed directly. Three cases are:</p>
<ul>
<li><p>The quartet is equivalent, by point group symmetry, to another quartet already treated.</p></li>
<li><p>The quartet can be ignored on the basis of cheaply computed ERI bounds<span class="citation" data-cites="Gill:1994b"></span> on the largest quartet bra-ket.</p></li>
<li><p>On the basis of an incremental Fock matrix build, the largest density matrix element which will multiply any of the bra-kets associated with the quartet may be negligibly small.</p></li>
</ul>
<p><strong>Note: </strong></p>
<p>Significance and negligibility is always based on the level of integral threshold set by the <em>$rem</em> variable .</p>

<h2 id="fundamental-eri">Fundamental ERI</h2>
<p>The fundamental ERI and the basis of all ERI algorithms is usually represented<span class="citation" data-cites="Gill:1994a"></span> <br /><span class="math display">$$\begin{aligned}
 [\ensuremath{\mathbf{0}}]^{(0)} &amp; \!\!\!=\!\!\! &amp; [ss\vert ss]^{(0)} \nonumber\\ 
               &amp; \!\!\!=\!\!\! &amp; D_A D_B D_C D_D \int {e^{-\alpha 
\left| {{\rm {\bf r}}_{\rm {\bf 1}} 
-{\rm {\bf A}}} \right|^2}e^{-\beta \left| {{\rm {\bf r}}_{\rm {\bf 1}} 
-{\rm {\bf B}}} \right|^2}\left[ {\frac{1}{r_{12} }} \right]e^{-\gamma 
\left| {{\rm {\bf r}}_{\rm {\bf 2}} -{\rm {\bf C}}} \right|^2}e^{-\delta 
\left| {{\rm {\bf r}}_2 -{\rm {\bf D}}} \right|^2}d{\rm {\bf r}}_{\rm {\bf 
1}} d{\rm {\bf r}}_{\rm {\bf 2}} }\end{aligned}$$</span><br /> which can be reduced to a one-dimensional integral of the form <br /><span class="math display">$$\label{eq:b4}
[{\rm {\bf 0}}]^{(0)}=U(2\,{\vartheta}^2)^{1/2}
\left({\frac{2}{\pi }}\right)^{1/2} 
\, \int\limits_0^1 {e^{-Tu^2}du}$$</span><br /> and can be efficiently computed using a modified Chebyshev interpolation scheme.<span class="citation" data-cites="Gill:1991a"></span> Equation <a href="#eq:b4" data-reference-type="eqref" data-reference="eq:b4">[eq:b4]</a> can also be adapted for the general case <span class="math inline">[<strong>0</strong>]<sup>(<em>m</em>)</sup></span> integrals required for most calculations. Following the fundamental ERI, building up to the full bra-ket ERI (or intermediary matrix elements, see later) are the problems of angular momentum and contraction.</p>
<p><strong>Note: </strong></p>
<p>Square brackets denote primitive integrals and parentheses denote fully-contracted integrals.</p>

<h2 id="angular-momentum-problem">Angular Momentum Problem</h2>
<p>The fundamental integral is essentially an integral without angular momentum (<em>i.e.</em>, it is an integral of the type <span class="math inline">[<em>s</em><em>s</em>|<em>s</em><em>s</em>]</span>). Angular momentum, usually depicted by <span class="math inline"><em>L</em></span>, has been problematic for efficient ERI formation, evident in the above time line. Initially, angular momentum was calculated by taking derivatives of the fundamental ERI with respect to one of the Cartesian coordinates of the nuclear center. This is an extremely inefficient route, but it works and was appropriate in the early development of ERI methods. Recursion relations<span class="citation" data-cites="Obara:1986 Obara:1988"></span> and the newly developed tensor equations<span class="citation" data-cites="Adams:1997"></span> are the basis for the modern approaches.</p>
<h2 id="contraction-problem">Contraction Problem</h2>
<p>The contraction problem may be described by considering a general contracted ERI of <span class="math inline"><em>s</em></span>-type functions derived from the STO-3G basis set. Each basis function has degree of contraction <span class="math inline"><em>K</em></span> = 3. Thus, the ERI may be written <br /><span class="math display">$$\begin{aligned}
 (ss\vert ss) &amp; = &amp; \sum\limits_{i=1}^3 {\sum\limits_{j=1}^3 {\sum\limits_{k=1}^3 
{\sum\limits_{l=1}^3 {D_{Ai} D_{Bj} D_{Ck} D_{Dl}} } } } \nonumber \\ 
&amp;&amp; \times\int {e^{-\alpha _i \left| {{\rm {\bf r}}_{\rm {\bf 1}} -{\rm {\bf A}}} 
\right|^2}e^{-\beta _j \left| {{\rm {\bf r}}_{\rm {\bf 1}} -{\rm {\bf B}}} 
\right|^2}\left[ {\frac{1}{r_{12} }} \right]e^{-\gamma _k \left| {{\rm {\bf 
r}}_{\rm {\bf 2}} -{\rm {\bf C}}} \right|^2}e^{-\delta _l \left| {{\rm {\bf 
r}}_2 -{\rm {\bf D}}} \right|^2}d{\rm {\bf r}}_{\rm {\bf 1}} d{\rm {\bf 
r}}_{\rm {\bf 2}} } \nonumber\\ 
 &amp; = &amp;\sum\limits_{i=1}^3 {\sum\limits_{j=1}^3 {\sum\limits_{k=1}^3 
{\sum\limits_{l=1}^3 {[s_i s_j \vert s_k s_l ]} } } } \end{aligned}$$</span><br /> and requires 81 primitive integrals for the single ERI. The problem escalates dramatically for more highly contracted sets (STO-6G, 6-311G) and has been the motivation for the development of techniques for shell-pair modeling,<span class="citation" data-cites="Adamson:1995"></span> in which a second shell-pair is constructed with fewer primitives that the first, but introduces no extra error relative to the integral threshold sought.</p>
<p>The Pople-Hehre axis-switch method<span class="citation" data-cites="Pople:1978"></span> is excellent for high contraction low angular momentum integral classes.</p>
<h2 id="quadratic-scaling">Quadratic Scaling</h2>
<p>The success of quantitative modern quantum chemistry, relative to its primitive, qualitative beginnings, can be traced to two sources: better algorithms and better computers. While the two technologies continue to improve rapidly, efforts are heavily thwarted by the fact that the total number of ERIs increases quadratically with the size of the molecular system. Even large increases in ERI algorithm efficiency yield only moderate increases in applicability, hindering the more widespread application of <em>ab initio</em> methods to areas of, perhaps, biochemical significance where semi-empirical techniques<span class="citation" data-cites="Dewar:1969 Dewar:1993"></span> have already proven so valuable.</p>
<p>Thus, the elimination of quadratic scaling algorithms has been the theme of many research efforts in quantum chemistry throughout the 1990s and has seen the construction of many alternative algorithms to alleviate the problem. Johnson was the first to implement DFT exchange/correlation functionals whose computational cost scaled linearly with system size.<span class="citation" data-cites="Johnson:1993c"></span> This paved the way for the most significant breakthrough in the area with the linear scaling CFMM algorithm<span class="citation" data-cites="White:1994a"></span> leading to linear scaling DFT calculations.<span class="citation" data-cites="White:1996a"></span> Further breakthroughs have been made with traditional theory in the form of the QCTC<span class="citation" data-cites="Challacombe:1996a Challacombe:1996b Challacombe:1997"></span> and ONX<span class="citation" data-cites="Schwegler:1996a Schwegler:1996b"></span> algorithms, while more radical approaches<span class="citation" data-cites="Adamson:1996 Dombroski:1996"></span> may lead to entirely new approaches to <em>ab initio</em> calculations. Investigations into the quadratic Coulomb problem has not only yielded linear scaling algorithms, but is also providing large insights into the significance of many molecular energy components.</p>
<p>Linear scaling Coulomb and SCF exchange/correlation algorithms are not the end of the story as the diagonalization step has been rate limiting in semi-empirical techniques and, been predicted to become rate limiting in <em>ab initio</em> approaches in the medium term.<span class="citation" data-cites="Strout:1995"></span> However, divide-and-conquer techniques<span class="citation" data-cites="Yang:1991a Yang:1991b Yang:1995 Lee:1996"></span> and the recently developed quadratically convergent SCF algorithm<span class="citation" data-cites="Ochsenfeld:1997"></span> show great promise for reducing this problem.</p>
<h2 id="algorithm-selection">Algorithm Selection</h2>
<p>No single ERI algorithm is available to efficiently handle all integral classes; rather, each tends to have specific integral classes where the specific algorithm outperforms the alternatives. The PRISM algorithm<span class="citation" data-cites="Gill:1991b"></span> is an intricate collection of pathways and steps in which the path chosen is that which is the most efficient for a given class. It appears that the most appropriate path for a given integral class depends on the relative position of the contraction step (lowly contracted bra-kets prefer late contraction, highly contracted bra-kets are most efficient with early contraction steps).</p>
<p>Careful studies have provided FLOP counts which are the current basis of integral algorithm selection, although care must be taken to ensure that algorithms are not rate limited by MOPs.<span class="citation" data-cites="Frisch:1993"></span> Future algorithm selection criteria will take greater account of memory, disk, chip architecture, cache size, vectorization and parallelization characteristics of the hardware, many of which are already exist within <span><span class="smallcaps">Q-Chem</span></span>.</p>
<h2 id="more-efficient-hartreefock-gradient-and-hessian-evaluations">More Efficient Hartree–Fock Gradient and Hessian Evaluations</h2>
<p><span><span class="smallcaps">Q-Chem</span></span> combines the Head-Gordon–Pople (HGP) method<span class="citation" data-cites="Head-Gordon:1988b"></span> and the COLD prism method<span class="citation" data-cites="Adams:1997"></span> for Hartree-Fock gradient and Hessian evaluations. All two-electron four-center integrals are classified according to their angular momentum types and degrees of contraction. For each type of integrals, the program chooses one with a lower cost. In practice, the HGP method is chosen for most integral classes in a gradient or Hessian calculation, and thus it dominates the total CPU time.</p>
<p>Recently the HGP codes within <span><span class="smallcaps">Q-Chem</span></span> were completely rewritten for the evaluation of the P II<span class="math inline"><em></em><sup><em>x</em></sup></span> P term in the gradient evaluation, and the P II<span class="math inline"><em></em><sup><em>x</em><em>y</em></sup></span> P term in the Hessian evaluation. Our emphasis is to improve code efficiency by reducing cache misses rather than by reducing FLOP counts. Some timing results from a Hartree-Fock calculation on azt are shown below.</p>
<table>
<caption>AIX timings were obtained on an IBM RS/6000 workstation with AIX4 operating system, and Linux timings on an Opteron cluster where the <span><span class="smallcaps">Q-Chem</span></span> executable was compiled with an Intel 32-bit compiler.</caption>
<thead>
<tr class="header">
<th style="text-align: left;"><strong>Basis Set</strong></th>
<th style="text-align: right;"></th>
<th style="text-align: right;"></th>
<th style="text-align: right;"></th>
<th style="text-align: right;"></th>
<th style="text-align: right;"></th>
<th style="text-align: right;"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: right;">Old</td>
<td style="text-align: right;">New</td>
<td style="text-align: right;">New/Old</td>
<td style="text-align: right;">Old</td>
<td style="text-align: right;">New</td>
<td style="text-align: right;">New/Old</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><p>3-21G</p></td>
<td style="text-align: right;">34 s</td>
<td style="text-align: right;">20 s</td>
<td style="text-align: right;">0.58  </td>
<td style="text-align: right;">25 s</td>
<td style="text-align: right;">14 s</td>
<td style="text-align: right;">0.56  </td>
</tr>
<tr class="even">
<td style="text-align: left;">6-31G**</td>
<td style="text-align: right;">259 s</td>
<td style="text-align: right;">147 s</td>
<td style="text-align: right;">0.57  </td>
<td style="text-align: right;">212 s</td>
<td style="text-align: right;">120 s</td>
<td style="text-align: right;">0.57  </td>
</tr>
<tr class="odd">
<td style="text-align: left;">DZ</td>
<td style="text-align: right;">128 s</td>
<td style="text-align: right;">118 s</td>
<td style="text-align: right;">0.92  </td>
<td style="text-align: right;">72 s</td>
<td style="text-align: right;">62 s</td>
<td style="text-align: right;">0.86  </td>
</tr>
<tr class="even">
<td style="text-align: left;">cc-pVDZ</td>
<td style="text-align: right;">398 s</td>
<td style="text-align: right;">274 s</td>
<td style="text-align: right;">0.69  </td>
<td style="text-align: right;">308 s</td>
<td style="text-align: right;">185 s</td>
<td style="text-align: right;">0.60  </td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: right;">Old</td>
<td style="text-align: right;">New</td>
<td style="text-align: right;">New/Old</td>
<td style="text-align: right;">Old</td>
<td style="text-align: right;">New</td>
<td style="text-align: right;">New/Old</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><p>3-21G</p></td>
<td style="text-align: right;">294 s</td>
<td style="text-align: right;">136 s</td>
<td style="text-align: right;">0.46  </td>
<td style="text-align: right;">238 s</td>
<td style="text-align: right;">100 s</td>
<td style="text-align: right;">0.42  </td>
</tr>
<tr class="even">
<td style="text-align: left;">6-31G**</td>
<td style="text-align: right;">2520 s</td>
<td style="text-align: right;">976 s</td>
<td style="text-align: right;">0.39  </td>
<td style="text-align: right;">2065 s</td>
<td style="text-align: right;">828 s</td>
<td style="text-align: right;">0.40  </td>
</tr>
<tr class="odd">
<td style="text-align: left;">DZ</td>
<td style="text-align: right;">631 s</td>
<td style="text-align: right;">332 s</td>
<td style="text-align: right;">0.53  </td>
<td style="text-align: right;">600 s</td>
<td style="text-align: right;">230 s</td>
<td style="text-align: right;">0.38  </td>
</tr>
<tr class="even">
<td style="text-align: left;">cc-pVDZ</td>
<td style="text-align: right;">3202 s</td>
<td style="text-align: right;">1192 s</td>
<td style="text-align: right;">0.37  </td>
<td style="text-align: right;">2715 s</td>
<td style="text-align: right;">866 s</td>
<td style="text-align: right;">0.32  </td>
</tr>
</tbody>
</table>
<h2 id="user-controllable-variables">User-Controllable Variables</h2>
<p><span><span class="smallcaps">AOInts</span></span> has been optimally constructed so that the fastest integral algorithm for ERI calculation is chosen for the given integral class and batch. Thus, the user has not been provided with the necessary variables for overriding the program’s selection process. The user is, however, able to control the accuracy of the cutoff used during shell-pair formation () and the integral threshold (). In addition, the user can force the use of the direct SCF algorithm () and increase the default size of the integrals storage buffer ().</p>
<p>Currently, some of <span><span class="smallcaps">Q-Chem</span></span>’s linear scaling algorithms, such as QCTC and ONX algorithms, require the user to specify their use. It is anticipated that further research developments will lead to the identification of situations in which these, or combinations of these and other algorithms, will be selected automatically by <span><span class="smallcaps">Q-Chem</span></span> in much the same way that PRISM algorithms choose the most efficient pathway for given integral classes.</p>






<h1 id="app:quick_ref"><span><span class="smallcaps">Q-Chem</span></span> Quick Reference</h1>
<h2 id="q-chem-text-input-summary"><span><span class="smallcaps">Q-Chem</span></span> Text Input Summary</h2>
<ul>
<li><p>Users are able to enter input sections in any order; see Table <a href="#table:input_sections" data-reference-type="ref" data-reference="table:input_sections">[table:input_sections]</a> for a complete list of input sections.</p></li>
<li><p>Each input section must be terminated with <em>$end</em>.</p></li>
<li><p>Not all input sections are required, but <em>$rem</em> and <em>$molecule</em> are compulsory.</p></li>
<li><p>The entire <span><span class="smallcaps">Q-Chem</span></span> input is case-insensitive.</p></li>
<li><p>Multiple jobs are separated by the string <code>@@@</code> on a single line.</p></li>
</ul>
<h3 id="keyword-molecule">Keyword: <em>$molecule</em></h3>
<p>Four methods are available for inputing geometry information:</p>
<ul>
<li><p><em>Z</em>-matrix (<span>Å</span>ngstroms and degrees):<br />
<code>$molecule</code><br />
{<em>Z</em>-matrix}<br />
{blank line, if parameters are being used}<br />
{<em>Z</em>-matrix parameters, if used}<br />
<code>$end</code><br />
</p></li>
<li><p>Cartesian Coordinates (<span>Å</span>ngstroms):<br />
<code>$molecule</code><br />
{Cartesian coordinates}<br />
{blank line, if parameter are being used}<br />
{Coordinate parameters, if used}<br />
<code>$end</code><br />
</p></li>
<li><p>Read from a previous calculation:<br />
<code>$molecule</code><br />
<code>read</code><br />
<code>$end</code><br />
</p></li>
<li><p>Read from a file:<br />
<code>$molecule</code><br />
<code>read</code> <em>filename</em><br />
<code>$end</code><br />
</p></li>
</ul>
<h3 id="keyword-rem">Keyword: <em>$rem</em></h3>
<p>See also the list of <em>$rem</em> variables at the end of this Appendix. The general format is:</p>
<blockquote>
<pre><code>$rem
   REM_VARIABLE   VALUE   [optional comment]
$end</code></pre>
</blockquote>
<h3 id="keyword-basis">Keyword: <em>$basis</em></h3>
<p>The format for the user–defined basis section is as follows:</p>
<blockquote>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"><code>$basis</code></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: center;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"><span class="math inline"><em>X</em></span></td>
<td style="text-align: left;">0</td>
<td style="text-align: left;"></td>
<td style="text-align: center;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"><span class="math inline"><em>L</em></span></td>
<td style="text-align: left;"><span class="math inline"><em>K</em></span></td>
<td style="text-align: left;"><span class="math inline"><em>s</em><em>c</em><em>a</em><em>l</em><em>e</em></span></td>
<td style="text-align: center;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"><span class="math inline"><em>α</em><sub>1</sub></span></td>
<td style="text-align: left;"><span class="math inline"><em>C</em><sub>1</sub><sup><em>L</em><sub><em>m</em><em>i</em><em>n</em></sub></sup></span></td>
<td style="text-align: left;"><span class="math inline"><em>C</em><sub>1</sub><sup><em>L</em><sub><em>m</em><em>i</em><em>n</em></sub> + 1</sup></span></td>
<td style="text-align: center;"><span class="math inline">…</span></td>
<td style="text-align: left;"><span class="math inline"><em>C</em><sub>1</sub><sup><em>L</em><sub><em>m</em><em>a</em><em>x</em></sub></sup></span></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"><span class="math inline"><em>α</em><sub>2</sub></span></td>
<td style="text-align: left;"><span class="math inline"><em>C</em><sub>2</sub><sup><em>L</em><sub><em>m</em><em>i</em><em>n</em></sub></sup></span></td>
<td style="text-align: left;"><span class="math inline"><em>C</em><sub>2</sub><sup><em>L</em><sub><em>m</em><em>i</em><em>n</em></sub> + 1</sup></span></td>
<td style="text-align: center;"><span class="math inline">…</span></td>
<td style="text-align: left;"><span class="math inline"><em>C</em><sub>2</sub><sup><em>L</em><sub><em>m</em><em>a</em><em>x</em></sub></sup></span></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"><span class="math inline">⋮</span></td>
<td style="text-align: left;"><span class="math inline">⋮</span></td>
<td style="text-align: left;"><span class="math inline">⋮</span></td>
<td style="text-align: center;"><span class="math inline">⋱</span></td>
<td style="text-align: left;"><span class="math inline">⋮</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"><span class="math inline"><em>α</em><sub><em>K</em></sub></span></td>
<td style="text-align: left;"><span class="math inline"><em>C</em><sub><em>K</em></sub><sup><em>L</em><sub><em>m</em><em>i</em><em>n</em></sub></sup></span></td>
<td style="text-align: left;"><span class="math inline"><em>C</em><sub><em>K</em></sub><sup><em>L</em><sub><em>m</em><em>i</em><em>n</em></sub> + 1</sup></span></td>
<td style="text-align: center;"><span class="math inline">…</span></td>
<td style="text-align: left;"><span class="math inline"><em>C</em><sub><em>K</em></sub><sup><em>L</em><sub><em>m</em><em>a</em><em>x</em></sub></sup></span></td>
</tr>
<tr class="even">
<td style="text-align: left;"><code>****</code></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: center;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"><code>$end</code></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: center;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
</blockquote>
<p>where</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"><span class="math inline"><em>X</em></span></td>
<td style="text-align: left;">Atomic symbol of the atom (atomic number not accepted)</td>
</tr>
<tr class="even">
<td style="text-align: left;"><span class="math inline"><em>L</em></span></td>
<td style="text-align: left;">Angular momentum symbol (S, P, SP, D, F, G)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><span class="math inline"><em>K</em></span></td>
<td style="text-align: left;">Degree of contraction of the shell (integer)</td>
</tr>
<tr class="even">
<td style="text-align: left;"><span class="math inline"><em>s</em><em>c</em><em>a</em><em>l</em><em>e</em></span></td>
<td style="text-align: left;">Scaling to be applied to exponents (default is 1.00)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><span class="math inline"><em>a</em><sub><em>i</em></sub></span></td>
<td style="text-align: left;">Gaussian primitive exponent (positive real number)</td>
</tr>
<tr class="even">
<td style="text-align: left;"><span class="math inline"><em>C</em><sub><em>i</em></sub><sup><em>L</em></sup></span></td>
<td style="text-align: left;">Contraction coefficient for each angular momentum (non–zero real numbers).</td>
</tr>
</tbody>
</table>
<p>Atoms are terminated with <code>****</code> and the complete basis set is terminated with the <em>$end</em> keyword terminator. No blank lines can be incorporated within the general basis set input. Note that more than one contraction coefficient per line is one required for compound shells like SP. As with all <span><span class="smallcaps">Q-Chem</span></span> input deck information, all input is case–insensitive.</p>
<h3 id="keyword-comment">Keyword: <em>$comment</em></h3>
<p>Note that the entire input deck is echoed to the output file, thus making the <em>$comment</em> keyword largely redundant.</p>
<blockquote>
<pre><code>$comment
   User comments - copied to output file
$end</code></pre>
</blockquote>
<h3 id="keyword-ecp">Keyword: <em>$ecp</em></h3>
<p><code>$ecp</code><br />
For each atom that will bear an ECP<br />
Chemical symbol for the atom<br />
ECP name; the <span class="math inline"><em>L</em></span> value for the ECP; number of core electrons removed<br />
For each ECP component (in the order unprojected, <span class="math inline"><em>P̂</em><sub>0</sub></span>, <span class="math inline"><em>P̂</em><sub>1</sub></span>, , <span class="math inline"><em>P̂</em><sub><em>L</em> − 1</sub></span><br />
The component name<br />
The number of Gaussians in the component<br />
For each Gaussian in the component<br />
The power of <span class="math inline"><em>r</em></span>; the exponent; the contraction coefficient<br />
<code>****</code><br />
<code>$end</code></p>
<p><strong>Note: </strong></p>
<p>(1) All of the information in the <em>$ecp</em> block is case–insensitive.<br />
(2) The <span class="math inline"><em>L</em></span> value may not exceed 4. That is, nothing beyond <span class="math inline"><em>G</em></span> projectors is allowed.<br />
(3) The power of <span class="math inline"><em>r</em></span> (which includes the Jacobian <span class="math inline"><em>r</em><sup>2</sup></span> factor) must be 0, 1 or 2.</p>

<h3 id="keyword-empirical_dispersion">Keyword: <em>$empirical_dispersion</em></h3>
<blockquote>
<pre><code>$empirical_dispersion
S6 S6_value
D D_value
C6 element_1 C6_value_for_element_1 element_2 C6_value_for_element_2
VDW_RADII element_1 radii_for_element_1 element_2 radii_for_element_2
$end</code></pre>
</blockquote>
<p><strong>Note: </strong></p>
<p>This section is only for values that the user wants to change from the default values recommended by Grimme.</p>

<h3 id="keyword-external_charges">Keyword: <em>$external_charges</em></h3>
<p>All input should be given in atomic units.</p>
<p>Update: While charges should indeed be listed in atomic units, the units for distances depend on the user input. If the structure is specified in <span>Å</span>ngstroms (the default), the coordinates for external charges should also be in <span>Å</span>ngstroms. If the structure is specified in atomic units, the coordinates for external charges should also be in atomic units. (See .)</p>
<blockquote>
<pre><code>$external_charges
   x-coord1   y-coord1   z-coord1   charge1
   x-coord2   y-coord2   z-coord2   charge2
$end</code></pre>
</blockquote>
<h3 id="keyword-intracule">Keyword: <em>$intracule</em></h3>
<blockquote>
<p><code>$intracule</code><br />
</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"><em>int_type</em></td>
<td style="text-align: left;">0</td>
<td style="text-align: left;">Compute <span class="math inline"><em>P</em>(<em>u</em>)</span> only</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">Compute <span class="math inline"><em>M</em>(<em>v</em>)</span> only</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">Compute <span class="math inline"><em>W</em>(<em>u</em>, <em>v</em>)</span> only</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">3</td>
<td style="text-align: left;">Compute <span class="math inline"><em>P</em>(<em>u</em>)</span>, <span class="math inline"><em>M</em>(<em>v</em>)</span> and <span class="math inline"><em>W</em>(<em>u</em>, <em>v</em>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">4</td>
<td style="text-align: left;">Compute <span class="math inline"><em>P</em>(<em>u</em>)</span> and <span class="math inline"><em>M</em>(<em>v</em>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">5</td>
<td style="text-align: left;">Compute <span class="math inline"><em>P</em>(<em>u</em>)</span> and <span class="math inline"><em>W</em>(<em>u</em>, <em>v</em>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">6</td>
<td style="text-align: left;">Compute <span class="math inline"><em>M</em>(<em>v</em>)</span> and <span class="math inline"><em>W</em>(<em>u</em>, <em>v</em>)</span></td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>u_points</em></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Number of points, start, end.</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>v_points</em></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">Number of points, start, end.</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>moments</em></td>
<td style="text-align: left;">0–4</td>
<td style="text-align: left;">Order of moments to be computed (<span class="math inline"><em>P</em>(<em>u</em>)</span> only).</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><em>derivs</em></td>
<td style="text-align: left;">0–4</td>
<td style="text-align: left;">order of derivatives to be computed (<span class="math inline"><em>P</em>(<em>u</em>)</span> only).</td>
</tr>
<tr class="even">
<td style="text-align: left;"><em>accuracy</em></td>
<td style="text-align: left;"><span class="math inline"><em>n</em></span></td>
<td style="text-align: left;">(<span class="math inline">10<sup> − <em>n</em></sup>)</span> specify accuracy of intracule interpolation table (<span class="math inline"><em>P</em>(<em>u</em>)</span> only).</td>
</tr>
</tbody>
</table>
<p><code>$end</code></p>
</blockquote>
<h3 id="keyword-isotopes">Keyword: <em>$isotopes</em></h3>
<p>Note that masses should be given in atomic units.</p>
<blockquote>
<pre><code>$isotopes
   number_extra_loops  tp_flag
      number_of_atoms  [temp pressure]
      atom_number1 mass1
      atom_number2 mass2
   ...
$end</code></pre>
</blockquote>
<h3 id="keyword-multipole_field">Keyword: <em>$multipole_field</em></h3>
<p>Multipole fields are all in atomic units.</p>
<blockquote>
<pre><code>$multipole_field
   field_component1   value1
   field_component2   value2
   ...
$end</code></pre>
</blockquote>
<h3 id="keyword-nbo">Keyword: <em>$nbo</em></h3>
<p>Refer to Chapter <a href="#chap:properties" data-reference-type="ref" data-reference="chap:properties">11</a> and the NBO manual for further information. Note that the <em>$rem</em> variable must be set to to initiate the NBO package.</p>
<blockquote>
<pre><code>$nbo
   [ NBO options ]
$end</code></pre>
</blockquote>
<h3 id="keyword-occupied">Keyword: <em>$occupied</em></h3>
<blockquote>
<pre><code>$occupied
   1  2  3  4 ...  nalpha
   1  2  3  4 ...  nbeta  
$end  </code></pre>
</blockquote>
<h3 id="keyword-opt">Keyword: <em>$opt</em></h3>
<p>Note that units are in <span>Å</span>ngstroms and degrees. Also see the summary in the next section of this Appendix.</p>
<blockquote>
<pre><code>$opt
CONSTRAINT
stre  atom1  atom2  value
...
bend  atom1  atom2  atom3  value
...
outp  atom1  atom2  atom3  atom4  value
...
tors  atom1  atom2  atom3  atom4  value
...
linc  atom1  atom2  atom3  atom4  value
...
linp  atom1  atom2  atom3  atom4  value
...
ENDCONSTRAINT

FIXED
atom   coordinate_reference
...
ENDFIXED

DUMMY
idum   type   list_length   defining_list
...
ENDDUMMY

CONNECT
atom   list_length   list
...
ENDCONNECT
$end</code></pre>
</blockquote>
<h3 id="keyword-svp">Keyword: <em>$svp</em></h3>
<pre><code>$svp
&lt;KEYWORD&gt;=&lt;VALUE&gt;, &lt;KEYWORD&gt;=&lt;VALUE&gt;,...
&lt;KEYWORD&gt;=&lt;VALUE&gt;
$end</code></pre>
<p>For example, the section may look like this:</p>
<pre><code>$svp
    RHOISO=0.001, DIELST=78.39, NPTLEB=110
$end </code></pre>
<h3 id="keyword-svpirf">Keyword: <em>$svpirf</em></h3>
<pre><code>$svpirf
    &lt;# point&gt; &lt;x point&gt; &lt;y point&gt; &lt;z point&gt; &lt;charge&gt; &lt;grid weight&gt;
    &lt;# point&gt; &lt;x normal&gt; &lt;y normal&gt; &lt;z normal&gt;
$end</code></pre>
<h3 id="keyword-plots">Keyword: <em>$plots</em></h3>
<p><code>$plots</code><br />
One comment line<br />
Specification of the 3–D mesh of points on 3 lines:<br />
<span class="math inline"><em>N</em><sub><em>x</em></sub> <em>x</em><sub><em>m</em><em>i</em><em>n</em></sub> <em>x</em><sub><em>m</em><em>a</em><em>x</em></sub></span><br />
<span class="math inline"><em>N</em><sub><em>y</em></sub> <em>y</em><sub><em>m</em><em>i</em><em>n</em></sub> <em>y</em><sub><em>m</em><em>a</em><em>x</em></sub></span><br />
<span class="math inline"><em>N</em><sub><em>z</em></sub> <em>z</em><sub><em>m</em><em>i</em><em>n</em></sub> <em>z</em><sub><em>m</em><em>a</em><em>x</em></sub></span><br />
A line with 4 integers indicating how many things to plot:<br />
<span class="math inline"><em>N</em><sub><em>M</em><em>O</em></sub> <em>N</em><sub><em>R</em><em>h</em><em>o</em></sub> <em>N</em><sub><em>T</em><em>r</em><em>a</em><em>n</em><em>s</em></sub> <em>N</em><sub><em>D</em><em>A</em></sub></span><br />
An optional line with the integer list of MO’s to evaluate (only if <span class="math inline"><em>N</em><sub><em>M</em><em>O</em></sub> &gt; 0</span>)<br />
MO(1)  MO(2) <span class="math inline">…</span> MO(<span class="math inline"><em>N</em><sub><em>M</em><em>O</em></sub></span>)<br />
An optional line with the integer list of densities to evaluate (only if <span class="math inline"><em>N</em><sub><em>R</em><em>h</em><em>o</em></sub> &gt; 0</span>)<br />
Rho(1)  Rho(2) <span class="math inline">…</span> Rho(<span class="math inline"><em>N</em><sub><em>R</em><em>h</em><em>o</em></sub></span>)<br />
An optional line with the integer list of transition densities (only if <span class="math inline"><em>N</em><sub><em>T</em><em>r</em><em>a</em><em>n</em><em>s</em></sub> &gt; 0</span>)<br />
Trans(1)  Trans(2) <span class="math inline">…</span> Trans(<span class="math inline"><em>N</em><sub><em>T</em><em>r</em><em>a</em><em>n</em><em>s</em></sub></span>)<br />
An optional line with states for detachment/attachment densities (if <span class="math inline"><em>N</em><sub><em>D</em><em>A</em></sub> &gt; 0</span>)<br />
DA(1)  DA(2) <span class="math inline">…</span> DA(<span class="math inline"><em>N</em><sub><em>D</em><em>A</em></sub></span>)<br />
<code>$end</code></p>
<h3 id="keyword-localized_diabatization">Keyword: <em>$localized_diabatization</em></h3>
<p><code>$plots</code><br />
One comment line.<br />
One line with an an array of adiabatic states to mix together.<br />
<span class="math inline"> &lt; <em>a</em><em>d</em><em>i</em><em>a</em><em>b</em><em>a</em><em>t</em>1&gt;       &lt; <em>a</em><em>d</em><em>i</em><em>a</em><em>b</em><em>a</em><em>t</em>2&gt;        &lt; <em>a</em><em>d</em><em>i</em><em>a</em><em>b</em><em>a</em><em>t</em>3&gt;</span> <span class="math inline">…</span><br />
<code>$end</code><br />
<strong>Note: </strong></p>
<p>We count adiabatic states such that the first excited state is <span class="math inline"> &lt; <em>a</em><em>d</em><em>i</em><em>a</em><em>b</em><em>a</em><em>t</em> &gt;  = 1</span>, the fifth is <span class="math inline"> &lt; <em>a</em><em>d</em><em>i</em><em>a</em><em>b</em><em>a</em><em>t</em> &gt;  = 5</span>, and so forth.</p>
<h3 id="keyword-van_der_waals">Keyword: <em>$van_der_waals</em></h3>
<p><strong>Note: </strong></p>
<p>All radii are given in<span>Å</span>ngstroms.</p>
<blockquote>
<pre><code>$van_der_waals
   1
   atomic_number   VdW_radius
$end</code></pre>
</blockquote>
<p>(alternative format)</p>
<blockquote>
<pre><code>$van_der_waals
   2
   sequential_atom_number   VdW_radius 
$end</code></pre>
</blockquote>
<h3 id="keyword-xc_functional">Keyword: <em>$xc_functional</em></h3>
<blockquote>
<pre><code>$xc_functional
   X  exchange_symbol  coefficient
   X  exchange_symbol  coefficient
   ...
   C  correlation_symbol  coefficient
   C  correlation_symbol  coefficient
   ...
   K  coefficient
$end</code></pre>
</blockquote>
<h2 id="geometry-optimization-with-general-constraints">Geometry Optimization with General Constraints</h2>
<p>and define the beginning and end, respectively, of the constraint section of <em>$opt</em> within which users may specify up to six different types of constraints:</p>






<p>Absolute atom positions can be frozen with the <em>FIXED</em> section. The section starts with the <em>FIXED</em> keyword as the first line and ends with the <em>ENDFIXED</em> keyword on the last. The format to fix a coordinate or coordinates of an atom is:</p>
<p><code>atom   coordinate_reference</code></p>
<p><em>coordinate_reference</em> can be any combination of up to three characters <span class="math inline"><em>X</em></span>, <span class="math inline"><em>Y</em></span> and <span class="math inline"><em>Z</em></span> to specify the coordinate(s) to be fixed: <span class="math inline"><em>X</em></span>, <span class="math inline"><em>Y</em></span>, <span class="math inline"><em>Z</em></span>, <em>XY</em>, <em>XZ</em>, <em>YZ</em>, <em>XYZ</em>. The fixing characters must be next to each other. <em>e.g.</em>,</p>
<pre><code>FIXED
2 XY
ENDFIXED</code></pre>
<h2 id="rem-variable-list"><em>$rem</em> Variable List</h2>
<p>The general format of the <em>$rem</em> input for <span><span class="smallcaps">Q-Chem</span></span> text input files is simply as follows:</p>
<blockquote>
<pre><code>$rem
   rem_variable   rem_option   [comment]
   rem_variable   rem_option   [comment]
$end</code></pre>
</blockquote>
<p>This input is not case sensitive. The following sections contain the names and options of available <em>$rem</em> variables for users. The format for describing each <em>$rem</em> variable is as follows:</p>
<blockquote>

</blockquote>
<h3 id="general">General</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="scf-control">SCF Control</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="dft-options">DFT Options</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="large-molecules">Large Molecules</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="correlated-methods">Correlated Methods</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="correlated-methods-handled-by-ccman-and-ccman2">Correlated Methods Handled by CCMAN and CCMAN2</h3>
<p>Most of these <em>$rem</em> variables that start .</p>
<p>These are relevant for CCSD and other CC methods (OD, VOD, CCD, QCCD, CCVB-SD, etc).</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<p>These options are only relevant to methods involving orbital optimization (OOCD, VOD, QCCD, VQCCD):</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<p>Properties and optimization:</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="perfect-pairing-coupled-cluster-valence-bond-and-related-methods">Perfect pairing, Coupled cluster valence bond, and related methods</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="excited-states-cis-tddft-sf-xcis-and-sos-cisd">Excited States: CIS, TDDFT, SF-XCIS and SOS-CIS(D)</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="excited-states-eom-cc-and-ci-methods">Excited States: EOM-CC and CI Methods</h3>
<p>Those are keywords relevant to EOM-CC and CI methods handled by CCMAN/CCMAN2. Most of these <em>$rem</em> variables that start and .</p>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="geometry-optimizations">Geometry Optimizations</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="vibrational-analysis">Vibrational Analysis</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="reaction-coordinate-following">Reaction Coordinate Following</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="nmr-calculations">NMR Calculations</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="wave-function-analysis-and-molecular-properties">Wave function Analysis and Molecular Properties</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="symmetry">Symmetry</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="printing-options">Printing Options</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h3 id="resource-control">Resource Control</h3>
<table>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
</tr>
</tbody>
</table>
<h2 id="alphabetical-listing-of-rem-variables">Alphabetical Listing of <em>$rem</em> Variables</h2>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>
<p><span>Additional equations (for the left EOM-CCSD eigenvectors plus lambda CCSD equations in case if transition properties between the CCSD reference and EOM-CCSD target states are requested) need to be solved for transition properties, approximately doubling the computational cost. The cost of the transition properties calculation itself is low.</span></p>
</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>
<p><span>None</span></p>
</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>
<blockquote>

</blockquote>









